{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Import PDF and analyze\n",
    "\n",
    "Example notebook for how to read and analyze PDF files"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Python environment installation instructions\n",
    "\n",
    "General Packages:\n",
    "\n",
    "    conda install numpy scipy matplotlib jupyter\n",
    "\n",
    "PDF-specific packages:\n",
    "\n",
    "    pip install pypdf2\n",
    "    conda install tika\n",
    "\n",
    "Text analysis:\n",
    "\n",
    "    conda install nltk\n",
    "\n",
    "    Download additional nltk ressources in python shell\n",
    "\n",
    "        python nltk.download('punkt')\n",
    "        python nltk.download('averaged_perceptron_tagger')\n",
    "        python nltk.download('maxent_ne_chunker')\n",
    "        python nltk.download('words')\n",
    "        python nltk.download('treebank')\n",
    "        python nltk.download('stopwords')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Get Text and Metadata from PDF using PyPDF2\n",
    "\n",
    "https://pythonhosted.org/PyPDF2/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import PyPDF2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "# read an example pdf called 'nchemex.pdf'\n",
    "file = open('nchemex.pdf', 'rb')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "PdfReadWarning: Xref table not zero-indexed. ID numbers for objects will be corrected. [pdf.py:1736]\n"
     ]
    }
   ],
   "source": [
    "# creating a pdf reader object\n",
    "fileReader = PyPDF2.PdfFileReader(file)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "7\n"
     ]
    }
   ],
   "source": [
    "# print the number of pages in pdf file\n",
    "print(fileReader.numPages)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'/Author': 'Nuria Plattner',\n",
       " '/CreationDate': \"D:20170524121757+05'30'\",\n",
       " '/Creator': 'Arbortext Advanced Print Publisher 10.0.1465/W Unicode',\n",
       " '/ModDate': \"D:20170524134328+05'30'\",\n",
       " '/Producer': 'Acrobat Distiller 8.1.0 (Windows)',\n",
       " '/Subject': 'Nature Chemistry (2017). doi:10.1038/nchem.2785',\n",
       " '/Title': 'Complete protein-protein association kinetics in atomic detail revealed by molecular dynamics simulations and Markov modelling'}"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# get document metadata\n",
    "fileReader.documentInfo"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "page = fileReader.getPage(0)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "page_content = page.extractText()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "b'Completeprotein\\n\\xc5\\x92proteinassociationkineticsin\\natomicdetailrevealedbymoleculardynamics\\n\\nsimulationsandMarkovmodelling\\nNuriaPlattner\\n1*,StefanDoerr\\n2,GianniDeFabritiis\\n2,3\\nandFrankNo\\xc3\\xa9\\n1*Protein\\n\\xc5\\x92proteinassociationisfundamentaltomanylifeprocesses.However,amicroscopicmodeldescribingthestructures\\nandkineticsduringassociationanddissociationislackingonaccountofthelonglifetimesofassociatedstates,whichhave\\n\\npreventedef\\ncientsamplingbydirectmoleculardynamics(MD)simulations.Herewedemonstrateprotein\\n\\xc5\\x92protein\\nassociationanddissociationinatomisticresolutionfortheribonucleasebarnaseanditsinhibitorbarstarbycombining\\nadaptivehigh-throughputMDsimulationsandhiddenMarkovmodelling.Themodelrevealsexperimentallyconsistent\\n\\nintermediatestructures,energeticsandkineticsontimescalesfrommicrosecondstohours.Avarietyof\\nexiblyattached\\nintermediatesandmisboundstatesfunneldowntoatransitionstateandanativebasinconsistingofthelooselybound\\nnear-nativestateandthetightlyboundcrystallographicstate.Theseresultsofferadeeperlevelofinsightinto\\nmacromolecularrecognitionandourapproachopensthedoorforunderstandingandmanipulatingawiderangeof\\n\\nmacromolecularassociationprocesses.\\nLifeprocessesarecriticallydependentontheformationofbio-\\n\\nmolecularcomplexes,andinparticular,protein\\n\\xc5\\x92proteinassoci-\\nation,whichmediatestheassemblyofcellularstructures,signal\\ntransductionandinhibition,immuneresponseandproteinrecruit-\\n\\nmenttomembranescaffolds.Interactionsofproteinswithother\\nmacromoleculesalsoplaykeyrolesinapplicationssuchasinhibition\\nofproteinactivitybydrugs\\n1,geneediting\\n2andtheassemblyof\\nnewmaterials\\n3.Adeepunderstandingofsuchprocessesandtheabilitytofully\\nplanandcontroltheirdesignwouldarisefromafullmapofthe\\nconformationspaceoftheinteractingproteinpairs.Suchamicro-\\nscopicmodelwouldinvolverepresentativesamplesofallstructures\\n\\n(bound,dissociated,intermediates,etc.)inatomicresolution,their\\n\\nequilibriumpopulationsandthetransitionratesbetweenthem.\\nHowever,suchacompletemicroscopicpicturehasnotyetbeen\\nachievedforprotein\\n\\xc5\\x92proteininteraction.Experimentscanprobe\\neitherindividualaspectsofthemicroscopicpictureoraveraged\\n\\nensembles,suchasthemoststablestructures,associationkinetics\\n\\nandbindingaf\\nnities.Mutationalanalysescanidentifyamino\\nacidsstabilizingassociatedstatesortransitionstates\\n4.Thestructure\\nofthetransientencountercomplex,thatis,thesetofstructures\\n\\nfoundimmediatelybeforethetransitionstateofproteinassociation,\\n\\nhasbeenprobedusingNMR\\n5andcoarse-grainedsimulations\\n6\\xc5\\x929.Bindingfreeenergiescanbecomputedfromatomisticsimulations\\nwhentheboundposeispreciselyknown\\n10,andthebindingfree\\nenergylandscapecanbereconstructedusingmetadynamicsgiven\\n\\nasetofpre-de\\nnedcollectivevariables\\n11.Inferenceofkinetics\\nthenrequiresakineticmodel\\n12oradditionalunbiasedsimulations\\n13.Acandidatetorevealacompletemicroscopicmodelofprotein\\n\\xc5\\x92proteininteractionisunbiasedatomic-detailmoleculardynamics\\n(MD)simulation.Microscopicmodelsofbiomolecularprocesses\\n\\nhaverecentlybeenachievedduetocombinedadvancesinhigh-\\n\\nperformancecomputingandMarkovmodelling\\n14\\xc5\\x9219.Previous\\nbindingstudies,however,havebeenlimitedtotheassociationof\\nsmallligandstoproteinsandtheco-foldingofshorterpeptide\\nchains16,19,20,orsmallpartsoftheassociationlandscapesuchas\\nthedownhillbindingstepfromapre-boundpose\\n21.Thesimulation\\nofprotein\\n\\xc5\\x92proteinassociationischallengedbythefactthat\\ndissociationtimesexceedaccessibleMDsimulationlengthsbyfar,\\nsuchthatlongMDsimulationswillgetstuckafterasingle\\nbindingevent,possiblyeveninnon-nativetraps.\\nHerewedemonstratehowreversibleprotein\\n\\xc5\\x92proteinassociation\\ncanbestudiedinatomisticdetailbycombininganextensiveand\\n\\nadaptivelygeneratedensembleofunbiasedmoleculardynamics\\nsimulationswithhiddenMarkovmodelling.Wereproduce\\nmeasuredstructures,energeticsandkineticsquantitativelyand\\n\\nobtainafullmicroscopicmodeloftheproteincomplexstructures,\\n\\nthermodynamicsandkinetics.\\nResultsanddiscussion\\nAll-atomsimulationofreversibleprotein\\n\\xc5\\x92proteinassociation.\\nAsaprototypicalexample,westudythediffusion-limitedassociationof\\n\\ntheexperimentallywell-characterizedbacterialribonucleasebarnase\\nwithitsinhibitorbarstar\\n22,23.All-atommodelsofbarnasewildtype\\nandbarstarpseudowildtype\\n24wereimmersedinasolventbox\\n(approximately90,000atoms).Severalthousandunbiasedmolecular\\n\\ndynamics(MD)simulationsof2millisecondsaggregatelengthwere\\nconductedinanautomatedhigh-throughputmanner.Thesedata\\ninclude1.7millisecondsofindependenttrajectoriesstartingfrom\\n\\nthedissociatedstateand0.3millisecondsinadaptiveMDsimulations.\\nTheadaptiveMDlaunchedperiodsofmultipleparallelMD\\ntrajectories,andthestartingpointsofeachperiodwereselectedso\\nastofavourtheexplorationofnewstates(Methods,Supplementary\\nMethods1).Thisstrategycanyieldlargespeed-upscomparedto\\n\\nnon-adaptiveMDrunsbecauseslowprocesseswithlargeenergy\\n\\nbarriersarebrokendownintostepswithindividuallysmaller\\n\\nbarriersthatcanbesampledef\\nciently\\n25\\xc5\\x9227(Fig.1a,b).Whenstarting\\nfromthedissociatedstate,bothadaptiveandnon-adaptive\\n1DepartmentofMathematicsandComputerScience,FreieUniversit\\xc3\\xa4tBerlin,Arnimallee6,14195Berlin,Germany.\\n2ComputationalBiophysicsLaboratory\\n(GRIB-IMIM),UniversitatPompeuFabra,BarcelonaBiomedicalResearchPark(PRBB),C/DoctorAiguader88,08003Barcelona,Spain.\\n3Instituci\\xc3\\xb3Catalana\\ndeRecercaiEstudisAvan\\xc3\\xa7ats(ICREA),PasseigLluisCompanys23,08010Barcelona,Spain.\\n*e-mail:\\nnuria.plattner@fu-berlin.de\\n;frank.noe@fu-berlin.de\\nARTICLESPUBLISHEDONLINE:5JUNE2017|\\nDOI:10.1038/NCHEM.2785\\nNATURECHEMISTRY\\n|ADVANCEONLINEPUBLICATION|\\nwww.nature.com/naturechemistry\\n1\\xc2\\xa92017MacmillanPublishersLimited,partofSpringerNature.Allrightsreserved.\\n'\n"
     ]
    }
   ],
   "source": [
    "print (page_content.encode('utf-8'))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'Completeprotein\\nŒproteinassociationkineticsin\\natomicdetailrevealedbymoleculardynamics\\n\\nsimulationsandMarkovmodelling\\nNuriaPlattner\\n1*,StefanDoerr\\n2,GianniDeFabritiis\\n2,3\\nandFrankNoé\\n1*Protein\\nŒproteinassociationisfundamentaltomanylifeprocesses.However,amicroscopicmodeldescribingthestructures\\nandkineticsduringassociationanddissociationislackingonaccountofthelonglifetimesofassociatedstates,whichhave\\n\\npreventedef\\ncientsamplingbydirectmoleculardynamics(MD)simulations.Herewedemonstrateprotein\\nŒprotein\\nassociationanddissociationinatomisticresolutionfortheribonucleasebarnaseanditsinhibitorbarstarbycombining\\nadaptivehigh-throughputMDsimulationsandhiddenMarkovmodelling.Themodelrevealsexperimentallyconsistent\\n\\nintermediatestructures,energeticsandkineticsontimescalesfrommicrosecondstohours.Avarietyof\\nexiblyattached\\nintermediatesandmisboundstatesfunneldowntoatransitionstateandanativebasinconsistingofthelooselybound\\nnear-nativestateandthetightlyboundcrystallographicstate.Theseresultsofferadeeperlevelofinsightinto\\nmacromolecularrecognitionandourapproachopensthedoorforunderstandingandmanipulatingawiderangeof\\n\\nmacromolecularassociationprocesses.\\nLifeprocessesarecriticallydependentontheformationofbio-\\n\\nmolecularcomplexes,andinparticular,protein\\nŒproteinassoci-\\nation,whichmediatestheassemblyofcellularstructures,signal\\ntransductionandinhibition,immuneresponseandproteinrecruit-\\n\\nmenttomembranescaffolds.Interactionsofproteinswithother\\nmacromoleculesalsoplaykeyrolesinapplicationssuchasinhibition\\nofproteinactivitybydrugs\\n1,geneediting\\n2andtheassemblyof\\nnewmaterials\\n3.Adeepunderstandingofsuchprocessesandtheabilitytofully\\nplanandcontroltheirdesignwouldarisefromafullmapofthe\\nconformationspaceoftheinteractingproteinpairs.Suchamicro-\\nscopicmodelwouldinvolverepresentativesamplesofallstructures\\n\\n(bound,dissociated,intermediates,etc.)inatomicresolution,their\\n\\nequilibriumpopulationsandthetransitionratesbetweenthem.\\nHowever,suchacompletemicroscopicpicturehasnotyetbeen\\nachievedforprotein\\nŒproteininteraction.Experimentscanprobe\\neitherindividualaspectsofthemicroscopicpictureoraveraged\\n\\nensembles,suchasthemoststablestructures,associationkinetics\\n\\nandbindingaf\\nnities.Mutationalanalysescanidentifyamino\\nacidsstabilizingassociatedstatesortransitionstates\\n4.Thestructure\\nofthetransientencountercomplex,thatis,thesetofstructures\\n\\nfoundimmediatelybeforethetransitionstateofproteinassociation,\\n\\nhasbeenprobedusingNMR\\n5andcoarse-grainedsimulations\\n6Œ9.Bindingfreeenergiescanbecomputedfromatomisticsimulations\\nwhentheboundposeispreciselyknown\\n10,andthebindingfree\\nenergylandscapecanbereconstructedusingmetadynamicsgiven\\n\\nasetofpre-de\\nnedcollectivevariables\\n11.Inferenceofkinetics\\nthenrequiresakineticmodel\\n12oradditionalunbiasedsimulations\\n13.Acandidatetorevealacompletemicroscopicmodelofprotein\\nŒproteininteractionisunbiasedatomic-detailmoleculardynamics\\n(MD)simulation.Microscopicmodelsofbiomolecularprocesses\\n\\nhaverecentlybeenachievedduetocombinedadvancesinhigh-\\n\\nperformancecomputingandMarkovmodelling\\n14Œ19.Previous\\nbindingstudies,however,havebeenlimitedtotheassociationof\\nsmallligandstoproteinsandtheco-foldingofshorterpeptide\\nchains16,19,20,orsmallpartsoftheassociationlandscapesuchas\\nthedownhillbindingstepfromapre-boundpose\\n21.Thesimulation\\nofprotein\\nŒproteinassociationischallengedbythefactthat\\ndissociationtimesexceedaccessibleMDsimulationlengthsbyfar,\\nsuchthatlongMDsimulationswillgetstuckafterasingle\\nbindingevent,possiblyeveninnon-nativetraps.\\nHerewedemonstratehowreversibleprotein\\nŒproteinassociation\\ncanbestudiedinatomisticdetailbycombininganextensiveand\\n\\nadaptivelygeneratedensembleofunbiasedmoleculardynamics\\nsimulationswithhiddenMarkovmodelling.Wereproduce\\nmeasuredstructures,energeticsandkineticsquantitativelyand\\n\\nobtainafullmicroscopicmodeloftheproteincomplexstructures,\\n\\nthermodynamicsandkinetics.\\nResultsanddiscussion\\nAll-atomsimulationofreversibleprotein\\nŒproteinassociation.\\nAsaprototypicalexample,westudythediffusion-limitedassociationof\\n\\ntheexperimentallywell-characterizedbacterialribonucleasebarnase\\nwithitsinhibitorbarstar\\n22,23.All-atommodelsofbarnasewildtype\\nandbarstarpseudowildtype\\n24wereimmersedinasolventbox\\n(approximately90,000atoms).Severalthousandunbiasedmolecular\\n\\ndynamics(MD)simulationsof2millisecondsaggregatelengthwere\\nconductedinanautomatedhigh-throughputmanner.Thesedata\\ninclude1.7millisecondsofindependenttrajectoriesstartingfrom\\n\\nthedissociatedstateand0.3millisecondsinadaptiveMDsimulations.\\nTheadaptiveMDlaunchedperiodsofmultipleparallelMD\\ntrajectories,andthestartingpointsofeachperiodwereselectedso\\nastofavourtheexplorationofnewstates(Methods,Supplementary\\nMethods1).Thisstrategycanyieldlargespeed-upscomparedto\\n\\nnon-adaptiveMDrunsbecauseslowprocesseswithlargeenergy\\n\\nbarriersarebrokendownintostepswithindividuallysmaller\\n\\nbarriersthatcanbesampledef\\nciently\\n25Œ27(Fig.1a,b).Whenstarting\\nfromthedissociatedstate,bothadaptiveandnon-adaptive\\n1DepartmentofMathematicsandComputerScience,FreieUniversitätBerlin,Arnimallee6,14195Berlin,Germany.\\n2ComputationalBiophysicsLaboratory\\n(GRIB-IMIM),UniversitatPompeuFabra,BarcelonaBiomedicalResearchPark(PRBB),C/DoctorAiguader88,08003Barcelona,Spain.\\n3InstitucióCatalana\\ndeRecercaiEstudisAvançats(ICREA),PasseigLluisCompanys23,08010Barcelona,Spain.\\n*e-mail:\\nnuria.plattner@fu-berlin.de\\n;frank.noe@fu-berlin.de\\nARTICLESPUBLISHEDONLINE:5JUNE2017|\\nDOI:10.1038/NCHEM.2785\\nNATURECHEMISTRY\\n|ADVANCEONLINEPUBLICATION|\\nwww.nature.com/naturechemistry\\n1©2017MacmillanPublishersLimited,partofSpringerNature.Allrightsreserved.\\n'"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "page_content"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Get text and metadata from PDF using TIKA\n",
    "\n",
    "General information about TIKA\n",
    "https://cwiki.apache.org/confluence/display/TIKA/TikaServer\n",
    "\n",
    "TIKA python API\n",
    "https://github.com/chrismattmann/tika-python"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "from tika import parser"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "# read an example pdf called 'nchemex.pdf'\n",
    "parsedPDF = parser.from_file(\"nchemex.pdf\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'content': '\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\nComplete protein-protein association kinetics in atomic detail revealed by molecular dynamics simulations and Markov modelling\\n\\n\\nComplete protein–protein association kinetics in\\natomic detail revealed by molecular dynamics\\nsimulations and Markov modelling\\nNuria Plattner1*, Stefan Doerr2, Gianni De Fabritiis2,3 and Frank Noé1*\\n\\nProtein–protein association is fundamental to many life processes. However, a microscopic model describing the structures\\nand kinetics during association and dissociation is lacking on account of the long lifetimes of associated states, which have\\nprevented efficient sampling by direct molecular dynamics (MD) simulations. Here we demonstrate protein–protein\\nassociation and dissociation in atomistic resolution for the ribonuclease barnase and its inhibitor barstar by combining\\nadaptive high-throughput MD simulations and hidden Markov modelling. The model reveals experimentally consistent\\nintermediate structures, energetics and kinetics on timescales from microseconds to hours. A variety of flexibly attached\\nintermediates and misbound states funnel down to a transition state and a native basin consisting of the loosely bound\\nnear-native state and the tightly bound crystallographic state. These results offer a deeper level of insight into\\nmacromolecular recognition and our approach opens the door for understanding and manipulating a wide range of\\nmacromolecular association processes.\\n\\nLife processes are critically dependent on the formation of bio-molecular complexes, and in particular, protein–protein associ-ation, which mediates the assembly of cellular structures, signal\\ntransduction and inhibition, immune response and protein recruit-\\nment to membrane scaffolds. Interactions of proteins with other\\nmacromolecules also play key roles in applications such as inhibition\\nof protein activity by drugs1, gene editing2 and the assembly of\\nnew materials3.\\n\\nA deep understanding of such processes and the ability to fully\\nplan and control their design would arise from a full map of the\\nconformation space of the interacting protein pairs. Such a micro-\\nscopic model would involve representative samples of all structures\\n(bound, dissociated, intermediates, etc.) in atomic resolution, their\\nequilibrium populations and the transition rates between them.\\n\\nHowever, such a complete microscopic picture has not yet been\\nachieved for protein–protein interaction. Experiments can probe\\neither individual aspects of the microscopic picture or averaged\\nensembles, such as the most stable structures, association kinetics\\nand binding affinities. Mutational analyses can identify amino\\nacids stabilizing associated states or transition states4. The structure\\nof the transient encounter complex, that is, the set of structures\\nfound immediately before the transition state of protein association,\\nhas been probed using NMR5 and coarse-grained simulations6–9.\\nBinding free energies can be computed from atomistic simulations\\nwhen the bound pose is precisely known10, and the binding free\\nenergy landscape can be reconstructed using metadynamics given\\na set of pre-defined collective variables11. Inference of kinetics\\nthen requires a kinetic model12 or additional unbiased simulations13.\\n\\nA candidate to reveal a complete microscopic model of protein–\\nprotein interaction is unbiased atomic-detail molecular dynamics\\n(MD) simulation. Microscopic models of biomolecular processes\\nhave recently been achieved due to combined advances in high-\\nperformance computing and Markov modelling14–19. Previous\\nbinding studies, however, have been limited to the association of\\n\\nsmall ligands to proteins and the co-folding of shorter peptide\\nchains16,19,20, or small parts of the association landscape such as\\nthe downhill binding step from a pre-bound pose21. The simulation\\nof protein–protein association is challenged by the fact that\\ndissociation times exceed accessible MD simulation lengths by far,\\nsuch that long MD simulations will get stuck after a single\\nbinding event, possibly even in non-native traps.\\n\\nHere we demonstrate how reversible protein–protein association\\ncan be studied in atomistic detail by combining an extensive and\\nadaptively generated ensemble of unbiased molecular dynamics\\nsimulations with hidden Markov modelling. We reproduce\\nmeasured structures, energetics and kinetics quantitatively and\\nobtain a full microscopic model of the protein complex structures,\\nthermodynamics and kinetics.\\n\\nResults and discussion\\nAll-atom simulation of reversible protein–protein association. As\\na prototypical example, we study the diffusion-limited association of\\nthe experimentally well-characterized bacterial ribonuclease barnase\\nwith its inhibitor barstar22,23. All-atom models of barnase wild type\\nand barstar pseudo wild type24 were immersed in a solvent box\\n(approximately 90,000 atoms). Several thousand unbiased molecular\\ndynamics (MD) simulations of 2 milliseconds aggregate length were\\nconducted in an automated high-throughput manner. These data\\ninclude 1.7 milliseconds of independent trajectories starting from\\nthe dissociated state and 0.3 milliseconds in adaptive MD simulations.\\n\\nThe adaptive MD launched periods of multiple parallel MD\\ntrajectories, and the starting points of each period were selected so\\nas to favour the exploration of new states (Methods, Supplementary\\nMethods 1). This strategy can yield large speed-ups compared to\\nnon-adaptive MD runs because slow processes with large energy\\nbarriers are broken down into steps with individually smaller\\nbarriers that can be sampled efficiently25–27 (Fig. 1a,b). When starting\\nfrom the dissociated state, both adaptive and non-adaptive\\n\\n1Department of Mathematics and Computer Science, Freie Universität Berlin, Arnimallee 6, 14195 Berlin, Germany. 2Computational Biophysics Laboratory\\n(GRIB-IMIM), Universitat Pompeu Fabra, Barcelona Biomedical Research Park (PRBB), C/ Doctor Aiguader 88, 08003 Barcelona, Spain. 3Institució Catalana\\nde Recerca i Estudis Avançats (ICREA), Passeig Lluis Companys 23, 08010 Barcelona, Spain. *e-mail: nuria.plattner@fu-berlin.de; frank.noe@fu-berlin.de\\n\\nARTICLES\\nPUBLISHED ONLINE: 5 JUNE 2017 | DOI: 10.1038/NCHEM.2785\\n\\nNATURE CHEMISTRY | ADVANCE ONLINE PUBLICATION | www.nature.com/naturechemistry 1\\n\\n© 2017 Macmillan Publishers Limited, part of Springer Nature. All rights reserved.\\n\\nmailto:nuria.plattner@fu-berlin.de\\nmailto:frank.noe@fu-berlin.de\\nhttp://dx.doi.org/10.1038/nchem.2785\\nhttp://www.nature.com/naturechemistry\\n\\n\\nsimulations reproducibly found the native bound complex. Adaptive\\nsimulations required a total simulation effort of a few microseconds,\\nroughly an order of magnitude speed-up compared to an equilibrium\\nbinding time of a tens of microseconds (Supplementary Fig. 1).\\nSurprisingly, the adaptive MD runs could also reproducibly reach\\nthe dissociated state when starting from the crystal structure using\\na total simulation effort of a few hundred microseconds, although\\nthe expected dissociation time is hours (Supplementary Fig. 1).\\n\\nA first analysis shows that the total dataset contains several\\nthousand unspecific association and dissociation events, about 90\\nbinding events from non-native associated states to native and 10\\nunbinding events (Fig. 1c).\\n\\nEquilibrium kinetics model and validation. Although the\\nindividual simulations are not started from an equilibrium\\ndistribution and are shorter than binding and dissociation\\ntimescales, Markov modelling approaches can combine such a\\ntrajectory ensemble towards unbiased equilibrium models of the\\nlong-timescale kinetics and thermodynamics (Fig. 1a)28,29. Here, a\\nhidden Markov model (HMM) was built to estimate a network of\\n\\nstates, associated protein structures, and transition rates30. The\\nHMM is coupled to a diffusive association model inspired by a\\nprevious study31 in order to be able to predict concentration-\\ndependent association rates and binding free energies (Methods,\\nSupplementary Methods 3–7).\\n\\nAs the data contains only a few unbinding events, the following\\ncomputed quantities have large uncertainties but are statistically\\nconsistent with experiment, which will be exploited to compute\\nconsensus models of experimental and simulation data (intervals\\nare 95% confidence): (i) binding free energy lies between 12 and\\n19 kcal mol–1 (experimental 16.8 kcal mol–1; see Supplementary\\nMethods 5), (ii) the dissociation rate lies in the range of 3 × 10–6\\n\\nto 1 × 10–1 s–1 (measurements lie in the range of 4.8 × 10–5 to\\n5.0 × 10–4 s–1; see Supplementary Methods 6).\\n\\nThe model successfully predicts the following quantities: (iii)\\ncrystal structure 1BRS is predicted to be the most stable state, and\\nthe simulated structures assigned to it have an average RMSD of\\n0.21 nm to 1BRS, (iv) association rate is predicted to be 2.63 × 108\\n\\nto 2.65 × 108 s–1 M–1 (1 × 108 to 3.4 × 108 s–1 M–1 measured pre-\\nviously23,32), (v) using perturbation theory33, binding free energy\\nchanges of 35 mutant sequences reported in23,32 were predicted\\nwithin statistical error (Fig. 2a and Supplementary Methods 9),\\n(vi) although this approach cannot compute absolute values of\\nassociation rates of mutants, the computed change in transition\\nstate energy is an excellent predictor for the change in the experimental\\non-rate (Fig. 2b).\\n\\nOverall binding mechanism. Having validated our model with a\\nlarge set of experimental data we proceeded to examine properties\\nthat are more difficult to probe directly by experiments. Figure 2c,\\nd displays the overall mechanisms and structures found along the\\nbinding pathways. The proteins initially associate to non-native\\n(large RMSD) states. At that point, they can dissociate and re-\\nassociate or stay in contact and rearrange towards the native\\ncomplex (Fig. 2c–f ).\\n\\nAt the high protein concentration in the simulation, initial\\nassociation occurs within tens of nanoseconds and with no detect-\\nable barrier. The initial association of barstar can occur at all\\nlocations on the barnase surface. During the first microsecond,\\nthe complex preferentially populates early intermediates with\\nbarstar located opposite of the barnase binding site (orange states\\nin Fig. 2d). However, association can also occur directly into the\\nlate intermediate and mis-bound states (blue/red states in Fig. 2d),\\nwhere barstar is located in or near the barnase binding groove,\\nbut not in the native orientation. The preference for initially associ-\\nating to locations away from the binding pocket can be explained by\\nthe fact that these states cover a larger fraction of the surface. The\\ninitial association preference depends on the salt concentration;\\nwith enhanced steering to the binding site being expected when\\ncharge screening is reduced7,22. Knocking out intermediate states\\nby removing them from the HMM showed that individual early\\nintermediates have no significant effect on the association rate,\\nand even removing all early intermediates only reduces the\\nassociation rate by 20% (Supplementary Table 2).\\n\\nOn timescales of 1–10 µs, the population shifts toward the late\\nintermediates and mis-bound states (blue/red states in Figs 2d\\nand 5c), which have higher equilibrium populations than the\\nearly intermediates. Of special interest is the mis-bound state\\n(red), that has barstar bound ‘upside down’. This state does not\\nhave a well-defined binding interface, but a few non-native inter-\\nactions such as barnase Arg 83 or Arg 87 with barstar Glu 118\\noccur with significant probability (Supplementary Figs 3 and 4).\\nIntermediate states are kinetically well connected and have lifetimes\\non the order of one microsecond each (Supplementary Fig. 2).\\nRemoving the late intermediates would reduce the binding rate by\\na factor of 2.5 (Supplementary Table 2)\\n\\n1 2\\n\\n3\\n4\\n\\n5 Sampling strategy \\n\\nMarkov model \\n\\nTrajectories\\n\\nTasks\\n\\nData \\ngeneration \\n\\nModel building \\n\\nCompute clusterClientba\\n\\nc\\n\\nAssociatedDissociated Bound\\n\\nTime (μs) Time (μs)\\n\\n2,867\\n\\nD\\nis\\n\\nta\\nnc\\n\\ne \\n(n\\n\\nm\\n)\\n\\nR\\nM\\n\\nS\\nD\\n\\n (\\nnm\\n\\n)\\n\\n86\\n6\\n\\n0\\n0 0.5 0\\n\\n0\\n\\n1.5\\n\\n5\\n\\nTime (μs) Time (μs)\\n\\nR\\nM\\n\\nS\\nD\\n\\n (\\nnm\\n\\n)\\n\\nD\\nis\\n\\nta\\nnc\\n\\ne \\n(n\\n\\nm\\n) 101,465\\n\\n6\\n\\n0\\n0 3.5 0 7\\n\\n0\\n\\n1.5\\n\\nFigure 1 | Simulation of protein–protein association and dissociation.\\na, Scheme of a large-barrier process such as protein–protein dissociation that\\ncan be subdivided into multiple steps. Such processes can be sampled by\\ncombining adaptively seeded simulation trajectories between individual steps\\nvia Markov modelling. b, Schematic outline of adaptive sampling process.\\nc, Total number of full association–dissociation and binding–unbinding\\ntransitions evaluated by raw trajectory statistics of all barnase–barstar\\nsimulation trajectories used subsequently. An association–dissociation\\ntransition was counted when a trajectory moved from a state where the\\nproteins had a minimum distance between heavy atom pairs of >2 nm to\\n<0.3 nm, and vice versa. A binding–unbinding event was counted when a\\ntrajectory moved from a state with minimum heavy-atom RMSD from the\\ncrystallographic structure >1 nm to RMSD <0.3 nm, and vice versa. For each\\ntrajectory only one count was made in order to avoid duplicate counting\\nof recrossing events. Three sample trajectory snippets are shown for each\\ntype of transition.\\n\\nARTICLES NATURE CHEMISTRY DOI: 10.1038/NCHEM.2785\\n\\nNATURE CHEMISTRY | ADVANCE ONLINE PUBLICATION | www.nature.com/naturechemistry2\\n\\n© 2017 Macmillan Publishers Limited, part of Springer Nature. All rights reserved.\\n\\nhttp://dx.doi.org/10.1038/nchem.2785\\nhttp://www.nature.com/naturechemistry\\n\\n\\nMisbound\\n\\nEarly intermediates\\n\\nLate\\nintermediate 2\\n\\nLate\\nintermediate 1\\n\\nPre-bound\\n\\nLoosely\\nbound\\n\\nTightly\\nbound\\n\\na\\n\\nd\\n\\nb c e\\n\\nf\\n\\ng\\n\\nDissociated\\n\\nAssociation\\n\\nBinding\\n\\nUnbinding Rebinding\\n\\n12\\nΔΔ\\n\\nG\\n e\\n\\nxp\\ner\\n\\nim\\nen\\n\\nt (\\nkc\\n\\nal\\n m\\n\\nol\\n–1\\n\\n)\\n\\n8\\n\\n4\\n\\n–4 0\\n\\nΔΔG model (kcal mol–1)\\n\\n4 8 12\\n\\n–4\\n\\n0\\n\\n3\\n\\n2\\n\\n2\\n\\n0\\n\\nΔΔ\\nG\\n\\non\\n e\\n\\nxp\\ner\\n\\nim\\nen\\n\\nt (\\nkc\\n\\nal\\n m\\n\\nol\\n–1\\n\\n)\\n\\n–5 0 5 10\\n\\nΔΔGTS model (kcal mol–1)\\n\\n–2\\n\\n–1 F\\nre\\n\\ne \\nen\\n\\ner\\ngy\\n\\n (\\nkT\\n\\n)\\n\\nMinimal distance (nm)\\n\\n3.0 24\\n\\n21\\n\\n18\\n\\n15\\n\\n12\\n\\n9\\n\\n6\\n\\n3\\n\\n0\\n\\n2.5\\n\\n2.0\\n\\nR\\nM\\n\\nS\\nD\\n\\n (\\nnm\\n\\n)\\n\\n0.3\\n0.2\\n0.1\\n\\n0.00 0.02 0.04 0.06 0.08 0.10\\n\\n5\\n\\nt t+50 t+100 t+150 t+200\\n\\n10 15 20 25 30\\n\\n0.3\\n0.2\\n0.1\\n\\n0.3\\n0.2\\n0.1\\n\\nR\\nM\\n\\nS\\nD\\n\\n (\\nnm\\n\\n)\\nR\\n\\nM\\nS\\n\\nD\\n (\\n\\nnm\\n)\\n\\nR\\nM\\n\\nS\\nD\\n\\n (\\nnm\\n\\n)\\n\\n1.5\\n\\n1.0\\n\\n0.5\\n\\n0.25 0.50 1.00 2.00 4.00\\n\\nTime (μs)\\n\\nFigure 2 | Kinetic model of barnase–barstar association. a,b, Comparison of experimental and predicted changes in binding free energies and association\\nrates for all 35 single and double alanine mutations from Table II of ref. 23 and Table I of ref. 32. Statistical confidence intervals of the model are indicated in\\nwhite (1-σ, 68% confidence) and grey (2-σ, 95% confidence). a, Changes in binding free energy (ΔΔG) are predicted within statistical error. b, Comparison\\nof the computed changes in barrier height of the binding transition with changes of apparent association transition state energy, computed as ΔΔGon,apparent\\n= –kBTln(kon\\n\\nmutant/kon\\nwt), where kBT= 0.598 kcal mol\\n\\n–1 and kon\\nmutant, kon\\n\\nwt are the experimental association rates of mutant and wild type, respectively. The model\\nprediction was computed as free energy difference between the pre-bound state as a proxy for the transition state, and all early intermediate, late\\nintermediate and mis-bound states. c, Free energy surface over protein–protein heavy atom minimal distance and minimal RMSD to the native complex.\\nShown is a contour plot of –kTlnμ(x) evaluated for each trajectory sample x, where kT is the thermal energy and μ(x) is the equilibrium weight of each\\nsample computed by the HMM (see Supplementary Methods 4). d, Kinetic network between metastable states with lifetimes on the order of microseconds.\\nOnly transition rate constants ≥0.2ms–1 are shown. Arrow thickness is proportional to rate constant, disc area is proportional to square root of state\\npopulation. Structures are centered on barnase (orange) with its binding groove on top and its N-terminal helix on the bottom. Barstar is shown in grey with\\nits binding domain in green. Grey clouds show overlay of approximately 20 barstar conformations, sampled randomly from the corresponding metastable\\nstates. e–g, Minimal RMSD to crystallographic complex shown for sample trajectories generated from the HMM (see Supplementary Methods 4). Initial\\nassociation (e), binding to the native complex (f), and geminate rebinding after being bound for a long time t (one hour on average; g).\\n\\nNATURE CHEMISTRY DOI: 10.1038/NCHEM.2785 ARTICLES\\n\\nNATURE CHEMISTRY | ADVANCE ONLINE PUBLICATION | www.nature.com/naturechemistry 3\\n\\n© 2017 Macmillan Publishers Limited, part of Springer Nature. All rights reserved.\\n\\nhttp://dx.doi.org/10.1038/nchem.2785\\nhttp://www.nature.com/naturechemistry\\n\\n\\nOn timescales of 10–100 µs, the complex passes through a\\npre-bound state (magenta in Figs 2d and 5c) and then continues\\nto the natively bound complex (grey states in Figs 2d and 5c),\\nwhich consists of a loosely bound and a tightly bound substate\\nthat exchange on the microsecond timescale.\\n\\nNature of the transition state of binding. The kinetic bottleneck\\nbetween the intermediate states and natively bound states is the\\nmandatory passage through the pre-bound state (magenta in Fig. 2d),\\nthat is hence an on-pathway intermediate. Both the transition into\\nthe pre-bound state and the binding from the pre-bound state\\ninvolve barriers (Supplementary Fig. 6). We refer to these barriers\\ncollectively as transition state of binding as they are rate-limiting at\\nsufficiently high protein concentrations (Supplementary Fig. 5).\\n\\nThe transition from the late intermediates through the pre-bound\\nstate into the loosely bound state is characterized by two microscopic\\npathways, in which barstar interacts with either of the two barnase\\nloops surrounding the binding groove: the ‘left’ loop including Ser\\n\\n38 and the ‘right’ RNA recognition loop of barnase including Arg\\n59 (Fig. 3) whose importance for steering of the two protein has\\nbeen shown before6,7. Due to the flexibility of both loop regions,\\nbarstar can “swing” around its attachment point, giving rise to a\\nquite flexible set of structures. These pathways are not separated by\\nsignificant barriers, i.e. the complex can change between “left” and\\n“right” attachment point multiple times before binding.\\n\\nThe minor left pathway (5% population, Fig. 3a→ c→ e) involves\\ndifferent possible protein contacts, including hydrophobic contacts\\nand does not appear to have well-defined structural features. In the\\nmajor right pathway (95% population, Fig. 3b→ d→ e), barnase Arg\\n59 interacts with barstar Glu 76, Asp 35, or both, in line with34,35.\\nArg 59 may additionally form a hydrophobic contact with barstar\\nTrp 38. While no individual pair of residues is strongly formed in the\\ntransition state22, Arg 59 is involved in 95% of all transition state con-\\nfigurations. This observation confirms the pivotal role of Arg 59 inmed-\\niating the binding transition34, and explains that Arg 59 mutants\\nreduced the binding rate by a factor of 10 (ref. 32). We expect that a\\nbarstar double mutant of Glu 76 and Asp 35 will have a similar effect.\\n\\nThe pre-bound state has a lifetime of about two microseconds\\n(Supplementary Fig. 2). In all binding events simulated, the\\ntransition from the pre-bound to the loosely bound state trajectories\\noccurs within a single saving interval of our simulations, that is less\\nthan 100 picoseconds. Thus the last step of docking into the barnase\\nbinding pocket and forming the binding interface with its many\\ncontacts is rare, but occurs quickly when it happens21,36. In line with\\nprevious predictions22, the majority of the barnase–barstar short-\\nrange interactions is formed after the transition state. In the limit of\\nlow electrostatic interactions, the basal binding rate may be governed\\nby the bimolecular rate to form a ready-to-dock structure similar to\\nour pre-bound state22, and it is likely that the barrier of desolvation\\nand docking is reduced or even removed under these conditions.\\n\\nConformational change in the complex. The bound states are in the\\nrange of 5–12 kcal mol–1 more stable than intermediate, mis-bound\\nand pre-bound states and thus contain almost all of the equilibrium\\npopulation. We find two conformational substates in the bound\\nstate: The loosely bound state with about 5% of the population\\nand an average heavy-atom RMSD to the crystal structure of\\n0.3 nm (Fig. 4a); and the tightly bound state with the remaining\\n95% of the population (Fig. 4b). The density of the tightly bound\\nstate contains the crystal structure (Fig. 4c, average RMSD\\n0.21 nm), which is thus correctly predicted as the most stable state.\\n\\nUpon binding, the loosely bound state is reached first. This state\\ntransitions into the tightly bound state within 5 µs on average, that\\nitself has an average lifetime of 130 µs before transitioning back to\\nthe loosely bound state. These microsecond transitions occur\\nmany times before the complex undergoes geminate rebinding\\n(Fig. 2g) or finally dissociates.\\n\\nStructurally, the native-state transition consists of a slight\\nrotation of barstar in the barnase binding pocket (compare\\nFig. 4a,b). While both substates have similar residue contact patterns\\n(Supplementary Fig. 3), the most apparent difference is that the\\nbarnase 35–41 loop is inserted between the barstar binding helix\\nand the large helix of barstar in the loosely bound state, while it is\\ninserted between the binding helix and binding loop in the tightly\\nbound state. The loosely bound state is stabilized by an interaction\\nbetween barnase Lys 27 (whose functional role is to stabilize the\\nphosphate group in the RNA substrate) and barstar Glu 80, while\\nthe tightly bound state is stabilized by a salt bridge between\\nbarnase Arg 83 and barstar Asp 39. Due to the latter salt bridge,\\nbarnase loop 77–85 is stabilized in the tightly bound state, while it\\nis very flexible in the loosely bound state. Overall, the loosely bound\\nstate is more flexible than the tightly bound state (compare Fig. 4a,b).\\n\\nThese native substate dynamics may be observable experimen-\\ntally, for example, by fluorescence correlation microscopy or by\\n\\na b\\n\\nc d\\n\\ne\\n\\nArg 59\\nAsp 35\\n\\nTrp 38 Glu 76\\n\\nGly 43\\nTrp 44\\n\\nAla 37\\nSer 38\\n\\n45° Late\\nintermediate 2\\n\\n \\n\\nPre-bound\\n\\nLoosely\\nbound\\n\\n45°\\n\\nFigure 3 | Late steps in binding from late intermediates to the loosely\\nbound state. a,b, Two sets of structures of late intermediate state 2, shown\\nin 45° counterclockwise/clockwise rotation compared to Fig. 2d. c,d, Two\\nsets of structures of the pre-bound state. e, Loosely bound state. Grey clouds\\nshow overlay of approximately 20 barstar conformations, sampled randomly\\nfrom the corresponding metastable states. Two binding pathways exist, in\\nwhich the binding domain of barstar interacts with barnase at its ‘left’ loop\\naround Ala 37 and Ser 38 (a,c), or at its ‘right’ loop (b,d), before docking into\\nthe bound state (e). In the major pathway (b→ d→ e, approximately 95% of\\nthe flux), barnase Arg 59 ‘anchors’ into barstar via Asp 35, Glu 76 (polar)\\nand Trp 38 (hydrophobic). Barstar can swing around barnase with the\\nanchor staying attached until final docking into the bound pose.\\n\\nARTICLES NATURE CHEMISTRY DOI: 10.1038/NCHEM.2785\\n\\nNATURE CHEMISTRY | ADVANCE ONLINE PUBLICATION | www.nature.com/naturechemistry4\\n\\n© 2017 Macmillan Publishers Limited, part of Springer Nature. All rights reserved.\\n\\nhttp://dx.doi.org/10.1038/nchem.2785\\nhttp://www.nature.com/naturechemistry\\n\\n\\nfast temperature jump experiments that have been employed to\\nprobe loose–tight binding transitions37. In particular, barstar Trp\\n44 changes solvent accessibility during the native substate transition,\\nrendering intrinsic tryptophan fluorescence a promising probe.\\n\\nAssociation kinetics of wild-type and a faster-associating mutant.\\nThe very large association rate of barnase–barstar can be further\\nincreased by specific mutations. To understand the microscopic\\nmechanisms of such perturbations, we conducted additional\\n0.3 milliseconds of MD simulations using the barnase mutant Glu\\n73 to Ala (E73A, see Methods), which associates a factor of 1.5–4\\nfaster than the wild type23.\\n\\nAs both the wild-type and the E73A HMM suffered from a\\nstatistical bottleneck at the unbinding transition, the uncertain\\ntransition rate from the loosely bound to pre-bound state was set\\nsuch that the models produce the experimentally known binding\\nfree energy. The resulting consensus models still make independent\\npredictions of the binding rate and mechanism, while reproducing\\nexperimental thermodynamics.\\n\\nTheE73Amodel predicts anassociation rate of 4 to 4.7 × 108 s–1M–1,\\nwhich correctly reflects an increase by a factor of 1.6–1.8 (ref. 23),\\nand can be explained by the increased electrostatic complementarity.\\nE73 does not participate in the binding interface, and indeed the\\nwild-type native complex structure is still predicted to be the most\\n\\n95%5%RMSD 0.3 nm\\nLifetime 5 ms\\n\\nTightly bound\\nRMSD 0.21 nm\\nLifetime 130 ms\\n\\n1BRS\\ncrystal structure \\n\\na b c\\n\\nArg 83\\nAsp 39Lys 27\\n\\nGlu 80\\n\\nTrp 44\\nTrp 44\\n\\nLoosely bound\\n\\nFigure 4 | Native substate conformational dynamics. Native substates seen from ‘left’, that is, in a 90° clockwise rotation around the image vertical with\\nrespect to Fig. 2d. a, Loosely bound state with lower population and higher structural flexibility. b, Tightly bound state with higher population and small\\nstructural flexibility. c, Crystal structure 1BRS shown in the structural density of the tightly bound simulation state, for comparison with B. Grey cloud is\\nan overlay of approximately 20 barstar conformations, sampled randomly from the corresponding metastable states.\\n\\nWild type E73A\\n\\na b\\n\\nC\\nom\\n\\nm\\nitt\\n\\nor\\n\\n1\\n\\n0\\n\\n10–1 100 101 102 103 104\\n\\n0.5\\n\\nc d\\n\\nLoosely\\nbound\\n\\nPrebound\\n\\nLate intermediates\\n\\nEarly intermediates\\n\\nDissociated\\n\\nMisbound\\n\\nTightly\\nbound\\n\\n1.0\\n0.8\\n0.6\\n0.4\\n0.2\\n0.0\\n\\nP\\nro\\n\\nba\\nbi\\n\\nlit\\ny\\n\\nTime (μs)\\n10–1 100 101 102 103 104\\n\\n1.0\\n0.8\\n0.6\\n0.4\\n0.2\\n0.0\\n\\nP\\nro\\n\\nba\\nbi\\n\\nlit\\ny\\n\\nTime (μs)\\n\\nFigure 5 | Comparison of kinetics between wild type and E73A mutant. a,b, Kinetic networks of the consensus models that are consistent with MD\\nsimulation data and the experimental binding free energy. Network plots as in Fig. 2d, but the vertical position of states indicates the binding probability\\n(committor) and disc area is proportional to fourth root of state population. The E73A mutant shows a deeper funnel that leads to an increased association\\nrate. c,d, Kinetic evolution of probabilities when starting from dissociated state, computed from the HMM (see Supplementary Methods 4). Despite a higher\\nassociation rate, the E73A mutant takes longer to bind to the native state, as the increased association rate is due to rapid trapping in a mis-bound state that\\nis committed to further native binding. Colour scheme as in Fig. 2d.\\n\\nNATURE CHEMISTRY DOI: 10.1038/NCHEM.2785 ARTICLES\\n\\nNATURE CHEMISTRY | ADVANCE ONLINE PUBLICATION | www.nature.com/naturechemistry 5\\n\\n© 2017 Macmillan Publishers Limited, part of Springer Nature. All rights reserved.\\n\\nhttp://dx.doi.org/10.1038/nchem.2785\\nhttp://www.nature.com/naturechemistry\\n\\n\\nstable state in the mutant (Fig. 5b). Figure 5a,b shows kinetic\\nnetworks of the wild type and E73A mutant, similar to Fig. 2d but\\nwith the y-positions indicating the probability of binding (committor)\\nof each state. Both systems are characterized by binding funnels,\\nthat are broad on top but narrow down rapidly as the complex pro-\\nceeds through late intermediates towards binding. The E73A funnel\\nis deeper in the sense that early intermediates have larger committor\\nprobabilities, that is, it is more difficult to escape the complex, once\\nit is associated. E73A has a reduced stability of the bound state23.\\nThe state that experiences the most profound increase of probability\\nis the mis-bound state. The time to form the native bound complex\\nis much longer in E73A compared to the wild type, as the system\\ngets trapped in the mis-bound state for 100s of microseconds\\n(Fig. 5d). An increase of the association rate does therefore not\\nnecessarily imply more efficient binding in a functional sense.\\n\\nBased on the consensus models the experimental association rate\\nis simulated as a function of protein concentration. At micromolar\\nconcentrations commonly used in experiments, the association rates\\nare linear in the concentration, that is, binding is limited by the\\nbimolecular encounter (Supplementary Fig. 5). In this condition\\nthe dominant energy barrier is characterized by balancing the entro-\\npic cost of bringing two molecules together at low concentration and\\nthe onset of the electrostatic interactions. At concentrations of 10s to\\n100s of μM, this barrier disappears, the transition state of binding\\nbecomes rate-limiting4, and we observe a concentration-independent\\nrate constant of 18,000 s–1 for the wild type and 1,500 s–1 for the E73A\\nmutant due to the transition state of binding (Supplementary Fig. 5).\\n\\nAfter an initial contact has been formed, the complex will\\ntypically move through a sequence of other non-native inter-\\nmediates until it dissociates or proceeds to native binding with\\nabout 50% probability each (Fig. 5a,b). The major bottleneck in\\ndissociation from the native state is the separation of the loosely\\nbound complex into the pre-bound, mis-bound and intermediate\\nstates (Supplementary Fig. 6), from where the complex can dissociate\\nor perform geminate rebinding (Fig. 2g).\\n\\nConclusion\\nBy combining recent breakthroughs in simulation and analysis tech-\\nnology, we have demonstrated that a quantitatively accurate micro-\\nscopic model of protein–protein association can be obtained using\\nunbiased all-atom MD simulations and Markov modelling. This\\nmodel reveals kinetic and mechanistic details of the association mech-\\nanism and has enabled the discovery of previously unknown substates.\\n\\nThe presented combination of simulation and analysis techniques\\nare applicable to any pair of macromolecules. While the available\\ncomputational power limits the size of the systems being studied to\\ncurrently a few hundred amino acids per protein, biological timescales\\ncan be reached that are far beyond direct MD simulation times. This\\nopens the door to study how the parts of macromolecular assemblies\\nform, how viral and host proteins interact and how protein–DNA\\ninteractions depend on DNA sequence and epigenetic modifications.\\n\\nRather than a black-box model, the present approach provides\\npopulations for every sampled atomistic structure and transition\\nrates between them. We have demonstrated that combining such an\\nextensive structural ensemble with an appropriate reweighting\\nmethod, the thermodynamics and kinetics of similar protein sequences\\ncan bepredicted. In the future, protein sequences andpost-translational\\nmodifications, or sequence and epigenetic modifications of interacting\\nnucleic acids can be designed to stabilize desirable intermediates,\\nchange the kinetics or control the mechanism, with far-reaching\\nconsequences for pharmaceutics, biotechnology andmaterial sciences.\\n\\nMethods\\nMD simulation setup. All-atom models of barnase wild type and barstar pseudo\\nwild type with C40,82A double mutation24 were constructed starting from crystal\\nstructure 1BRS (Protein Data Bank ID)38 (chain B for barnase and chain F\\n\\nsuperimposed to chain E for barstar) and immersed in a water box with ∼10 nm\\nedge length at 50 mM salt concentration (approximately 90,000 atoms). Protonation\\nstates were chosen using AmberTools39. His102 was in un-protonated (U-tautomer)\\nform that is dominant at pH 7. The Amber ff99SB force field40, the TIP3P water\\nmodel41, and the ACEMD program42 were used for simulations. The system was\\nminimized and equilibrated first in the NpT ensemble; box size was fixed and\\nsimulations were conducted in NVT at 300 K.\\n\\nWild-type GPUgrid simulations. 1,892 independent MD simulations starting from\\ndissociated structures (generated by random translation and rotation of barstar from\\nthe complex, excluding overlapping structures) were distributed on the GPUgrid\\ncomputing network43. Trajectories were up to 1 english μs long (distribution shown\\nin Supplementary Fig. 1), and the aggregate simulation time was 1.7 milliseconds. 74\\nof these simulations found the native barnase–barstar complex.\\n\\nWild-type adaptive MD simulations. Multiple rounds of short MD trajectories are\\nrun and a selection strategy based on available trajectory statistics is employed to select\\nthe starting coordinates for the next round of trajectories25–27 (Supplementary Methods\\n1 and Fig. 1b for details). Different combinations of selection strategies were tested.\\nAssociation and binding into the native complex succeeded with various strategies,\\nwhereas dissociation only succeeded by combining regular spatial clustering with a\\nminimum RMSD metric14, and selecting new MD frames with a probability anti-\\nproportional to the number of frames in each cluster. To confirm that the observed\\ndissociation is robust with respect to the force field used, we reproduced adaptive\\ndissociation using CHARMM36 (ref. 44) following the setup reported previously10.\\n\\nAdaptive trajectories were 50 or 100 ns long when generated. Selected adaptive\\nruns (Supplementary Methods 2) were extended to 200 ns for the analysis. 797\\nadaptive MD simulations starting from the natively bound complex, and 756 from\\ndissociated and non-native associated states were used, comprising an aggregate\\nsimulation time of 0.3 milliseconds.\\n\\nE73A mutant simulations. 25 MD frames were sampled from each of the 15\\nwild-type HMM states (Fig. 2d). For each structure, Glu 73 was mutated to Ala,\\nstructures were rebuild using HTMD45, and four copies were run on GPUgrid,\\nresulting in 1,500 simulations of 200 ns length, summing up to 300 µs.\\n\\nFeature selection and discretization. Following feature selection and discretization\\nwas carried out for the wild-type simulations using PyEMMA (http://pyemma.org)46\\n\\nand HTMD (http://htmd.org)45. 4,094 ‘relevant’ residue–residue contacts between\\nproteins, defined by nearest-neighbour heavy-atom distances <0.4 nm, were\\nmeasured. A contact was considered relevant when the longest time it had been\\nformed continuously exceeded 10 ns. The following features were used for\\nanalysis: (i) the minimum heavy-atom RMSD between each frame and the native\\nbarnase/barstar complex, (ii) the minimum distance between the two proteins,\\nand (iii) the seven slowest linear combinations of the 4,094 relevant contacts,\\ncomputed variationally50 by the TICA method47,48 (lagtime 100 ns). This space was\\ndiscretized using k-means clustering (k = 2,000).\\n\\nE73A mutant simulation trajectories were mapped on the same features and\\nassigned to the same cluster discretization.\\n\\nHidden Markov model. Only simulations conducted with Amber ff99SB were\\nused to estimate the HMM. Discrete HMMs as described previously30 are built on\\nthe k-means clustering and find a coarse-grained kinetic model by optimally\\ngrouping fast mixing states into metastable sets, and estimating the transition rates\\nbetween them. See Supplementary Methods 4 and Supplementary Fig. 2 for\\nself-consistent validation of the model.\\n\\nHMMs with 15 metastable states were estimated from the discretized simulation\\ndata using the msm package of PyEMMA. The HMM with lag time 110 ns was\\nselected for analysis because the free energies and relaxation timescales were\\napproximately independent of the lag at this point (Supplementary Fig. 2). For the\\nE73A mutant trajectories, we also estimated an HMM at lag 110 ns. Uncertainties\\nwere computed using the reversible Bayesian estimator described previously49.\\nHMM analysis was conducted using the methods provided by PyEMMA’s HMM\\nclass, and some functionalities of the msmtools package (Supplementary Methods 4).\\n\\nProtein association model. While the HMM automatically identified a dissociated\\nstate (lower green state in Fig. 2d), calculation of association rates and binding free\\nenergies comparable to experiment requires the definition of a bulk dissociated\\nstate with concentration-dependent association rate. Following an idea by\\nNorthrup et al.31, we attached a bulk dissociated state to the HMM as follows:\\n\\nbulk dissociated state ⇌\\n4πDb\\n\\nku\\nHMM dissociated state⇌\\n\\nkij\\n\\nkji\\nother HMM stateswhere kij\\n\\ndenotes the transition rates between HMM states. The bimolecular association rate\\n4πDb is the Smoluchowski rate of hitting a sphere of radius b = 4 nm around the\\norigin that is the effective size of the HMM dissociated state. The diffusion constant\\nD = 28.5 nm2 μs–1 has been fitted from the MD simulations. The bulk dissociation\\nrate is given by ku = ( f∞\\n\\n–1 – 1) Σjkji, where f∞ is the probability to dissociate to the bulk\\nwhen starting from b, which was computed by combining trajectory statistics and a\\nBrownian dynamics model (Supplementary Methods 9).\\n\\nARTICLES NATURE CHEMISTRY DOI: 10.1038/NCHEM.2785\\n\\nNATURE CHEMISTRY | ADVANCE ONLINE PUBLICATION | www.nature.com/naturechemistry6\\n\\n© 2017 Macmillan Publishers Limited, part of Springer Nature. All rights reserved.\\n\\nhttp://pyemma.org\\nhttp://htmd.org\\nhttp://dx.doi.org/10.1038/nchem.2785\\nhttp://www.nature.com/naturechemistry\\n\\n\\nCode availability. PyEMMA (http://pyemma.org) and HTMD (http://htmd.org) are\\navailable free of charge. See http://acellera.com for the availability of ACEMD.\\n\\nReceived 19 October 2016; accepted 21 April 2017;\\npublished online 5 June 2017\\n\\nReferences\\n1. Scott, D. E., Bayly, A. R., Abell, C. & Skidmore, J. Small molecules, big targets:\\n\\ndrug discovery faces the protein–protein interaction challenge. Nat. Rev. Drug.\\nDiscov. 15, 533–550 (2016).\\n\\n2. Doench, J. G. et al. Rational design of highly active sgrnas for crispr–cas9-\\nmediated gene inactivation. Nat. Biotechnol. 32, 1262–1267 (2014).\\n\\n3. King, N. P. et al. Computational design of self-assembling protein nanomaterials\\nwith atomic level accuracy. Science 336, 1171–1174 (2012).\\n\\n4. Schreiber, G., Haran, G. & Zhou, H.-X. Fundamental aspects of protein–protein\\nassociation kinetics. Chem. Rev. 109, 839–860 (2009).\\n\\n5. Tang, C., Iwahara, J. & Clore, G. M. Visualization of transient encounter\\ncomplexes in protein–protein association. Nature 444, 383–386 (2006).\\n\\n6. Gabdoulline, R. R. & Wade, R. C. Simulation of the diffusional association of\\nbarnase and barstar. Biophys. J. 72, 1917–1929 (1997).\\n\\n7. Spaar, A., Dammer, C., Gabdoulline, R. R., Wade, R. C. & Helms, V. Diffusional\\nencounter of barnase and barstar. Biophys. J. 90, 1913–1924 (2006).\\n\\n8. Levy, Y., Wolynes, P. G. & Onuchic, J. N. Protein topology determines binding\\nmechanism. Proc. Natl Acad. Sci. USA 101, 511–516 (2004).\\n\\n9. Schluttig, J., Alamanova, D., Helms, V. & Schwarz, U. S. Dynamics of protein–\\nprotein encounter: a langevin equation approach with reaction patches. J. Chem.\\nPhys. 129, 155106 (2008).\\n\\n10. Gumbart, J. C., Roux, B. & Chipot, C. Efficient determination of protein–protein\\nstandard binding free energies from first principles. J. Chem. Theory Comput. 9,\\n3789–3798 (2013).\\n\\n11. Barducci, A., Bonomi, M., Prakash, M. K. & Parrinello, M. Free-energy\\nlandscape of protein oligomerization from atomistic simulations. Proc. Natl\\nAcad. Sci. USA 110, E4708–E4713 (2013).\\n\\n12. Tiwary, P. & Parrinello, M. From metadynamics to dynamics. Phys. Rev. Lett.\\n111, 230602 (2013).\\n\\n13. Wu, H., Paul, F., Wehmeyer, C. & Noé, F. Multiensemble Markov models of\\nmolecular thermodynamics and kinetics. Proc. Natl Acad. Sci. USA 113,\\nE3221–E3230 (2016).\\n\\n14. Prinz, J.-H. et al. Markov models of molecular kinetics: generation and\\nvalidation. J. Chem. Phys. 134, 174105 (2011).\\n\\n15. Lindorff-Larsen, K., Piana, S., Dror, R. O. & Shaw, D. E. How fast-folding\\nproteins fold. Science 334, 517–520 (2011).\\n\\n16. Buch, I., Giorgino, T. & De Fabritiis, G. Complete reconstruction of an enzyme-\\ninhibitor binding process by molecular dynamics simulations. Proc. Natl Acad.\\nSci. USA 108, 10184–10189 (2011).\\n\\n17. Kohlhoff, K. J. et al. Cloud-based simulations on Google Exacycle reveal ligand\\nmodulation of GPCR activation pathways. Nat. Chem. 6, 15–21 (2014).\\n\\n18. Plattner, N. & Noé, F. Protein conformational plasticity and complex ligand\\nbinding kinetics explored by atomistic simulations and Markov models. Nat.\\nCommun. 6, 7653 (2015).\\n\\n19. Silva, D.-A., Bowman, G. R., Sosa-Peinado, A. & Huang, X. A role for both\\nconformational selection and induced fit in ligand binding by the Lao protein.\\nPLoS Comput. Biol. 7, e1002054 (2011).\\n\\n20. Piana, S., Lindorff-Larsen, K. & Shaw, D. E. Atomistic description of the folding\\nof a dimeric protein. J. Phys. Chem. B 117, 12935–12942 (2013).\\n\\n21. Ahmad, M., Gu, W., Geyer, T. & Helms, V. Adhesive water networks facilitate\\nbinding of protein interfaces. Nat. Commun. 2, 261 (2011).\\n\\n22. Schreiber, G. & Fersht, A. R. Rapid, electrostatically assisted association of\\nproteins. Nat. Struct. Biol. 3, 427–431 (1996).\\n\\n23. Schreiber, G. & Fersht, A. R. Interaction of barnase with its polypeptide\\ninhibitor barstar studied by protein engineering. Biochemistry 32,\\n5145–5150 (1993).\\n\\n24. Hartley, R. W. Directed mutagenesis and barnase–barstar recognition.\\nBiochemistry 32, 5978–5984 (1993).\\n\\n25. Doerr, S. & Fabritiis, G. D. On-the-fly learning and sampling of ligand binding\\nby high-throughput molecular simulations. J. Chem. Theory Comput. 10,\\n2064–2069 (2014).\\n\\n26. Bowman, G. R., Ensign, D. L. & Pande, V. S. Enhanced modeling via network\\ntheory: adaptive sampling of Markov state models. J. Chem. Theory Comput. 6,\\n787–794 (2010).\\n\\n27. Preto, J. & Clementi, C. Fast recovery of free energy landscapes via\\ndiffusion-map-directed molecular dynamics. Phys. Chem. Chem. Phys. 16,\\n19181–19191 (2014).\\n\\n28. Bowman, G. R., Pande, V. S. & Noé, F. (eds.) An Introduction to Markov State\\nModels and Their Application to Long Timescale Molecular Simulation (Vol. 797\\nof Advances in Experimental Medicine and Biology, Springer, 2014).\\n\\n29. Sarich, M. & Schütte, C. Metastability and Markov State Models in Molecular\\nDynamics (Courant Lecture Notes, American Mathematical Society, 2013).\\n\\n30. Noé, F., Wu, H., Prinz, J.-H. & Plattner, N. Projected and hidden Markov models\\nfor calculating kinetics and metastable states of complex molecules. J. Chem.\\nPhys. 139, 184114 (2013).\\n\\n31. Northrup, S. H., Allison, S. & McCammon, J. Brownian dynamics of diffusion-\\ninfluenced bimolecular reactions. J. Chem. Phys. 80, 1517–1524 (1984).\\n\\n32. Schreiber, G. & Fersht, A. R. Energetics of protein–protein interactions: analysis\\nof the barnase–barstar interface by single mutations and double mutant cycles.\\nJ. Mol. Biol. 248, 478–486 (1995).\\n\\n33. Matysiak, S. & Clementi, C. Optimal combination of theory and experiment for\\nthe characterization of the protein folding landscape of S6: how far can a\\nminimalist model go? J. Mol. Biol. 343, 235–248 (2004).\\n\\n34. Frisch, C., Fersht, A. R. & Schreiber, G. Experimental assignment of the structure\\nof the transition state for the association of barnase and barstar. J. Mol. Biol. 308,\\n69–77 (2001).\\n\\n35. Harel, M., Cohen, M. & Schreiber, G. On the dynamic nature of the transition\\nstate for protein–protein association as determined by double-mutant cycle\\nanalysis and simulation. J. Mol. Biol. 371, 180–196 (2007).\\n\\n36. Chung, H. S., Louis, J. M. & Eaton,W. A. Single-molecule fluorescence experiments\\ndetermine protein folding transition path times. Science 335, 981–984 (2012).\\n\\n37. Anunciado, D., Dhar, A., Gruebele, M. & Baranger, A. M. Multistep kinetics of\\nthe U1A–SL2 RNA complex dissociation. J. Mol. Biol. 408, 896–908 (2011).\\n\\n38. Buckle, A. M., Schreiber, G. & Fersht, A. R. Protein–protein recognition: crystal\\nstructural analysis of a barnase–barstar complex at 2.0-Å resolution.\\nBiochemistry 33, 8878–8889 (1994).\\n\\n39. Case, D. A. et al. The Amber biomolecular simulation programs. J. Comput.\\nChem. 26, 1668–1688 (2005).\\n\\n40. Hornak, V. et al. Comparison of multiple Amber force fields and development of\\nimproved protein backbone parameters. Proteins 65, 712–725 (2006).\\n\\n41. Jorgensen, W. L., Chandrasekhar, J., Madura, J. D., Impey, R. W. & Klein, M. L.\\nComparison of simple potential functions for simulating liquid water. J. Chem.\\nPhys. 79, 926–935 (1983).\\n\\n42. Harvey, M. J., Giupponi, G. & De Fabritiis, G. ACEMD: Accelerated molecular\\ndynamics simulations in the microseconds timescale. J. Chem. Theory Comp. 5,\\n1632–1639 (2009).\\n\\n43. Buch, I., Harvey, M. J., Giorgino, T., Anderson, D. P. & De Fabritiis, G.\\nHigh-throughput all-atom molecular dynamics simulations using distributed\\ncomputing. J. Chem. Inf. Model. 50, 397–403 (2010).\\n\\n44. Best, R. B. et al. Optimization of the additive CHARMM all-atom protein force\\nfield targeting improved sampling of the backbone ϕ, ψ and side-chain χ1 and χ2\\ndihedral angles. J. Chem. Theo. Comp. 8, 3257–3273 (2012).\\n\\n45. Doerr, S., Harvey, M. J., Noé, F. & Fabritiis, G. D. HTMD: high-throughput\\nmolecular dynamics for molecular discovery. J. Chem. Theory Comput. 12,\\n1845–1852 (2016).\\n\\n46. Scherer, M. K. et al. PyEMMA 2: a software package for estimation, validation\\nand analysis of Markov models. J. Chem. Theory Comput. 11, 5525–5542 (2015).\\n\\n47. Perez-Hernandez, G., Paul, F., Giogino, T., De Fabritiis, G. & Noé, F.\\nIdentification of slow molecular order parameters for Markov model\\nconstruction. J. Chem. Phys. 139, 015102 (2013).\\n\\n48. Molgedey, L. & Schuster, H. G. Separation of a mixture of independent signals\\nusing time delayed correlations. Phys. Rev. Lett. 72, 3634–3637 (1994).\\n\\n49. Trendelkamp-Schroer, B., Wu, H., Paul, F. & Noé, F. Estimation and uncertainty\\nof reversible Markov models. J. Chem. Phys. 143, 174101 (2015).\\n\\n50. Noé, F. & Nüske, F. A variational approach to modeling slow processes in\\nstochastic dynamical systems. Multiscale Model. Simul. 11, 635–655 (2013).\\n\\nAcknowledgements\\nWe are grateful to the following researchers for inspiring discussions and extensive\\nfeedback on the manuscript: J. Clark, C. Clementi, O. Daumke, W. A. Eaton, M. Gruebele,\\nK. Lindorff-Larsen, S. Olsson, F. Paul, B. Roux, U. Schwarz, S. Sukenik, R. Wade and\\nT.Weikl. Funding is acknowledged from the European Commission (ERC StG pcCells and\\nPRACE project 2014102337 to F.N.), Deutsche Forschungsgemeinschaft (NO 825/3-1, SFB\\n958/A4, SFB 740/D7 to F.N.), Einstein Foundation Berlin (project SoOPic to N.P.),\\nMINECO (BIO2014-53095-P) and FEDER (to G.D.F.), Acellera Ltd. (to S.D.). We thank\\nthe volunteers of GPUGRID for donating computing time for simulations.\\n\\nAuthor contributions\\nN.P., S.D., G.D.F and F.N. designed research and developed/implemented software.\\nN.P. and S.D. conducted simulations. N.P., S.D. and F.N. analysed simulations. F.N.\\ndeveloped methods. N.P. and F.N. wrote the paper.\\n\\nAdditional information\\nSupplementary information is available in the online version of the paper. Reprints and\\npermissions information is available online at www.nature.com/reprints. Publisher’s note:\\nSpringer Nature remains neutral with regard to jurisdictional claims in published maps and\\ninstitutional affiliations. Correspondence and requests for materials should be addressed\\nto N.P. or F.N.\\n\\nCompeting financial interests\\nThe authors declare no competing financial interests.\\n\\nNATURE CHEMISTRY DOI: 10.1038/NCHEM.2785 ARTICLES\\n\\nNATURE CHEMISTRY | ADVANCE ONLINE PUBLICATION | www.nature.com/naturechemistry 7\\n\\n© 2017 Macmillan Publishers Limited, part of Springer Nature. All rights reserved.\\n\\nhttp://pyemma.org\\nhttp://htmd.org\\nhttp://acellera.com\\nhttp://dx.doi.org/10.1038/nchem.2785\\nhttp://www.nature.com/reprints\\nhttp://dx.doi.org/10.1038/nchem.2785\\nhttp://www.nature.com/naturechemistry\\n\\n\\tComplete protein–protein association kinetics in atomic detail revealed by molecular dynamics simulations and Markov modelling\\n\\tResults and discussion\\n\\tAll-atom simulation of reversible protein–protein association\\n\\tEquilibrium kinetics model and validation\\n\\tOverall binding mechanism\\n\\tNature of the transition state of binding\\n\\tConformational change in the complex\\n\\tAssociation kinetics of wild-type and a faster-associating mutant\\n\\n\\tConclusion\\n\\tMethods\\n\\tMD simulation setup\\n\\tWild-type GPUgrid simulations\\n\\tWild-type adaptive MD simulations\\n\\tE73A mutant simulations\\n\\tFeature selection and discretization\\n\\tHidden Markov model\\n\\tProtein association model\\n\\tCode availability\\n\\n\\tFigure 1  Simulation of protein–protein association and dissociation.\\n\\tFigure 2  Kinetic model of barnase–barstar association.\\n\\tFigure 3  Late steps in binding from late intermediates to the loosely bound state.\\n\\tFigure 4  Native substate conformational dynamics.\\n\\tFigure 5  Comparison of kinetics between wild type and E73A mutant.\\n\\tReferences\\n\\tAcknowledgements\\n\\tAuthor contributions\\n\\tAdditional information\\n\\tCompeting financial interests\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n<<\\n  /ASCII85EncodePages false\\n  /AllowTransparency false\\n  /AutoPositionEPSFiles true\\n  /AutoRotatePages /None\\n  /Binding /Left\\n  /CalGrayProfile ()\\n  /CalRGBProfile (sRGB IEC61966-2.1)\\n  /CalCMYKProfile (U.S. Web Coated \\\\050SWOP\\\\051 v2)\\n  /sRGBProfile (sRGB IEC61966-2.1)\\n  /CannotEmbedFontPolicy /Error\\n  /CompatibilityLevel 1.3\\n  /CompressObjects /Off\\n  /CompressPages true\\n  /ConvertImagesToIndexed true\\n  /PassThroughJPEGImages true\\n  /CreateJobTicket false\\n  /DefaultRenderingIntent /Default\\n  /DetectBlends true\\n  /DetectCurves 0.1000\\n  /ColorConversionStrategy /LeaveColorUnchanged\\n  /DoThumbnails false\\n  /EmbedAllFonts true\\n  /EmbedOpenType false\\n  /ParseICCProfilesInComments true\\n  /EmbedJobOptions true\\n  /DSCReportingLevel 0\\n  /EmitDSCWarnings false\\n  /EndPage -1\\n  /ImageMemory 1048576\\n  /LockDistillerParams true\\n  /MaxSubsetPct 100\\n  /Optimize true\\n  /OPM 1\\n  /ParseDSCComments true\\n  /ParseDSCCommentsForDocInfo true\\n  /PreserveCopyPage false\\n  /PreserveDICMYKValues true\\n  /PreserveEPSInfo true\\n  /PreserveFlatness true\\n  /PreserveHalftoneInfo false\\n  /PreserveOPIComments false\\n  /PreserveOverprintSettings true\\n  /StartPage 1\\n  /SubsetFonts true\\n  /TransferFunctionInfo /Apply\\n  /UCRandBGInfo /Remove\\n  /UsePrologue true\\n  /ColorSettingsFile (None)\\n  /AlwaysEmbed [ true\\n  ]\\n  /NeverEmbed [ true\\n  ]\\n  /AntiAliasColorImages false\\n  /CropColorImages true\\n  /ColorImageMinResolution 150\\n  /ColorImageMinResolutionPolicy /OK\\n  /DownsampleColorImages true\\n  /ColorImageDownsampleType /Bicubic\\n  /ColorImageResolution 450\\n  /ColorImageDepth -1\\n  /ColorImageMinDownsampleDepth 1\\n  /ColorImageDownsampleThreshold 1.00000\\n  /EncodeColorImages true\\n  /ColorImageFilter /DCTEncode\\n  /AutoFilterColorImages true\\n  /ColorImageAutoFilterStrategy /JPEG\\n  /ColorACSImageDict <<\\n    /QFactor 0.15\\n    /HSamples [1 1 1 1] /VSamples [1 1 1 1]\\n  >>\\n  /ColorImageDict <<\\n    /QFactor 0.15\\n    /HSamples [1 1 1 1] /VSamples [1 1 1 1]\\n  >>\\n  /JPEG2000ColorACSImageDict <<\\n    /TileWidth 256\\n    /TileHeight 256\\n    /Quality 30\\n  >>\\n  /JPEG2000ColorImageDict <<\\n    /TileWidth 256\\n    /TileHeight 256\\n    /Quality 30\\n  >>\\n  /AntiAliasGrayImages false\\n  /CropGrayImages true\\n  /GrayImageMinResolution 150\\n  /GrayImageMinResolutionPolicy /OK\\n  /DownsampleGrayImages true\\n  /GrayImageDownsampleType /Bicubic\\n  /GrayImageResolution 450\\n  /GrayImageDepth -1\\n  /GrayImageMinDownsampleDepth 2\\n  /GrayImageDownsampleThreshold 1.00000\\n  /EncodeGrayImages true\\n  /GrayImageFilter /DCTEncode\\n  /AutoFilterGrayImages true\\n  /GrayImageAutoFilterStrategy /JPEG\\n  /GrayACSImageDict <<\\n    /QFactor 0.15\\n    /HSamples [1 1 1 1] /VSamples [1 1 1 1]\\n  >>\\n  /GrayImageDict <<\\n    /QFactor 0.15\\n    /HSamples [1 1 1 1] /VSamples [1 1 1 1]\\n  >>\\n  /JPEG2000GrayACSImageDict <<\\n    /TileWidth 256\\n    /TileHeight 256\\n    /Quality 30\\n  >>\\n  /JPEG2000GrayImageDict <<\\n    /TileWidth 256\\n    /TileHeight 256\\n    /Quality 30\\n  >>\\n  /AntiAliasMonoImages false\\n  /CropMonoImages true\\n  /MonoImageMinResolution 1200\\n  /MonoImageMinResolutionPolicy /OK\\n  /DownsampleMonoImages true\\n  /MonoImageDownsampleType /Bicubic\\n  /MonoImageResolution 2400\\n  /MonoImageDepth -1\\n  /MonoImageDownsampleThreshold 1.00000\\n  /EncodeMonoImages true\\n  /MonoImageFilter /CCITTFaxEncode\\n  /MonoImageDict <<\\n    /K -1\\n  >>\\n  /AllowPSXObjects false\\n  /CheckCompliance [\\n    /None\\n  ]\\n  /PDFX1aCheck true\\n  /PDFX3Check false\\n  /PDFXCompliantPDFOnly false\\n  /PDFXNoTrimBoxError false\\n  /PDFXTrimBoxToMediaBoxOffset [\\n    35.29000\\n    35.29000\\n    36.28000\\n    36.28000\\n  ]\\n  /PDFXSetBleedBoxToMediaBox false\\n  /PDFXBleedBoxToTrimBoxOffset [\\n    8.50000\\n    8.50000\\n    8.50000\\n    8.50000\\n  ]\\n  /PDFXOutputIntentProfile (OFCOM_PO_P1_F60)\\n  /PDFXOutputConditionIdentifier ()\\n  /PDFXOutputCondition (OFCOM_PO_P1_F60)\\n  /PDFXRegistryName ()\\n  /PDFXTrapped /False\\n\\n  /CreateJDFFile false\\n  /SyntheticBoldness 1.000000\\n  /Description <<\\n    /JPN <FEFF3053306e8a2d5b9a306f300130d330b830cd30b9658766f8306e8868793a304a3088307353705237306b90693057305f00200050004400460020658766f830924f5c62103059308b3068304d306b4f7f75283057307e305930023053306e8a2d5b9a30674f5c62103057305f00200050004400460020658766f8306f0020004100630072006f0062006100740020304a30883073002000520065006100640065007200200035002e003000204ee5964d30678868793a3067304d307e30593002>\\n    /DEU <FEFF00560065007200770065006e00640065006e0020005300690065002000640069006500730065002000450069006e007300740065006c006c0075006e00670065006e0020007a0075006d002000450072007300740065006c006c0065006e00200076006f006e0020005000440046002d0044006f006b0075006d0065006e00740065006e002c00200075006d002000650069006e00650020007a0075007600650072006c00e40073007300690067006500200041006e007a006500690067006500200075006e00640020004100750073006700610062006500200076006f006e00200047006500730063006800e40066007400730064006f006b0075006d0065006e00740065006e0020007a0075002000650072007a00690065006c0065006e002e00200044006900650020005000440046002d0044006f006b0075006d0065006e007400650020006b00f6006e006e0065006e0020006d006900740020004100630072006f0062006100740020006f0064006500720020006d00690074002000640065006d002000520065006100640065007200200035002e003000200075006e00640020006800f600680065007200200067006500f600660066006e00650074002000770065007200640065006e002e>\\n    /FRA <FEFF004f007000740069006f006e00730020007000650072006d0065007400740061006e007400200064006500200063007200e900650072002000640065007300200064006f00630075006d0065006e007400730020005000440046002000700072006f00660065007300730069006f006e006e0065006c007300200066006900610062006c0065007300200070006f007500720020006c0061002000760069007300750061006c00690073006100740069006f006e0020006500740020006c00270069006d007000720065007300730069006f006e002e00200049006c002000650073007400200070006f0073007300690062006c0065002000640027006f00750076007200690072002000630065007300200064006f00630075006d0065006e007400730020005000440046002000640061006e00730020004100630072006f0062006100740020006500740020005200650061006400650072002c002000760065007200730069006f006e002000200035002e00300020006f007500200075006c007400e9007200690065007500720065002e>\\n    /PTB <FEFF005500740069006c0069007a006500200065007300740061007300200063006f006e00660069006700750072006100e700f5006500730020007000610072006100200063007200690061007200200064006f00630075006d0065006e0074006f0073002000500044004600200063006f006d00200075006d0061002000760069007300750061006c0069007a006100e700e3006f0020006500200069006d0070007200650073007300e3006f00200061006400650071007500610064006100730020007000610072006100200064006f00630075006d0065006e0074006f007300200063006f006d0065007200630069006100690073002e0020004f007300200064006f00630075006d0065006e0074006f0073002000500044004600200070006f00640065006d0020007300650072002000610062006500720074006f007300200063006f006d0020006f0020004100630072006f006200610074002c002000520065006100640065007200200035002e00300020006500200070006f00730074006500720069006f0072002e>\\n    /DAN <FEFF004200720075006700200064006900730073006500200069006e0064007300740069006c006c0069006e006700650072002000740069006c0020006100740020006f0070007200650074007400650020005000440046002d0064006f006b0075006d0065006e007400650072002c0020006400650072002000650072002000650067006e006500640065002000740069006c0020007000e5006c006900640065006c006900670020007600690073006e0069006e00670020006f00670020007500640073006b007200690076006e0069006e006700200061006600200066006f0072007200650074006e0069006e006700730064006f006b0075006d0065006e007400650072002e0020005000440046002d0064006f006b0075006d0065006e007400650072006e00650020006b0061006e002000e50062006e006500730020006d006500640020004100630072006f0062006100740020006f0067002000520065006100640065007200200035002e00300020006f00670020006e0079006500720065002e>\\n    /NLD <FEFF004700650062007200750069006b002000640065007a006500200069006e007300740065006c006c0069006e00670065006e0020006f006d0020005000440046002d0064006f00630075006d0065006e00740065006e0020007400650020006d0061006b0065006e00200064006900650020006700650073006300680069006b00740020007a0069006a006e0020006f006d0020007a0061006b0065006c0069006a006b006500200064006f00630075006d0065006e00740065006e00200062006500740072006f0075007700620061006100720020007700650065007200200074006500200067006500760065006e00200065006e0020006100660020007400650020006400720075006b006b0065006e002e0020004400650020005000440046002d0064006f00630075006d0065006e00740065006e0020006b0075006e006e0065006e00200077006f007200640065006e002000670065006f00700065006e00640020006d006500740020004100630072006f00620061007400200065006e002000520065006100640065007200200035002e003000200065006e00200068006f006700650072002e>\\n    /ESP <FEFF0055007300650020006500730074006100730020006f007000630069006f006e006500730020007000610072006100200063007200650061007200200064006f00630075006d0065006e0074006f0073002000500044004600200071007500650020007000650072006d006900740061006e002000760069007300750061006c0069007a006100720020006500200069006d007000720069006d0069007200200063006f007200720065006300740061006d0065006e0074006500200064006f00630075006d0065006e0074006f007300200065006d00700072006500730061007200690061006c00650073002e0020004c006f007300200064006f00630075006d0065006e0074006f00730020005000440046002000730065002000700075006500640065006e00200061006200720069007200200063006f006e0020004100630072006f00620061007400200079002000520065006100640065007200200035002e003000200079002000760065007200730069006f006e0065007300200070006f00730074006500720069006f007200650073002e>\\n    /SUO <FEFF004e00e4006900640065006e002000610073006500740075007300740065006e0020006100760075006c006c006100200076006f006900740020006c0075006f006400610020006a0061002000740075006c006f00730074006100610020005000440046002d0061007300690061006b00690072006a006f006a0061002c0020006a006f006900640065006e0020006500730069006b0061007400730065006c00750020006e00e400790074007400e400e40020006c0075006f00740065007400740061007600610073007400690020006c006f00700070007500740075006c006f006b00730065006e002e0020005000440046002d0061007300690061006b00690072006a0061007400200076006f0069006400610061006e0020006100760061007400610020004100630072006f006200610074002d0020006a0061002000520065006100640065007200200035002e00300020002d006f0068006a0065006c006d0061006c006c0061002000740061006900200075007500640065006d006d0061006c006c0061002000760065007200730069006f006c006c0061002e>\\n    /ITA <FEFF00550073006100720065002000710075006500730074006500200069006d0070006f007300740061007a0069006f006e00690020007000650072002000630072006500610072006500200064006f00630075006d0065006e007400690020005000440046002000610064006100740074006900200070006500720020006c00610020007300740061006d00700061002000650020006c0061002000760069007300750061006c0069007a007a0061007a0069006f006e006500200064006900200064006f00630075006d0065006e0074006900200061007a00690065006e00640061006c0069002e0020004900200064006f00630075006d0065006e00740069002000500044004600200070006f00730073006f006e006f0020006500730073006500720065002000610070006500720074006900200063006f006e0020004100630072006f00620061007400200065002000520065006100640065007200200035002e003000200065002000760065007200730069006f006e006900200073007500630063006500730073006900760065002e>\\n    /NOR <FEFF004200720075006b00200064006900730073006500200069006e006e007300740069006c006c0069006e00670065006e0065002000740069006c002000e50020006f00700070007200650074007400650020005000440046002d0064006f006b0075006d0065006e00740065007200200073006f006d002000700061007300730065007200200066006f00720020007000e5006c006900740065006c006900670020007600690073006e0069006e00670020006f00670020007500740073006b007200690066007400200061007600200066006f0072007200650074006e0069006e006700730064006f006b0075006d0065006e007400650072002e0020005000440046002d0064006f006b0075006d0065006e00740065006e00650020006b0061006e002000e50070006e006500730020006d006500640020004100630072006f0062006100740020006f0067002000520065006100640065007200200035002e00300020006f0067002000730065006e006500720065002e>\\n    /SVE <FEFF0041006e007600e4006e00640020006400650020006800e4007200200069006e0073007400e4006c006c006e0069006e006700610072006e00610020006e00e40072002000640075002000760069006c006c00200073006b0061007000610020005000440046002d0064006f006b0075006d0065006e007400200073006f006d00200070006100730073006100720020006600f600720020007000e5006c00690074006c006900670020007600690073006e0069006e00670020006f006300680020007500740073006b0072006900660074002000610076002000610066006600e4007200730064006f006b0075006d0065006e0074002e0020005000440046002d0064006f006b0075006d0065006e00740065006e0020006b0061006e002000f600700070006e006100730020006d006500640020004100630072006f0062006100740020006f00630068002000520065006100640065007200200035002e003000200065006c006c00650072002000730065006e006100720065002e>\\n    /ENU (NPG PRINT PDF Job Options. 4th September 2006. PDF 1.3 Compatibility. Adds Trim and Bleed boxes top Nature pages where none exist.)\\n  >>\\n>> setdistillerparams\\n<<\\n  /HWResolution [2400 2400]\\n  /PageSize [665.858 854.929]\\n>> setpagedevice\\n\\n',\n",
       " 'metadata': {'Author': ['Stefan Doerr',\n",
       "   'Gianni De Fabritiis',\n",
       "   'Frank Noé',\n",
       "   'Nuria Plattner'],\n",
       "  'Content-Encoding': 'ISO-8859-1',\n",
       "  'Content-Length': '-1',\n",
       "  'Content-Type': ['application/pdf', 'text/plain; charset=ISO-8859-1'],\n",
       "  'Creation-Date': '2017-05-24T06:47:57Z',\n",
       "  'Last-Modified': '2017-05-24T08:13:28Z',\n",
       "  'Last-Save-Date': '2017-05-24T08:13:28Z',\n",
       "  'X-Parsed-By': ['org.apache.tika.parser.DefaultParser',\n",
       "   'org.apache.tika.parser.pdf.PDFParser',\n",
       "   ['org.apache.tika.parser.DefaultParser',\n",
       "    'org.apache.tika.parser.txt.TXTParser']],\n",
       "  'X-TIKA:embedded_resource_path': '/folder.joboptions',\n",
       "  'X-TIKA:origResourceName': 'folder.joboptions',\n",
       "  'X-TIKA:parse_time_millis': ['338', '3'],\n",
       "  'access_permission:assemble_document': 'true',\n",
       "  'access_permission:can_modify': 'true',\n",
       "  'access_permission:can_print': 'true',\n",
       "  'access_permission:can_print_degraded': 'true',\n",
       "  'access_permission:extract_content': 'true',\n",
       "  'access_permission:extract_for_accessibility': 'true',\n",
       "  'access_permission:fill_in_form': 'true',\n",
       "  'access_permission:modify_annotations': 'true',\n",
       "  'cp:subject': 'Nature Chemistry (2017). doi:10.1038/nchem.2785',\n",
       "  'created': '2017-05-24T06:47:57Z',\n",
       "  'creator': ['Stefan Doerr',\n",
       "   'Gianni De Fabritiis',\n",
       "   'Frank Noé',\n",
       "   'Nuria Plattner'],\n",
       "  'date': '2017-05-24T08:13:28Z',\n",
       "  'dc:creator': ['Stefan Doerr',\n",
       "   'Gianni De Fabritiis',\n",
       "   'Frank Noé',\n",
       "   'Nuria Plattner'],\n",
       "  'dc:description': 'Nature Chemistry (2017). doi:10.1038/nchem.2785',\n",
       "  'dc:format': 'application/pdf; version=1.3',\n",
       "  'dc:title': 'Complete protein-protein association kinetics in atomic detail revealed by molecular dynamics simulations and Markov modelling',\n",
       "  'dcterms:created': '2017-05-24T06:47:57Z',\n",
       "  'dcterms:modified': '2017-05-24T08:13:28Z',\n",
       "  'description': 'Nature Chemistry (2017). doi:10.1038/nchem.2785',\n",
       "  'embeddedResourceType': 'ATTACHMENT',\n",
       "  'meta:author': ['Stefan Doerr',\n",
       "   'Gianni De Fabritiis',\n",
       "   'Frank Noé',\n",
       "   'Nuria Plattner'],\n",
       "  'meta:creation-date': '2017-05-24T06:47:57Z',\n",
       "  'meta:save-date': '2017-05-24T08:13:28Z',\n",
       "  'modified': '2017-05-24T08:13:28Z',\n",
       "  'pdf:PDFVersion': '1.3',\n",
       "  'pdf:docinfo:created': '2017-05-24T06:47:57Z',\n",
       "  'pdf:docinfo:creator': 'Nuria Plattner',\n",
       "  'pdf:docinfo:creator_tool': 'Arbortext Advanced Print Publisher 10.0.1465/W Unicode',\n",
       "  'pdf:docinfo:modified': '2017-05-24T08:13:28Z',\n",
       "  'pdf:docinfo:producer': 'Acrobat Distiller 8.1.0 (Windows)',\n",
       "  'pdf:docinfo:subject': 'Nature Chemistry (2017). doi:10.1038/nchem.2785',\n",
       "  'pdf:docinfo:title': 'Complete protein-protein association kinetics in atomic detail revealed by molecular dynamics simulations and Markov modelling',\n",
       "  'pdf:encrypted': 'false',\n",
       "  'producer': 'Acrobat Distiller 8.1.0 (Windows)',\n",
       "  'resourceName': ['nchemex.pdf', 'folder.joboptions'],\n",
       "  'subject': 'Nature Chemistry (2017). doi:10.1038/nchem.2785',\n",
       "  'title': 'Complete protein-protein association kinetics in atomic detail revealed by molecular dynamics simulations and Markov modelling',\n",
       "  'xmp:CreatorTool': 'Arbortext Advanced Print Publisher 10.0.1465/W Unicode',\n",
       "  'xmpMM:DocumentID': 'uuid:03d04121-f0e7-347f-b6a1-e56cb26a998e',\n",
       "  'xmpTPg:NPages': '7'},\n",
       " 'status': 200}"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "parsedPDF"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "dict_keys(['content', 'status', 'metadata'])"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "parsedPDF.keys()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\nComplete protein-protein association kinetics in atomic detail revealed by molecular dynamics simulations and Markov modelling\\n\\n\\nComplete protein–protein association kinetics in\\natomic detail revealed by molecular dynamics\\nsimulations and Markov modelling\\nNuria Plattner1*, Stefan Doerr2, Gianni De Fabritiis2,3 and Frank Noé1*\\n\\nProtein–protein association is fundamental to many life processes. However, a microscopic model describing the structures\\nand kinetics during association and dissociation is lacking on account of the long lifetimes of associated states, which have\\nprevented efficient sampling by direct molecular dynamics (MD) simulations. Here we demonstrate protein–protein\\nassociation and dissociation in atomistic resolution for the ribonuclease barnase and its inhibitor barstar by combining\\nadaptive high-throughput MD simulations and hidden Markov modelling. The model reveals experimentally consistent\\nintermediate structures, energetics and kinetics on timescales from microseconds to hours. A variety of flexibly attached\\nintermediates and misbound states funnel down to a transition state and a native basin consisting of the loosely bound\\nnear-native state and the tightly bound crystallographic state. These results offer a deeper level of insight into\\nmacromolecular recognition and our approach opens the door for understanding and manipulating a wide range of\\nmacromolecular association processes.\\n\\nLife processes are critically dependent on the formation of bio-molecular complexes, and in particular, protein–protein associ-ation, which mediates the assembly of cellular structures, signal\\ntransduction and inhibition, immune response and protein recruit-\\nment to membrane scaffolds. Interactions of proteins with other\\nmacromolecules also play key roles in applications such as inhibition\\nof protein activity by drugs1, gene editing2 and the assembly of\\nnew materials3.\\n\\nA deep understanding of such processes and the ability to fully\\nplan and control their design would arise from a full map of the\\nconformation space of the interacting protein pairs. Such a micro-\\nscopic model would involve representative samples of all structures\\n(bound, dissociated, intermediates, etc.) in atomic resolution, their\\nequilibrium populations and the transition rates between them.\\n\\nHowever, such a complete microscopic picture has not yet been\\nachieved for protein–protein interaction. Experiments can probe\\neither individual aspects of the microscopic picture or averaged\\nensembles, such as the most stable structures, association kinetics\\nand binding affinities. Mutational analyses can identify amino\\nacids stabilizing associated states or transition states4. The structure\\nof the transient encounter complex, that is, the set of structures\\nfound immediately before the transition state of protein association,\\nhas been probed using NMR5 and coarse-grained simulations6–9.\\nBinding free energies can be computed from atomistic simulations\\nwhen the bound pose is precisely known10, and the binding free\\nenergy landscape can be reconstructed using metadynamics given\\na set of pre-defined collective variables11. Inference of kinetics\\nthen requires a kinetic model12 or additional unbiased simulations13.\\n\\nA candidate to reveal a complete microscopic model of protein–\\nprotein interaction is unbiased atomic-detail molecular dynamics\\n(MD) simulation. Microscopic models of biomolecular processes\\nhave recently been achieved due to combined advances in high-\\nperformance computing and Markov modelling14–19. Previous\\nbinding studies, however, have been limited to the association of\\n\\nsmall ligands to proteins and the co-folding of shorter peptide\\nchains16,19,20, or small parts of the association landscape such as\\nthe downhill binding step from a pre-bound pose21. The simulation\\nof protein–protein association is challenged by the fact that\\ndissociation times exceed accessible MD simulation lengths by far,\\nsuch that long MD simulations will get stuck after a single\\nbinding event, possibly even in non-native traps.\\n\\nHere we demonstrate how reversible protein–protein association\\ncan be studied in atomistic detail by combining an extensive and\\nadaptively generated ensemble of unbiased molecular dynamics\\nsimulations with hidden Markov modelling. We reproduce\\nmeasured structures, energetics and kinetics quantitatively and\\nobtain a full microscopic model of the protein complex structures,\\nthermodynamics and kinetics.\\n\\nResults and discussion\\nAll-atom simulation of reversible protein–protein association. As\\na prototypical example, we study the diffusion-limited association of\\nthe experimentally well-characterized bacterial ribonuclease barnase\\nwith its inhibitor barstar22,23. All-atom models of barnase wild type\\nand barstar pseudo wild type24 were immersed in a solvent box\\n(approximately 90,000 atoms). Several thousand unbiased molecular\\ndynamics (MD) simulations of 2 milliseconds aggregate length were\\nconducted in an automated high-throughput manner. These data\\ninclude 1.7 milliseconds of independent trajectories starting from\\nthe dissociated state and 0.3 milliseconds in adaptive MD simulations.\\n\\nThe adaptive MD launched periods of multiple parallel MD\\ntrajectories, and the starting points of each period were selected so\\nas to favour the exploration of new states (Methods, Supplementary\\nMethods 1). This strategy can yield large speed-ups compared to\\nnon-adaptive MD runs because slow processes with large energy\\nbarriers are broken down into steps with individually smaller\\nbarriers that can be sampled efficiently25–27 (Fig. 1a,b). When starting\\nfrom the dissociated state, both adaptive and non-adaptive\\n\\n1Department of Mathematics and Computer Science, Freie Universität Berlin, Arnimallee 6, 14195 Berlin, Germany. 2Computational Biophysics Laboratory\\n(GRIB-IMIM), Universitat Pompeu Fabra, Barcelona Biomedical Research Park (PRBB), C/ Doctor Aiguader 88, 08003 Barcelona, Spain. 3Institució Catalana\\nde Recerca i Estudis Avançats (ICREA), Passeig Lluis Companys 23, 08010 Barcelona, Spain. *e-mail: nuria.plattner@fu-berlin.de; frank.noe@fu-berlin.de\\n\\nARTICLES\\nPUBLISHED ONLINE: 5 JUNE 2017 | DOI: 10.1038/NCHEM.2785\\n\\nNATURE CHEMISTRY | ADVANCE ONLINE PUBLICATION | www.nature.com/naturechemistry 1\\n\\n© 2017 Macmillan Publishers Limited, part of Springer Nature. All rights reserved.\\n\\nmailto:nuria.plattner@fu-berlin.de\\nmailto:frank.noe@fu-berlin.de\\nhttp://dx.doi.org/10.1038/nchem.2785\\nhttp://www.nature.com/naturechemistry\\n\\n\\nsimulations reproducibly found the native bound complex. Adaptive\\nsimulations required a total simulation effort of a few microseconds,\\nroughly an order of magnitude speed-up compared to an equilibrium\\nbinding time of a tens of microseconds (Supplementary Fig. 1).\\nSurprisingly, the adaptive MD runs could also reproducibly reach\\nthe dissociated state when starting from the crystal structure using\\na total simulation effort of a few hundred microseconds, although\\nthe expected dissociation time is hours (Supplementary Fig. 1).\\n\\nA first analysis shows that the total dataset contains several\\nthousand unspecific association and dissociation events, about 90\\nbinding events from non-native associated states to native and 10\\nunbinding events (Fig. 1c).\\n\\nEquilibrium kinetics model and validation. Although the\\nindividual simulations are not started from an equilibrium\\ndistribution and are shorter than binding and dissociation\\ntimescales, Markov modelling approaches can combine such a\\ntrajectory ensemble towards unbiased equilibrium models of the\\nlong-timescale kinetics and thermodynamics (Fig. 1a)28,29. Here, a\\nhidden Markov model (HMM) was built to estimate a network of\\n\\nstates, associated protein structures, and transition rates30. The\\nHMM is coupled to a diffusive association model inspired by a\\nprevious study31 in order to be able to predict concentration-\\ndependent association rates and binding free energies (Methods,\\nSupplementary Methods 3–7).\\n\\nAs the data contains only a few unbinding events, the following\\ncomputed quantities have large uncertainties but are statistically\\nconsistent with experiment, which will be exploited to compute\\nconsensus models of experimental and simulation data (intervals\\nare 95% confidence): (i) binding free energy lies between 12 and\\n19 kcal mol–1 (experimental 16.8 kcal mol–1; see Supplementary\\nMethods 5), (ii) the dissociation rate lies in the range of 3 × 10–6\\n\\nto 1 × 10–1 s–1 (measurements lie in the range of 4.8 × 10–5 to\\n5.0 × 10–4 s–1; see Supplementary Methods 6).\\n\\nThe model successfully predicts the following quantities: (iii)\\ncrystal structure 1BRS is predicted to be the most stable state, and\\nthe simulated structures assigned to it have an average RMSD of\\n0.21 nm to 1BRS, (iv) association rate is predicted to be 2.63 × 108\\n\\nto 2.65 × 108 s–1 M–1 (1 × 108 to 3.4 × 108 s–1 M–1 measured pre-\\nviously23,32), (v) using perturbation theory33, binding free energy\\nchanges of 35 mutant sequences reported in23,32 were predicted\\nwithin statistical error (Fig. 2a and Supplementary Methods 9),\\n(vi) although this approach cannot compute absolute values of\\nassociation rates of mutants, the computed change in transition\\nstate energy is an excellent predictor for the change in the experimental\\non-rate (Fig. 2b).\\n\\nOverall binding mechanism. Having validated our model with a\\nlarge set of experimental data we proceeded to examine properties\\nthat are more difficult to probe directly by experiments. Figure 2c,\\nd displays the overall mechanisms and structures found along the\\nbinding pathways. The proteins initially associate to non-native\\n(large RMSD) states. At that point, they can dissociate and re-\\nassociate or stay in contact and rearrange towards the native\\ncomplex (Fig. 2c–f ).\\n\\nAt the high protein concentration in the simulation, initial\\nassociation occurs within tens of nanoseconds and with no detect-\\nable barrier. The initial association of barstar can occur at all\\nlocations on the barnase surface. During the first microsecond,\\nthe complex preferentially populates early intermediates with\\nbarstar located opposite of the barnase binding site (orange states\\nin Fig. 2d). However, association can also occur directly into the\\nlate intermediate and mis-bound states (blue/red states in Fig. 2d),\\nwhere barstar is located in or near the barnase binding groove,\\nbut not in the native orientation. The preference for initially associ-\\nating to locations away from the binding pocket can be explained by\\nthe fact that these states cover a larger fraction of the surface. The\\ninitial association preference depends on the salt concentration;\\nwith enhanced steering to the binding site being expected when\\ncharge screening is reduced7,22. Knocking out intermediate states\\nby removing them from the HMM showed that individual early\\nintermediates have no significant effect on the association rate,\\nand even removing all early intermediates only reduces the\\nassociation rate by 20% (Supplementary Table 2).\\n\\nOn timescales of 1–10 µs, the population shifts toward the late\\nintermediates and mis-bound states (blue/red states in Figs 2d\\nand 5c), which have higher equilibrium populations than the\\nearly intermediates. Of special interest is the mis-bound state\\n(red), that has barstar bound ‘upside down’. This state does not\\nhave a well-defined binding interface, but a few non-native inter-\\nactions such as barnase Arg 83 or Arg 87 with barstar Glu 118\\noccur with significant probability (Supplementary Figs 3 and 4).\\nIntermediate states are kinetically well connected and have lifetimes\\non the order of one microsecond each (Supplementary Fig. 2).\\nRemoving the late intermediates would reduce the binding rate by\\na factor of 2.5 (Supplementary Table 2)\\n\\n1 2\\n\\n3\\n4\\n\\n5 Sampling strategy \\n\\nMarkov model \\n\\nTrajectories\\n\\nTasks\\n\\nData \\ngeneration \\n\\nModel building \\n\\nCompute clusterClientba\\n\\nc\\n\\nAssociatedDissociated Bound\\n\\nTime (μs) Time (μs)\\n\\n2,867\\n\\nD\\nis\\n\\nta\\nnc\\n\\ne \\n(n\\n\\nm\\n)\\n\\nR\\nM\\n\\nS\\nD\\n\\n (\\nnm\\n\\n)\\n\\n86\\n6\\n\\n0\\n0 0.5 0\\n\\n0\\n\\n1.5\\n\\n5\\n\\nTime (μs) Time (μs)\\n\\nR\\nM\\n\\nS\\nD\\n\\n (\\nnm\\n\\n)\\n\\nD\\nis\\n\\nta\\nnc\\n\\ne \\n(n\\n\\nm\\n) 101,465\\n\\n6\\n\\n0\\n0 3.5 0 7\\n\\n0\\n\\n1.5\\n\\nFigure 1 | Simulation of protein–protein association and dissociation.\\na, Scheme of a large-barrier process such as protein–protein dissociation that\\ncan be subdivided into multiple steps. Such processes can be sampled by\\ncombining adaptively seeded simulation trajectories between individual steps\\nvia Markov modelling. b, Schematic outline of adaptive sampling process.\\nc, Total number of full association–dissociation and binding–unbinding\\ntransitions evaluated by raw trajectory statistics of all barnase–barstar\\nsimulation trajectories used subsequently. An association–dissociation\\ntransition was counted when a trajectory moved from a state where the\\nproteins had a minimum distance between heavy atom pairs of >2 nm to\\n<0.3 nm, and vice versa. A binding–unbinding event was counted when a\\ntrajectory moved from a state with minimum heavy-atom RMSD from the\\ncrystallographic structure >1 nm to RMSD <0.3 nm, and vice versa. For each\\ntrajectory only one count was made in order to avoid duplicate counting\\nof recrossing events. Three sample trajectory snippets are shown for each\\ntype of transition.\\n\\nARTICLES NATURE CHEMISTRY DOI: 10.1038/NCHEM.2785\\n\\nNATURE CHEMISTRY | ADVANCE ONLINE PUBLICATION | www.nature.com/naturechemistry2\\n\\n© 2017 Macmillan Publishers Limited, part of Springer Nature. All rights reserved.\\n\\nhttp://dx.doi.org/10.1038/nchem.2785\\nhttp://www.nature.com/naturechemistry\\n\\n\\nMisbound\\n\\nEarly intermediates\\n\\nLate\\nintermediate 2\\n\\nLate\\nintermediate 1\\n\\nPre-bound\\n\\nLoosely\\nbound\\n\\nTightly\\nbound\\n\\na\\n\\nd\\n\\nb c e\\n\\nf\\n\\ng\\n\\nDissociated\\n\\nAssociation\\n\\nBinding\\n\\nUnbinding Rebinding\\n\\n12\\nΔΔ\\n\\nG\\n e\\n\\nxp\\ner\\n\\nim\\nen\\n\\nt (\\nkc\\n\\nal\\n m\\n\\nol\\n–1\\n\\n)\\n\\n8\\n\\n4\\n\\n–4 0\\n\\nΔΔG model (kcal mol–1)\\n\\n4 8 12\\n\\n–4\\n\\n0\\n\\n3\\n\\n2\\n\\n2\\n\\n0\\n\\nΔΔ\\nG\\n\\non\\n e\\n\\nxp\\ner\\n\\nim\\nen\\n\\nt (\\nkc\\n\\nal\\n m\\n\\nol\\n–1\\n\\n)\\n\\n–5 0 5 10\\n\\nΔΔGTS model (kcal mol–1)\\n\\n–2\\n\\n–1 F\\nre\\n\\ne \\nen\\n\\ner\\ngy\\n\\n (\\nkT\\n\\n)\\n\\nMinimal distance (nm)\\n\\n3.0 24\\n\\n21\\n\\n18\\n\\n15\\n\\n12\\n\\n9\\n\\n6\\n\\n3\\n\\n0\\n\\n2.5\\n\\n2.0\\n\\nR\\nM\\n\\nS\\nD\\n\\n (\\nnm\\n\\n)\\n\\n0.3\\n0.2\\n0.1\\n\\n0.00 0.02 0.04 0.06 0.08 0.10\\n\\n5\\n\\nt t+50 t+100 t+150 t+200\\n\\n10 15 20 25 30\\n\\n0.3\\n0.2\\n0.1\\n\\n0.3\\n0.2\\n0.1\\n\\nR\\nM\\n\\nS\\nD\\n\\n (\\nnm\\n\\n)\\nR\\n\\nM\\nS\\n\\nD\\n (\\n\\nnm\\n)\\n\\nR\\nM\\n\\nS\\nD\\n\\n (\\nnm\\n\\n)\\n\\n1.5\\n\\n1.0\\n\\n0.5\\n\\n0.25 0.50 1.00 2.00 4.00\\n\\nTime (μs)\\n\\nFigure 2 | Kinetic model of barnase–barstar association. a,b, Comparison of experimental and predicted changes in binding free energies and association\\nrates for all 35 single and double alanine mutations from Table II of ref. 23 and Table I of ref. 32. Statistical confidence intervals of the model are indicated in\\nwhite (1-σ, 68% confidence) and grey (2-σ, 95% confidence). a, Changes in binding free energy (ΔΔG) are predicted within statistical error. b, Comparison\\nof the computed changes in barrier height of the binding transition with changes of apparent association transition state energy, computed as ΔΔGon,apparent\\n= –kBTln(kon\\n\\nmutant/kon\\nwt), where kBT= 0.598 kcal mol\\n\\n–1 and kon\\nmutant, kon\\n\\nwt are the experimental association rates of mutant and wild type, respectively. The model\\nprediction was computed as free energy difference between the pre-bound state as a proxy for the transition state, and all early intermediate, late\\nintermediate and mis-bound states. c, Free energy surface over protein–protein heavy atom minimal distance and minimal RMSD to the native complex.\\nShown is a contour plot of –kTlnμ(x) evaluated for each trajectory sample x, where kT is the thermal energy and μ(x) is the equilibrium weight of each\\nsample computed by the HMM (see Supplementary Methods 4). d, Kinetic network between metastable states with lifetimes on the order of microseconds.\\nOnly transition rate constants ≥0.2ms–1 are shown. Arrow thickness is proportional to rate constant, disc area is proportional to square root of state\\npopulation. Structures are centered on barnase (orange) with its binding groove on top and its N-terminal helix on the bottom. Barstar is shown in grey with\\nits binding domain in green. Grey clouds show overlay of approximately 20 barstar conformations, sampled randomly from the corresponding metastable\\nstates. e–g, Minimal RMSD to crystallographic complex shown for sample trajectories generated from the HMM (see Supplementary Methods 4). Initial\\nassociation (e), binding to the native complex (f), and geminate rebinding after being bound for a long time t (one hour on average; g).\\n\\nNATURE CHEMISTRY DOI: 10.1038/NCHEM.2785 ARTICLES\\n\\nNATURE CHEMISTRY | ADVANCE ONLINE PUBLICATION | www.nature.com/naturechemistry 3\\n\\n© 2017 Macmillan Publishers Limited, part of Springer Nature. All rights reserved.\\n\\nhttp://dx.doi.org/10.1038/nchem.2785\\nhttp://www.nature.com/naturechemistry\\n\\n\\nOn timescales of 10–100 µs, the complex passes through a\\npre-bound state (magenta in Figs 2d and 5c) and then continues\\nto the natively bound complex (grey states in Figs 2d and 5c),\\nwhich consists of a loosely bound and a tightly bound substate\\nthat exchange on the microsecond timescale.\\n\\nNature of the transition state of binding. The kinetic bottleneck\\nbetween the intermediate states and natively bound states is the\\nmandatory passage through the pre-bound state (magenta in Fig. 2d),\\nthat is hence an on-pathway intermediate. Both the transition into\\nthe pre-bound state and the binding from the pre-bound state\\ninvolve barriers (Supplementary Fig. 6). We refer to these barriers\\ncollectively as transition state of binding as they are rate-limiting at\\nsufficiently high protein concentrations (Supplementary Fig. 5).\\n\\nThe transition from the late intermediates through the pre-bound\\nstate into the loosely bound state is characterized by two microscopic\\npathways, in which barstar interacts with either of the two barnase\\nloops surrounding the binding groove: the ‘left’ loop including Ser\\n\\n38 and the ‘right’ RNA recognition loop of barnase including Arg\\n59 (Fig. 3) whose importance for steering of the two protein has\\nbeen shown before6,7. Due to the flexibility of both loop regions,\\nbarstar can “swing” around its attachment point, giving rise to a\\nquite flexible set of structures. These pathways are not separated by\\nsignificant barriers, i.e. the complex can change between “left” and\\n“right” attachment point multiple times before binding.\\n\\nThe minor left pathway (5% population, Fig. 3a→ c→ e) involves\\ndifferent possible protein contacts, including hydrophobic contacts\\nand does not appear to have well-defined structural features. In the\\nmajor right pathway (95% population, Fig. 3b→ d→ e), barnase Arg\\n59 interacts with barstar Glu 76, Asp 35, or both, in line with34,35.\\nArg 59 may additionally form a hydrophobic contact with barstar\\nTrp 38. While no individual pair of residues is strongly formed in the\\ntransition state22, Arg 59 is involved in 95% of all transition state con-\\nfigurations. This observation confirms the pivotal role of Arg 59 inmed-\\niating the binding transition34, and explains that Arg 59 mutants\\nreduced the binding rate by a factor of 10 (ref. 32). We expect that a\\nbarstar double mutant of Glu 76 and Asp 35 will have a similar effect.\\n\\nThe pre-bound state has a lifetime of about two microseconds\\n(Supplementary Fig. 2). In all binding events simulated, the\\ntransition from the pre-bound to the loosely bound state trajectories\\noccurs within a single saving interval of our simulations, that is less\\nthan 100 picoseconds. Thus the last step of docking into the barnase\\nbinding pocket and forming the binding interface with its many\\ncontacts is rare, but occurs quickly when it happens21,36. In line with\\nprevious predictions22, the majority of the barnase–barstar short-\\nrange interactions is formed after the transition state. In the limit of\\nlow electrostatic interactions, the basal binding rate may be governed\\nby the bimolecular rate to form a ready-to-dock structure similar to\\nour pre-bound state22, and it is likely that the barrier of desolvation\\nand docking is reduced or even removed under these conditions.\\n\\nConformational change in the complex. The bound states are in the\\nrange of 5–12 kcal mol–1 more stable than intermediate, mis-bound\\nand pre-bound states and thus contain almost all of the equilibrium\\npopulation. We find two conformational substates in the bound\\nstate: The loosely bound state with about 5% of the population\\nand an average heavy-atom RMSD to the crystal structure of\\n0.3 nm (Fig. 4a); and the tightly bound state with the remaining\\n95% of the population (Fig. 4b). The density of the tightly bound\\nstate contains the crystal structure (Fig. 4c, average RMSD\\n0.21 nm), which is thus correctly predicted as the most stable state.\\n\\nUpon binding, the loosely bound state is reached first. This state\\ntransitions into the tightly bound state within 5 µs on average, that\\nitself has an average lifetime of 130 µs before transitioning back to\\nthe loosely bound state. These microsecond transitions occur\\nmany times before the complex undergoes geminate rebinding\\n(Fig. 2g) or finally dissociates.\\n\\nStructurally, the native-state transition consists of a slight\\nrotation of barstar in the barnase binding pocket (compare\\nFig. 4a,b). While both substates have similar residue contact patterns\\n(Supplementary Fig. 3), the most apparent difference is that the\\nbarnase 35–41 loop is inserted between the barstar binding helix\\nand the large helix of barstar in the loosely bound state, while it is\\ninserted between the binding helix and binding loop in the tightly\\nbound state. The loosely bound state is stabilized by an interaction\\nbetween barnase Lys 27 (whose functional role is to stabilize the\\nphosphate group in the RNA substrate) and barstar Glu 80, while\\nthe tightly bound state is stabilized by a salt bridge between\\nbarnase Arg 83 and barstar Asp 39. Due to the latter salt bridge,\\nbarnase loop 77–85 is stabilized in the tightly bound state, while it\\nis very flexible in the loosely bound state. Overall, the loosely bound\\nstate is more flexible than the tightly bound state (compare Fig. 4a,b).\\n\\nThese native substate dynamics may be observable experimen-\\ntally, for example, by fluorescence correlation microscopy or by\\n\\na b\\n\\nc d\\n\\ne\\n\\nArg 59\\nAsp 35\\n\\nTrp 38 Glu 76\\n\\nGly 43\\nTrp 44\\n\\nAla 37\\nSer 38\\n\\n45° Late\\nintermediate 2\\n\\n \\n\\nPre-bound\\n\\nLoosely\\nbound\\n\\n45°\\n\\nFigure 3 | Late steps in binding from late intermediates to the loosely\\nbound state. a,b, Two sets of structures of late intermediate state 2, shown\\nin 45° counterclockwise/clockwise rotation compared to Fig. 2d. c,d, Two\\nsets of structures of the pre-bound state. e, Loosely bound state. Grey clouds\\nshow overlay of approximately 20 barstar conformations, sampled randomly\\nfrom the corresponding metastable states. Two binding pathways exist, in\\nwhich the binding domain of barstar interacts with barnase at its ‘left’ loop\\naround Ala 37 and Ser 38 (a,c), or at its ‘right’ loop (b,d), before docking into\\nthe bound state (e). In the major pathway (b→ d→ e, approximately 95% of\\nthe flux), barnase Arg 59 ‘anchors’ into barstar via Asp 35, Glu 76 (polar)\\nand Trp 38 (hydrophobic). Barstar can swing around barnase with the\\nanchor staying attached until final docking into the bound pose.\\n\\nARTICLES NATURE CHEMISTRY DOI: 10.1038/NCHEM.2785\\n\\nNATURE CHEMISTRY | ADVANCE ONLINE PUBLICATION | www.nature.com/naturechemistry4\\n\\n© 2017 Macmillan Publishers Limited, part of Springer Nature. All rights reserved.\\n\\nhttp://dx.doi.org/10.1038/nchem.2785\\nhttp://www.nature.com/naturechemistry\\n\\n\\nfast temperature jump experiments that have been employed to\\nprobe loose–tight binding transitions37. In particular, barstar Trp\\n44 changes solvent accessibility during the native substate transition,\\nrendering intrinsic tryptophan fluorescence a promising probe.\\n\\nAssociation kinetics of wild-type and a faster-associating mutant.\\nThe very large association rate of barnase–barstar can be further\\nincreased by specific mutations. To understand the microscopic\\nmechanisms of such perturbations, we conducted additional\\n0.3 milliseconds of MD simulations using the barnase mutant Glu\\n73 to Ala (E73A, see Methods), which associates a factor of 1.5–4\\nfaster than the wild type23.\\n\\nAs both the wild-type and the E73A HMM suffered from a\\nstatistical bottleneck at the unbinding transition, the uncertain\\ntransition rate from the loosely bound to pre-bound state was set\\nsuch that the models produce the experimentally known binding\\nfree energy. The resulting consensus models still make independent\\npredictions of the binding rate and mechanism, while reproducing\\nexperimental thermodynamics.\\n\\nTheE73Amodel predicts anassociation rate of 4 to 4.7 × 108 s–1M–1,\\nwhich correctly reflects an increase by a factor of 1.6–1.8 (ref. 23),\\nand can be explained by the increased electrostatic complementarity.\\nE73 does not participate in the binding interface, and indeed the\\nwild-type native complex structure is still predicted to be the most\\n\\n95%5%RMSD 0.3 nm\\nLifetime 5 ms\\n\\nTightly bound\\nRMSD 0.21 nm\\nLifetime 130 ms\\n\\n1BRS\\ncrystal structure \\n\\na b c\\n\\nArg 83\\nAsp 39Lys 27\\n\\nGlu 80\\n\\nTrp 44\\nTrp 44\\n\\nLoosely bound\\n\\nFigure 4 | Native substate conformational dynamics. Native substates seen from ‘left’, that is, in a 90° clockwise rotation around the image vertical with\\nrespect to Fig. 2d. a, Loosely bound state with lower population and higher structural flexibility. b, Tightly bound state with higher population and small\\nstructural flexibility. c, Crystal structure 1BRS shown in the structural density of the tightly bound simulation state, for comparison with B. Grey cloud is\\nan overlay of approximately 20 barstar conformations, sampled randomly from the corresponding metastable states.\\n\\nWild type E73A\\n\\na b\\n\\nC\\nom\\n\\nm\\nitt\\n\\nor\\n\\n1\\n\\n0\\n\\n10–1 100 101 102 103 104\\n\\n0.5\\n\\nc d\\n\\nLoosely\\nbound\\n\\nPrebound\\n\\nLate intermediates\\n\\nEarly intermediates\\n\\nDissociated\\n\\nMisbound\\n\\nTightly\\nbound\\n\\n1.0\\n0.8\\n0.6\\n0.4\\n0.2\\n0.0\\n\\nP\\nro\\n\\nba\\nbi\\n\\nlit\\ny\\n\\nTime (μs)\\n10–1 100 101 102 103 104\\n\\n1.0\\n0.8\\n0.6\\n0.4\\n0.2\\n0.0\\n\\nP\\nro\\n\\nba\\nbi\\n\\nlit\\ny\\n\\nTime (μs)\\n\\nFigure 5 | Comparison of kinetics between wild type and E73A mutant. a,b, Kinetic networks of the consensus models that are consistent with MD\\nsimulation data and the experimental binding free energy. Network plots as in Fig. 2d, but the vertical position of states indicates the binding probability\\n(committor) and disc area is proportional to fourth root of state population. The E73A mutant shows a deeper funnel that leads to an increased association\\nrate. c,d, Kinetic evolution of probabilities when starting from dissociated state, computed from the HMM (see Supplementary Methods 4). Despite a higher\\nassociation rate, the E73A mutant takes longer to bind to the native state, as the increased association rate is due to rapid trapping in a mis-bound state that\\nis committed to further native binding. Colour scheme as in Fig. 2d.\\n\\nNATURE CHEMISTRY DOI: 10.1038/NCHEM.2785 ARTICLES\\n\\nNATURE CHEMISTRY | ADVANCE ONLINE PUBLICATION | www.nature.com/naturechemistry 5\\n\\n© 2017 Macmillan Publishers Limited, part of Springer Nature. All rights reserved.\\n\\nhttp://dx.doi.org/10.1038/nchem.2785\\nhttp://www.nature.com/naturechemistry\\n\\n\\nstable state in the mutant (Fig. 5b). Figure 5a,b shows kinetic\\nnetworks of the wild type and E73A mutant, similar to Fig. 2d but\\nwith the y-positions indicating the probability of binding (committor)\\nof each state. Both systems are characterized by binding funnels,\\nthat are broad on top but narrow down rapidly as the complex pro-\\nceeds through late intermediates towards binding. The E73A funnel\\nis deeper in the sense that early intermediates have larger committor\\nprobabilities, that is, it is more difficult to escape the complex, once\\nit is associated. E73A has a reduced stability of the bound state23.\\nThe state that experiences the most profound increase of probability\\nis the mis-bound state. The time to form the native bound complex\\nis much longer in E73A compared to the wild type, as the system\\ngets trapped in the mis-bound state for 100s of microseconds\\n(Fig. 5d). An increase of the association rate does therefore not\\nnecessarily imply more efficient binding in a functional sense.\\n\\nBased on the consensus models the experimental association rate\\nis simulated as a function of protein concentration. At micromolar\\nconcentrations commonly used in experiments, the association rates\\nare linear in the concentration, that is, binding is limited by the\\nbimolecular encounter (Supplementary Fig. 5). In this condition\\nthe dominant energy barrier is characterized by balancing the entro-\\npic cost of bringing two molecules together at low concentration and\\nthe onset of the electrostatic interactions. At concentrations of 10s to\\n100s of μM, this barrier disappears, the transition state of binding\\nbecomes rate-limiting4, and we observe a concentration-independent\\nrate constant of 18,000 s–1 for the wild type and 1,500 s–1 for the E73A\\nmutant due to the transition state of binding (Supplementary Fig. 5).\\n\\nAfter an initial contact has been formed, the complex will\\ntypically move through a sequence of other non-native inter-\\nmediates until it dissociates or proceeds to native binding with\\nabout 50% probability each (Fig. 5a,b). The major bottleneck in\\ndissociation from the native state is the separation of the loosely\\nbound complex into the pre-bound, mis-bound and intermediate\\nstates (Supplementary Fig. 6), from where the complex can dissociate\\nor perform geminate rebinding (Fig. 2g).\\n\\nConclusion\\nBy combining recent breakthroughs in simulation and analysis tech-\\nnology, we have demonstrated that a quantitatively accurate micro-\\nscopic model of protein–protein association can be obtained using\\nunbiased all-atom MD simulations and Markov modelling. This\\nmodel reveals kinetic and mechanistic details of the association mech-\\nanism and has enabled the discovery of previously unknown substates.\\n\\nThe presented combination of simulation and analysis techniques\\nare applicable to any pair of macromolecules. While the available\\ncomputational power limits the size of the systems being studied to\\ncurrently a few hundred amino acids per protein, biological timescales\\ncan be reached that are far beyond direct MD simulation times. This\\nopens the door to study how the parts of macromolecular assemblies\\nform, how viral and host proteins interact and how protein–DNA\\ninteractions depend on DNA sequence and epigenetic modifications.\\n\\nRather than a black-box model, the present approach provides\\npopulations for every sampled atomistic structure and transition\\nrates between them. We have demonstrated that combining such an\\nextensive structural ensemble with an appropriate reweighting\\nmethod, the thermodynamics and kinetics of similar protein sequences\\ncan bepredicted. In the future, protein sequences andpost-translational\\nmodifications, or sequence and epigenetic modifications of interacting\\nnucleic acids can be designed to stabilize desirable intermediates,\\nchange the kinetics or control the mechanism, with far-reaching\\nconsequences for pharmaceutics, biotechnology andmaterial sciences.\\n\\nMethods\\nMD simulation setup. All-atom models of barnase wild type and barstar pseudo\\nwild type with C40,82A double mutation24 were constructed starting from crystal\\nstructure 1BRS (Protein Data Bank ID)38 (chain B for barnase and chain F\\n\\nsuperimposed to chain E for barstar) and immersed in a water box with ∼10 nm\\nedge length at 50 mM salt concentration (approximately 90,000 atoms). Protonation\\nstates were chosen using AmberTools39. His102 was in un-protonated (U-tautomer)\\nform that is dominant at pH 7. The Amber ff99SB force field40, the TIP3P water\\nmodel41, and the ACEMD program42 were used for simulations. The system was\\nminimized and equilibrated first in the NpT ensemble; box size was fixed and\\nsimulations were conducted in NVT at 300 K.\\n\\nWild-type GPUgrid simulations. 1,892 independent MD simulations starting from\\ndissociated structures (generated by random translation and rotation of barstar from\\nthe complex, excluding overlapping structures) were distributed on the GPUgrid\\ncomputing network43. Trajectories were up to 1 english μs long (distribution shown\\nin Supplementary Fig. 1), and the aggregate simulation time was 1.7 milliseconds. 74\\nof these simulations found the native barnase–barstar complex.\\n\\nWild-type adaptive MD simulations. Multiple rounds of short MD trajectories are\\nrun and a selection strategy based on available trajectory statistics is employed to select\\nthe starting coordinates for the next round of trajectories25–27 (Supplementary Methods\\n1 and Fig. 1b for details). Different combinations of selection strategies were tested.\\nAssociation and binding into the native complex succeeded with various strategies,\\nwhereas dissociation only succeeded by combining regular spatial clustering with a\\nminimum RMSD metric14, and selecting new MD frames with a probability anti-\\nproportional to the number of frames in each cluster. To confirm that the observed\\ndissociation is robust with respect to the force field used, we reproduced adaptive\\ndissociation using CHARMM36 (ref. 44) following the setup reported previously10.\\n\\nAdaptive trajectories were 50 or 100 ns long when generated. Selected adaptive\\nruns (Supplementary Methods 2) were extended to 200 ns for the analysis. 797\\nadaptive MD simulations starting from the natively bound complex, and 756 from\\ndissociated and non-native associated states were used, comprising an aggregate\\nsimulation time of 0.3 milliseconds.\\n\\nE73A mutant simulations. 25 MD frames were sampled from each of the 15\\nwild-type HMM states (Fig. 2d). For each structure, Glu 73 was mutated to Ala,\\nstructures were rebuild using HTMD45, and four copies were run on GPUgrid,\\nresulting in 1,500 simulations of 200 ns length, summing up to 300 µs.\\n\\nFeature selection and discretization. Following feature selection and discretization\\nwas carried out for the wild-type simulations using PyEMMA (http://pyemma.org)46\\n\\nand HTMD (http://htmd.org)45. 4,094 ‘relevant’ residue–residue contacts between\\nproteins, defined by nearest-neighbour heavy-atom distances <0.4 nm, were\\nmeasured. A contact was considered relevant when the longest time it had been\\nformed continuously exceeded 10 ns. The following features were used for\\nanalysis: (i) the minimum heavy-atom RMSD between each frame and the native\\nbarnase/barstar complex, (ii) the minimum distance between the two proteins,\\nand (iii) the seven slowest linear combinations of the 4,094 relevant contacts,\\ncomputed variationally50 by the TICA method47,48 (lagtime 100 ns). This space was\\ndiscretized using k-means clustering (k = 2,000).\\n\\nE73A mutant simulation trajectories were mapped on the same features and\\nassigned to the same cluster discretization.\\n\\nHidden Markov model. Only simulations conducted with Amber ff99SB were\\nused to estimate the HMM. Discrete HMMs as described previously30 are built on\\nthe k-means clustering and find a coarse-grained kinetic model by optimally\\ngrouping fast mixing states into metastable sets, and estimating the transition rates\\nbetween them. See Supplementary Methods 4 and Supplementary Fig. 2 for\\nself-consistent validation of the model.\\n\\nHMMs with 15 metastable states were estimated from the discretized simulation\\ndata using the msm package of PyEMMA. The HMM with lag time 110 ns was\\nselected for analysis because the free energies and relaxation timescales were\\napproximately independent of the lag at this point (Supplementary Fig. 2). For the\\nE73A mutant trajectories, we also estimated an HMM at lag 110 ns. Uncertainties\\nwere computed using the reversible Bayesian estimator described previously49.\\nHMM analysis was conducted using the methods provided by PyEMMA’s HMM\\nclass, and some functionalities of the msmtools package (Supplementary Methods 4).\\n\\nProtein association model. While the HMM automatically identified a dissociated\\nstate (lower green state in Fig. 2d), calculation of association rates and binding free\\nenergies comparable to experiment requires the definition of a bulk dissociated\\nstate with concentration-dependent association rate. Following an idea by\\nNorthrup et al.31, we attached a bulk dissociated state to the HMM as follows:\\n\\nbulk dissociated state ⇌\\n4πDb\\n\\nku\\nHMM dissociated state⇌\\n\\nkij\\n\\nkji\\nother HMM stateswhere kij\\n\\ndenotes the transition rates between HMM states. The bimolecular association rate\\n4πDb is the Smoluchowski rate of hitting a sphere of radius b = 4 nm around the\\norigin that is the effective size of the HMM dissociated state. The diffusion constant\\nD = 28.5 nm2 μs–1 has been fitted from the MD simulations. The bulk dissociation\\nrate is given by ku = ( f∞\\n\\n–1 – 1) Σjkji, where f∞ is the probability to dissociate to the bulk\\nwhen starting from b, which was computed by combining trajectory statistics and a\\nBrownian dynamics model (Supplementary Methods 9).\\n\\nARTICLES NATURE CHEMISTRY DOI: 10.1038/NCHEM.2785\\n\\nNATURE CHEMISTRY | ADVANCE ONLINE PUBLICATION | www.nature.com/naturechemistry6\\n\\n© 2017 Macmillan Publishers Limited, part of Springer Nature. All rights reserved.\\n\\nhttp://pyemma.org\\nhttp://htmd.org\\nhttp://dx.doi.org/10.1038/nchem.2785\\nhttp://www.nature.com/naturechemistry\\n\\n\\nCode availability. PyEMMA (http://pyemma.org) and HTMD (http://htmd.org) are\\navailable free of charge. See http://acellera.com for the availability of ACEMD.\\n\\nReceived 19 October 2016; accepted 21 April 2017;\\npublished online 5 June 2017\\n\\nReferences\\n1. Scott, D. E., Bayly, A. R., Abell, C. & Skidmore, J. Small molecules, big targets:\\n\\ndrug discovery faces the protein–protein interaction challenge. Nat. Rev. Drug.\\nDiscov. 15, 533–550 (2016).\\n\\n2. Doench, J. G. et al. Rational design of highly active sgrnas for crispr–cas9-\\nmediated gene inactivation. Nat. Biotechnol. 32, 1262–1267 (2014).\\n\\n3. King, N. P. et al. Computational design of self-assembling protein nanomaterials\\nwith atomic level accuracy. Science 336, 1171–1174 (2012).\\n\\n4. Schreiber, G., Haran, G. & Zhou, H.-X. Fundamental aspects of protein–protein\\nassociation kinetics. Chem. Rev. 109, 839–860 (2009).\\n\\n5. Tang, C., Iwahara, J. & Clore, G. M. Visualization of transient encounter\\ncomplexes in protein–protein association. Nature 444, 383–386 (2006).\\n\\n6. Gabdoulline, R. R. & Wade, R. C. Simulation of the diffusional association of\\nbarnase and barstar. Biophys. J. 72, 1917–1929 (1997).\\n\\n7. Spaar, A., Dammer, C., Gabdoulline, R. R., Wade, R. C. & Helms, V. Diffusional\\nencounter of barnase and barstar. Biophys. J. 90, 1913–1924 (2006).\\n\\n8. Levy, Y., Wolynes, P. G. & Onuchic, J. N. Protein topology determines binding\\nmechanism. Proc. Natl Acad. Sci. USA 101, 511–516 (2004).\\n\\n9. Schluttig, J., Alamanova, D., Helms, V. & Schwarz, U. S. Dynamics of protein–\\nprotein encounter: a langevin equation approach with reaction patches. J. Chem.\\nPhys. 129, 155106 (2008).\\n\\n10. Gumbart, J. C., Roux, B. & Chipot, C. Efficient determination of protein–protein\\nstandard binding free energies from first principles. J. Chem. Theory Comput. 9,\\n3789–3798 (2013).\\n\\n11. Barducci, A., Bonomi, M., Prakash, M. K. & Parrinello, M. Free-energy\\nlandscape of protein oligomerization from atomistic simulations. Proc. Natl\\nAcad. Sci. USA 110, E4708–E4713 (2013).\\n\\n12. Tiwary, P. & Parrinello, M. From metadynamics to dynamics. Phys. Rev. Lett.\\n111, 230602 (2013).\\n\\n13. Wu, H., Paul, F., Wehmeyer, C. & Noé, F. Multiensemble Markov models of\\nmolecular thermodynamics and kinetics. Proc. Natl Acad. Sci. USA 113,\\nE3221–E3230 (2016).\\n\\n14. Prinz, J.-H. et al. Markov models of molecular kinetics: generation and\\nvalidation. J. Chem. Phys. 134, 174105 (2011).\\n\\n15. Lindorff-Larsen, K., Piana, S., Dror, R. O. & Shaw, D. E. How fast-folding\\nproteins fold. Science 334, 517–520 (2011).\\n\\n16. Buch, I., Giorgino, T. & De Fabritiis, G. Complete reconstruction of an enzyme-\\ninhibitor binding process by molecular dynamics simulations. Proc. Natl Acad.\\nSci. USA 108, 10184–10189 (2011).\\n\\n17. Kohlhoff, K. J. et al. Cloud-based simulations on Google Exacycle reveal ligand\\nmodulation of GPCR activation pathways. Nat. Chem. 6, 15–21 (2014).\\n\\n18. Plattner, N. & Noé, F. Protein conformational plasticity and complex ligand\\nbinding kinetics explored by atomistic simulations and Markov models. Nat.\\nCommun. 6, 7653 (2015).\\n\\n19. Silva, D.-A., Bowman, G. R., Sosa-Peinado, A. & Huang, X. A role for both\\nconformational selection and induced fit in ligand binding by the Lao protein.\\nPLoS Comput. Biol. 7, e1002054 (2011).\\n\\n20. Piana, S., Lindorff-Larsen, K. & Shaw, D. E. Atomistic description of the folding\\nof a dimeric protein. J. Phys. Chem. B 117, 12935–12942 (2013).\\n\\n21. Ahmad, M., Gu, W., Geyer, T. & Helms, V. Adhesive water networks facilitate\\nbinding of protein interfaces. Nat. Commun. 2, 261 (2011).\\n\\n22. Schreiber, G. & Fersht, A. R. Rapid, electrostatically assisted association of\\nproteins. Nat. Struct. Biol. 3, 427–431 (1996).\\n\\n23. Schreiber, G. & Fersht, A. R. Interaction of barnase with its polypeptide\\ninhibitor barstar studied by protein engineering. Biochemistry 32,\\n5145–5150 (1993).\\n\\n24. Hartley, R. W. Directed mutagenesis and barnase–barstar recognition.\\nBiochemistry 32, 5978–5984 (1993).\\n\\n25. Doerr, S. & Fabritiis, G. D. On-the-fly learning and sampling of ligand binding\\nby high-throughput molecular simulations. J. Chem. Theory Comput. 10,\\n2064–2069 (2014).\\n\\n26. Bowman, G. R., Ensign, D. L. & Pande, V. S. Enhanced modeling via network\\ntheory: adaptive sampling of Markov state models. J. Chem. Theory Comput. 6,\\n787–794 (2010).\\n\\n27. Preto, J. & Clementi, C. Fast recovery of free energy landscapes via\\ndiffusion-map-directed molecular dynamics. Phys. Chem. Chem. Phys. 16,\\n19181–19191 (2014).\\n\\n28. Bowman, G. R., Pande, V. S. & Noé, F. (eds.) An Introduction to Markov State\\nModels and Their Application to Long Timescale Molecular Simulation (Vol. 797\\nof Advances in Experimental Medicine and Biology, Springer, 2014).\\n\\n29. Sarich, M. & Schütte, C. Metastability and Markov State Models in Molecular\\nDynamics (Courant Lecture Notes, American Mathematical Society, 2013).\\n\\n30. Noé, F., Wu, H., Prinz, J.-H. & Plattner, N. Projected and hidden Markov models\\nfor calculating kinetics and metastable states of complex molecules. J. Chem.\\nPhys. 139, 184114 (2013).\\n\\n31. Northrup, S. H., Allison, S. & McCammon, J. Brownian dynamics of diffusion-\\ninfluenced bimolecular reactions. J. Chem. Phys. 80, 1517–1524 (1984).\\n\\n32. Schreiber, G. & Fersht, A. R. Energetics of protein–protein interactions: analysis\\nof the barnase–barstar interface by single mutations and double mutant cycles.\\nJ. Mol. Biol. 248, 478–486 (1995).\\n\\n33. Matysiak, S. & Clementi, C. Optimal combination of theory and experiment for\\nthe characterization of the protein folding landscape of S6: how far can a\\nminimalist model go? J. Mol. Biol. 343, 235–248 (2004).\\n\\n34. Frisch, C., Fersht, A. R. & Schreiber, G. Experimental assignment of the structure\\nof the transition state for the association of barnase and barstar. J. Mol. Biol. 308,\\n69–77 (2001).\\n\\n35. Harel, M., Cohen, M. & Schreiber, G. On the dynamic nature of the transition\\nstate for protein–protein association as determined by double-mutant cycle\\nanalysis and simulation. J. Mol. Biol. 371, 180–196 (2007).\\n\\n36. Chung, H. S., Louis, J. M. & Eaton,W. A. Single-molecule fluorescence experiments\\ndetermine protein folding transition path times. Science 335, 981–984 (2012).\\n\\n37. Anunciado, D., Dhar, A., Gruebele, M. & Baranger, A. M. Multistep kinetics of\\nthe U1A–SL2 RNA complex dissociation. J. Mol. Biol. 408, 896–908 (2011).\\n\\n38. Buckle, A. M., Schreiber, G. & Fersht, A. R. Protein–protein recognition: crystal\\nstructural analysis of a barnase–barstar complex at 2.0-Å resolution.\\nBiochemistry 33, 8878–8889 (1994).\\n\\n39. Case, D. A. et al. The Amber biomolecular simulation programs. J. Comput.\\nChem. 26, 1668–1688 (2005).\\n\\n40. Hornak, V. et al. Comparison of multiple Amber force fields and development of\\nimproved protein backbone parameters. Proteins 65, 712–725 (2006).\\n\\n41. Jorgensen, W. L., Chandrasekhar, J., Madura, J. D., Impey, R. W. & Klein, M. L.\\nComparison of simple potential functions for simulating liquid water. J. Chem.\\nPhys. 79, 926–935 (1983).\\n\\n42. Harvey, M. J., Giupponi, G. & De Fabritiis, G. ACEMD: Accelerated molecular\\ndynamics simulations in the microseconds timescale. J. Chem. Theory Comp. 5,\\n1632–1639 (2009).\\n\\n43. Buch, I., Harvey, M. J., Giorgino, T., Anderson, D. P. & De Fabritiis, G.\\nHigh-throughput all-atom molecular dynamics simulations using distributed\\ncomputing. J. Chem. Inf. Model. 50, 397–403 (2010).\\n\\n44. Best, R. B. et al. Optimization of the additive CHARMM all-atom protein force\\nfield targeting improved sampling of the backbone ϕ, ψ and side-chain χ1 and χ2\\ndihedral angles. J. Chem. Theo. Comp. 8, 3257–3273 (2012).\\n\\n45. Doerr, S., Harvey, M. J., Noé, F. & Fabritiis, G. D. HTMD: high-throughput\\nmolecular dynamics for molecular discovery. J. Chem. Theory Comput. 12,\\n1845–1852 (2016).\\n\\n46. Scherer, M. K. et al. PyEMMA 2: a software package for estimation, validation\\nand analysis of Markov models. J. Chem. Theory Comput. 11, 5525–5542 (2015).\\n\\n47. Perez-Hernandez, G., Paul, F., Giogino, T., De Fabritiis, G. & Noé, F.\\nIdentification of slow molecular order parameters for Markov model\\nconstruction. J. Chem. Phys. 139, 015102 (2013).\\n\\n48. Molgedey, L. & Schuster, H. G. Separation of a mixture of independent signals\\nusing time delayed correlations. Phys. Rev. Lett. 72, 3634–3637 (1994).\\n\\n49. Trendelkamp-Schroer, B., Wu, H., Paul, F. & Noé, F. Estimation and uncertainty\\nof reversible Markov models. J. Chem. Phys. 143, 174101 (2015).\\n\\n50. Noé, F. & Nüske, F. A variational approach to modeling slow processes in\\nstochastic dynamical systems. Multiscale Model. Simul. 11, 635–655 (2013).\\n\\nAcknowledgements\\nWe are grateful to the following researchers for inspiring discussions and extensive\\nfeedback on the manuscript: J. Clark, C. Clementi, O. Daumke, W. A. Eaton, M. Gruebele,\\nK. Lindorff-Larsen, S. Olsson, F. Paul, B. Roux, U. Schwarz, S. Sukenik, R. Wade and\\nT.Weikl. Funding is acknowledged from the European Commission (ERC StG pcCells and\\nPRACE project 2014102337 to F.N.), Deutsche Forschungsgemeinschaft (NO 825/3-1, SFB\\n958/A4, SFB 740/D7 to F.N.), Einstein Foundation Berlin (project SoOPic to N.P.),\\nMINECO (BIO2014-53095-P) and FEDER (to G.D.F.), Acellera Ltd. (to S.D.). We thank\\nthe volunteers of GPUGRID for donating computing time for simulations.\\n\\nAuthor contributions\\nN.P., S.D., G.D.F and F.N. designed research and developed/implemented software.\\nN.P. and S.D. conducted simulations. N.P., S.D. and F.N. analysed simulations. F.N.\\ndeveloped methods. N.P. and F.N. wrote the paper.\\n\\nAdditional information\\nSupplementary information is available in the online version of the paper. Reprints and\\npermissions information is available online at www.nature.com/reprints. Publisher’s note:\\nSpringer Nature remains neutral with regard to jurisdictional claims in published maps and\\ninstitutional affiliations. Correspondence and requests for materials should be addressed\\nto N.P. or F.N.\\n\\nCompeting financial interests\\nThe authors declare no competing financial interests.\\n\\nNATURE CHEMISTRY DOI: 10.1038/NCHEM.2785 ARTICLES\\n\\nNATURE CHEMISTRY | ADVANCE ONLINE PUBLICATION | www.nature.com/naturechemistry 7\\n\\n© 2017 Macmillan Publishers Limited, part of Springer Nature. All rights reserved.\\n\\nhttp://pyemma.org\\nhttp://htmd.org\\nhttp://acellera.com\\nhttp://dx.doi.org/10.1038/nchem.2785\\nhttp://www.nature.com/reprints\\nhttp://dx.doi.org/10.1038/nchem.2785\\nhttp://www.nature.com/naturechemistry\\n\\n\\tComplete protein–protein association kinetics in atomic detail revealed by molecular dynamics simulations and Markov modelling\\n\\tResults and discussion\\n\\tAll-atom simulation of reversible protein–protein association\\n\\tEquilibrium kinetics model and validation\\n\\tOverall binding mechanism\\n\\tNature of the transition state of binding\\n\\tConformational change in the complex\\n\\tAssociation kinetics of wild-type and a faster-associating mutant\\n\\n\\tConclusion\\n\\tMethods\\n\\tMD simulation setup\\n\\tWild-type GPUgrid simulations\\n\\tWild-type adaptive MD simulations\\n\\tE73A mutant simulations\\n\\tFeature selection and discretization\\n\\tHidden Markov model\\n\\tProtein association model\\n\\tCode availability\\n\\n\\tFigure 1  Simulation of protein–protein association and dissociation.\\n\\tFigure 2  Kinetic model of barnase–barstar association.\\n\\tFigure 3  Late steps in binding from late intermediates to the loosely bound state.\\n\\tFigure 4  Native substate conformational dynamics.\\n\\tFigure 5  Comparison of kinetics between wild type and E73A mutant.\\n\\tReferences\\n\\tAcknowledgements\\n\\tAuthor contributions\\n\\tAdditional information\\n\\tCompeting financial interests\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n<<\\n  /ASCII85EncodePages false\\n  /AllowTransparency false\\n  /AutoPositionEPSFiles true\\n  /AutoRotatePages /None\\n  /Binding /Left\\n  /CalGrayProfile ()\\n  /CalRGBProfile (sRGB IEC61966-2.1)\\n  /CalCMYKProfile (U.S. Web Coated \\\\050SWOP\\\\051 v2)\\n  /sRGBProfile (sRGB IEC61966-2.1)\\n  /CannotEmbedFontPolicy /Error\\n  /CompatibilityLevel 1.3\\n  /CompressObjects /Off\\n  /CompressPages true\\n  /ConvertImagesToIndexed true\\n  /PassThroughJPEGImages true\\n  /CreateJobTicket false\\n  /DefaultRenderingIntent /Default\\n  /DetectBlends true\\n  /DetectCurves 0.1000\\n  /ColorConversionStrategy /LeaveColorUnchanged\\n  /DoThumbnails false\\n  /EmbedAllFonts true\\n  /EmbedOpenType false\\n  /ParseICCProfilesInComments true\\n  /EmbedJobOptions true\\n  /DSCReportingLevel 0\\n  /EmitDSCWarnings false\\n  /EndPage -1\\n  /ImageMemory 1048576\\n  /LockDistillerParams true\\n  /MaxSubsetPct 100\\n  /Optimize true\\n  /OPM 1\\n  /ParseDSCComments true\\n  /ParseDSCCommentsForDocInfo true\\n  /PreserveCopyPage false\\n  /PreserveDICMYKValues true\\n  /PreserveEPSInfo true\\n  /PreserveFlatness true\\n  /PreserveHalftoneInfo false\\n  /PreserveOPIComments false\\n  /PreserveOverprintSettings true\\n  /StartPage 1\\n  /SubsetFonts true\\n  /TransferFunctionInfo /Apply\\n  /UCRandBGInfo /Remove\\n  /UsePrologue true\\n  /ColorSettingsFile (None)\\n  /AlwaysEmbed [ true\\n  ]\\n  /NeverEmbed [ true\\n  ]\\n  /AntiAliasColorImages false\\n  /CropColorImages true\\n  /ColorImageMinResolution 150\\n  /ColorImageMinResolutionPolicy /OK\\n  /DownsampleColorImages true\\n  /ColorImageDownsampleType /Bicubic\\n  /ColorImageResolution 450\\n  /ColorImageDepth -1\\n  /ColorImageMinDownsampleDepth 1\\n  /ColorImageDownsampleThreshold 1.00000\\n  /EncodeColorImages true\\n  /ColorImageFilter /DCTEncode\\n  /AutoFilterColorImages true\\n  /ColorImageAutoFilterStrategy /JPEG\\n  /ColorACSImageDict <<\\n    /QFactor 0.15\\n    /HSamples [1 1 1 1] /VSamples [1 1 1 1]\\n  >>\\n  /ColorImageDict <<\\n    /QFactor 0.15\\n    /HSamples [1 1 1 1] /VSamples [1 1 1 1]\\n  >>\\n  /JPEG2000ColorACSImageDict <<\\n    /TileWidth 256\\n    /TileHeight 256\\n    /Quality 30\\n  >>\\n  /JPEG2000ColorImageDict <<\\n    /TileWidth 256\\n    /TileHeight 256\\n    /Quality 30\\n  >>\\n  /AntiAliasGrayImages false\\n  /CropGrayImages true\\n  /GrayImageMinResolution 150\\n  /GrayImageMinResolutionPolicy /OK\\n  /DownsampleGrayImages true\\n  /GrayImageDownsampleType /Bicubic\\n  /GrayImageResolution 450\\n  /GrayImageDepth -1\\n  /GrayImageMinDownsampleDepth 2\\n  /GrayImageDownsampleThreshold 1.00000\\n  /EncodeGrayImages true\\n  /GrayImageFilter /DCTEncode\\n  /AutoFilterGrayImages true\\n  /GrayImageAutoFilterStrategy /JPEG\\n  /GrayACSImageDict <<\\n    /QFactor 0.15\\n    /HSamples [1 1 1 1] /VSamples [1 1 1 1]\\n  >>\\n  /GrayImageDict <<\\n    /QFactor 0.15\\n    /HSamples [1 1 1 1] /VSamples [1 1 1 1]\\n  >>\\n  /JPEG2000GrayACSImageDict <<\\n    /TileWidth 256\\n    /TileHeight 256\\n    /Quality 30\\n  >>\\n  /JPEG2000GrayImageDict <<\\n    /TileWidth 256\\n    /TileHeight 256\\n    /Quality 30\\n  >>\\n  /AntiAliasMonoImages false\\n  /CropMonoImages true\\n  /MonoImageMinResolution 1200\\n  /MonoImageMinResolutionPolicy /OK\\n  /DownsampleMonoImages true\\n  /MonoImageDownsampleType /Bicubic\\n  /MonoImageResolution 2400\\n  /MonoImageDepth -1\\n  /MonoImageDownsampleThreshold 1.00000\\n  /EncodeMonoImages true\\n  /MonoImageFilter /CCITTFaxEncode\\n  /MonoImageDict <<\\n    /K -1\\n  >>\\n  /AllowPSXObjects false\\n  /CheckCompliance [\\n    /None\\n  ]\\n  /PDFX1aCheck true\\n  /PDFX3Check false\\n  /PDFXCompliantPDFOnly false\\n  /PDFXNoTrimBoxError false\\n  /PDFXTrimBoxToMediaBoxOffset [\\n    35.29000\\n    35.29000\\n    36.28000\\n    36.28000\\n  ]\\n  /PDFXSetBleedBoxToMediaBox false\\n  /PDFXBleedBoxToTrimBoxOffset [\\n    8.50000\\n    8.50000\\n    8.50000\\n    8.50000\\n  ]\\n  /PDFXOutputIntentProfile (OFCOM_PO_P1_F60)\\n  /PDFXOutputConditionIdentifier ()\\n  /PDFXOutputCondition (OFCOM_PO_P1_F60)\\n  /PDFXRegistryName ()\\n  /PDFXTrapped /False\\n\\n  /CreateJDFFile false\\n  /SyntheticBoldness 1.000000\\n  /Description <<\\n    /JPN <FEFF3053306e8a2d5b9a306f300130d330b830cd30b9658766f8306e8868793a304a3088307353705237306b90693057305f00200050004400460020658766f830924f5c62103059308b3068304d306b4f7f75283057307e305930023053306e8a2d5b9a30674f5c62103057305f00200050004400460020658766f8306f0020004100630072006f0062006100740020304a30883073002000520065006100640065007200200035002e003000204ee5964d30678868793a3067304d307e30593002>\\n    /DEU <FEFF00560065007200770065006e00640065006e0020005300690065002000640069006500730065002000450069006e007300740065006c006c0075006e00670065006e0020007a0075006d002000450072007300740065006c006c0065006e00200076006f006e0020005000440046002d0044006f006b0075006d0065006e00740065006e002c00200075006d002000650069006e00650020007a0075007600650072006c00e40073007300690067006500200041006e007a006500690067006500200075006e00640020004100750073006700610062006500200076006f006e00200047006500730063006800e40066007400730064006f006b0075006d0065006e00740065006e0020007a0075002000650072007a00690065006c0065006e002e00200044006900650020005000440046002d0044006f006b0075006d0065006e007400650020006b00f6006e006e0065006e0020006d006900740020004100630072006f0062006100740020006f0064006500720020006d00690074002000640065006d002000520065006100640065007200200035002e003000200075006e00640020006800f600680065007200200067006500f600660066006e00650074002000770065007200640065006e002e>\\n    /FRA <FEFF004f007000740069006f006e00730020007000650072006d0065007400740061006e007400200064006500200063007200e900650072002000640065007300200064006f00630075006d0065006e007400730020005000440046002000700072006f00660065007300730069006f006e006e0065006c007300200066006900610062006c0065007300200070006f007500720020006c0061002000760069007300750061006c00690073006100740069006f006e0020006500740020006c00270069006d007000720065007300730069006f006e002e00200049006c002000650073007400200070006f0073007300690062006c0065002000640027006f00750076007200690072002000630065007300200064006f00630075006d0065006e007400730020005000440046002000640061006e00730020004100630072006f0062006100740020006500740020005200650061006400650072002c002000760065007200730069006f006e002000200035002e00300020006f007500200075006c007400e9007200690065007500720065002e>\\n    /PTB <FEFF005500740069006c0069007a006500200065007300740061007300200063006f006e00660069006700750072006100e700f5006500730020007000610072006100200063007200690061007200200064006f00630075006d0065006e0074006f0073002000500044004600200063006f006d00200075006d0061002000760069007300750061006c0069007a006100e700e3006f0020006500200069006d0070007200650073007300e3006f00200061006400650071007500610064006100730020007000610072006100200064006f00630075006d0065006e0074006f007300200063006f006d0065007200630069006100690073002e0020004f007300200064006f00630075006d0065006e0074006f0073002000500044004600200070006f00640065006d0020007300650072002000610062006500720074006f007300200063006f006d0020006f0020004100630072006f006200610074002c002000520065006100640065007200200035002e00300020006500200070006f00730074006500720069006f0072002e>\\n    /DAN <FEFF004200720075006700200064006900730073006500200069006e0064007300740069006c006c0069006e006700650072002000740069006c0020006100740020006f0070007200650074007400650020005000440046002d0064006f006b0075006d0065006e007400650072002c0020006400650072002000650072002000650067006e006500640065002000740069006c0020007000e5006c006900640065006c006900670020007600690073006e0069006e00670020006f00670020007500640073006b007200690076006e0069006e006700200061006600200066006f0072007200650074006e0069006e006700730064006f006b0075006d0065006e007400650072002e0020005000440046002d0064006f006b0075006d0065006e007400650072006e00650020006b0061006e002000e50062006e006500730020006d006500640020004100630072006f0062006100740020006f0067002000520065006100640065007200200035002e00300020006f00670020006e0079006500720065002e>\\n    /NLD <FEFF004700650062007200750069006b002000640065007a006500200069006e007300740065006c006c0069006e00670065006e0020006f006d0020005000440046002d0064006f00630075006d0065006e00740065006e0020007400650020006d0061006b0065006e00200064006900650020006700650073006300680069006b00740020007a0069006a006e0020006f006d0020007a0061006b0065006c0069006a006b006500200064006f00630075006d0065006e00740065006e00200062006500740072006f0075007700620061006100720020007700650065007200200074006500200067006500760065006e00200065006e0020006100660020007400650020006400720075006b006b0065006e002e0020004400650020005000440046002d0064006f00630075006d0065006e00740065006e0020006b0075006e006e0065006e00200077006f007200640065006e002000670065006f00700065006e00640020006d006500740020004100630072006f00620061007400200065006e002000520065006100640065007200200035002e003000200065006e00200068006f006700650072002e>\\n    /ESP <FEFF0055007300650020006500730074006100730020006f007000630069006f006e006500730020007000610072006100200063007200650061007200200064006f00630075006d0065006e0074006f0073002000500044004600200071007500650020007000650072006d006900740061006e002000760069007300750061006c0069007a006100720020006500200069006d007000720069006d0069007200200063006f007200720065006300740061006d0065006e0074006500200064006f00630075006d0065006e0074006f007300200065006d00700072006500730061007200690061006c00650073002e0020004c006f007300200064006f00630075006d0065006e0074006f00730020005000440046002000730065002000700075006500640065006e00200061006200720069007200200063006f006e0020004100630072006f00620061007400200079002000520065006100640065007200200035002e003000200079002000760065007200730069006f006e0065007300200070006f00730074006500720069006f007200650073002e>\\n    /SUO <FEFF004e00e4006900640065006e002000610073006500740075007300740065006e0020006100760075006c006c006100200076006f006900740020006c0075006f006400610020006a0061002000740075006c006f00730074006100610020005000440046002d0061007300690061006b00690072006a006f006a0061002c0020006a006f006900640065006e0020006500730069006b0061007400730065006c00750020006e00e400790074007400e400e40020006c0075006f00740065007400740061007600610073007400690020006c006f00700070007500740075006c006f006b00730065006e002e0020005000440046002d0061007300690061006b00690072006a0061007400200076006f0069006400610061006e0020006100760061007400610020004100630072006f006200610074002d0020006a0061002000520065006100640065007200200035002e00300020002d006f0068006a0065006c006d0061006c006c0061002000740061006900200075007500640065006d006d0061006c006c0061002000760065007200730069006f006c006c0061002e>\\n    /ITA <FEFF00550073006100720065002000710075006500730074006500200069006d0070006f007300740061007a0069006f006e00690020007000650072002000630072006500610072006500200064006f00630075006d0065006e007400690020005000440046002000610064006100740074006900200070006500720020006c00610020007300740061006d00700061002000650020006c0061002000760069007300750061006c0069007a007a0061007a0069006f006e006500200064006900200064006f00630075006d0065006e0074006900200061007a00690065006e00640061006c0069002e0020004900200064006f00630075006d0065006e00740069002000500044004600200070006f00730073006f006e006f0020006500730073006500720065002000610070006500720074006900200063006f006e0020004100630072006f00620061007400200065002000520065006100640065007200200035002e003000200065002000760065007200730069006f006e006900200073007500630063006500730073006900760065002e>\\n    /NOR <FEFF004200720075006b00200064006900730073006500200069006e006e007300740069006c006c0069006e00670065006e0065002000740069006c002000e50020006f00700070007200650074007400650020005000440046002d0064006f006b0075006d0065006e00740065007200200073006f006d002000700061007300730065007200200066006f00720020007000e5006c006900740065006c006900670020007600690073006e0069006e00670020006f00670020007500740073006b007200690066007400200061007600200066006f0072007200650074006e0069006e006700730064006f006b0075006d0065006e007400650072002e0020005000440046002d0064006f006b0075006d0065006e00740065006e00650020006b0061006e002000e50070006e006500730020006d006500640020004100630072006f0062006100740020006f0067002000520065006100640065007200200035002e00300020006f0067002000730065006e006500720065002e>\\n    /SVE <FEFF0041006e007600e4006e00640020006400650020006800e4007200200069006e0073007400e4006c006c006e0069006e006700610072006e00610020006e00e40072002000640075002000760069006c006c00200073006b0061007000610020005000440046002d0064006f006b0075006d0065006e007400200073006f006d00200070006100730073006100720020006600f600720020007000e5006c00690074006c006900670020007600690073006e0069006e00670020006f006300680020007500740073006b0072006900660074002000610076002000610066006600e4007200730064006f006b0075006d0065006e0074002e0020005000440046002d0064006f006b0075006d0065006e00740065006e0020006b0061006e002000f600700070006e006100730020006d006500640020004100630072006f0062006100740020006f00630068002000520065006100640065007200200035002e003000200065006c006c00650072002000730065006e006100720065002e>\\n    /ENU (NPG PRINT PDF Job Options. 4th September 2006. PDF 1.3 Compatibility. Adds Trim and Bleed boxes top Nature pages where none exist.)\\n  >>\\n>> setdistillerparams\\n<<\\n  /HWResolution [2400 2400]\\n  /PageSize [665.858 854.929]\\n>> setpagedevice\\n\\n'"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "parsedPDF['content']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'Author': ['Stefan Doerr',\n",
       "  'Gianni De Fabritiis',\n",
       "  'Frank Noé',\n",
       "  'Nuria Plattner'],\n",
       " 'Content-Encoding': 'ISO-8859-1',\n",
       " 'Content-Length': '-1',\n",
       " 'Content-Type': ['application/pdf', 'text/plain; charset=ISO-8859-1'],\n",
       " 'Creation-Date': '2017-05-24T06:47:57Z',\n",
       " 'Last-Modified': '2017-05-24T08:13:28Z',\n",
       " 'Last-Save-Date': '2017-05-24T08:13:28Z',\n",
       " 'X-Parsed-By': ['org.apache.tika.parser.DefaultParser',\n",
       "  'org.apache.tika.parser.pdf.PDFParser',\n",
       "  ['org.apache.tika.parser.DefaultParser',\n",
       "   'org.apache.tika.parser.txt.TXTParser']],\n",
       " 'X-TIKA:embedded_resource_path': '/folder.joboptions',\n",
       " 'X-TIKA:origResourceName': 'folder.joboptions',\n",
       " 'X-TIKA:parse_time_millis': ['338', '3'],\n",
       " 'access_permission:assemble_document': 'true',\n",
       " 'access_permission:can_modify': 'true',\n",
       " 'access_permission:can_print': 'true',\n",
       " 'access_permission:can_print_degraded': 'true',\n",
       " 'access_permission:extract_content': 'true',\n",
       " 'access_permission:extract_for_accessibility': 'true',\n",
       " 'access_permission:fill_in_form': 'true',\n",
       " 'access_permission:modify_annotations': 'true',\n",
       " 'cp:subject': 'Nature Chemistry (2017). doi:10.1038/nchem.2785',\n",
       " 'created': '2017-05-24T06:47:57Z',\n",
       " 'creator': ['Stefan Doerr',\n",
       "  'Gianni De Fabritiis',\n",
       "  'Frank Noé',\n",
       "  'Nuria Plattner'],\n",
       " 'date': '2017-05-24T08:13:28Z',\n",
       " 'dc:creator': ['Stefan Doerr',\n",
       "  'Gianni De Fabritiis',\n",
       "  'Frank Noé',\n",
       "  'Nuria Plattner'],\n",
       " 'dc:description': 'Nature Chemistry (2017). doi:10.1038/nchem.2785',\n",
       " 'dc:format': 'application/pdf; version=1.3',\n",
       " 'dc:title': 'Complete protein-protein association kinetics in atomic detail revealed by molecular dynamics simulations and Markov modelling',\n",
       " 'dcterms:created': '2017-05-24T06:47:57Z',\n",
       " 'dcterms:modified': '2017-05-24T08:13:28Z',\n",
       " 'description': 'Nature Chemistry (2017). doi:10.1038/nchem.2785',\n",
       " 'embeddedResourceType': 'ATTACHMENT',\n",
       " 'meta:author': ['Stefan Doerr',\n",
       "  'Gianni De Fabritiis',\n",
       "  'Frank Noé',\n",
       "  'Nuria Plattner'],\n",
       " 'meta:creation-date': '2017-05-24T06:47:57Z',\n",
       " 'meta:save-date': '2017-05-24T08:13:28Z',\n",
       " 'modified': '2017-05-24T08:13:28Z',\n",
       " 'pdf:PDFVersion': '1.3',\n",
       " 'pdf:docinfo:created': '2017-05-24T06:47:57Z',\n",
       " 'pdf:docinfo:creator': 'Nuria Plattner',\n",
       " 'pdf:docinfo:creator_tool': 'Arbortext Advanced Print Publisher 10.0.1465/W Unicode',\n",
       " 'pdf:docinfo:modified': '2017-05-24T08:13:28Z',\n",
       " 'pdf:docinfo:producer': 'Acrobat Distiller 8.1.0 (Windows)',\n",
       " 'pdf:docinfo:subject': 'Nature Chemistry (2017). doi:10.1038/nchem.2785',\n",
       " 'pdf:docinfo:title': 'Complete protein-protein association kinetics in atomic detail revealed by molecular dynamics simulations and Markov modelling',\n",
       " 'pdf:encrypted': 'false',\n",
       " 'producer': 'Acrobat Distiller 8.1.0 (Windows)',\n",
       " 'resourceName': ['nchemex.pdf', 'folder.joboptions'],\n",
       " 'subject': 'Nature Chemistry (2017). doi:10.1038/nchem.2785',\n",
       " 'title': 'Complete protein-protein association kinetics in atomic detail revealed by molecular dynamics simulations and Markov modelling',\n",
       " 'xmp:CreatorTool': 'Arbortext Advanced Print Publisher 10.0.1465/W Unicode',\n",
       " 'xmpMM:DocumentID': 'uuid:03d04121-f0e7-347f-b6a1-e56cb26a998e',\n",
       " 'xmpTPg:NPages': '7'}"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "parsedPDF['metadata']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [],
   "source": [
    "cont = parsedPDF['content']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\nComplete prot',\n",
       " 'n-prot',\n",
       " 'n association kinetics in atomic detail revealed by molecular dynamics simulations and Markov modelling\\n\\n\\nComplete prot',\n",
       " 'n–prot',\n",
       " 'n association kinetics in\\natomic detail revealed by molecular dynamics\\nsimulations and Markov modelling\\nNuria Plattner1*, Stefan Doerr2, Gianni De Fabritiis2,3 and Frank Noé1*\\n\\nProt',\n",
       " 'n–prot',\n",
       " 'n association is fundamental to many life processes. However, a microscopic model describing the structures\\nand kinetics during association and dissociation is lacking on account of the long lifetimes of associated states, which have\\nprevented efficient sampling by direct molecular dynamics (MD) simulations. Here we demonstrate prot',\n",
       " 'n–prot',\n",
       " 'n\\nassociation and dissociation in atomistic resolution for the ribonuclease barnase and its inhibitor barstar by combining\\nadaptive high-throughput MD simulations and hidden Markov modelling. The model reveals experimentally consistent\\nintermediate structures, energetics and kinetics on timescales from microseconds to hours. A variety of flexibly attached\\nintermediates and misbound states funnel down to a transition state and a native basin consisting of the loosely bound\\nnear-native state and the tightly bound crystallographic state. These results offer a deeper level of insight into\\nmacromolecular recognition and our approach opens the door for understanding and manipulating a wide range of\\nmacromolecular association processes.\\n\\nLife processes are critically dependent on the formation of bio-molecular complexes, and in particular, prot',\n",
       " 'n–prot',\n",
       " 'n associ-ation, which mediates the assembly of cellular structures, signal\\ntransduction and inhibition, immune response and prot',\n",
       " 'n recruit-\\nment to membrane scaffolds. Interactions of prot',\n",
       " 'ns with other\\nmacromolecules also play key roles in applications such as inhibition\\nof prot',\n",
       " 'n activity by drugs1, gene editing2 and the assembly of\\nnew materials3.\\n\\nA deep understanding of such processes and the ability to fully\\nplan and control th',\n",
       " 'r design would arise from a full map of the\\nconformation space of the interacting prot',\n",
       " 'n pairs. Such a micro-\\nscopic model would involve representative samples of all structures\\n(bound, dissociated, intermediates, etc.) in atomic resolution, th',\n",
       " 'r\\nequilibrium populations and the transition rates between them.\\n\\nHowever, such a complete microscopic picture has not yet been\\nachieved for prot',\n",
       " 'n–prot',\n",
       " 'n interaction. Experiments can probe\\n',\n",
       " 'ther individual aspects of the microscopic picture or averaged\\nensembles, such as the most stable structures, association kinetics\\nand binding affinities. Mutational analyses can identify amino\\nacids stabilizing associated states or transition states4. The structure\\nof the transient encounter complex, that is, the set of structures\\nfound immediately before the transition state of prot',\n",
       " 'n association,\\nhas been probed using NMR5 and coarse-grained simulations6–9.\\nBinding free energies can be computed from atomistic simulations\\nwhen the bound pose is precisely known10, and the binding free\\nenergy landscape can be reconstructed using metadynamics given\\na set of pre-defined collective variables11. Inference of kinetics\\nthen requires a kinetic model12 or additional unbiased simulations13.\\n\\nA candidate to reveal a complete microscopic model of prot',\n",
       " 'n–\\nprot',\n",
       " 'n interaction is unbiased atomic-detail molecular dynamics\\n(MD) simulation. Microscopic models of biomolecular processes\\nhave recently been achieved due to combined advances in high-\\nperformance computing and Markov modelling14–19. Previous\\nbinding studies, however, have been limited to the association of\\n\\nsmall ligands to prot',\n",
       " 'ns and the co-folding of shorter peptide\\nchains16,19,20, or small parts of the association landscape such as\\nthe downhill binding step from a pre-bound pose21. The simulation\\nof prot',\n",
       " 'n–prot',\n",
       " 'n association is challenged by the fact that\\ndissociation times exceed accessible MD simulation lengths by far,\\nsuch that long MD simulations will get stuck after a single\\nbinding event, possibly even in non-native traps.\\n\\nHere we demonstrate how reversible prot',\n",
       " 'n–prot',\n",
       " 'n association\\ncan be studied in atomistic detail by combining an extensive and\\nadaptively generated ensemble of unbiased molecular dynamics\\nsimulations with hidden Markov modelling. We reproduce\\nmeasured structures, energetics and kinetics quantitatively and\\nobtain a full microscopic model of the prot',\n",
       " 'n complex structures,\\nthermodynamics and kinetics.\\n\\nResults and discussion\\nAll-atom simulation of reversible prot',\n",
       " 'n–prot',\n",
       " 'n association. As\\na prototypical example, we study the diffusion-limited association of\\nthe experimentally well-characterized bacterial ribonuclease barnase\\nwith its inhibitor barstar22,23. All-atom models of barnase wild type\\nand barstar pseudo wild type24 were immersed in a solvent box\\n(approximately 90,000 atoms). Several thousand unbiased molecular\\ndynamics (MD) simulations of 2 milliseconds aggregate length were\\nconducted in an automated high-throughput manner. These data\\ninclude 1.7 milliseconds of independent trajectories starting from\\nthe dissociated state and 0.3 milliseconds in adaptive MD simulations.\\n\\nThe adaptive MD launched periods of multiple parallel MD\\ntrajectories, and the starting points of each period were selected so\\nas to favour the exploration of new states (Methods, Supplementary\\nMethods 1). This strategy can yield large speed-ups compared to\\nnon-adaptive MD runs because slow processes with large energy\\nbarriers are broken down into steps with individually smaller\\nbarriers that can be sampled efficiently25–27 (Fig. 1a,b). When starting\\nfrom the dissociated state, both adaptive and non-adaptive\\n\\n1Department of Mathematics and Computer Science, Fr',\n",
       " 'e Universität Berlin, Arnimallee 6, 14195 Berlin, Germany. 2Computational Biophysics Laboratory\\n(GRIB-IMIM), Universitat Pompeu Fabra, Barcelona Biomedical Research Park (PRBB), C/ Doctor Aiguader 88, 08003 Barcelona, Spain. 3Institució Catalana\\nde Recerca i Estudis Avançats (ICREA), Pass',\n",
       " 'g Lluis Companys 23, 08010 Barcelona, Spain. *e-mail: nuria.plattner@fu-berlin.de; frank.noe@fu-berlin.de\\n\\nARTICLES\\nPUBLISHED ONLINE: 5 JUNE 2017 | DOI: 10.1038/NCHEM.2785\\n\\nNATURE CHEMISTRY | ADVANCE ONLINE PUBLICATION | www.nature.com/naturechemistry 1\\n\\n© 2017 Macmillan Publishers Limited, part of Springer Nature. All rights reserved.\\n\\nmailto:nuria.plattner@fu-berlin.de\\nmailto:frank.noe@fu-berlin.de\\nhttp://dx.doi.org/10.1038/nchem.2785\\nhttp://www.nature.com/naturechemistry\\n\\n\\nsimulations reproducibly found the native bound complex. Adaptive\\nsimulations required a total simulation effort of a few microseconds,\\nroughly an order of magnitude speed-up compared to an equilibrium\\nbinding time of a tens of microseconds (Supplementary Fig. 1).\\nSurprisingly, the adaptive MD runs could also reproducibly reach\\nthe dissociated state when starting from the crystal structure using\\na total simulation effort of a few hundred microseconds, although\\nthe expected dissociation time is hours (Supplementary Fig. 1).\\n\\nA first analysis shows that the total dataset contains several\\nthousand unspecific association and dissociation events, about 90\\nbinding events from non-native associated states to native and 10\\nunbinding events (Fig. 1c).\\n\\nEquilibrium kinetics model and validation. Although the\\nindividual simulations are not started from an equilibrium\\ndistribution and are shorter than binding and dissociation\\ntimescales, Markov modelling approaches can combine such a\\ntrajectory ensemble towards unbiased equilibrium models of the\\nlong-timescale kinetics and thermodynamics (Fig. 1a)28,29. Here, a\\nhidden Markov model (HMM) was built to estimate a network of\\n\\nstates, associated prot',\n",
       " 'n structures, and transition rates30. The\\nHMM is coupled to a diffusive association model inspired by a\\nprevious study31 in order to be able to predict concentration-\\ndependent association rates and binding free energies (Methods,\\nSupplementary Methods 3–7).\\n\\nAs the data contains only a few unbinding events, the following\\ncomputed quantities have large uncertainties but are statistically\\nconsistent with experiment, which will be exploited to compute\\nconsensus models of experimental and simulation data (intervals\\nare 95% confidence): (i) binding free energy lies between 12 and\\n19 kcal mol–1 (experimental 16.8 kcal mol–1; see Supplementary\\nMethods 5), (ii) the dissociation rate lies in the range of 3 × 10–6\\n\\nto 1 × 10–1 s–1 (measurements lie in the range of 4.8 × 10–5 to\\n5.0 × 10–4 s–1; see Supplementary Methods 6).\\n\\nThe model successfully predicts the following quantities: (iii)\\ncrystal structure 1BRS is predicted to be the most stable state, and\\nthe simulated structures assigned to it have an average RMSD of\\n0.21 nm to 1BRS, (iv) association rate is predicted to be 2.63 × 108\\n\\nto 2.65 × 108 s–1 M–1 (1 × 108 to 3.4 × 108 s–1 M–1 measured pre-\\nviously23,32), (v) using perturbation theory33, binding free energy\\nchanges of 35 mutant sequences reported in23,32 were predicted\\nwithin statistical error (Fig. 2a and Supplementary Methods 9),\\n(vi) although this approach cannot compute absolute values of\\nassociation rates of mutants, the computed change in transition\\nstate energy is an excellent predictor for the change in the experimental\\non-rate (Fig. 2b).\\n\\nOverall binding mechanism. Having validated our model with a\\nlarge set of experimental data we proceeded to examine properties\\nthat are more difficult to probe directly by experiments. Figure 2c,\\nd displays the overall mechanisms and structures found along the\\nbinding pathways. The prot',\n",
       " 'ns initially associate to non-native\\n(large RMSD) states. At that point, they can dissociate and re-\\nassociate or stay in contact and rearrange towards the native\\ncomplex (Fig. 2c–f ).\\n\\nAt the high prot',\n",
       " 'n concentration in the simulation, initial\\nassociation occurs within tens of nanoseconds and with no detect-\\nable barrier. The initial association of barstar can occur at all\\nlocations on the barnase surface. During the first microsecond,\\nthe complex preferentially populates early intermediates with\\nbarstar located opposite of the barnase binding site (orange states\\nin Fig. 2d). However, association can also occur directly into the\\nlate intermediate and mis-bound states (blue/red states in Fig. 2d),\\nwhere barstar is located in or near the barnase binding groove,\\nbut not in the native orientation. The preference for initially associ-\\nating to locations away from the binding pocket can be explained by\\nthe fact that these states cover a larger fraction of the surface. The\\ninitial association preference depends on the salt concentration;\\nwith enhanced steering to the binding site b',\n",
       " 'ng expected when\\ncharge screening is reduced7,22. Knocking out intermediate states\\nby removing them from the HMM showed that individual early\\nintermediates have no significant effect on the association rate,\\nand even removing all early intermediates only reduces the\\nassociation rate by 20% (Supplementary Table 2).\\n\\nOn timescales of 1–10 µs, the population shifts toward the late\\nintermediates and mis-bound states (blue/red states in Figs 2d\\nand 5c), which have higher equilibrium populations than the\\nearly intermediates. Of special interest is the mis-bound state\\n(red), that has barstar bound ‘upside down’. This state does not\\nhave a well-defined binding interface, but a few non-native inter-\\nactions such as barnase Arg 83 or Arg 87 with barstar Glu 118\\noccur with significant probability (Supplementary Figs 3 and 4).\\nIntermediate states are kinetically well connected and have lifetimes\\non the order of one microsecond each (Supplementary Fig. 2).\\nRemoving the late intermediates would reduce the binding rate by\\na factor of 2.5 (Supplementary Table 2)\\n\\n1 2\\n\\n3\\n4\\n\\n5 Sampling strategy \\n\\nMarkov model \\n\\nTrajectories\\n\\nTasks\\n\\nData \\ngeneration \\n\\nModel building \\n\\nCompute clusterClientba\\n\\nc\\n\\nAssociatedDissociated Bound\\n\\nTime (μs) Time (μs)\\n\\n2,867\\n\\nD\\nis\\n\\nta\\nnc\\n\\ne \\n(n\\n\\nm\\n)\\n\\nR\\nM\\n\\nS\\nD\\n\\n (\\nnm\\n\\n)\\n\\n86\\n6\\n\\n0\\n0 0.5 0\\n\\n0\\n\\n1.5\\n\\n5\\n\\nTime (μs) Time (μs)\\n\\nR\\nM\\n\\nS\\nD\\n\\n (\\nnm\\n\\n)\\n\\nD\\nis\\n\\nta\\nnc\\n\\ne \\n(n\\n\\nm\\n) 101,465\\n\\n6\\n\\n0\\n0 3.5 0 7\\n\\n0\\n\\n1.5\\n\\nFigure 1 | Simulation of prot',\n",
       " 'n–prot',\n",
       " 'n association and dissociation.\\na, Scheme of a large-barrier process such as prot',\n",
       " 'n–prot',\n",
       " 'n dissociation that\\ncan be subdivided into multiple steps. Such processes can be sampled by\\ncombining adaptively seeded simulation trajectories between individual steps\\nvia Markov modelling. b, Schematic outline of adaptive sampling process.\\nc, Total number of full association–dissociation and binding–unbinding\\ntransitions evaluated by raw trajectory statistics of all barnase–barstar\\nsimulation trajectories used subsequently. An association–dissociation\\ntransition was counted when a trajectory moved from a state where the\\nprot',\n",
       " 'ns had a minimum distance between heavy atom pairs of >2 nm to\\n<0.3 nm, and vice versa. A binding–unbinding event was counted when a\\ntrajectory moved from a state with minimum heavy-atom RMSD from the\\ncrystallographic structure >1 nm to RMSD <0.3 nm, and vice versa. For each\\ntrajectory only one count was made in order to avoid duplicate counting\\nof recrossing events. Three sample trajectory snippets are shown for each\\ntype of transition.\\n\\nARTICLES NATURE CHEMISTRY DOI: 10.1038/NCHEM.2785\\n\\nNATURE CHEMISTRY | ADVANCE ONLINE PUBLICATION | www.nature.com/naturechemistry2\\n\\n© 2017 Macmillan Publishers Limited, part of Springer Nature. All rights reserved.\\n\\nhttp://dx.doi.org/10.1038/nchem.2785\\nhttp://www.nature.com/naturechemistry\\n\\n\\nMisbound\\n\\nEarly intermediates\\n\\nLate\\nintermediate 2\\n\\nLate\\nintermediate 1\\n\\nPre-bound\\n\\nLoosely\\nbound\\n\\nTightly\\nbound\\n\\na\\n\\nd\\n\\nb c e\\n\\nf\\n\\ng\\n\\nDissociated\\n\\nAssociation\\n\\nBinding\\n\\nUnbinding Rebinding\\n\\n12\\nΔΔ\\n\\nG\\n e\\n\\nxp\\ner\\n\\nim\\nen\\n\\nt (\\nkc\\n\\nal\\n m\\n\\nol\\n–1\\n\\n)\\n\\n8\\n\\n4\\n\\n–4 0\\n\\nΔΔG model (kcal mol–1)\\n\\n4 8 12\\n\\n–4\\n\\n0\\n\\n3\\n\\n2\\n\\n2\\n\\n0\\n\\nΔΔ\\nG\\n\\non\\n e\\n\\nxp\\ner\\n\\nim\\nen\\n\\nt (\\nkc\\n\\nal\\n m\\n\\nol\\n–1\\n\\n)\\n\\n–5 0 5 10\\n\\nΔΔGTS model (kcal mol–1)\\n\\n–2\\n\\n–1 F\\nre\\n\\ne \\nen\\n\\ner\\ngy\\n\\n (\\nkT\\n\\n)\\n\\nMinimal distance (nm)\\n\\n3.0 24\\n\\n21\\n\\n18\\n\\n15\\n\\n12\\n\\n9\\n\\n6\\n\\n3\\n\\n0\\n\\n2.5\\n\\n2.0\\n\\nR\\nM\\n\\nS\\nD\\n\\n (\\nnm\\n\\n)\\n\\n0.3\\n0.2\\n0.1\\n\\n0.00 0.02 0.04 0.06 0.08 0.10\\n\\n5\\n\\nt t+50 t+100 t+150 t+200\\n\\n10 15 20 25 30\\n\\n0.3\\n0.2\\n0.1\\n\\n0.3\\n0.2\\n0.1\\n\\nR\\nM\\n\\nS\\nD\\n\\n (\\nnm\\n\\n)\\nR\\n\\nM\\nS\\n\\nD\\n (\\n\\nnm\\n)\\n\\nR\\nM\\n\\nS\\nD\\n\\n (\\nnm\\n\\n)\\n\\n1.5\\n\\n1.0\\n\\n0.5\\n\\n0.25 0.50 1.00 2.00 4.00\\n\\nTime (μs)\\n\\nFigure 2 | Kinetic model of barnase–barstar association. a,b, Comparison of experimental and predicted changes in binding free energies and association\\nrates for all 35 single and double alanine mutations from Table II of ref. 23 and Table I of ref. 32. Statistical confidence intervals of the model are indicated in\\nwhite (1-σ, 68% confidence) and grey (2-σ, 95% confidence). a, Changes in binding free energy (ΔΔG) are predicted within statistical error. b, Comparison\\nof the computed changes in barrier h',\n",
       " 'ght of the binding transition with changes of apparent association transition state energy, computed as ΔΔGon,apparent\\n= –kBTln(kon\\n\\nmutant/kon\\nwt), where kBT= 0.598 kcal mol\\n\\n–1 and kon\\nmutant, kon\\n\\nwt are the experimental association rates of mutant and wild type, respectively. The model\\nprediction was computed as free energy difference between the pre-bound state as a proxy for the transition state, and all early intermediate, late\\nintermediate and mis-bound states. c, Free energy surface over prot',\n",
       " 'n–prot',\n",
       " 'n heavy atom minimal distance and minimal RMSD to the native complex.\\nShown is a contour plot of –kTlnμ(x) evaluated for each trajectory sample x, where kT is the thermal energy and μ(x) is the equilibrium w',\n",
       " 'ght of each\\nsample computed by the HMM (see Supplementary Methods 4). d, Kinetic network between metastable states with lifetimes on the order of microseconds.\\nOnly transition rate constants ≥0.2ms–1 are shown. Arrow thickness is proportional to rate constant, disc area is proportional to square root of state\\npopulation. Structures are centered on barnase (orange) with its binding groove on top and its N-terminal helix on the bottom. Barstar is shown in grey with\\nits binding domain in green. Grey clouds show overlay of approximately 20 barstar conformations, sampled randomly from the corresponding metastable\\nstates. e–g, Minimal RMSD to crystallographic complex shown for sample trajectories generated from the HMM (see Supplementary Methods 4). Initial\\nassociation (e), binding to the native complex (f), and geminate rebinding after b',\n",
       " 'ng bound for a long time t (one hour on average; g).\\n\\nNATURE CHEMISTRY DOI: 10.1038/NCHEM.2785 ARTICLES\\n\\nNATURE CHEMISTRY | ADVANCE ONLINE PUBLICATION | www.nature.com/naturechemistry 3\\n\\n© 2017 Macmillan Publishers Limited, part of Springer Nature. All rights reserved.\\n\\nhttp://dx.doi.org/10.1038/nchem.2785\\nhttp://www.nature.com/naturechemistry\\n\\n\\nOn timescales of 10–100 µs, the complex passes through a\\npre-bound state (magenta in Figs 2d and 5c) and then continues\\nto the natively bound complex (grey states in Figs 2d and 5c),\\nwhich consists of a loosely bound and a tightly bound substate\\nthat exchange on the microsecond timescale.\\n\\nNature of the transition state of binding. The kinetic bottleneck\\nbetween the intermediate states and natively bound states is the\\nmandatory passage through the pre-bound state (magenta in Fig. 2d),\\nthat is hence an on-pathway intermediate. Both the transition into\\nthe pre-bound state and the binding from the pre-bound state\\ninvolve barriers (Supplementary Fig. 6). We refer to these barriers\\ncollectively as transition state of binding as they are rate-limiting at\\nsufficiently high prot',\n",
       " 'n concentrations (Supplementary Fig. 5).\\n\\nThe transition from the late intermediates through the pre-bound\\nstate into the loosely bound state is characterized by two microscopic\\npathways, in which barstar interacts with ',\n",
       " 'ther of the two barnase\\nloops surrounding the binding groove: the ‘left’ loop including Ser\\n\\n38 and the ‘right’ RNA recognition loop of barnase including Arg\\n59 (Fig. 3) whose importance for steering of the two prot',\n",
       " 'n has\\nbeen shown before6,7. Due to the flexibility of both loop regions,\\nbarstar can “swing” around its attachment point, giving rise to a\\nquite flexible set of structures. These pathways are not separated by\\nsignificant barriers, i.e. the complex can change between “left” and\\n“right” attachment point multiple times before binding.\\n\\nThe minor left pathway (5% population, Fig. 3a→ c→ e) involves\\ndifferent possible prot',\n",
       " 'n contacts, including hydrophobic contacts\\nand does not appear to have well-defined structural features. In the\\nmajor right pathway (95% population, Fig. 3b→ d→ e), barnase Arg\\n59 interacts with barstar Glu 76, Asp 35, or both, in line with34,35.\\nArg 59 may additionally form a hydrophobic contact with barstar\\nTrp 38. While no individual pair of residues is strongly formed in the\\ntransition state22, Arg 59 is involved in 95% of all transition state con-\\nfigurations. This observation confirms the pivotal role of Arg 59 inmed-\\niating the binding transition34, and explains that Arg 59 mutants\\nreduced the binding rate by a factor of 10 (ref. 32). We expect that a\\nbarstar double mutant of Glu 76 and Asp 35 will have a similar effect.\\n\\nThe pre-bound state has a lifetime of about two microseconds\\n(Supplementary Fig. 2). In all binding events simulated, the\\ntransition from the pre-bound to the loosely bound state trajectories\\noccurs within a single saving interval of our simulations, that is less\\nthan 100 picoseconds. Thus the last step of docking into the barnase\\nbinding pocket and forming the binding interface with its many\\ncontacts is rare, but occurs quickly when it happens21,36. In line with\\nprevious predictions22, the majority of the barnase–barstar short-\\nrange interactions is formed after the transition state. In the limit of\\nlow electrostatic interactions, the basal binding rate may be governed\\nby the bimolecular rate to form a ready-to-dock structure similar to\\nour pre-bound state22, and it is likely that the barrier of desolvation\\nand docking is reduced or even removed under these conditions.\\n\\nConformational change in the complex. The bound states are in the\\nrange of 5–12 kcal mol–1 more stable than intermediate, mis-bound\\nand pre-bound states and thus contain almost all of the equilibrium\\npopulation. We find two conformational substates in the bound\\nstate: The loosely bound state with about 5% of the population\\nand an average heavy-atom RMSD to the crystal structure of\\n0.3 nm (Fig. 4a); and the tightly bound state with the remaining\\n95% of the population (Fig. 4b). The density of the tightly bound\\nstate contains the crystal structure (Fig. 4c, average RMSD\\n0.21 nm), which is thus correctly predicted as the most stable state.\\n\\nUpon binding, the loosely bound state is reached first. This state\\ntransitions into the tightly bound state within 5 µs on average, that\\nitself has an average lifetime of 130 µs before transitioning back to\\nthe loosely bound state. These microsecond transitions occur\\nmany times before the complex undergoes geminate rebinding\\n(Fig. 2g) or finally dissociates.\\n\\nStructurally, the native-state transition consists of a slight\\nrotation of barstar in the barnase binding pocket (compare\\nFig. 4a,b). While both substates have similar residue contact patterns\\n(Supplementary Fig. 3), the most apparent difference is that the\\nbarnase 35–41 loop is inserted between the barstar binding helix\\nand the large helix of barstar in the loosely bound state, while it is\\ninserted between the binding helix and binding loop in the tightly\\nbound state. The loosely bound state is stabilized by an interaction\\nbetween barnase Lys 27 (whose functional role is to stabilize the\\nphosphate group in the RNA substrate) and barstar Glu 80, while\\nthe tightly bound state is stabilized by a salt bridge between\\nbarnase Arg 83 and barstar Asp 39. Due to the latter salt bridge,\\nbarnase loop 77–85 is stabilized in the tightly bound state, while it\\nis very flexible in the loosely bound state. Overall, the loosely bound\\nstate is more flexible than the tightly bound state (compare Fig. 4a,b).\\n\\nThese native substate dynamics may be observable experimen-\\ntally, for example, by fluorescence correlation microscopy or by\\n\\na b\\n\\nc d\\n\\ne\\n\\nArg 59\\nAsp 35\\n\\nTrp 38 Glu 76\\n\\nGly 43\\nTrp 44\\n\\nAla 37\\nSer 38\\n\\n45° Late\\nintermediate 2\\n\\n \\n\\nPre-bound\\n\\nLoosely\\nbound\\n\\n45°\\n\\nFigure 3 | Late steps in binding from late intermediates to the loosely\\nbound state. a,b, Two sets of structures of late intermediate state 2, shown\\nin 45° counterclockwise/clockwise rotation compared to Fig. 2d. c,d, Two\\nsets of structures of the pre-bound state. e, Loosely bound state. Grey clouds\\nshow overlay of approximately 20 barstar conformations, sampled randomly\\nfrom the corresponding metastable states. Two binding pathways exist, in\\nwhich the binding domain of barstar interacts with barnase at its ‘left’ loop\\naround Ala 37 and Ser 38 (a,c), or at its ‘right’ loop (b,d), before docking into\\nthe bound state (e). In the major pathway (b→ d→ e, approximately 95% of\\nthe flux), barnase Arg 59 ‘anchors’ into barstar via Asp 35, Glu 76 (polar)\\nand Trp 38 (hydrophobic). Barstar can swing around barnase with the\\nanchor staying attached until final docking into the bound pose.\\n\\nARTICLES NATURE CHEMISTRY DOI: 10.1038/NCHEM.2785\\n\\nNATURE CHEMISTRY | ADVANCE ONLINE PUBLICATION | www.nature.com/naturechemistry4\\n\\n© 2017 Macmillan Publishers Limited, part of Springer Nature. All rights reserved.\\n\\nhttp://dx.doi.org/10.1038/nchem.2785\\nhttp://www.nature.com/naturechemistry\\n\\n\\nfast temperature jump experiments that have been employed to\\nprobe loose–tight binding transitions37. In particular, barstar Trp\\n44 changes solvent accessibility during the native substate transition,\\nrendering intrinsic tryptophan fluorescence a promising probe.\\n\\nAssociation kinetics of wild-type and a faster-associating mutant.\\nThe very large association rate of barnase–barstar can be further\\nincreased by specific mutations. To understand the microscopic\\nmechanisms of such perturbations, we conducted additional\\n0.3 milliseconds of MD simulations using the barnase mutant Glu\\n73 to Ala (E73A, see Methods), which associates a factor of 1.5–4\\nfaster than the wild type23.\\n\\nAs both the wild-type and the E73A HMM suffered from a\\nstatistical bottleneck at the unbinding transition, the uncertain\\ntransition rate from the loosely bound to pre-bound state was set\\nsuch that the models produce the experimentally known binding\\nfree energy. The resulting consensus models still make independent\\npredictions of the binding rate and mechanism, while reproducing\\nexperimental thermodynamics.\\n\\nTheE73Amodel predicts anassociation rate of 4 to 4.7 × 108 s–1M–1,\\nwhich correctly reflects an increase by a factor of 1.6–1.8 (ref. 23),\\nand can be explained by the increased electrostatic complementarity.\\nE73 does not participate in the binding interface, and indeed the\\nwild-type native complex structure is still predicted to be the most\\n\\n95%5%RMSD 0.3 nm\\nLifetime 5 ms\\n\\nTightly bound\\nRMSD 0.21 nm\\nLifetime 130 ms\\n\\n1BRS\\ncrystal structure \\n\\na b c\\n\\nArg 83\\nAsp 39Lys 27\\n\\nGlu 80\\n\\nTrp 44\\nTrp 44\\n\\nLoosely bound\\n\\nFigure 4 | Native substate conformational dynamics. Native substates seen from ‘left’, that is, in a 90° clockwise rotation around the image vertical with\\nrespect to Fig. 2d. a, Loosely bound state with lower population and higher structural flexibility. b, Tightly bound state with higher population and small\\nstructural flexibility. c, Crystal structure 1BRS shown in the structural density of the tightly bound simulation state, for comparison with B. Grey cloud is\\nan overlay of approximately 20 barstar conformations, sampled randomly from the corresponding metastable states.\\n\\nWild type E73A\\n\\na b\\n\\nC\\nom\\n\\nm\\nitt\\n\\nor\\n\\n1\\n\\n0\\n\\n10–1 100 101 102 103 104\\n\\n0.5\\n\\nc d\\n\\nLoosely\\nbound\\n\\nPrebound\\n\\nLate intermediates\\n\\nEarly intermediates\\n\\nDissociated\\n\\nMisbound\\n\\nTightly\\nbound\\n\\n1.0\\n0.8\\n0.6\\n0.4\\n0.2\\n0.0\\n\\nP\\nro\\n\\nba\\nbi\\n\\nlit\\ny\\n\\nTime (μs)\\n10–1 100 101 102 103 104\\n\\n1.0\\n0.8\\n0.6\\n0.4\\n0.2\\n0.0\\n\\nP\\nro\\n\\nba\\nbi\\n\\nlit\\ny\\n\\nTime (μs)\\n\\nFigure 5 | Comparison of kinetics between wild type and E73A mutant. a,b, Kinetic networks of the consensus models that are consistent with MD\\nsimulation data and the experimental binding free energy. Network plots as in Fig. 2d, but the vertical position of states indicates the binding probability\\n(committor) and disc area is proportional to fourth root of state population. The E73A mutant shows a deeper funnel that leads to an increased association\\nrate. c,d, Kinetic evolution of probabilities when starting from dissociated state, computed from the HMM (see Supplementary Methods 4). Despite a higher\\nassociation rate, the E73A mutant takes longer to bind to the native state, as the increased association rate is due to rapid trapping in a mis-bound state that\\nis committed to further native binding. Colour scheme as in Fig. 2d.\\n\\nNATURE CHEMISTRY DOI: 10.1038/NCHEM.2785 ARTICLES\\n\\nNATURE CHEMISTRY | ADVANCE ONLINE PUBLICATION | www.nature.com/naturechemistry 5\\n\\n© 2017 Macmillan Publishers Limited, part of Springer Nature. All rights reserved.\\n\\nhttp://dx.doi.org/10.1038/nchem.2785\\nhttp://www.nature.com/naturechemistry\\n\\n\\nstable state in the mutant (Fig. 5b). Figure 5a,b shows kinetic\\nnetworks of the wild type and E73A mutant, similar to Fig. 2d but\\nwith the y-positions indicating the probability of binding (committor)\\nof each state. Both systems are characterized by binding funnels,\\nthat are broad on top but narrow down rapidly as the complex pro-\\nceeds through late intermediates towards binding. The E73A funnel\\nis deeper in the sense that early intermediates have larger committor\\nprobabilities, that is, it is more difficult to escape the complex, once\\nit is associated. E73A has a reduced stability of the bound state23.\\nThe state that experiences the most profound increase of probability\\nis the mis-bound state. The time to form the native bound complex\\nis much longer in E73A compared to the wild type, as the system\\ngets trapped in the mis-bound state for 100s of microseconds\\n(Fig. 5d). An increase of the association rate does therefore not\\nnecessarily imply more efficient binding in a functional sense.\\n\\nBased on the consensus models the experimental association rate\\nis simulated as a function of prot',\n",
       " 'n concentration. At micromolar\\nconcentrations commonly used in experiments, the association rates\\nare linear in the concentration, that is, binding is limited by the\\nbimolecular encounter (Supplementary Fig. 5). In this condition\\nthe dominant energy barrier is characterized by balancing the entro-\\npic cost of bringing two molecules together at low concentration and\\nthe onset of the electrostatic interactions. At concentrations of 10s to\\n100s of μM, this barrier disappears, the transition state of binding\\nbecomes rate-limiting4, and we observe a concentration-independent\\nrate constant of 18,000 s–1 for the wild type and 1,500 s–1 for the E73A\\nmutant due to the transition state of binding (Supplementary Fig. 5).\\n\\nAfter an initial contact has been formed, the complex will\\ntypically move through a sequence of other non-native inter-\\nmediates until it dissociates or proceeds to native binding with\\nabout 50% probability each (Fig. 5a,b). The major bottleneck in\\ndissociation from the native state is the separation of the loosely\\nbound complex into the pre-bound, mis-bound and intermediate\\nstates (Supplementary Fig. 6), from where the complex can dissociate\\nor perform geminate rebinding (Fig. 2g).\\n\\nConclusion\\nBy combining recent breakthroughs in simulation and analysis tech-\\nnology, we have demonstrated that a quantitatively accurate micro-\\nscopic model of prot',\n",
       " 'n–prot',\n",
       " 'n association can be obtained using\\nunbiased all-atom MD simulations and Markov modelling. This\\nmodel reveals kinetic and mechanistic details of the association mech-\\nanism and has enabled the discovery of previously unknown substates.\\n\\nThe presented combination of simulation and analysis techniques\\nare applicable to any pair of macromolecules. While the available\\ncomputational power limits the size of the systems b',\n",
       " 'ng studied to\\ncurrently a few hundred amino acids per prot',\n",
       " 'n, biological timescales\\ncan be reached that are far beyond direct MD simulation times. This\\nopens the door to study how the parts of macromolecular assemblies\\nform, how viral and host prot',\n",
       " 'ns interact and how prot',\n",
       " 'n–DNA\\ninteractions depend on DNA sequence and epigenetic modifications.\\n\\nRather than a black-box model, the present approach provides\\npopulations for every sampled atomistic structure and transition\\nrates between them. We have demonstrated that combining such an\\nextensive structural ensemble with an appropriate rew',\n",
       " 'ghting\\nmethod, the thermodynamics and kinetics of similar prot',\n",
       " 'n sequences\\ncan bepredicted. In the future, prot',\n",
       " 'n sequences andpost-translational\\nmodifications, or sequence and epigenetic modifications of interacting\\nnucl',\n",
       " 'c acids can be designed to stabilize desirable intermediates,\\nchange the kinetics or control the mechanism, with far-reaching\\nconsequences for pharmaceutics, biotechnology andmaterial sciences.\\n\\nMethods\\nMD simulation setup. All-atom models of barnase wild type and barstar pseudo\\nwild type with C40,82A double mutation24 were constructed starting from crystal\\nstructure 1BRS (Prot',\n",
       " 'n Data Bank ID)38 (chain B for barnase and chain F\\n\\nsuperimposed to chain E for barstar) and immersed in a water box with ∼10 nm\\nedge length at 50 mM salt concentration (approximately 90,000 atoms). Protonation\\nstates were chosen using AmberTools39. His102 was in un-protonated (U-tautomer)\\nform that is dominant at pH 7. The Amber ff99SB force field40, the TIP3P water\\nmodel41, and the ACEMD program42 were used for simulations. The system was\\nminimized and equilibrated first in the NpT ensemble; box size was fixed and\\nsimulations were conducted in NVT at 300 K.\\n\\nWild-type GPUgrid simulations. 1,892 independent MD simulations starting from\\ndissociated structures (generated by random translation and rotation of barstar from\\nthe complex, excluding overlapping structures) were distributed on the GPUgrid\\ncomputing network43. Trajectories were up to 1 english μs long (distribution shown\\nin Supplementary Fig. 1), and the aggregate simulation time was 1.7 milliseconds. 74\\nof these simulations found the native barnase–barstar complex.\\n\\nWild-type adaptive MD simulations. Multiple rounds of short MD trajectories are\\nrun and a selection strategy based on available trajectory statistics is employed to select\\nthe starting coordinates for the next round of trajectories25–27 (Supplementary Methods\\n1 and Fig. 1b for details). Different combinations of selection strategies were tested.\\nAssociation and binding into the native complex succeeded with various strategies,\\nwhereas dissociation only succeeded by combining regular spatial clustering with a\\nminimum RMSD metric14, and selecting new MD frames with a probability anti-\\nproportional to the number of frames in each cluster. To confirm that the observed\\ndissociation is robust with respect to the force field used, we reproduced adaptive\\ndissociation using CHARMM36 (ref. 44) following the setup reported previously10.\\n\\nAdaptive trajectories were 50 or 100 ns long when generated. Selected adaptive\\nruns (Supplementary Methods 2) were extended to 200 ns for the analysis. 797\\nadaptive MD simulations starting from the natively bound complex, and 756 from\\ndissociated and non-native associated states were used, comprising an aggregate\\nsimulation time of 0.3 milliseconds.\\n\\nE73A mutant simulations. 25 MD frames were sampled from each of the 15\\nwild-type HMM states (Fig. 2d). For each structure, Glu 73 was mutated to Ala,\\nstructures were rebuild using HTMD45, and four copies were run on GPUgrid,\\nresulting in 1,500 simulations of 200 ns length, summing up to 300 µs.\\n\\nFeature selection and discretization. Following feature selection and discretization\\nwas carried out for the wild-type simulations using PyEMMA (http://pyemma.org)46\\n\\nand HTMD (http://htmd.org)45. 4,094 ‘relevant’ residue–residue contacts between\\nprot',\n",
       " 'ns, defined by nearest-n',\n",
       " 'ghbour heavy-atom distances <0.4 nm, were\\nmeasured. A contact was considered relevant when the longest time it had been\\nformed continuously exceeded 10 ns. The following features were used for\\nanalysis: (i) the minimum heavy-atom RMSD between each frame and the native\\nbarnase/barstar complex, (ii) the minimum distance between the two prot',\n",
       " 'ns,\\nand (iii) the seven slowest linear combinations of the 4,094 relevant contacts,\\ncomputed variationally50 by the TICA method47,48 (lagtime 100 ns). This space was\\ndiscretized using k-means clustering (k = 2,000).\\n\\nE73A mutant simulation trajectories were mapped on the same features and\\nassigned to the same cluster discretization.\\n\\nHidden Markov model. Only simulations conducted with Amber ff99SB were\\nused to estimate the HMM. Discrete HMMs as described previously30 are built on\\nthe k-means clustering and find a coarse-grained kinetic model by optimally\\ngrouping fast mixing states into metastable sets, and estimating the transition rates\\nbetween them. See Supplementary Methods 4 and Supplementary Fig. 2 for\\nself-consistent validation of the model.\\n\\nHMMs with 15 metastable states were estimated from the discretized simulation\\ndata using the msm package of PyEMMA. The HMM with lag time 110 ns was\\nselected for analysis because the free energies and relaxation timescales were\\napproximately independent of the lag at this point (Supplementary Fig. 2). For the\\nE73A mutant trajectories, we also estimated an HMM at lag 110 ns. Uncertainties\\nwere computed using the reversible Bayesian estimator described previously49.\\nHMM analysis was conducted using the methods provided by PyEMMA’s HMM\\nclass, and some functionalities of the msmtools package (Supplementary Methods 4).\\n\\nProt',\n",
       " 'n association model. While the HMM automatically identified a dissociated\\nstate (lower green state in Fig. 2d), calculation of association rates and binding free\\nenergies comparable to experiment requires the definition of a bulk dissociated\\nstate with concentration-dependent association rate. Following an idea by\\nNorthrup et al.31, we attached a bulk dissociated state to the HMM as follows:\\n\\nbulk dissociated state ⇌\\n4πDb\\n\\nku\\nHMM dissociated state⇌\\n\\nkij\\n\\nkji\\nother HMM stateswhere kij\\n\\ndenotes the transition rates between HMM states. The bimolecular association rate\\n4πDb is the Smoluchowski rate of hitting a sphere of radius b = 4 nm around the\\norigin that is the effective size of the HMM dissociated state. The diffusion constant\\nD = 28.5 nm2 μs–1 has been fitted from the MD simulations. The bulk dissociation\\nrate is given by ku = ( f∞\\n\\n–1 – 1) Σjkji, where f∞ is the probability to dissociate to the bulk\\nwhen starting from b, which was computed by combining trajectory statistics and a\\nBrownian dynamics model (Supplementary Methods 9).\\n\\nARTICLES NATURE CHEMISTRY DOI: 10.1038/NCHEM.2785\\n\\nNATURE CHEMISTRY | ADVANCE ONLINE PUBLICATION | www.nature.com/naturechemistry6\\n\\n© 2017 Macmillan Publishers Limited, part of Springer Nature. All rights reserved.\\n\\nhttp://pyemma.org\\nhttp://htmd.org\\nhttp://dx.doi.org/10.1038/nchem.2785\\nhttp://www.nature.com/naturechemistry\\n\\n\\nCode availability. PyEMMA (http://pyemma.org) and HTMD (http://htmd.org) are\\navailable free of charge. See http://acellera.com for the availability of ACEMD.\\n\\nRec',\n",
       " 'ved 19 October 2016; accepted 21 April 2017;\\npublished online 5 June 2017\\n\\nReferences\\n1. Scott, D. E., Bayly, A. R., Abell, C. & Skidmore, J. Small molecules, big targets:\\n\\ndrug discovery faces the prot',\n",
       " 'n–prot',\n",
       " 'n interaction challenge. Nat. Rev. Drug.\\nDiscov. 15, 533–550 (2016).\\n\\n2. Doench, J. G. et al. Rational design of highly active sgrnas for crispr–cas9-\\nmediated gene inactivation. Nat. Biotechnol. 32, 1262–1267 (2014).\\n\\n3. King, N. P. et al. Computational design of self-assembling prot',\n",
       " 'n nanomaterials\\nwith atomic level accuracy. Science 336, 1171–1174 (2012).\\n\\n4. Schr',\n",
       " 'ber, G., Haran, G. & Zhou, H.-X. Fundamental aspects of prot',\n",
       " 'n–prot',\n",
       " 'n\\nassociation kinetics. Chem. Rev. 109, 839–860 (2009).\\n\\n5. Tang, C., Iwahara, J. & Clore, G. M. Visualization of transient encounter\\ncomplexes in prot',\n",
       " 'n–prot',\n",
       " 'n association. Nature 444, 383–386 (2006).\\n\\n6. Gabdoulline, R. R. & Wade, R. C. Simulation of the diffusional association of\\nbarnase and barstar. Biophys. J. 72, 1917–1929 (1997).\\n\\n7. Spaar, A., Dammer, C., Gabdoulline, R. R., Wade, R. C. & Helms, V. Diffusional\\nencounter of barnase and barstar. Biophys. J. 90, 1913–1924 (2006).\\n\\n8. Levy, Y., Wolynes, P. G. & Onuchic, J. N. Prot',\n",
       " 'n topology determines binding\\nmechanism. Proc. Natl Acad. Sci. USA 101, 511–516 (2004).\\n\\n9. Schluttig, J., Alamanova, D., Helms, V. & Schwarz, U. S. Dynamics of prot',\n",
       " 'n–\\nprot',\n",
       " 'n encounter: a langevin equation approach with reaction patches. J. Chem.\\nPhys. 129, 155106 (2008).\\n\\n10. Gumbart, J. C., Roux, B. & Chipot, C. Efficient determination of prot',\n",
       " 'n–prot',\n",
       " 'n\\nstandard binding free energies from first principles. J. Chem. Theory Comput. 9,\\n3789–3798 (2013).\\n\\n11. Barducci, A., Bonomi, M., Prakash, M. K. & Parrinello, M. Free-energy\\nlandscape of prot',\n",
       " 'n oligomerization from atomistic simulations. Proc. Natl\\nAcad. Sci. USA 110, E4708–E4713 (2013).\\n\\n12. Tiwary, P. & Parrinello, M. From metadynamics to dynamics. Phys. Rev. Lett.\\n111, 230602 (2013).\\n\\n13. Wu, H., Paul, F., Wehmeyer, C. & Noé, F. Multiensemble Markov models of\\nmolecular thermodynamics and kinetics. Proc. Natl Acad. Sci. USA 113,\\nE3221–E3230 (2016).\\n\\n14. Prinz, J.-H. et al. Markov models of molecular kinetics: generation and\\nvalidation. J. Chem. Phys. 134, 174105 (2011).\\n\\n15. Lindorff-Larsen, K., Piana, S., Dror, R. O. & Shaw, D. E. How fast-folding\\nprot',\n",
       " 'ns fold. Science 334, 517–520 (2011).\\n\\n16. Buch, I., Giorgino, T. & De Fabritiis, G. Complete reconstruction of an enzyme-\\ninhibitor binding process by molecular dynamics simulations. Proc. Natl Acad.\\nSci. USA 108, 10184–10189 (2011).\\n\\n17. Kohlhoff, K. J. et al. Cloud-based simulations on Google Exacycle reveal ligand\\nmodulation of GPCR activation pathways. Nat. Chem. 6, 15–21 (2014).\\n\\n18. Plattner, N. & Noé, F. Prot',\n",
       " 'n conformational plasticity and complex ligand\\nbinding kinetics explored by atomistic simulations and Markov models. Nat.\\nCommun. 6, 7653 (2015).\\n\\n19. Silva, D.-A., Bowman, G. R., Sosa-P',\n",
       " 'nado, A. & Huang, X. A role for both\\nconformational selection and induced fit in ligand binding by the Lao prot',\n",
       " 'n.\\nPLoS Comput. Biol. 7, e1002054 (2011).\\n\\n20. Piana, S., Lindorff-Larsen, K. & Shaw, D. E. Atomistic description of the folding\\nof a dimeric prot',\n",
       " 'n. J. Phys. Chem. B 117, 12935–12942 (2013).\\n\\n21. Ahmad, M., Gu, W., Geyer, T. & Helms, V. Adhesive water networks facilitate\\nbinding of prot',\n",
       " 'n interfaces. Nat. Commun. 2, 261 (2011).\\n\\n22. Schr',\n",
       " 'ber, G. & Fersht, A. R. Rapid, electrostatically assisted association of\\nprot',\n",
       " 'ns. Nat. Struct. Biol. 3, 427–431 (1996).\\n\\n23. Schr',\n",
       " 'ber, G. & Fersht, A. R. Interaction of barnase with its polypeptide\\ninhibitor barstar studied by prot',\n",
       " 'n engineering. Biochemistry 32,\\n5145–5150 (1993).\\n\\n24. Hartley, R. W. Directed mutagenesis and barnase–barstar recognition.\\nBiochemistry 32, 5978–5984 (1993).\\n\\n25. Doerr, S. & Fabritiis, G. D. On-the-fly learning and sampling of ligand binding\\nby high-throughput molecular simulations. J. Chem. Theory Comput. 10,\\n2064–2069 (2014).\\n\\n26. Bowman, G. R., Ensign, D. L. & Pande, V. S. Enhanced modeling via network\\ntheory: adaptive sampling of Markov state models. J. Chem. Theory Comput. 6,\\n787–794 (2010).\\n\\n27. Preto, J. & Clementi, C. Fast recovery of free energy landscapes via\\ndiffusion-map-directed molecular dynamics. Phys. Chem. Chem. Phys. 16,\\n19181–19191 (2014).\\n\\n28. Bowman, G. R., Pande, V. S. & Noé, F. (eds.) An Introduction to Markov State\\nModels and Th',\n",
       " 'r Application to Long Timescale Molecular Simulation (Vol. 797\\nof Advances in Experimental Medicine and Biology, Springer, 2014).\\n\\n29. Sarich, M. & Schütte, C. Metastability and Markov State Models in Molecular\\nDynamics (Courant Lecture Notes, American Mathematical Society, 2013).\\n\\n30. Noé, F., Wu, H., Prinz, J.-H. & Plattner, N. Projected and hidden Markov models\\nfor calculating kinetics and metastable states of complex molecules. J. Chem.\\nPhys. 139, 184114 (2013).\\n\\n31. Northrup, S. H., Allison, S. & McCammon, J. Brownian dynamics of diffusion-\\ninfluenced bimolecular reactions. J. Chem. Phys. 80, 1517–1524 (1984).\\n\\n32. Schr',\n",
       " 'ber, G. & Fersht, A. R. Energetics of prot',\n",
       " 'n–prot',\n",
       " 'n interactions: analysis\\nof the barnase–barstar interface by single mutations and double mutant cycles.\\nJ. Mol. Biol. 248, 478–486 (1995).\\n\\n33. Matysiak, S. & Clementi, C. Optimal combination of theory and experiment for\\nthe characterization of the prot',\n",
       " 'n folding landscape of S6: how far can a\\nminimalist model go? J. Mol. Biol. 343, 235–248 (2004).\\n\\n34. Frisch, C., Fersht, A. R. & Schr',\n",
       " 'ber, G. Experimental assignment of the structure\\nof the transition state for the association of barnase and barstar. J. Mol. Biol. 308,\\n69–77 (2001).\\n\\n35. Harel, M., Cohen, M. & Schr',\n",
       " 'ber, G. On the dynamic nature of the transition\\nstate for prot',\n",
       " 'n–prot',\n",
       " 'n association as determined by double-mutant cycle\\nanalysis and simulation. J. Mol. Biol. 371, 180–196 (2007).\\n\\n36. Chung, H. S., Louis, J. M. & Eaton,W. A. Single-molecule fluorescence experiments\\ndetermine prot',\n",
       " 'n folding transition path times. Science 335, 981–984 (2012).\\n\\n37. Anunciado, D., Dhar, A., Gruebele, M. & Baranger, A. M. Multistep kinetics of\\nthe U1A–SL2 RNA complex dissociation. J. Mol. Biol. 408, 896–908 (2011).\\n\\n38. Buckle, A. M., Schr',\n",
       " 'ber, G. & Fersht, A. R. Prot',\n",
       " 'n–prot',\n",
       " 'n recognition: crystal\\nstructural analysis of a barnase–barstar complex at 2.0-Å resolution.\\nBiochemistry 33, 8878–8889 (1994).\\n\\n39. Case, D. A. et al. The Amber biomolecular simulation programs. J. Comput.\\nChem. 26, 1668–1688 (2005).\\n\\n40. Hornak, V. et al. Comparison of multiple Amber force fields and development of\\nimproved prot',\n",
       " 'n backbone parameters. Prot',\n",
       " 'ns 65, 712–725 (2006).\\n\\n41. Jorgensen, W. L., Chandrasekhar, J., Madura, J. D., Impey, R. W. & Kl',\n",
       " 'n, M. L.\\nComparison of simple potential functions for simulating liquid water. J. Chem.\\nPhys. 79, 926–935 (1983).\\n\\n42. Harvey, M. J., Giupponi, G. & De Fabritiis, G. ACEMD: Accelerated molecular\\ndynamics simulations in the microseconds timescale. J. Chem. Theory Comp. 5,\\n1632–1639 (2009).\\n\\n43. Buch, I., Harvey, M. J., Giorgino, T., Anderson, D. P. & De Fabritiis, G.\\nHigh-throughput all-atom molecular dynamics simulations using distributed\\ncomputing. J. Chem. Inf. Model. 50, 397–403 (2010).\\n\\n44. Best, R. B. et al. Optimization of the additive CHARMM all-atom prot',\n",
       " 'n force\\nfield targeting improved sampling of the backbone ϕ, ψ and side-chain χ1 and χ2\\ndihedral angles. J. Chem. Theo. Comp. 8, 3257–3273 (2012).\\n\\n45. Doerr, S., Harvey, M. J., Noé, F. & Fabritiis, G. D. HTMD: high-throughput\\nmolecular dynamics for molecular discovery. J. Chem. Theory Comput. 12,\\n1845–1852 (2016).\\n\\n46. Scherer, M. K. et al. PyEMMA 2: a software package for estimation, validation\\nand analysis of Markov models. J. Chem. Theory Comput. 11, 5525–5542 (2015).\\n\\n47. Perez-Hernandez, G., Paul, F., Giogino, T., De Fabritiis, G. & Noé, F.\\nIdentification of slow molecular order parameters for Markov model\\nconstruction. J. Chem. Phys. 139, 015102 (2013).\\n\\n48. Molgedey, L. & Schuster, H. G. Separation of a mixture of independent signals\\nusing time delayed correlations. Phys. Rev. Lett. 72, 3634–3637 (1994).\\n\\n49. Trendelkamp-Schroer, B., Wu, H., Paul, F. & Noé, F. Estimation and uncertainty\\nof reversible Markov models. J. Chem. Phys. 143, 174101 (2015).\\n\\n50. Noé, F. & Nüske, F. A variational approach to modeling slow processes in\\nstochastic dynamical systems. Multiscale Model. Simul. 11, 635–655 (2013).\\n\\nAcknowledgements\\nWe are grateful to the following researchers for inspiring discussions and extensive\\nfeedback on the manuscript: J. Clark, C. Clementi, O. Daumke, W. A. Eaton, M. Gruebele,\\nK. Lindorff-Larsen, S. Olsson, F. Paul, B. Roux, U. Schwarz, S. Sukenik, R. Wade and\\nT.W',\n",
       " 'kl. Funding is acknowledged from the European Commission (ERC StG pcCells and\\nPRACE project 2014102337 to F.N.), Deutsche Forschungsgem',\n",
       " 'nschaft (NO 825/3-1, SFB\\n958/A4, SFB 740/D7 to F.N.), Einst',\n",
       " 'n Foundation Berlin (project SoOPic to N.P.),\\nMINECO (BIO2014-53095-P) and FEDER (to G.D.F.), Acellera Ltd. (to S.D.). We thank\\nthe volunteers of GPUGRID for donating computing time for simulations.\\n\\nAuthor contributions\\nN.P., S.D., G.D.F and F.N. designed research and developed/implemented software.\\nN.P. and S.D. conducted simulations. N.P., S.D. and F.N. analysed simulations. F.N.\\ndeveloped methods. N.P. and F.N. wrote the paper.\\n\\nAdditional information\\nSupplementary information is available in the online version of the paper. Reprints and\\npermissions information is available online at www.nature.com/reprints. Publisher’s note:\\nSpringer Nature remains neutral with regard to jurisdictional claims in published maps and\\ninstitutional affiliations. Correspondence and requests for materials should be addressed\\nto N.P. or F.N.\\n\\nCompeting financial interests\\nThe authors declare no competing financial interests.\\n\\nNATURE CHEMISTRY DOI: 10.1038/NCHEM.2785 ARTICLES\\n\\nNATURE CHEMISTRY | ADVANCE ONLINE PUBLICATION | www.nature.com/naturechemistry 7\\n\\n© 2017 Macmillan Publishers Limited, part of Springer Nature. All rights reserved.\\n\\nhttp://pyemma.org\\nhttp://htmd.org\\nhttp://acellera.com\\nhttp://dx.doi.org/10.1038/nchem.2785\\nhttp://www.nature.com/reprints\\nhttp://dx.doi.org/10.1038/nchem.2785\\nhttp://www.nature.com/naturechemistry\\n\\n\\tComplete prot',\n",
       " 'n–prot',\n",
       " 'n association kinetics in atomic detail revealed by molecular dynamics simulations and Markov modelling\\n\\tResults and discussion\\n\\tAll-atom simulation of reversible prot',\n",
       " 'n–prot',\n",
       " 'n association\\n\\tEquilibrium kinetics model and validation\\n\\tOverall binding mechanism\\n\\tNature of the transition state of binding\\n\\tConformational change in the complex\\n\\tAssociation kinetics of wild-type and a faster-associating mutant\\n\\n\\tConclusion\\n\\tMethods\\n\\tMD simulation setup\\n\\tWild-type GPUgrid simulations\\n\\tWild-type adaptive MD simulations\\n\\tE73A mutant simulations\\n\\tFeature selection and discretization\\n\\tHidden Markov model\\n\\tProt',\n",
       " 'n association model\\n\\tCode availability\\n\\n\\tFigure 1  Simulation of prot',\n",
       " 'n–prot',\n",
       " 'n association and dissociation.\\n\\tFigure 2  Kinetic model of barnase–barstar association.\\n\\tFigure 3  Late steps in binding from late intermediates to the loosely bound state.\\n\\tFigure 4  Native substate conformational dynamics.\\n\\tFigure 5  Comparison of kinetics between wild type and E73A mutant.\\n\\tReferences\\n\\tAcknowledgements\\n\\tAuthor contributions\\n\\tAdditional information\\n\\tCompeting financial interests\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n<<\\n  /ASCII85EncodePages false\\n  /AllowTransparency false\\n  /AutoPositionEPSFiles true\\n  /AutoRotatePages /None\\n  /Binding /Left\\n  /CalGrayProfile ()\\n  /CalRGBProfile (sRGB IEC61966-2.1)\\n  /CalCMYKProfile (U.S. Web Coated \\\\050SWOP\\\\051 v2)\\n  /sRGBProfile (sRGB IEC61966-2.1)\\n  /CannotEmbedFontPolicy /Error\\n  /CompatibilityLevel 1.3\\n  /CompressObjects /Off\\n  /CompressPages true\\n  /ConvertImagesToIndexed true\\n  /PassThroughJPEGImages true\\n  /CreateJobTicket false\\n  /DefaultRenderingIntent /Default\\n  /DetectBlends true\\n  /DetectCurves 0.1000\\n  /ColorConversionStrategy /LeaveColorUnchanged\\n  /DoThumbnails false\\n  /EmbedAllFonts true\\n  /EmbedOpenType false\\n  /ParseICCProfilesInComments true\\n  /EmbedJobOptions true\\n  /DSCReportingLevel 0\\n  /EmitDSCWarnings false\\n  /EndPage -1\\n  /ImageMemory 1048576\\n  /LockDistillerParams true\\n  /MaxSubsetPct 100\\n  /Optimize true\\n  /OPM 1\\n  /ParseDSCComments true\\n  /ParseDSCCommentsForDocInfo true\\n  /PreserveCopyPage false\\n  /PreserveDICMYKValues true\\n  /PreserveEPSInfo true\\n  /PreserveFlatness true\\n  /PreserveHalftoneInfo false\\n  /PreserveOPIComments false\\n  /PreserveOverprintSettings true\\n  /StartPage 1\\n  /SubsetFonts true\\n  /TransferFunctionInfo /Apply\\n  /UCRandBGInfo /Remove\\n  /UsePrologue true\\n  /ColorSettingsFile (None)\\n  /AlwaysEmbed [ true\\n  ]\\n  /NeverEmbed [ true\\n  ]\\n  /AntiAliasColorImages false\\n  /CropColorImages true\\n  /ColorImageMinResolution 150\\n  /ColorImageMinResolutionPolicy /OK\\n  /DownsampleColorImages true\\n  /ColorImageDownsampleType /Bicubic\\n  /ColorImageResolution 450\\n  /ColorImageDepth -1\\n  /ColorImageMinDownsampleDepth 1\\n  /ColorImageDownsampleThreshold 1.00000\\n  /EncodeColorImages true\\n  /ColorImageFilter /DCTEncode\\n  /AutoFilterColorImages true\\n  /ColorImageAutoFilterStrategy /JPEG\\n  /ColorACSImageDict <<\\n    /QFactor 0.15\\n    /HSamples [1 1 1 1] /VSamples [1 1 1 1]\\n  >>\\n  /ColorImageDict <<\\n    /QFactor 0.15\\n    /HSamples [1 1 1 1] /VSamples [1 1 1 1]\\n  >>\\n  /JPEG2000ColorACSImageDict <<\\n    /TileWidth 256\\n    /TileH',\n",
       " 'ght 256\\n    /Quality 30\\n  >>\\n  /JPEG2000ColorImageDict <<\\n    /TileWidth 256\\n    /TileH',\n",
       " 'ght 256\\n    /Quality 30\\n  >>\\n  /AntiAliasGrayImages false\\n  /CropGrayImages true\\n  /GrayImageMinResolution 150\\n  /GrayImageMinResolutionPolicy /OK\\n  /DownsampleGrayImages true\\n  /GrayImageDownsampleType /Bicubic\\n  /GrayImageResolution 450\\n  /GrayImageDepth -1\\n  /GrayImageMinDownsampleDepth 2\\n  /GrayImageDownsampleThreshold 1.00000\\n  /EncodeGrayImages true\\n  /GrayImageFilter /DCTEncode\\n  /AutoFilterGrayImages true\\n  /GrayImageAutoFilterStrategy /JPEG\\n  /GrayACSImageDict <<\\n    /QFactor 0.15\\n    /HSamples [1 1 1 1] /VSamples [1 1 1 1]\\n  >>\\n  /GrayImageDict <<\\n    /QFactor 0.15\\n    /HSamples [1 1 1 1] /VSamples [1 1 1 1]\\n  >>\\n  /JPEG2000GrayACSImageDict <<\\n    /TileWidth 256\\n    /TileH',\n",
       " 'ght 256\\n    /Quality 30\\n  >>\\n  /JPEG2000GrayImageDict <<\\n    /TileWidth 256\\n    /TileH',\n",
       " 'ght 256\\n    /Quality 30\\n  >>\\n  /AntiAliasMonoImages false\\n  /CropMonoImages true\\n  /MonoImageMinResolution 1200\\n  /MonoImageMinResolutionPolicy /OK\\n  /DownsampleMonoImages true\\n  /MonoImageDownsampleType /Bicubic\\n  /MonoImageResolution 2400\\n  /MonoImageDepth -1\\n  /MonoImageDownsampleThreshold 1.00000\\n  /EncodeMonoImages true\\n  /MonoImageFilter /CCITTFaxEncode\\n  /MonoImageDict <<\\n    /K -1\\n  >>\\n  /AllowPSXObjects false\\n  /CheckCompliance [\\n    /None\\n  ]\\n  /PDFX1aCheck true\\n  /PDFX3Check false\\n  /PDFXCompliantPDFOnly false\\n  /PDFXNoTrimBoxError false\\n  /PDFXTrimBoxToMediaBoxOffset [\\n    35.29000\\n    35.29000\\n    36.28000\\n    36.28000\\n  ]\\n  /PDFXSetBleedBoxToMediaBox false\\n  /PDFXBleedBoxToTrimBoxOffset [\\n    8.50000\\n    8.50000\\n    8.50000\\n    8.50000\\n  ]\\n  /PDFXOutputIntentProfile (OFCOM_PO_P1_F60)\\n  /PDFXOutputConditionIdentifier ()\\n  /PDFXOutputCondition (OFCOM_PO_P1_F60)\\n  /PDFXRegistryName ()\\n  /PDFXTrapped /False\\n\\n  /CreateJDFFile false\\n  /SyntheticBoldness 1.000000\\n  /Description <<\\n    /JPN <FEFF3053306e8a2d5b9a306f300130d330b830cd30b9658766f8306e8868793a304a3088307353705237306b90693057305f00200050004400460020658766f830924f5c62103059308b3068304d306b4f7f75283057307e305930023053306e8a2d5b9a30674f5c62103057305f00200050004400460020658766f8306f0020004100630072006f0062006100740020304a30883073002000520065006100640065007200200035002e003000204ee5964d30678868793a3067304d307e30593002>\\n    /DEU <FEFF00560065007200770065006e00640065006e0020005300690065002000640069006500730065002000450069006e007300740065006c006c0075006e00670065006e0020007a0075006d002000450072007300740065006c006c0065006e00200076006f006e0020005000440046002d0044006f006b0075006d0065006e00740065006e002c00200075006d002000650069006e00650020007a0075007600650072006c00e40073007300690067006500200041006e007a006500690067006500200075006e00640020004100750073006700610062006500200076006f006e00200047006500730063006800e40066007400730064006f006b0075006d0065006e00740065006e0020007a0075002000650072007a00690065006c0065006e002e00200044006900650020005000440046002d0044006f006b0075006d0065006e007400650020006b00f6006e006e0065006e0020006d006900740020004100630072006f0062006100740020006f0064006500720020006d00690074002000640065006d002000520065006100640065007200200035002e003000200075006e00640020006800f600680065007200200067006500f600660066006e00650074002000770065007200640065006e002e>\\n    /FRA <FEFF004f007000740069006f006e00730020007000650072006d0065007400740061006e007400200064006500200063007200e900650072002000640065007300200064006f00630075006d0065006e007400730020005000440046002000700072006f00660065007300730069006f006e006e0065006c007300200066006900610062006c0065007300200070006f007500720020006c0061002000760069007300750061006c00690073006100740069006f006e0020006500740020006c00270069006d007000720065007300730069006f006e002e00200049006c002000650073007400200070006f0073007300690062006c0065002000640027006f00750076007200690072002000630065007300200064006f00630075006d0065006e007400730020005000440046002000640061006e00730020004100630072006f0062006100740020006500740020005200650061006400650072002c002000760065007200730069006f006e002000200035002e00300020006f007500200075006c007400e9007200690065007500720065002e>\\n    /PTB <FEFF005500740069006c0069007a006500200065007300740061007300200063006f006e00660069006700750072006100e700f5006500730020007000610072006100200063007200690061007200200064006f00630075006d0065006e0074006f0073002000500044004600200063006f006d00200075006d0061002000760069007300750061006c0069007a006100e700e3006f0020006500200069006d0070007200650073007300e3006f00200061006400650071007500610064006100730020007000610072006100200064006f00630075006d0065006e0074006f007300200063006f006d0065007200630069006100690073002e0020004f007300200064006f00630075006d0065006e0074006f0073002000500044004600200070006f00640065006d0020007300650072002000610062006500720074006f007300200063006f006d0020006f0020004100630072006f006200610074002c002000520065006100640065007200200035002e00300020006500200070006f00730074006500720069006f0072002e>\\n    /DAN <FEFF004200720075006700200064006900730073006500200069006e0064007300740069006c006c0069006e006700650072002000740069006c0020006100740020006f0070007200650074007400650020005000440046002d0064006f006b0075006d0065006e007400650072002c0020006400650072002000650072002000650067006e006500640065002000740069006c0020007000e5006c006900640065006c006900670020007600690073006e0069006e00670020006f00670020007500640073006b007200690076006e0069006e006700200061006600200066006f0072007200650074006e0069006e006700730064006f006b0075006d0065006e007400650072002e0020005000440046002d0064006f006b0075006d0065006e007400650072006e00650020006b0061006e002000e50062006e006500730020006d006500640020004100630072006f0062006100740020006f0067002000520065006100640065007200200035002e00300020006f00670020006e0079006500720065002e>\\n    /NLD <FEFF004700650062007200750069006b002000640065007a006500200069006e007300740065006c006c0069006e00670065006e0020006f006d0020005000440046002d0064006f00630075006d0065006e00740065006e0020007400650020006d0061006b0065006e00200064006900650020006700650073006300680069006b00740020007a0069006a006e0020006f006d0020007a0061006b0065006c0069006a006b006500200064006f00630075006d0065006e00740065006e00200062006500740072006f0075007700620061006100720020007700650065007200200074006500200067006500760065006e00200065006e0020006100660020007400650020006400720075006b006b0065006e002e0020004400650020005000440046002d0064006f00630075006d0065006e00740065006e0020006b0075006e006e0065006e00200077006f007200640065006e002000670065006f00700065006e00640020006d006500740020004100630072006f00620061007400200065006e002000520065006100640065007200200035002e003000200065006e00200068006f006700650072002e>\\n    /ESP <FEFF0055007300650020006500730074006100730020006f007000630069006f006e006500730020007000610072006100200063007200650061007200200064006f00630075006d0065006e0074006f0073002000500044004600200071007500650020007000650072006d006900740061006e002000760069007300750061006c0069007a006100720020006500200069006d007000720069006d0069007200200063006f007200720065006300740061006d0065006e0074006500200064006f00630075006d0065006e0074006f007300200065006d00700072006500730061007200690061006c00650073002e0020004c006f007300200064006f00630075006d0065006e0074006f00730020005000440046002000730065002000700075006500640065006e00200061006200720069007200200063006f006e0020004100630072006f00620061007400200079002000520065006100640065007200200035002e003000200079002000760065007200730069006f006e0065007300200070006f00730074006500720069006f007200650073002e>\\n    /SUO <FEFF004e00e4006900640065006e002000610073006500740075007300740065006e0020006100760075006c006c006100200076006f006900740020006c0075006f006400610020006a0061002000740075006c006f00730074006100610020005000440046002d0061007300690061006b00690072006a006f006a0061002c0020006a006f006900640065006e0020006500730069006b0061007400730065006c00750020006e00e400790074007400e400e40020006c0075006f00740065007400740061007600610073007400690020006c006f00700070007500740075006c006f006b00730065006e002e0020005000440046002d0061007300690061006b00690072006a0061007400200076006f0069006400610061006e0020006100760061007400610020004100630072006f006200610074002d0020006a0061002000520065006100640065007200200035002e00300020002d006f0068006a0065006c006d0061006c006c0061002000740061006900200075007500640065006d006d0061006c006c0061002000760065007200730069006f006c006c0061002e>\\n    /ITA <FEFF00550073006100720065002000710075006500730074006500200069006d0070006f007300740061007a0069006f006e00690020007000650072002000630072006500610072006500200064006f00630075006d0065006e007400690020005000440046002000610064006100740074006900200070006500720020006c00610020007300740061006d00700061002000650020006c0061002000760069007300750061006c0069007a007a0061007a0069006f006e006500200064006900200064006f00630075006d0065006e0074006900200061007a00690065006e00640061006c0069002e0020004900200064006f00630075006d0065006e00740069002000500044004600200070006f00730073006f006e006f0020006500730073006500720065002000610070006500720074006900200063006f006e0020004100630072006f00620061007400200065002000520065006100640065007200200035002e003000200065002000760065007200730069006f006e006900200073007500630063006500730073006900760065002e>\\n    /NOR <FEFF004200720075006b00200064006900730073006500200069006e006e007300740069006c006c0069006e00670065006e0065002000740069006c002000e50020006f00700070007200650074007400650020005000440046002d0064006f006b0075006d0065006e00740065007200200073006f006d002000700061007300730065007200200066006f00720020007000e5006c006900740065006c006900670020007600690073006e0069006e00670020006f00670020007500740073006b007200690066007400200061007600200066006f0072007200650074006e0069006e006700730064006f006b0075006d0065006e007400650072002e0020005000440046002d0064006f006b0075006d0065006e00740065006e00650020006b0061006e002000e50070006e006500730020006d006500640020004100630072006f0062006100740020006f0067002000520065006100640065007200200035002e00300020006f0067002000730065006e006500720065002e>\\n    /SVE <FEFF0041006e007600e4006e00640020006400650020006800e4007200200069006e0073007400e4006c006c006e0069006e006700610072006e00610020006e00e40072002000640075002000760069006c006c00200073006b0061007000610020005000440046002d0064006f006b0075006d0065006e007400200073006f006d00200070006100730073006100720020006600f600720020007000e5006c00690074006c006900670020007600690073006e0069006e00670020006f006300680020007500740073006b0072006900660074002000610076002000610066006600e4007200730064006f006b0075006d0065006e0074002e0020005000440046002d0064006f006b0075006d0065006e00740065006e0020006b0061006e002000f600700070006e006100730020006d006500640020004100630072006f0062006100740020006f00630068002000520065006100640065007200200035002e003000200065006c006c00650072002000730065006e006100720065002e>\\n    /ENU (NPG PRINT PDF Job Options. 4th September 2006. PDF 1.3 Compatibility. Adds Trim and Bleed boxes top Nature pages where none exist.)\\n  >>\\n>> setdistillerparams\\n<<\\n  /HWResolution [2400 2400]\\n  /PageSize [665.858 854.929]\\n>> setpagedevice\\n\\n']"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# separate by a specified sequence of characters\n",
    "cont.split('ei')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['',\n",
       " '',\n",
       " '',\n",
       " '',\n",
       " '',\n",
       " '',\n",
       " '',\n",
       " '',\n",
       " '',\n",
       " '',\n",
       " '',\n",
       " '',\n",
       " '',\n",
       " '',\n",
       " '',\n",
       " '',\n",
       " '',\n",
       " '',\n",
       " '',\n",
       " '',\n",
       " '',\n",
       " '',\n",
       " '',\n",
       " '',\n",
       " '',\n",
       " '',\n",
       " '',\n",
       " '',\n",
       " '',\n",
       " '',\n",
       " '',\n",
       " '',\n",
       " '',\n",
       " '',\n",
       " '',\n",
       " '',\n",
       " '',\n",
       " '',\n",
       " '',\n",
       " '',\n",
       " '',\n",
       " '',\n",
       " '',\n",
       " '',\n",
       " '',\n",
       " '',\n",
       " '',\n",
       " '',\n",
       " '',\n",
       " '',\n",
       " '',\n",
       " '',\n",
       " '',\n",
       " '',\n",
       " '',\n",
       " '',\n",
       " '',\n",
       " '',\n",
       " '',\n",
       " 'Complete protein-protein association kinetics in atomic detail revealed by molecular dynamics simulations and Markov modelling',\n",
       " '',\n",
       " '',\n",
       " 'Complete protein–protein association kinetics in',\n",
       " 'atomic detail revealed by molecular dynamics',\n",
       " 'simulations and Markov modelling',\n",
       " 'Nuria Plattner1*, Stefan Doerr2, Gianni De Fabritiis2,3 and Frank Noé1*',\n",
       " '',\n",
       " 'Protein–protein association is fundamental to many life processes. However, a microscopic model describing the structures',\n",
       " 'and kinetics during association and dissociation is lacking on account of the long lifetimes of associated states, which have',\n",
       " 'prevented efficient sampling by direct molecular dynamics (MD) simulations. Here we demonstrate protein–protein',\n",
       " 'association and dissociation in atomistic resolution for the ribonuclease barnase and its inhibitor barstar by combining',\n",
       " 'adaptive high-throughput MD simulations and hidden Markov modelling. The model reveals experimentally consistent',\n",
       " 'intermediate structures, energetics and kinetics on timescales from microseconds to hours. A variety of flexibly attached',\n",
       " 'intermediates and misbound states funnel down to a transition state and a native basin consisting of the loosely bound',\n",
       " 'near-native state and the tightly bound crystallographic state. These results offer a deeper level of insight into',\n",
       " 'macromolecular recognition and our approach opens the door for understanding and manipulating a wide range of',\n",
       " 'macromolecular association processes.',\n",
       " '',\n",
       " 'Life processes are critically dependent on the formation of bio-molecular complexes, and in particular, protein–protein associ-ation, which mediates the assembly of cellular structures, signal',\n",
       " 'transduction and inhibition, immune response and protein recruit-',\n",
       " 'ment to membrane scaffolds. Interactions of proteins with other',\n",
       " 'macromolecules also play key roles in applications such as inhibition',\n",
       " 'of protein activity by drugs1, gene editing2 and the assembly of',\n",
       " 'new materials3.',\n",
       " '',\n",
       " 'A deep understanding of such processes and the ability to fully',\n",
       " 'plan and control their design would arise from a full map of the',\n",
       " 'conformation space of the interacting protein pairs. Such a micro-',\n",
       " 'scopic model would involve representative samples of all structures',\n",
       " '(bound, dissociated, intermediates, etc.) in atomic resolution, their',\n",
       " 'equilibrium populations and the transition rates between them.',\n",
       " '',\n",
       " 'However, such a complete microscopic picture has not yet been',\n",
       " 'achieved for protein–protein interaction. Experiments can probe',\n",
       " 'either individual aspects of the microscopic picture or averaged',\n",
       " 'ensembles, such as the most stable structures, association kinetics',\n",
       " 'and binding affinities. Mutational analyses can identify amino',\n",
       " 'acids stabilizing associated states or transition states4. The structure',\n",
       " 'of the transient encounter complex, that is, the set of structures',\n",
       " 'found immediately before the transition state of protein association,',\n",
       " 'has been probed using NMR5 and coarse-grained simulations6–9.',\n",
       " 'Binding free energies can be computed from atomistic simulations',\n",
       " 'when the bound pose is precisely known10, and the binding free',\n",
       " 'energy landscape can be reconstructed using metadynamics given',\n",
       " 'a set of pre-defined collective variables11. Inference of kinetics',\n",
       " 'then requires a kinetic model12 or additional unbiased simulations13.',\n",
       " '',\n",
       " 'A candidate to reveal a complete microscopic model of protein–',\n",
       " 'protein interaction is unbiased atomic-detail molecular dynamics',\n",
       " '(MD) simulation. Microscopic models of biomolecular processes',\n",
       " 'have recently been achieved due to combined advances in high-',\n",
       " 'performance computing and Markov modelling14–19. Previous',\n",
       " 'binding studies, however, have been limited to the association of',\n",
       " '',\n",
       " 'small ligands to proteins and the co-folding of shorter peptide',\n",
       " 'chains16,19,20, or small parts of the association landscape such as',\n",
       " 'the downhill binding step from a pre-bound pose21. The simulation',\n",
       " 'of protein–protein association is challenged by the fact that',\n",
       " 'dissociation times exceed accessible MD simulation lengths by far,',\n",
       " 'such that long MD simulations will get stuck after a single',\n",
       " 'binding event, possibly even in non-native traps.',\n",
       " '',\n",
       " 'Here we demonstrate how reversible protein–protein association',\n",
       " 'can be studied in atomistic detail by combining an extensive and',\n",
       " 'adaptively generated ensemble of unbiased molecular dynamics',\n",
       " 'simulations with hidden Markov modelling. We reproduce',\n",
       " 'measured structures, energetics and kinetics quantitatively and',\n",
       " 'obtain a full microscopic model of the protein complex structures,',\n",
       " 'thermodynamics and kinetics.',\n",
       " '',\n",
       " 'Results and discussion',\n",
       " 'All-atom simulation of reversible protein–protein association. As',\n",
       " 'a prototypical example, we study the diffusion-limited association of',\n",
       " 'the experimentally well-characterized bacterial ribonuclease barnase',\n",
       " 'with its inhibitor barstar22,23. All-atom models of barnase wild type',\n",
       " 'and barstar pseudo wild type24 were immersed in a solvent box',\n",
       " '(approximately 90,000 atoms). Several thousand unbiased molecular',\n",
       " 'dynamics (MD) simulations of 2 milliseconds aggregate length were',\n",
       " 'conducted in an automated high-throughput manner. These data',\n",
       " 'include 1.7 milliseconds of independent trajectories starting from',\n",
       " 'the dissociated state and 0.3 milliseconds in adaptive MD simulations.',\n",
       " '',\n",
       " 'The adaptive MD launched periods of multiple parallel MD',\n",
       " 'trajectories, and the starting points of each period were selected so',\n",
       " 'as to favour the exploration of new states (Methods, Supplementary',\n",
       " 'Methods 1). This strategy can yield large speed-ups compared to',\n",
       " 'non-adaptive MD runs because slow processes with large energy',\n",
       " 'barriers are broken down into steps with individually smaller',\n",
       " 'barriers that can be sampled efficiently25–27 (Fig. 1a,b). When starting',\n",
       " 'from the dissociated state, both adaptive and non-adaptive',\n",
       " '',\n",
       " '1Department of Mathematics and Computer Science, Freie Universität Berlin, Arnimallee 6, 14195 Berlin, Germany. 2Computational Biophysics Laboratory',\n",
       " '(GRIB-IMIM), Universitat Pompeu Fabra, Barcelona Biomedical Research Park (PRBB), C/ Doctor Aiguader 88, 08003 Barcelona, Spain. 3Institució Catalana',\n",
       " 'de Recerca i Estudis Avançats (ICREA), Passeig Lluis Companys 23, 08010 Barcelona, Spain. *e-mail: nuria.plattner@fu-berlin.de; frank.noe@fu-berlin.de',\n",
       " '',\n",
       " 'ARTICLES',\n",
       " 'PUBLISHED ONLINE: 5 JUNE 2017 | DOI: 10.1038/NCHEM.2785',\n",
       " '',\n",
       " 'NATURE CHEMISTRY | ADVANCE ONLINE PUBLICATION | www.nature.com/naturechemistry 1',\n",
       " '',\n",
       " '© 2017 Macmillan Publishers Limited, part of Springer Nature. All rights reserved.',\n",
       " '',\n",
       " 'mailto:nuria.plattner@fu-berlin.de',\n",
       " 'mailto:frank.noe@fu-berlin.de',\n",
       " 'http://dx.doi.org/10.1038/nchem.2785',\n",
       " 'http://www.nature.com/naturechemistry',\n",
       " '',\n",
       " '',\n",
       " 'simulations reproducibly found the native bound complex. Adaptive',\n",
       " 'simulations required a total simulation effort of a few microseconds,',\n",
       " 'roughly an order of magnitude speed-up compared to an equilibrium',\n",
       " 'binding time of a tens of microseconds (Supplementary Fig. 1).',\n",
       " 'Surprisingly, the adaptive MD runs could also reproducibly reach',\n",
       " 'the dissociated state when starting from the crystal structure using',\n",
       " 'a total simulation effort of a few hundred microseconds, although',\n",
       " 'the expected dissociation time is hours (Supplementary Fig. 1).',\n",
       " '',\n",
       " 'A first analysis shows that the total dataset contains several',\n",
       " 'thousand unspecific association and dissociation events, about 90',\n",
       " 'binding events from non-native associated states to native and 10',\n",
       " 'unbinding events (Fig. 1c).',\n",
       " '',\n",
       " 'Equilibrium kinetics model and validation. Although the',\n",
       " 'individual simulations are not started from an equilibrium',\n",
       " 'distribution and are shorter than binding and dissociation',\n",
       " 'timescales, Markov modelling approaches can combine such a',\n",
       " 'trajectory ensemble towards unbiased equilibrium models of the',\n",
       " 'long-timescale kinetics and thermodynamics (Fig. 1a)28,29. Here, a',\n",
       " 'hidden Markov model (HMM) was built to estimate a network of',\n",
       " '',\n",
       " 'states, associated protein structures, and transition rates30. The',\n",
       " 'HMM is coupled to a diffusive association model inspired by a',\n",
       " 'previous study31 in order to be able to predict concentration-',\n",
       " 'dependent association rates and binding free energies (Methods,',\n",
       " 'Supplementary Methods 3–7).',\n",
       " '',\n",
       " 'As the data contains only a few unbinding events, the following',\n",
       " 'computed quantities have large uncertainties but are statistically',\n",
       " 'consistent with experiment, which will be exploited to compute',\n",
       " 'consensus models of experimental and simulation data (intervals',\n",
       " 'are 95% confidence): (i) binding free energy lies between 12 and',\n",
       " '19 kcal mol–1 (experimental 16.8 kcal mol–1; see Supplementary',\n",
       " 'Methods 5), (ii) the dissociation rate lies in the range of 3 × 10–6',\n",
       " '',\n",
       " 'to 1 × 10–1 s–1 (measurements lie in the range of 4.8 × 10–5 to',\n",
       " '5.0 × 10–4 s–1; see Supplementary Methods 6).',\n",
       " '',\n",
       " 'The model successfully predicts the following quantities: (iii)',\n",
       " 'crystal structure 1BRS is predicted to be the most stable state, and',\n",
       " 'the simulated structures assigned to it have an average RMSD of',\n",
       " '0.21 nm to 1BRS, (iv) association rate is predicted to be 2.63 × 108',\n",
       " '',\n",
       " 'to 2.65 × 108 s–1 M–1 (1 × 108 to 3.4 × 108 s–1 M–1 measured pre-',\n",
       " 'viously23,32), (v) using perturbation theory33, binding free energy',\n",
       " 'changes of 35 mutant sequences reported in23,32 were predicted',\n",
       " 'within statistical error (Fig. 2a and Supplementary Methods 9),',\n",
       " '(vi) although this approach cannot compute absolute values of',\n",
       " 'association rates of mutants, the computed change in transition',\n",
       " 'state energy is an excellent predictor for the change in the experimental',\n",
       " 'on-rate (Fig. 2b).',\n",
       " '',\n",
       " 'Overall binding mechanism. Having validated our model with a',\n",
       " 'large set of experimental data we proceeded to examine properties',\n",
       " 'that are more difficult to probe directly by experiments. Figure 2c,',\n",
       " 'd displays the overall mechanisms and structures found along the',\n",
       " 'binding pathways. The proteins initially associate to non-native',\n",
       " '(large RMSD) states. At that point, they can dissociate and re-',\n",
       " 'associate or stay in contact and rearrange towards the native',\n",
       " 'complex (Fig. 2c–f ).',\n",
       " '',\n",
       " 'At the high protein concentration in the simulation, initial',\n",
       " 'association occurs within tens of nanoseconds and with no detect-',\n",
       " 'able barrier. The initial association of barstar can occur at all',\n",
       " 'locations on the barnase surface. During the first microsecond,',\n",
       " 'the complex preferentially populates early intermediates with',\n",
       " 'barstar located opposite of the barnase binding site (orange states',\n",
       " 'in Fig. 2d). However, association can also occur directly into the',\n",
       " 'late intermediate and mis-bound states (blue/red states in Fig. 2d),',\n",
       " 'where barstar is located in or near the barnase binding groove,',\n",
       " 'but not in the native orientation. The preference for initially associ-',\n",
       " 'ating to locations away from the binding pocket can be explained by',\n",
       " 'the fact that these states cover a larger fraction of the surface. The',\n",
       " 'initial association preference depends on the salt concentration;',\n",
       " 'with enhanced steering to the binding site being expected when',\n",
       " 'charge screening is reduced7,22. Knocking out intermediate states',\n",
       " 'by removing them from the HMM showed that individual early',\n",
       " 'intermediates have no significant effect on the association rate,',\n",
       " 'and even removing all early intermediates only reduces the',\n",
       " 'association rate by 20% (Supplementary Table 2).',\n",
       " '',\n",
       " 'On timescales of 1–10 µs, the population shifts toward the late',\n",
       " 'intermediates and mis-bound states (blue/red states in Figs 2d',\n",
       " 'and 5c), which have higher equilibrium populations than the',\n",
       " 'early intermediates. Of special interest is the mis-bound state',\n",
       " '(red), that has barstar bound ‘upside down’. This state does not',\n",
       " 'have a well-defined binding interface, but a few non-native inter-',\n",
       " 'actions such as barnase Arg 83 or Arg 87 with barstar Glu 118',\n",
       " 'occur with significant probability (Supplementary Figs 3 and 4).',\n",
       " 'Intermediate states are kinetically well connected and have lifetimes',\n",
       " 'on the order of one microsecond each (Supplementary Fig. 2).',\n",
       " 'Removing the late intermediates would reduce the binding rate by',\n",
       " 'a factor of 2.5 (Supplementary Table 2)',\n",
       " '',\n",
       " '1 2',\n",
       " '',\n",
       " '3',\n",
       " '4',\n",
       " '',\n",
       " '5 Sampling strategy ',\n",
       " '',\n",
       " 'Markov model ',\n",
       " '',\n",
       " 'Trajectories',\n",
       " '',\n",
       " 'Tasks',\n",
       " '',\n",
       " 'Data ',\n",
       " 'generation ',\n",
       " '',\n",
       " 'Model building ',\n",
       " '',\n",
       " 'Compute clusterClientba',\n",
       " '',\n",
       " 'c',\n",
       " '',\n",
       " 'AssociatedDissociated Bound',\n",
       " '',\n",
       " 'Time (μs) Time (μs)',\n",
       " '',\n",
       " '2,867',\n",
       " '',\n",
       " 'D',\n",
       " 'is',\n",
       " '',\n",
       " 'ta',\n",
       " 'nc',\n",
       " '',\n",
       " 'e ',\n",
       " '(n',\n",
       " '',\n",
       " 'm',\n",
       " ')',\n",
       " '',\n",
       " 'R',\n",
       " 'M',\n",
       " '',\n",
       " 'S',\n",
       " 'D',\n",
       " '',\n",
       " ' (',\n",
       " 'nm',\n",
       " '',\n",
       " ')',\n",
       " '',\n",
       " '86',\n",
       " '6',\n",
       " '',\n",
       " '0',\n",
       " '0 0.5 0',\n",
       " '',\n",
       " '0',\n",
       " '',\n",
       " '1.5',\n",
       " '',\n",
       " '5',\n",
       " '',\n",
       " 'Time (μs) Time (μs)',\n",
       " '',\n",
       " 'R',\n",
       " 'M',\n",
       " '',\n",
       " 'S',\n",
       " 'D',\n",
       " '',\n",
       " ' (',\n",
       " 'nm',\n",
       " '',\n",
       " ')',\n",
       " '',\n",
       " 'D',\n",
       " 'is',\n",
       " '',\n",
       " 'ta',\n",
       " 'nc',\n",
       " '',\n",
       " 'e ',\n",
       " '(n',\n",
       " '',\n",
       " 'm',\n",
       " ') 101,465',\n",
       " '',\n",
       " '6',\n",
       " '',\n",
       " '0',\n",
       " '0 3.5 0 7',\n",
       " '',\n",
       " '0',\n",
       " '',\n",
       " '1.5',\n",
       " '',\n",
       " 'Figure 1 | Simulation of protein–protein association and dissociation.',\n",
       " 'a, Scheme of a large-barrier process such as protein–protein dissociation that',\n",
       " 'can be subdivided into multiple steps. Such processes can be sampled by',\n",
       " 'combining adaptively seeded simulation trajectories between individual steps',\n",
       " 'via Markov modelling. b, Schematic outline of adaptive sampling process.',\n",
       " 'c, Total number of full association–dissociation and binding–unbinding',\n",
       " 'transitions evaluated by raw trajectory statistics of all barnase–barstar',\n",
       " 'simulation trajectories used subsequently. An association–dissociation',\n",
       " 'transition was counted when a trajectory moved from a state where the',\n",
       " 'proteins had a minimum distance between heavy atom pairs of >2 nm to',\n",
       " '<0.3 nm, and vice versa. A binding–unbinding event was counted when a',\n",
       " 'trajectory moved from a state with minimum heavy-atom RMSD from the',\n",
       " 'crystallographic structure >1 nm to RMSD <0.3 nm, and vice versa. For each',\n",
       " 'trajectory only one count was made in order to avoid duplicate counting',\n",
       " 'of recrossing events. Three sample trajectory snippets are shown for each',\n",
       " 'type of transition.',\n",
       " '',\n",
       " 'ARTICLES NATURE CHEMISTRY DOI: 10.1038/NCHEM.2785',\n",
       " '',\n",
       " 'NATURE CHEMISTRY | ADVANCE ONLINE PUBLICATION | www.nature.com/naturechemistry2',\n",
       " '',\n",
       " '© 2017 Macmillan Publishers Limited, part of Springer Nature. All rights reserved.',\n",
       " '',\n",
       " 'http://dx.doi.org/10.1038/nchem.2785',\n",
       " 'http://www.nature.com/naturechemistry',\n",
       " '',\n",
       " '',\n",
       " 'Misbound',\n",
       " '',\n",
       " 'Early intermediates',\n",
       " '',\n",
       " 'Late',\n",
       " 'intermediate 2',\n",
       " '',\n",
       " 'Late',\n",
       " 'intermediate 1',\n",
       " '',\n",
       " 'Pre-bound',\n",
       " '',\n",
       " 'Loosely',\n",
       " 'bound',\n",
       " '',\n",
       " 'Tightly',\n",
       " 'bound',\n",
       " '',\n",
       " 'a',\n",
       " '',\n",
       " 'd',\n",
       " '',\n",
       " 'b c e',\n",
       " '',\n",
       " 'f',\n",
       " '',\n",
       " 'g',\n",
       " '',\n",
       " 'Dissociated',\n",
       " '',\n",
       " 'Association',\n",
       " '',\n",
       " 'Binding',\n",
       " '',\n",
       " 'Unbinding Rebinding',\n",
       " '',\n",
       " '12',\n",
       " 'ΔΔ',\n",
       " '',\n",
       " 'G',\n",
       " ' e',\n",
       " '',\n",
       " 'xp',\n",
       " 'er',\n",
       " '',\n",
       " 'im',\n",
       " 'en',\n",
       " '',\n",
       " 't (',\n",
       " 'kc',\n",
       " '',\n",
       " 'al',\n",
       " ' m',\n",
       " '',\n",
       " 'ol',\n",
       " '–1',\n",
       " '',\n",
       " ')',\n",
       " '',\n",
       " '8',\n",
       " '',\n",
       " '4',\n",
       " '',\n",
       " '–4 0',\n",
       " '',\n",
       " 'ΔΔG model (kcal mol–1)',\n",
       " '',\n",
       " '4 8 12',\n",
       " '',\n",
       " '–4',\n",
       " '',\n",
       " '0',\n",
       " '',\n",
       " '3',\n",
       " '',\n",
       " '2',\n",
       " '',\n",
       " '2',\n",
       " '',\n",
       " '0',\n",
       " '',\n",
       " 'ΔΔ',\n",
       " 'G',\n",
       " '',\n",
       " 'on',\n",
       " ' e',\n",
       " '',\n",
       " 'xp',\n",
       " 'er',\n",
       " '',\n",
       " 'im',\n",
       " 'en',\n",
       " '',\n",
       " 't (',\n",
       " 'kc',\n",
       " '',\n",
       " 'al',\n",
       " ' m',\n",
       " '',\n",
       " 'ol',\n",
       " '–1',\n",
       " '',\n",
       " ')',\n",
       " '',\n",
       " '–5 0 5 10',\n",
       " '',\n",
       " 'ΔΔGTS model (kcal mol–1)',\n",
       " '',\n",
       " '–2',\n",
       " '',\n",
       " '–1 F',\n",
       " 're',\n",
       " '',\n",
       " 'e ',\n",
       " 'en',\n",
       " '',\n",
       " 'er',\n",
       " 'gy',\n",
       " '',\n",
       " ' (',\n",
       " 'kT',\n",
       " '',\n",
       " ')',\n",
       " '',\n",
       " 'Minimal distance (nm)',\n",
       " '',\n",
       " '3.0 24',\n",
       " '',\n",
       " '21',\n",
       " '',\n",
       " '18',\n",
       " '',\n",
       " '15',\n",
       " '',\n",
       " '12',\n",
       " '',\n",
       " '9',\n",
       " '',\n",
       " '6',\n",
       " '',\n",
       " '3',\n",
       " '',\n",
       " '0',\n",
       " '',\n",
       " '2.5',\n",
       " '',\n",
       " '2.0',\n",
       " '',\n",
       " 'R',\n",
       " 'M',\n",
       " '',\n",
       " 'S',\n",
       " 'D',\n",
       " '',\n",
       " ' (',\n",
       " 'nm',\n",
       " '',\n",
       " ')',\n",
       " '',\n",
       " '0.3',\n",
       " '0.2',\n",
       " '0.1',\n",
       " '',\n",
       " '0.00 0.02 0.04 0.06 0.08 0.10',\n",
       " '',\n",
       " '5',\n",
       " '',\n",
       " 't t+50 t+100 t+150 t+200',\n",
       " '',\n",
       " '10 15 20 25 30',\n",
       " '',\n",
       " '0.3',\n",
       " '0.2',\n",
       " '0.1',\n",
       " '',\n",
       " '0.3',\n",
       " '0.2',\n",
       " '0.1',\n",
       " '',\n",
       " 'R',\n",
       " 'M',\n",
       " '',\n",
       " 'S',\n",
       " 'D',\n",
       " '',\n",
       " ' (',\n",
       " 'nm',\n",
       " '',\n",
       " ')',\n",
       " 'R',\n",
       " '',\n",
       " 'M',\n",
       " 'S',\n",
       " '',\n",
       " 'D',\n",
       " ' (',\n",
       " '',\n",
       " 'nm',\n",
       " ')',\n",
       " '',\n",
       " 'R',\n",
       " 'M',\n",
       " '',\n",
       " 'S',\n",
       " 'D',\n",
       " '',\n",
       " ' (',\n",
       " 'nm',\n",
       " '',\n",
       " ')',\n",
       " '',\n",
       " '1.5',\n",
       " '',\n",
       " '1.0',\n",
       " '',\n",
       " '0.5',\n",
       " '',\n",
       " '0.25 0.50 1.00 2.00 4.00',\n",
       " '',\n",
       " 'Time (μs)',\n",
       " '',\n",
       " 'Figure 2 | Kinetic model of barnase–barstar association. a,b, Comparison of experimental and predicted changes in binding free energies and association',\n",
       " 'rates for all 35 single and double alanine mutations from Table II of ref. 23 and Table I of ref. 32. Statistical confidence intervals of the model are indicated in',\n",
       " 'white (1-σ, 68% confidence) and grey (2-σ, 95% confidence). a, Changes in binding free energy (ΔΔG) are predicted within statistical error. b, Comparison',\n",
       " 'of the computed changes in barrier height of the binding transition with changes of apparent association transition state energy, computed as ΔΔGon,apparent',\n",
       " '= –kBTln(kon',\n",
       " '',\n",
       " 'mutant/kon',\n",
       " 'wt), where kBT= 0.598 kcal mol',\n",
       " '',\n",
       " '–1 and kon',\n",
       " 'mutant, kon',\n",
       " '',\n",
       " 'wt are the experimental association rates of mutant and wild type, respectively. The model',\n",
       " 'prediction was computed as free energy difference between the pre-bound state as a proxy for the transition state, and all early intermediate, late',\n",
       " 'intermediate and mis-bound states. c, Free energy surface over protein–protein heavy atom minimal distance and minimal RMSD to the native complex.',\n",
       " 'Shown is a contour plot of –kTlnμ(x) evaluated for each trajectory sample x, where kT is the thermal energy and μ(x) is the equilibrium weight of each',\n",
       " 'sample computed by the HMM (see Supplementary Methods 4). d, Kinetic network between metastable states with lifetimes on the order of microseconds.',\n",
       " 'Only transition rate constants ≥0.2ms–1 are shown. Arrow thickness is proportional to rate constant, disc area is proportional to square root of state',\n",
       " 'population. Structures are centered on barnase (orange) with its binding groove on top and its N-terminal helix on the bottom. Barstar is shown in grey with',\n",
       " 'its binding domain in green. Grey clouds show overlay of approximately 20 barstar conformations, sampled randomly from the corresponding metastable',\n",
       " 'states. e–g, Minimal RMSD to crystallographic complex shown for sample trajectories generated from the HMM (see Supplementary Methods 4). Initial',\n",
       " 'association (e), binding to the native complex (f), and geminate rebinding after being bound for a long time t (one hour on average; g).',\n",
       " '',\n",
       " 'NATURE CHEMISTRY DOI: 10.1038/NCHEM.2785 ARTICLES',\n",
       " '',\n",
       " 'NATURE CHEMISTRY | ADVANCE ONLINE PUBLICATION | www.nature.com/naturechemistry 3',\n",
       " '',\n",
       " '© 2017 Macmillan Publishers Limited, part of Springer Nature. All rights reserved.',\n",
       " '',\n",
       " 'http://dx.doi.org/10.1038/nchem.2785',\n",
       " 'http://www.nature.com/naturechemistry',\n",
       " '',\n",
       " '',\n",
       " 'On timescales of 10–100 µs, the complex passes through a',\n",
       " 'pre-bound state (magenta in Figs 2d and 5c) and then continues',\n",
       " 'to the natively bound complex (grey states in Figs 2d and 5c),',\n",
       " 'which consists of a loosely bound and a tightly bound substate',\n",
       " 'that exchange on the microsecond timescale.',\n",
       " '',\n",
       " 'Nature of the transition state of binding. The kinetic bottleneck',\n",
       " 'between the intermediate states and natively bound states is the',\n",
       " 'mandatory passage through the pre-bound state (magenta in Fig. 2d),',\n",
       " 'that is hence an on-pathway intermediate. Both the transition into',\n",
       " 'the pre-bound state and the binding from the pre-bound state',\n",
       " 'involve barriers (Supplementary Fig. 6). We refer to these barriers',\n",
       " 'collectively as transition state of binding as they are rate-limiting at',\n",
       " 'sufficiently high protein concentrations (Supplementary Fig. 5).',\n",
       " '',\n",
       " 'The transition from the late intermediates through the pre-bound',\n",
       " 'state into the loosely bound state is characterized by two microscopic',\n",
       " 'pathways, in which barstar interacts with either of the two barnase',\n",
       " 'loops surrounding the binding groove: the ‘left’ loop including Ser',\n",
       " '',\n",
       " '38 and the ‘right’ RNA recognition loop of barnase including Arg',\n",
       " '59 (Fig. 3) whose importance for steering of the two protein has',\n",
       " 'been shown before6,7. Due to the flexibility of both loop regions,',\n",
       " 'barstar can “swing” around its attachment point, giving rise to a',\n",
       " 'quite flexible set of structures. These pathways are not separated by',\n",
       " 'significant barriers, i.e. the complex can change between “left” and',\n",
       " '“right” attachment point multiple times before binding.',\n",
       " '',\n",
       " 'The minor left pathway (5% population, Fig. 3a→ c→ e) involves',\n",
       " 'different possible protein contacts, including hydrophobic contacts',\n",
       " 'and does not appear to have well-defined structural features. In the',\n",
       " 'major right pathway (95% population, Fig. 3b→ d→ e), barnase Arg',\n",
       " '59 interacts with barstar Glu 76, Asp 35, or both, in line with34,35.',\n",
       " 'Arg 59 may additionally form a hydrophobic contact with barstar',\n",
       " 'Trp 38. While no individual pair of residues is strongly formed in the',\n",
       " 'transition state22, Arg 59 is involved in 95% of all transition state con-',\n",
       " 'figurations. This observation confirms the pivotal role of Arg 59 inmed-',\n",
       " 'iating the binding transition34, and explains that Arg 59 mutants',\n",
       " 'reduced the binding rate by a factor of 10 (ref. 32). We expect that a',\n",
       " 'barstar double mutant of Glu 76 and Asp 35 will have a similar effect.',\n",
       " '',\n",
       " 'The pre-bound state has a lifetime of about two microseconds',\n",
       " '(Supplementary Fig. 2). In all binding events simulated, the',\n",
       " 'transition from the pre-bound to the loosely bound state trajectories',\n",
       " 'occurs within a single saving interval of our simulations, that is less',\n",
       " 'than 100 picoseconds. Thus the last step of docking into the barnase',\n",
       " 'binding pocket and forming the binding interface with its many',\n",
       " 'contacts is rare, but occurs quickly when it happens21,36. In line with',\n",
       " 'previous predictions22, the majority of the barnase–barstar short-',\n",
       " 'range interactions is formed after the transition state. In the limit of',\n",
       " 'low electrostatic interactions, the basal binding rate may be governed',\n",
       " 'by the bimolecular rate to form a ready-to-dock structure similar to',\n",
       " 'our pre-bound state22, and it is likely that the barrier of desolvation',\n",
       " 'and docking is reduced or even removed under these conditions.',\n",
       " '',\n",
       " 'Conformational change in the complex. The bound states are in the',\n",
       " 'range of 5–12 kcal mol–1 more stable than intermediate, mis-bound',\n",
       " 'and pre-bound states and thus contain almost all of the equilibrium',\n",
       " 'population. We find two conformational substates in the bound',\n",
       " 'state: The loosely bound state with about 5% of the population',\n",
       " 'and an average heavy-atom RMSD to the crystal structure of',\n",
       " '0.3 nm (Fig. 4a); and the tightly bound state with the remaining',\n",
       " '95% of the population (Fig. 4b). The density of the tightly bound',\n",
       " 'state contains the crystal structure (Fig. 4c, average RMSD',\n",
       " '0.21 nm), which is thus correctly predicted as the most stable state.',\n",
       " '',\n",
       " 'Upon binding, the loosely bound state is reached first. This state',\n",
       " 'transitions into the tightly bound state within 5 µs on average, that',\n",
       " 'itself has an average lifetime of 130 µs before transitioning back to',\n",
       " 'the loosely bound state. These microsecond transitions occur',\n",
       " 'many times before the complex undergoes geminate rebinding',\n",
       " '(Fig. 2g) or finally dissociates.',\n",
       " '',\n",
       " 'Structurally, the native-state transition consists of a slight',\n",
       " 'rotation of barstar in the barnase binding pocket (compare',\n",
       " 'Fig. 4a,b). While both substates have similar residue contact patterns',\n",
       " '(Supplementary Fig. 3), the most apparent difference is that the',\n",
       " 'barnase 35–41 loop is inserted between the barstar binding helix',\n",
       " 'and the large helix of barstar in the loosely bound state, while it is',\n",
       " 'inserted between the binding helix and binding loop in the tightly',\n",
       " 'bound state. The loosely bound state is stabilized by an interaction',\n",
       " 'between barnase Lys 27 (whose functional role is to stabilize the',\n",
       " 'phosphate group in the RNA substrate) and barstar Glu 80, while',\n",
       " 'the tightly bound state is stabilized by a salt bridge between',\n",
       " 'barnase Arg 83 and barstar Asp 39. Due to the latter salt bridge,',\n",
       " 'barnase loop 77–85 is stabilized in the tightly bound state, while it',\n",
       " 'is very flexible in the loosely bound state. Overall, the loosely bound',\n",
       " 'state is more flexible than the tightly bound state (compare Fig. 4a,b).',\n",
       " '',\n",
       " 'These native substate dynamics may be observable experimen-',\n",
       " 'tally, for example, by fluorescence correlation microscopy or by',\n",
       " '',\n",
       " 'a b',\n",
       " '',\n",
       " 'c d',\n",
       " '',\n",
       " 'e',\n",
       " '',\n",
       " 'Arg 59',\n",
       " 'Asp 35',\n",
       " '',\n",
       " 'Trp 38 Glu 76',\n",
       " '',\n",
       " 'Gly 43',\n",
       " 'Trp 44',\n",
       " '',\n",
       " 'Ala 37',\n",
       " 'Ser 38',\n",
       " '',\n",
       " '45° Late',\n",
       " 'intermediate 2',\n",
       " '',\n",
       " ' ',\n",
       " '',\n",
       " 'Pre-bound',\n",
       " '',\n",
       " 'Loosely',\n",
       " 'bound',\n",
       " '',\n",
       " '45°',\n",
       " '',\n",
       " 'Figure 3 | Late steps in binding from late intermediates to the loosely',\n",
       " 'bound state. a,b, Two sets of structures of late intermediate state 2, shown',\n",
       " 'in 45° counterclockwise/clockwise rotation compared to Fig. 2d. c,d, Two',\n",
       " 'sets of structures of the pre-bound state. e, Loosely bound state. Grey clouds',\n",
       " 'show overlay of approximately 20 barstar conformations, sampled randomly',\n",
       " 'from the corresponding metastable states. Two binding pathways exist, in',\n",
       " 'which the binding domain of barstar interacts with barnase at its ‘left’ loop',\n",
       " 'around Ala 37 and Ser 38 (a,c), or at its ‘right’ loop (b,d), before docking into',\n",
       " 'the bound state (e). In the major pathway (b→ d→ e, approximately 95% of',\n",
       " 'the flux), barnase Arg 59 ‘anchors’ into barstar via Asp 35, Glu 76 (polar)',\n",
       " 'and Trp 38 (hydrophobic). Barstar can swing around barnase with the',\n",
       " 'anchor staying attached until final docking into the bound pose.',\n",
       " '',\n",
       " 'ARTICLES NATURE CHEMISTRY DOI: 10.1038/NCHEM.2785',\n",
       " '',\n",
       " 'NATURE CHEMISTRY | ADVANCE ONLINE PUBLICATION | www.nature.com/naturechemistry4',\n",
       " '',\n",
       " '© 2017 Macmillan Publishers Limited, part of Springer Nature. All rights reserved.',\n",
       " '',\n",
       " 'http://dx.doi.org/10.1038/nchem.2785',\n",
       " 'http://www.nature.com/naturechemistry',\n",
       " '',\n",
       " '',\n",
       " 'fast temperature jump experiments that have been employed to',\n",
       " 'probe loose–tight binding transitions37. In particular, barstar Trp',\n",
       " '44 changes solvent accessibility during the native substate transition,',\n",
       " 'rendering intrinsic tryptophan fluorescence a promising probe.',\n",
       " '',\n",
       " 'Association kinetics of wild-type and a faster-associating mutant.',\n",
       " 'The very large association rate of barnase–barstar can be further',\n",
       " 'increased by specific mutations. To understand the microscopic',\n",
       " 'mechanisms of such perturbations, we conducted additional',\n",
       " '0.3 milliseconds of MD simulations using the barnase mutant Glu',\n",
       " '73 to Ala (E73A, see Methods), which associates a factor of 1.5–4',\n",
       " 'faster than the wild type23.',\n",
       " '',\n",
       " 'As both the wild-type and the E73A HMM suffered from a',\n",
       " 'statistical bottleneck at the unbinding transition, the uncertain',\n",
       " 'transition rate from the loosely bound to pre-bound state was set',\n",
       " 'such that the models produce the experimentally known binding',\n",
       " 'free energy. The resulting consensus models still make independent',\n",
       " 'predictions of the binding rate and mechanism, while reproducing',\n",
       " 'experimental thermodynamics.',\n",
       " '',\n",
       " 'TheE73Amodel predicts anassociation rate of 4 to 4.7 × 108 s–1M–1,',\n",
       " 'which correctly reflects an increase by a factor of 1.6–1.8 (ref. 23),',\n",
       " 'and can be explained by the increased electrostatic complementarity.',\n",
       " 'E73 does not participate in the binding interface, and indeed the',\n",
       " 'wild-type native complex structure is still predicted to be the most',\n",
       " '',\n",
       " '95%5%RMSD 0.3 nm',\n",
       " 'Lifetime 5 ms',\n",
       " '',\n",
       " 'Tightly bound',\n",
       " 'RMSD 0.21 nm',\n",
       " 'Lifetime 130 ms',\n",
       " '',\n",
       " '1BRS',\n",
       " 'crystal structure ',\n",
       " '',\n",
       " 'a b c',\n",
       " '',\n",
       " 'Arg 83',\n",
       " 'Asp 39Lys 27',\n",
       " '',\n",
       " 'Glu 80',\n",
       " '',\n",
       " 'Trp 44',\n",
       " 'Trp 44',\n",
       " '',\n",
       " 'Loosely bound',\n",
       " '',\n",
       " 'Figure 4 | Native substate conformational dynamics. Native substates seen from ‘left’, that is, in a 90° clockwise rotation around the image vertical with',\n",
       " 'respect to Fig. 2d. a, Loosely bound state with lower population and higher structural flexibility. b, Tightly bound state with higher population and small',\n",
       " 'structural flexibility. c, Crystal structure 1BRS shown in the structural density of the tightly bound simulation state, for comparison with B. Grey cloud is',\n",
       " 'an overlay of approximately 20 barstar conformations, sampled randomly from the corresponding metastable states.',\n",
       " '',\n",
       " 'Wild type E73A',\n",
       " '',\n",
       " 'a b',\n",
       " '',\n",
       " 'C',\n",
       " 'om',\n",
       " '',\n",
       " 'm',\n",
       " 'itt',\n",
       " '',\n",
       " 'or',\n",
       " '',\n",
       " '1',\n",
       " '',\n",
       " '0',\n",
       " '',\n",
       " '10–1 100 101 102 103 104',\n",
       " '',\n",
       " '0.5',\n",
       " '',\n",
       " 'c d',\n",
       " '',\n",
       " 'Loosely',\n",
       " 'bound',\n",
       " '',\n",
       " 'Prebound',\n",
       " '',\n",
       " 'Late intermediates',\n",
       " '',\n",
       " 'Early intermediates',\n",
       " '',\n",
       " 'Dissociated',\n",
       " '',\n",
       " 'Misbound',\n",
       " '',\n",
       " 'Tightly',\n",
       " 'bound',\n",
       " '',\n",
       " '1.0',\n",
       " '0.8',\n",
       " '0.6',\n",
       " '0.4',\n",
       " '0.2',\n",
       " '0.0',\n",
       " '',\n",
       " 'P',\n",
       " 'ro',\n",
       " '',\n",
       " 'ba',\n",
       " 'bi',\n",
       " '',\n",
       " 'lit',\n",
       " 'y',\n",
       " '',\n",
       " 'Time (μs)',\n",
       " '10–1 100 101 102 103 104',\n",
       " '',\n",
       " '1.0',\n",
       " '0.8',\n",
       " '0.6',\n",
       " '0.4',\n",
       " '0.2',\n",
       " '0.0',\n",
       " '',\n",
       " 'P',\n",
       " 'ro',\n",
       " '',\n",
       " 'ba',\n",
       " 'bi',\n",
       " '',\n",
       " 'lit',\n",
       " 'y',\n",
       " '',\n",
       " 'Time (μs)',\n",
       " '',\n",
       " 'Figure 5 | Comparison of kinetics between wild type and E73A mutant. a,b, Kinetic networks of the consensus models that are consistent with MD',\n",
       " 'simulation data and the experimental binding free energy. Network plots as in Fig. 2d, but the vertical position of states indicates the binding probability',\n",
       " '(committor) and disc area is proportional to fourth root of state population. The E73A mutant shows a deeper funnel that leads to an increased association',\n",
       " 'rate. c,d, Kinetic evolution of probabilities when starting from dissociated state, computed from the HMM (see Supplementary Methods 4). Despite a higher',\n",
       " 'association rate, the E73A mutant takes longer to bind to the native state, as the increased association rate is due to rapid trapping in a mis-bound state that',\n",
       " 'is committed to further native binding. Colour scheme as in Fig. 2d.',\n",
       " '',\n",
       " 'NATURE CHEMISTRY DOI: 10.1038/NCHEM.2785 ARTICLES',\n",
       " '',\n",
       " 'NATURE CHEMISTRY | ADVANCE ONLINE PUBLICATION | www.nature.com/naturechemistry 5',\n",
       " '',\n",
       " '© 2017 Macmillan Publishers Limited, part of Springer Nature. All rights reserved.',\n",
       " '',\n",
       " 'http://dx.doi.org/10.1038/nchem.2785',\n",
       " 'http://www.nature.com/naturechemistry',\n",
       " '',\n",
       " '',\n",
       " 'stable state in the mutant (Fig. 5b). Figure 5a,b shows kinetic',\n",
       " 'networks of the wild type and E73A mutant, similar to Fig. 2d but',\n",
       " 'with the y-positions indicating the probability of binding (committor)',\n",
       " 'of each state. Both systems are characterized by binding funnels,',\n",
       " 'that are broad on top but narrow down rapidly as the complex pro-',\n",
       " 'ceeds through late intermediates towards binding. The E73A funnel',\n",
       " 'is deeper in the sense that early intermediates have larger committor',\n",
       " 'probabilities, that is, it is more difficult to escape the complex, once',\n",
       " 'it is associated. E73A has a reduced stability of the bound state23.',\n",
       " 'The state that experiences the most profound increase of probability',\n",
       " 'is the mis-bound state. The time to form the native bound complex',\n",
       " 'is much longer in E73A compared to the wild type, as the system',\n",
       " 'gets trapped in the mis-bound state for 100s of microseconds',\n",
       " '(Fig. 5d). An increase of the association rate does therefore not',\n",
       " 'necessarily imply more efficient binding in a functional sense.',\n",
       " '',\n",
       " 'Based on the consensus models the experimental association rate',\n",
       " 'is simulated as a function of protein concentration. At micromolar',\n",
       " 'concentrations commonly used in experiments, the association rates',\n",
       " 'are linear in the concentration, that is, binding is limited by the',\n",
       " 'bimolecular encounter (Supplementary Fig. 5). In this condition',\n",
       " 'the dominant energy barrier is characterized by balancing the entro-',\n",
       " 'pic cost of bringing two molecules together at low concentration and',\n",
       " 'the onset of the electrostatic interactions. At concentrations of 10s to',\n",
       " '100s of μM, this barrier disappears, the transition state of binding',\n",
       " 'becomes rate-limiting4, and we observe a concentration-independent',\n",
       " 'rate constant of 18,000 s–1 for the wild type and 1,500 s–1 for the E73A',\n",
       " 'mutant due to the transition state of binding (Supplementary Fig. 5).',\n",
       " '',\n",
       " 'After an initial contact has been formed, the complex will',\n",
       " 'typically move through a sequence of other non-native inter-',\n",
       " 'mediates until it dissociates or proceeds to native binding with',\n",
       " 'about 50% probability each (Fig. 5a,b). The major bottleneck in',\n",
       " 'dissociation from the native state is the separation of the loosely',\n",
       " 'bound complex into the pre-bound, mis-bound and intermediate',\n",
       " 'states (Supplementary Fig. 6), from where the complex can dissociate',\n",
       " 'or perform geminate rebinding (Fig. 2g).',\n",
       " '',\n",
       " 'Conclusion',\n",
       " 'By combining recent breakthroughs in simulation and analysis tech-',\n",
       " 'nology, we have demonstrated that a quantitatively accurate micro-',\n",
       " 'scopic model of protein–protein association can be obtained using',\n",
       " 'unbiased all-atom MD simulations and Markov modelling. This',\n",
       " 'model reveals kinetic and mechanistic details of the association mech-',\n",
       " 'anism and has enabled the discovery of previously unknown substates.',\n",
       " '',\n",
       " 'The presented combination of simulation and analysis techniques',\n",
       " 'are applicable to any pair of macromolecules. While the available',\n",
       " 'computational power limits the size of the systems being studied to',\n",
       " 'currently a few hundred amino acids per protein, biological timescales',\n",
       " 'can be reached that are far beyond direct MD simulation times. This',\n",
       " 'opens the door to study how the parts of macromolecular assemblies',\n",
       " 'form, how viral and host proteins interact and how protein–DNA',\n",
       " 'interactions depend on DNA sequence and epigenetic modifications.',\n",
       " '',\n",
       " 'Rather than a black-box model, the present approach provides',\n",
       " 'populations for every sampled atomistic structure and transition',\n",
       " 'rates between them. We have demonstrated that combining such an',\n",
       " 'extensive structural ensemble with an appropriate reweighting',\n",
       " 'method, the thermodynamics and kinetics of similar protein sequences',\n",
       " 'can bepredicted. In the future, protein sequences andpost-translational',\n",
       " 'modifications, or sequence and epigenetic modifications of interacting',\n",
       " 'nucleic acids can be designed to stabilize desirable intermediates,',\n",
       " 'change the kinetics or control the mechanism, with far-reaching',\n",
       " 'consequences for pharmaceutics, biotechnology andmaterial sciences.',\n",
       " '',\n",
       " 'Methods',\n",
       " 'MD simulation setup. All-atom models of barnase wild type and barstar pseudo',\n",
       " 'wild type with C40,82A double mutation24 were constructed starting from crystal',\n",
       " ...]"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# separate all lines in the PDF file\n",
    "cont.splitlines()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [],
   "source": [
    "# get a conent Partition at a specific word\n",
    "meth = cont.partition('Fabritiis')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\nComplete protein-protein association kinetics in atomic detail revealed by molecular dynamics simulations and Markov modelling\\n\\n\\nComplete protein–protein association kinetics in\\natomic detail revealed by molecular dynamics\\nsimulations and Markov modelling\\nNuria Plattner1*, Stefan Doerr2, Gianni De '"
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "meth[0]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Analyze text with Natural Language Toolkit\n",
    "\n",
    "Analysis of the text in the PDF can be done using NLTK: https://www.nltk.org"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [],
   "source": [
    "import nltk"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [],
   "source": [
    "sentence = cont"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [],
   "source": [
    "# first we tokenize the whole text\n",
    "tokens = nltk.word_tokenize(sentence)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['Complete',\n",
       " 'protein-protein',\n",
       " 'association',\n",
       " 'kinetics',\n",
       " 'in',\n",
       " 'atomic',\n",
       " 'detail',\n",
       " 'revealed',\n",
       " 'by',\n",
       " 'molecular',\n",
       " 'dynamics',\n",
       " 'simulations',\n",
       " 'and',\n",
       " 'Markov',\n",
       " 'modelling',\n",
       " 'Complete',\n",
       " 'protein–protein',\n",
       " 'association',\n",
       " 'kinetics',\n",
       " 'in',\n",
       " 'atomic',\n",
       " 'detail',\n",
       " 'revealed',\n",
       " 'by',\n",
       " 'molecular',\n",
       " 'dynamics',\n",
       " 'simulations',\n",
       " 'and',\n",
       " 'Markov',\n",
       " 'modelling',\n",
       " 'Nuria',\n",
       " 'Plattner1*',\n",
       " ',',\n",
       " 'Stefan',\n",
       " 'Doerr2',\n",
       " ',',\n",
       " 'Gianni',\n",
       " 'De',\n",
       " 'Fabritiis2,3',\n",
       " 'and',\n",
       " 'Frank',\n",
       " 'Noé1*',\n",
       " 'Protein–protein',\n",
       " 'association',\n",
       " 'is',\n",
       " 'fundamental',\n",
       " 'to',\n",
       " 'many',\n",
       " 'life',\n",
       " 'processes',\n",
       " '.',\n",
       " 'However',\n",
       " ',',\n",
       " 'a',\n",
       " 'microscopic',\n",
       " 'model',\n",
       " 'describing',\n",
       " 'the',\n",
       " 'structures',\n",
       " 'and',\n",
       " 'kinetics',\n",
       " 'during',\n",
       " 'association',\n",
       " 'and',\n",
       " 'dissociation',\n",
       " 'is',\n",
       " 'lacking',\n",
       " 'on',\n",
       " 'account',\n",
       " 'of',\n",
       " 'the',\n",
       " 'long',\n",
       " 'lifetimes',\n",
       " 'of',\n",
       " 'associated',\n",
       " 'states',\n",
       " ',',\n",
       " 'which',\n",
       " 'have',\n",
       " 'prevented',\n",
       " 'efficient',\n",
       " 'sampling',\n",
       " 'by',\n",
       " 'direct',\n",
       " 'molecular',\n",
       " 'dynamics',\n",
       " '(',\n",
       " 'MD',\n",
       " ')',\n",
       " 'simulations',\n",
       " '.',\n",
       " 'Here',\n",
       " 'we',\n",
       " 'demonstrate',\n",
       " 'protein–protein',\n",
       " 'association',\n",
       " 'and',\n",
       " 'dissociation',\n",
       " 'in',\n",
       " 'atomistic',\n",
       " 'resolution',\n",
       " 'for',\n",
       " 'the',\n",
       " 'ribonuclease',\n",
       " 'barnase',\n",
       " 'and',\n",
       " 'its',\n",
       " 'inhibitor',\n",
       " 'barstar',\n",
       " 'by',\n",
       " 'combining',\n",
       " 'adaptive',\n",
       " 'high-throughput',\n",
       " 'MD',\n",
       " 'simulations',\n",
       " 'and',\n",
       " 'hidden',\n",
       " 'Markov',\n",
       " 'modelling',\n",
       " '.',\n",
       " 'The',\n",
       " 'model',\n",
       " 'reveals',\n",
       " 'experimentally',\n",
       " 'consistent',\n",
       " 'intermediate',\n",
       " 'structures',\n",
       " ',',\n",
       " 'energetics',\n",
       " 'and',\n",
       " 'kinetics',\n",
       " 'on',\n",
       " 'timescales',\n",
       " 'from',\n",
       " 'microseconds',\n",
       " 'to',\n",
       " 'hours',\n",
       " '.',\n",
       " 'A',\n",
       " 'variety',\n",
       " 'of',\n",
       " 'flexibly',\n",
       " 'attached',\n",
       " 'intermediates',\n",
       " 'and',\n",
       " 'misbound',\n",
       " 'states',\n",
       " 'funnel',\n",
       " 'down',\n",
       " 'to',\n",
       " 'a',\n",
       " 'transition',\n",
       " 'state',\n",
       " 'and',\n",
       " 'a',\n",
       " 'native',\n",
       " 'basin',\n",
       " 'consisting',\n",
       " 'of',\n",
       " 'the',\n",
       " 'loosely',\n",
       " 'bound',\n",
       " 'near-native',\n",
       " 'state',\n",
       " 'and',\n",
       " 'the',\n",
       " 'tightly',\n",
       " 'bound',\n",
       " 'crystallographic',\n",
       " 'state',\n",
       " '.',\n",
       " 'These',\n",
       " 'results',\n",
       " 'offer',\n",
       " 'a',\n",
       " 'deeper',\n",
       " 'level',\n",
       " 'of',\n",
       " 'insight',\n",
       " 'into',\n",
       " 'macromolecular',\n",
       " 'recognition',\n",
       " 'and',\n",
       " 'our',\n",
       " 'approach',\n",
       " 'opens',\n",
       " 'the',\n",
       " 'door',\n",
       " 'for',\n",
       " 'understanding',\n",
       " 'and',\n",
       " 'manipulating',\n",
       " 'a',\n",
       " 'wide',\n",
       " 'range',\n",
       " 'of',\n",
       " 'macromolecular',\n",
       " 'association',\n",
       " 'processes',\n",
       " '.',\n",
       " 'Life',\n",
       " 'processes',\n",
       " 'are',\n",
       " 'critically',\n",
       " 'dependent',\n",
       " 'on',\n",
       " 'the',\n",
       " 'formation',\n",
       " 'of',\n",
       " 'bio-molecular',\n",
       " 'complexes',\n",
       " ',',\n",
       " 'and',\n",
       " 'in',\n",
       " 'particular',\n",
       " ',',\n",
       " 'protein–protein',\n",
       " 'associ-ation',\n",
       " ',',\n",
       " 'which',\n",
       " 'mediates',\n",
       " 'the',\n",
       " 'assembly',\n",
       " 'of',\n",
       " 'cellular',\n",
       " 'structures',\n",
       " ',',\n",
       " 'signal',\n",
       " 'transduction',\n",
       " 'and',\n",
       " 'inhibition',\n",
       " ',',\n",
       " 'immune',\n",
       " 'response',\n",
       " 'and',\n",
       " 'protein',\n",
       " 'recruit-',\n",
       " 'ment',\n",
       " 'to',\n",
       " 'membrane',\n",
       " 'scaffolds',\n",
       " '.',\n",
       " 'Interactions',\n",
       " 'of',\n",
       " 'proteins',\n",
       " 'with',\n",
       " 'other',\n",
       " 'macromolecules',\n",
       " 'also',\n",
       " 'play',\n",
       " 'key',\n",
       " 'roles',\n",
       " 'in',\n",
       " 'applications',\n",
       " 'such',\n",
       " 'as',\n",
       " 'inhibition',\n",
       " 'of',\n",
       " 'protein',\n",
       " 'activity',\n",
       " 'by',\n",
       " 'drugs1',\n",
       " ',',\n",
       " 'gene',\n",
       " 'editing2',\n",
       " 'and',\n",
       " 'the',\n",
       " 'assembly',\n",
       " 'of',\n",
       " 'new',\n",
       " 'materials3',\n",
       " '.',\n",
       " 'A',\n",
       " 'deep',\n",
       " 'understanding',\n",
       " 'of',\n",
       " 'such',\n",
       " 'processes',\n",
       " 'and',\n",
       " 'the',\n",
       " 'ability',\n",
       " 'to',\n",
       " 'fully',\n",
       " 'plan',\n",
       " 'and',\n",
       " 'control',\n",
       " 'their',\n",
       " 'design',\n",
       " 'would',\n",
       " 'arise',\n",
       " 'from',\n",
       " 'a',\n",
       " 'full',\n",
       " 'map',\n",
       " 'of',\n",
       " 'the',\n",
       " 'conformation',\n",
       " 'space',\n",
       " 'of',\n",
       " 'the',\n",
       " 'interacting',\n",
       " 'protein',\n",
       " 'pairs',\n",
       " '.',\n",
       " 'Such',\n",
       " 'a',\n",
       " 'micro-',\n",
       " 'scopic',\n",
       " 'model',\n",
       " 'would',\n",
       " 'involve',\n",
       " 'representative',\n",
       " 'samples',\n",
       " 'of',\n",
       " 'all',\n",
       " 'structures',\n",
       " '(',\n",
       " 'bound',\n",
       " ',',\n",
       " 'dissociated',\n",
       " ',',\n",
       " 'intermediates',\n",
       " ',',\n",
       " 'etc',\n",
       " '.',\n",
       " ')',\n",
       " 'in',\n",
       " 'atomic',\n",
       " 'resolution',\n",
       " ',',\n",
       " 'their',\n",
       " 'equilibrium',\n",
       " 'populations',\n",
       " 'and',\n",
       " 'the',\n",
       " 'transition',\n",
       " 'rates',\n",
       " 'between',\n",
       " 'them',\n",
       " '.',\n",
       " 'However',\n",
       " ',',\n",
       " 'such',\n",
       " 'a',\n",
       " 'complete',\n",
       " 'microscopic',\n",
       " 'picture',\n",
       " 'has',\n",
       " 'not',\n",
       " 'yet',\n",
       " 'been',\n",
       " 'achieved',\n",
       " 'for',\n",
       " 'protein–protein',\n",
       " 'interaction',\n",
       " '.',\n",
       " 'Experiments',\n",
       " 'can',\n",
       " 'probe',\n",
       " 'either',\n",
       " 'individual',\n",
       " 'aspects',\n",
       " 'of',\n",
       " 'the',\n",
       " 'microscopic',\n",
       " 'picture',\n",
       " 'or',\n",
       " 'averaged',\n",
       " 'ensembles',\n",
       " ',',\n",
       " 'such',\n",
       " 'as',\n",
       " 'the',\n",
       " 'most',\n",
       " 'stable',\n",
       " 'structures',\n",
       " ',',\n",
       " 'association',\n",
       " 'kinetics',\n",
       " 'and',\n",
       " 'binding',\n",
       " 'affinities',\n",
       " '.',\n",
       " 'Mutational',\n",
       " 'analyses',\n",
       " 'can',\n",
       " 'identify',\n",
       " 'amino',\n",
       " 'acids',\n",
       " 'stabilizing',\n",
       " 'associated',\n",
       " 'states',\n",
       " 'or',\n",
       " 'transition',\n",
       " 'states4',\n",
       " '.',\n",
       " 'The',\n",
       " 'structure',\n",
       " 'of',\n",
       " 'the',\n",
       " 'transient',\n",
       " 'encounter',\n",
       " 'complex',\n",
       " ',',\n",
       " 'that',\n",
       " 'is',\n",
       " ',',\n",
       " 'the',\n",
       " 'set',\n",
       " 'of',\n",
       " 'structures',\n",
       " 'found',\n",
       " 'immediately',\n",
       " 'before',\n",
       " 'the',\n",
       " 'transition',\n",
       " 'state',\n",
       " 'of',\n",
       " 'protein',\n",
       " 'association',\n",
       " ',',\n",
       " 'has',\n",
       " 'been',\n",
       " 'probed',\n",
       " 'using',\n",
       " 'NMR5',\n",
       " 'and',\n",
       " 'coarse-grained',\n",
       " 'simulations6–9',\n",
       " '.',\n",
       " 'Binding',\n",
       " 'free',\n",
       " 'energies',\n",
       " 'can',\n",
       " 'be',\n",
       " 'computed',\n",
       " 'from',\n",
       " 'atomistic',\n",
       " 'simulations',\n",
       " 'when',\n",
       " 'the',\n",
       " 'bound',\n",
       " 'pose',\n",
       " 'is',\n",
       " 'precisely',\n",
       " 'known10',\n",
       " ',',\n",
       " 'and',\n",
       " 'the',\n",
       " 'binding',\n",
       " 'free',\n",
       " 'energy',\n",
       " 'landscape',\n",
       " 'can',\n",
       " 'be',\n",
       " 'reconstructed',\n",
       " 'using',\n",
       " 'metadynamics',\n",
       " 'given',\n",
       " 'a',\n",
       " 'set',\n",
       " 'of',\n",
       " 'pre-defined',\n",
       " 'collective',\n",
       " 'variables11',\n",
       " '.',\n",
       " 'Inference',\n",
       " 'of',\n",
       " 'kinetics',\n",
       " 'then',\n",
       " 'requires',\n",
       " 'a',\n",
       " 'kinetic',\n",
       " 'model12',\n",
       " 'or',\n",
       " 'additional',\n",
       " 'unbiased',\n",
       " 'simulations13',\n",
       " '.',\n",
       " 'A',\n",
       " 'candidate',\n",
       " 'to',\n",
       " 'reveal',\n",
       " 'a',\n",
       " 'complete',\n",
       " 'microscopic',\n",
       " 'model',\n",
       " 'of',\n",
       " 'protein–',\n",
       " 'protein',\n",
       " 'interaction',\n",
       " 'is',\n",
       " 'unbiased',\n",
       " 'atomic-detail',\n",
       " 'molecular',\n",
       " 'dynamics',\n",
       " '(',\n",
       " 'MD',\n",
       " ')',\n",
       " 'simulation',\n",
       " '.',\n",
       " 'Microscopic',\n",
       " 'models',\n",
       " 'of',\n",
       " 'biomolecular',\n",
       " 'processes',\n",
       " 'have',\n",
       " 'recently',\n",
       " 'been',\n",
       " 'achieved',\n",
       " 'due',\n",
       " 'to',\n",
       " 'combined',\n",
       " 'advances',\n",
       " 'in',\n",
       " 'high-',\n",
       " 'performance',\n",
       " 'computing',\n",
       " 'and',\n",
       " 'Markov',\n",
       " 'modelling14–19',\n",
       " '.',\n",
       " 'Previous',\n",
       " 'binding',\n",
       " 'studies',\n",
       " ',',\n",
       " 'however',\n",
       " ',',\n",
       " 'have',\n",
       " 'been',\n",
       " 'limited',\n",
       " 'to',\n",
       " 'the',\n",
       " 'association',\n",
       " 'of',\n",
       " 'small',\n",
       " 'ligands',\n",
       " 'to',\n",
       " 'proteins',\n",
       " 'and',\n",
       " 'the',\n",
       " 'co-folding',\n",
       " 'of',\n",
       " 'shorter',\n",
       " 'peptide',\n",
       " 'chains16,19,20',\n",
       " ',',\n",
       " 'or',\n",
       " 'small',\n",
       " 'parts',\n",
       " 'of',\n",
       " 'the',\n",
       " 'association',\n",
       " 'landscape',\n",
       " 'such',\n",
       " 'as',\n",
       " 'the',\n",
       " 'downhill',\n",
       " 'binding',\n",
       " 'step',\n",
       " 'from',\n",
       " 'a',\n",
       " 'pre-bound',\n",
       " 'pose21',\n",
       " '.',\n",
       " 'The',\n",
       " 'simulation',\n",
       " 'of',\n",
       " 'protein–protein',\n",
       " 'association',\n",
       " 'is',\n",
       " 'challenged',\n",
       " 'by',\n",
       " 'the',\n",
       " 'fact',\n",
       " 'that',\n",
       " 'dissociation',\n",
       " 'times',\n",
       " 'exceed',\n",
       " 'accessible',\n",
       " 'MD',\n",
       " 'simulation',\n",
       " 'lengths',\n",
       " 'by',\n",
       " 'far',\n",
       " ',',\n",
       " 'such',\n",
       " 'that',\n",
       " 'long',\n",
       " 'MD',\n",
       " 'simulations',\n",
       " 'will',\n",
       " 'get',\n",
       " 'stuck',\n",
       " 'after',\n",
       " 'a',\n",
       " 'single',\n",
       " 'binding',\n",
       " 'event',\n",
       " ',',\n",
       " 'possibly',\n",
       " 'even',\n",
       " 'in',\n",
       " 'non-native',\n",
       " 'traps',\n",
       " '.',\n",
       " 'Here',\n",
       " 'we',\n",
       " 'demonstrate',\n",
       " 'how',\n",
       " 'reversible',\n",
       " 'protein–protein',\n",
       " 'association',\n",
       " 'can',\n",
       " 'be',\n",
       " 'studied',\n",
       " 'in',\n",
       " 'atomistic',\n",
       " 'detail',\n",
       " 'by',\n",
       " 'combining',\n",
       " 'an',\n",
       " 'extensive',\n",
       " 'and',\n",
       " 'adaptively',\n",
       " 'generated',\n",
       " 'ensemble',\n",
       " 'of',\n",
       " 'unbiased',\n",
       " 'molecular',\n",
       " 'dynamics',\n",
       " 'simulations',\n",
       " 'with',\n",
       " 'hidden',\n",
       " 'Markov',\n",
       " 'modelling',\n",
       " '.',\n",
       " 'We',\n",
       " 'reproduce',\n",
       " 'measured',\n",
       " 'structures',\n",
       " ',',\n",
       " 'energetics',\n",
       " 'and',\n",
       " 'kinetics',\n",
       " 'quantitatively',\n",
       " 'and',\n",
       " 'obtain',\n",
       " 'a',\n",
       " 'full',\n",
       " 'microscopic',\n",
       " 'model',\n",
       " 'of',\n",
       " 'the',\n",
       " 'protein',\n",
       " 'complex',\n",
       " 'structures',\n",
       " ',',\n",
       " 'thermodynamics',\n",
       " 'and',\n",
       " 'kinetics',\n",
       " '.',\n",
       " 'Results',\n",
       " 'and',\n",
       " 'discussion',\n",
       " 'All-atom',\n",
       " 'simulation',\n",
       " 'of',\n",
       " 'reversible',\n",
       " 'protein–protein',\n",
       " 'association',\n",
       " '.',\n",
       " 'As',\n",
       " 'a',\n",
       " 'prototypical',\n",
       " 'example',\n",
       " ',',\n",
       " 'we',\n",
       " 'study',\n",
       " 'the',\n",
       " 'diffusion-limited',\n",
       " 'association',\n",
       " 'of',\n",
       " 'the',\n",
       " 'experimentally',\n",
       " 'well-characterized',\n",
       " 'bacterial',\n",
       " 'ribonuclease',\n",
       " 'barnase',\n",
       " 'with',\n",
       " 'its',\n",
       " 'inhibitor',\n",
       " 'barstar22,23',\n",
       " '.',\n",
       " 'All-atom',\n",
       " 'models',\n",
       " 'of',\n",
       " 'barnase',\n",
       " 'wild',\n",
       " 'type',\n",
       " 'and',\n",
       " 'barstar',\n",
       " 'pseudo',\n",
       " 'wild',\n",
       " 'type24',\n",
       " 'were',\n",
       " 'immersed',\n",
       " 'in',\n",
       " 'a',\n",
       " 'solvent',\n",
       " 'box',\n",
       " '(',\n",
       " 'approximately',\n",
       " '90,000',\n",
       " 'atoms',\n",
       " ')',\n",
       " '.',\n",
       " 'Several',\n",
       " 'thousand',\n",
       " 'unbiased',\n",
       " 'molecular',\n",
       " 'dynamics',\n",
       " '(',\n",
       " 'MD',\n",
       " ')',\n",
       " 'simulations',\n",
       " 'of',\n",
       " '2',\n",
       " 'milliseconds',\n",
       " 'aggregate',\n",
       " 'length',\n",
       " 'were',\n",
       " 'conducted',\n",
       " 'in',\n",
       " 'an',\n",
       " 'automated',\n",
       " 'high-throughput',\n",
       " 'manner',\n",
       " '.',\n",
       " 'These',\n",
       " 'data',\n",
       " 'include',\n",
       " '1.7',\n",
       " 'milliseconds',\n",
       " 'of',\n",
       " 'independent',\n",
       " 'trajectories',\n",
       " 'starting',\n",
       " 'from',\n",
       " 'the',\n",
       " 'dissociated',\n",
       " 'state',\n",
       " 'and',\n",
       " '0.3',\n",
       " 'milliseconds',\n",
       " 'in',\n",
       " 'adaptive',\n",
       " 'MD',\n",
       " 'simulations',\n",
       " '.',\n",
       " 'The',\n",
       " 'adaptive',\n",
       " 'MD',\n",
       " 'launched',\n",
       " 'periods',\n",
       " 'of',\n",
       " 'multiple',\n",
       " 'parallel',\n",
       " 'MD',\n",
       " 'trajectories',\n",
       " ',',\n",
       " 'and',\n",
       " 'the',\n",
       " 'starting',\n",
       " 'points',\n",
       " 'of',\n",
       " 'each',\n",
       " 'period',\n",
       " 'were',\n",
       " 'selected',\n",
       " 'so',\n",
       " 'as',\n",
       " 'to',\n",
       " 'favour',\n",
       " 'the',\n",
       " 'exploration',\n",
       " 'of',\n",
       " 'new',\n",
       " 'states',\n",
       " '(',\n",
       " 'Methods',\n",
       " ',',\n",
       " 'Supplementary',\n",
       " 'Methods',\n",
       " '1',\n",
       " ')',\n",
       " '.',\n",
       " 'This',\n",
       " 'strategy',\n",
       " 'can',\n",
       " 'yield',\n",
       " 'large',\n",
       " 'speed-ups',\n",
       " 'compared',\n",
       " 'to',\n",
       " 'non-adaptive',\n",
       " 'MD',\n",
       " 'runs',\n",
       " 'because',\n",
       " 'slow',\n",
       " 'processes',\n",
       " 'with',\n",
       " 'large',\n",
       " 'energy',\n",
       " 'barriers',\n",
       " 'are',\n",
       " 'broken',\n",
       " 'down',\n",
       " 'into',\n",
       " 'steps',\n",
       " 'with',\n",
       " 'individually',\n",
       " 'smaller',\n",
       " 'barriers',\n",
       " 'that',\n",
       " 'can',\n",
       " 'be',\n",
       " 'sampled',\n",
       " 'efficiently25–27',\n",
       " '(',\n",
       " 'Fig',\n",
       " '.',\n",
       " '1a',\n",
       " ',',\n",
       " 'b',\n",
       " ')',\n",
       " '.',\n",
       " 'When',\n",
       " 'starting',\n",
       " 'from',\n",
       " 'the',\n",
       " 'dissociated',\n",
       " 'state',\n",
       " ',',\n",
       " 'both',\n",
       " 'adaptive',\n",
       " 'and',\n",
       " 'non-adaptive',\n",
       " '1Department',\n",
       " 'of',\n",
       " 'Mathematics',\n",
       " 'and',\n",
       " 'Computer',\n",
       " 'Science',\n",
       " ',',\n",
       " 'Freie',\n",
       " 'Universität',\n",
       " 'Berlin',\n",
       " ',',\n",
       " 'Arnimallee',\n",
       " '6',\n",
       " ',',\n",
       " '14195',\n",
       " 'Berlin',\n",
       " ',',\n",
       " 'Germany',\n",
       " '.',\n",
       " '2Computational',\n",
       " 'Biophysics',\n",
       " 'Laboratory',\n",
       " '(',\n",
       " 'GRIB-IMIM',\n",
       " ')',\n",
       " ',',\n",
       " 'Universitat',\n",
       " 'Pompeu',\n",
       " 'Fabra',\n",
       " ',',\n",
       " 'Barcelona',\n",
       " 'Biomedical',\n",
       " 'Research',\n",
       " 'Park',\n",
       " '(',\n",
       " 'PRBB',\n",
       " ')',\n",
       " ',',\n",
       " 'C/',\n",
       " 'Doctor',\n",
       " 'Aiguader',\n",
       " '88',\n",
       " ',',\n",
       " '08003',\n",
       " 'Barcelona',\n",
       " ',',\n",
       " 'Spain',\n",
       " '.',\n",
       " '3Institució',\n",
       " 'Catalana',\n",
       " 'de',\n",
       " 'Recerca',\n",
       " 'i',\n",
       " 'Estudis',\n",
       " 'Avançats',\n",
       " '(',\n",
       " 'ICREA',\n",
       " ')',\n",
       " ',',\n",
       " 'Passeig',\n",
       " 'Lluis',\n",
       " 'Companys',\n",
       " '23',\n",
       " ',',\n",
       " '08010',\n",
       " 'Barcelona',\n",
       " ',',\n",
       " 'Spain',\n",
       " '.',\n",
       " '*e-mail',\n",
       " ':',\n",
       " 'nuria.plattner',\n",
       " '@',\n",
       " 'fu-berlin.de',\n",
       " ';',\n",
       " 'frank.noe',\n",
       " '@',\n",
       " 'fu-berlin.de',\n",
       " 'ARTICLES',\n",
       " 'PUBLISHED',\n",
       " 'ONLINE',\n",
       " ':',\n",
       " '5',\n",
       " 'JUNE',\n",
       " '2017',\n",
       " '|',\n",
       " 'DOI',\n",
       " ':',\n",
       " '10.1038/NCHEM.2785',\n",
       " 'NATURE',\n",
       " 'CHEMISTRY',\n",
       " '|',\n",
       " 'ADVANCE',\n",
       " 'ONLINE',\n",
       " 'PUBLICATION',\n",
       " '|',\n",
       " 'www.nature.com/naturechemistry',\n",
       " '1',\n",
       " '©',\n",
       " '2017',\n",
       " 'Macmillan',\n",
       " 'Publishers',\n",
       " 'Limited',\n",
       " ',',\n",
       " 'part',\n",
       " 'of',\n",
       " 'Springer',\n",
       " 'Nature',\n",
       " '.',\n",
       " 'All',\n",
       " 'rights',\n",
       " 'reserved',\n",
       " '.',\n",
       " 'mailto',\n",
       " ':',\n",
       " 'nuria.plattner',\n",
       " '@',\n",
       " 'fu-berlin.de',\n",
       " 'mailto',\n",
       " ':',\n",
       " 'frank.noe',\n",
       " '@',\n",
       " 'fu-berlin.de',\n",
       " 'http',\n",
       " ':',\n",
       " '//dx.doi.org/10.1038/nchem.2785',\n",
       " 'http',\n",
       " ':',\n",
       " '//www.nature.com/naturechemistry',\n",
       " 'simulations',\n",
       " 'reproducibly',\n",
       " 'found',\n",
       " 'the',\n",
       " 'native',\n",
       " 'bound',\n",
       " 'complex',\n",
       " '.',\n",
       " 'Adaptive',\n",
       " 'simulations',\n",
       " 'required',\n",
       " 'a',\n",
       " 'total',\n",
       " 'simulation',\n",
       " 'effort',\n",
       " 'of',\n",
       " 'a',\n",
       " 'few',\n",
       " 'microseconds',\n",
       " ',',\n",
       " 'roughly',\n",
       " 'an',\n",
       " 'order',\n",
       " ...]"
      ]
     },
     "execution_count": 24,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "tokens"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [],
   "source": [
    "text = nltk.Text(tokens)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Complete protein-protein association kinetics in atomic detail revealed...\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "'Complete protein-protein association kinetics in atomic detail revealed...'"
      ]
     },
     "execution_count": 26,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "textname = text.name\n",
    "print(textname)\n",
    "text.name"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "NATURE CHEMISTRY; loosely bound; bound state; molecular dynamics;\n",
      "Supplementary Methods; wild type; Macmillan Publishers; Publishers\n",
      "Limited; rights reserved; protein–protein association\n"
     ]
    }
   ],
   "source": [
    "# Collocations: words occuring together\n",
    "text.collocations(num=10)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "the_rate\n"
     ]
    }
   ],
   "source": [
    "# find common context of two words\n",
    "text.common_contexts([\"binding\", \"association\"])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Displaying 3 of 5 matches:\n",
      " atomic detail revealed by molecular dynamic\n",
      "tein–protein association kinetics in atomic detail revealed by molecular dynamic\n",
      "ociated , intermediates , etc . ) in atomic resolution , their equilibrium popul\n"
     ]
    }
   ],
   "source": [
    "# occurances of expression within context\n",
    "text.concordance('Markov' and 'molecular' and 'atomic', width=100, lines=3)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0\n",
      "ConcordanceLine(left=[], query='atomic', right=['detail', 'revealed', 'by', 'molecular', 'dynamics', 'simulations', 'and', 'Markov', 'modelling', 'Complete', 'protein–protein', 'association', 'kinetics', 'in', 'atomic', 'detail', 'revealed', 'by'], offset=5, left_print='', right_print='detail revealed by molecular dynami', line=' atomic detail revealed by molecular dynami')\n",
      "1\n",
      "ConcordanceLine(left=['protein-protein', 'association', 'kinetics', 'in', 'atomic', 'detail', 'revealed', 'by', 'molecular', 'dynamics', 'simulations', 'and', 'Markov', 'modelling', 'Complete', 'protein–protein', 'association', 'kinetics', 'in'], query='atomic', right=['detail', 'revealed', 'by', 'molecular', 'dynamics', 'simulations', 'and', 'Markov', 'modelling', 'Nuria', 'Plattner1*', ',', 'Stefan', 'Doerr2', ',', 'Gianni', 'De', 'Fabritiis2,3'], offset=20, left_print='ein–protein association kinetics in', right_print='detail revealed by molecular dynami', line='ein–protein association kinetics in atomic detail revealed by molecular dynami')\n",
      "2\n",
      "ConcordanceLine(left=['model', 'would', 'involve', 'representative', 'samples', 'of', 'all', 'structures', '(', 'bound', ',', 'dissociated', ',', 'intermediates', ',', 'etc', '.', ')', 'in'], query='atomic', right=['resolution', ',', 'their', 'equilibrium', 'populations', 'and', 'the', 'transition', 'rates', 'between', 'them', '.', 'However', ',', 'such', 'a', 'complete', 'microscopic'], offset=327, left_print='ciated , intermediates , etc . ) in', right_print='resolution , their equilibrium popu', line='ciated , intermediates , etc . ) in atomic resolution , their equilibrium popu')\n",
      "3\n",
      "ConcordanceLine(left=['2014', ')', '.', '3', '.', 'King', ',', 'N.', 'P.', 'et', 'al', '.', 'Computational', 'design', 'of', 'self-assembling', 'protein', 'nanomaterials', 'with'], query='atomic', right=['level', 'accuracy', '.', 'Science', '336', ',', '1171–1174', '(', '2012', ')', '.', '4', '.', 'Schreiber', ',', 'G.', ',', 'Haran'], offset=6556, left_print='sembling protein nanomaterials with', right_print='level accuracy . Science 336 , 1171', line='sembling protein nanomaterials with atomic level accuracy . Science 336 , 1171')\n",
      "4\n",
      "ConcordanceLine(left=[':', '//acellera.com', 'http', ':', '//dx.doi.org/10.1038/nchem.2785', 'http', ':', '//www.nature.com/reprints', 'http', ':', '//dx.doi.org/10.1038/nchem.2785', 'http', ':', '//www.nature.com/naturechemistry', 'Complete', 'protein–protein', 'association', 'kinetics', 'in'], query='atomic', right=['detail', 'revealed', 'by', 'molecular', 'dynamics', 'simulations', 'and', 'Markov', 'modelling', 'Results', 'and', 'discussion', 'All-atom', 'simulation', 'of', 'reversible', 'protein–protein', 'association'], offset=8648, left_print='ein–protein association kinetics in', right_print='detail revealed by molecular dynami', line='ein–protein association kinetics in atomic detail revealed by molecular dynami')\n"
     ]
    }
   ],
   "source": [
    "# occurances of expression within context as list\n",
    "conclist = text.concordance_list('Markov' and 'molecular' and 'atomic')\n",
    "for n, item in enumerate(conclist):\n",
    "    print (n)\n",
    "    print (conclist[n])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "23"
      ]
     },
     "execution_count": 31,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# count number of occurences\n",
    "text.count('Markov')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "<Figure size 640x480 with 1 Axes>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# Dispersion plot for word occurences\n",
    "text.dispersion_plot(['simulation', 'kinetics', 'Markov'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "MD simulations; MD simulations; MD simulations; MD simulations; MD\n",
      "simulations; MD simulations; MD simulations; MD simulations; MD\n",
      "simulations; MD simulations\n"
     ]
    }
   ],
   "source": [
    "# find specific expression in text\n",
    "text.findall(\"<MD><simulations>\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "87"
      ]
     },
     "execution_count": 34,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "text.index('MD')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['by', 'direct', 'molecular', 'dynamics', '(', 'MD', ')']"
      ]
     },
     "execution_count": 35,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "text[82:89]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAYgAAAE5CAYAAACK1bf2AAAABHNCSVQICAgIfAhkiAAAAAlwSFlzAAALEgAACxIB0t1+/AAAADl0RVh0U29mdHdhcmUAbWF0cGxvdGxpYiB2ZXJzaW9uIDMuMC4wLCBodHRwOi8vbWF0cGxvdGxpYi5vcmcvqOYd8AAAIABJREFUeJzt3XmYXGWZ9/Hv3fuStbM2JCSBhCBLwHQDAZEtuCGoIOK4EZWRcR0ZZhyccRm312VUdGRGFEUEFRUVhIRFICZgICwdyIYEEkJCErJvnaTT+/3+cZ5KqivV3dXVXV3VXb/PddVVdU499Zy7uqvr7vNsx9wdERGRRAXZDkBERHKTEoSIiCSlBCEiIkkpQYiISFJKECIikpQShIiIJKUEISIiSSlBiIhIUkoQIiKSVFG2A+iN0aNH++TJk9N67cGDBykvL+/TsqpTdapO1ZlrdSazZMmSHe4+ptuC7j5gbzU1NZ6uurq6Pi+rOlWn6lSduVZnMkCdp/AdqyYmERFJSglCRESSUoIQEZGklCBERCQpJQgREUlKCUJERJLKywRR39jCI2sbaG/X1fRERDqTdwnC3bn8x09w05J6/rZmR7bDERHJWXmXIMyMy2ceDcAvH38ly9GIiOSuvEsQAP9w+jGUFMCCF7fzyo4D2Q5HRCQn5WWCqKos4Y2TojVMbntiXXaDERHJUXmZIAAunloBwB+XbGRfY0uWoxERyT15myAmjyjmzClV7G9q5U9LNmY7HBGRnJO3CQLgI2+YDMBti9dryKuISIK8ThAXvW4cR48o55UdB3j0pe3ZDkdEJKfkdYIoKizgqrMmAXCrOqtFRDrI6wQB8N7TJ1JWXMBjL21nzbb92Q5HRCRn5H2CGFFRwmWvnwDA7YvXZTUWEZFckvcJAuDDZ08GoiGv9RryKiICKEEAMH38UM4+bhQNzW3c+cyGbIcjIpITlCCC2FnE7YvX06YhryIiShAxs183jolV5by6q4EFq7ZlOxwRkaxTgggKC4w5Z00G4Jca8ioiogQR7z21EykvLmTRmh2s3rov2+GIiGSVEkSc4eXFvLsmulaEJs6JSL5TgkgQa2a669mN7G3QkFcRyV9KEAmmjRvKG6eNprGlnd/XvZrtcEREskYJIolDq7w+oSGvIpK/lCCSOP/4sUwaVcGmPQd55IWt2Q5HRCQrlCCSKIgf8vr4uqzGIiKSLRlNEGa2zsxWmNlSM6sL+6rM7GEzWx3uR4b9ZmY/MrM1ZrbczGZmMrbuXFE7gcqSQhav3cm6PeqsFpH80x9nEBe4+2nuXhu2Pw/Md/dpwPywDfA2YFq4XQPc1A+xdWpYWTFX1ESrvD6wpiGboYiIZEU2mpjeCdwWHt8GvCtu/+0eeRIYYWbVWYjvkPfUTgRgxbbmbIYhIpIV5p65UTpm9gqwG3Dgp+5+s5ntcfcRcWV2u/tIM5sHfNvdF4X984Hr3b0uoc5riM4wqK6urpk7d25asTU0NFBRUdFlmZZ25/13bcUdfnP5OEoLrdd19rSs6lSdqlN19rZsotra2iVxrTqdc/eM3YCjwv1YYBlwLrAnoczucH8fcE7c/vlATVf119TUeLrq6upSKnfB9xb4pOvn+cpNe/qszp6UVZ2qU3Wqzt6WTQTUeQrf4RltYnL318L9NuBu4Axga6zpKNzHlk7dCEyMe/kE4LVMxpeKaWOHAOhypCKSdzKWIMys0syGxh4DbwZWAvcCc0KxOcA94fG9wFVhNNMsYK+7b85UfKmaqgQhInmqKIN1jwPuNrPYce5w9wfN7BngTjO7GngVeE8ofz9wMbAGaAA+ksHYUjZt7FAAVm9VghCR/JKxBOHua4FTk+zfCcxOst+BT2UqnnQdOoPYrgQhIvlFM6m7cdyYIRiwbscBWtrasx2OiEi/UYLoRnlJIWMqCmltd9bvPJDtcERE+o0SRAomDIta4tQPISL5RAkiBROGFQIaySQi+UUJIgWHziCUIEQkjyhBpGBiSBA6gxCRfKIEkYKjQ4J4eft+XWFORPKGEkQKKosLGDeslKbWdjbtPpjtcERE+oUSRIpiE+ZWb9uX5UhERPqHEkSKDi25oX4IEckTShApOk6L9olInlGCSNG0Q01MShAikh+UIFIUSxAvb9sfu6CRiMigpgSRolFDShlZUcz+pla21DdmOxwRkYxTguiBWEe1+iFEJB8oQfRArKNai/aJSD5QguiBabp4kIjkESWIHjh0dTmdQYhIHlCC6IFp43QGISL5QwmiB8YPK2NIaRG7DjSzc39TtsMREckoJYgeMLPDHdUaySQig5wSRA9N05IbIpInlCB6aKoShIjkCSWIHtIZhIjkCyWIHtJ1IUQkXyhB9NCEkRWUFhWwtb6J+saWbIcjIpIxShA9VFhgHDtGzUwiMvgpQaRB/RAikg8yniDMrNDMnjOzeWF7ipk9ZWarzez3ZlYS9peG7TXh+cmZji1dGskkIvmgP84gPgu8ELf9HeAH7j4N2A1cHfZfDex296nAD0K5nKQzCBHJBxlNEGY2AXg78POwbcCFwB9DkduAd4XH7wzbhOdnh/I5J7Ymk0YyichgZpm8fKaZ/RH4FjAU+Dfgw8CT4SwBM5sIPODuJ5vZSuCt7r4xPPcycKa770io8xrgGoDq6uqauXPnphVbQ0MDFRUVaZVtbXfef9dW2h1+c9k4Sous13VmIk7VqTpVp+pMpra2dom713Zb0N0zcgMuAX4cHp8PzAPGAGviykwEVoTHzwMT4p57GRjV1TFqamo8XXV1db0qO/v7C33S9fN8xcY9fVZnb8qpTtWpOlVnqoA6T+F7PJNNTG8A3mFm64DfETUt/RAYYWZFocwE4LXweGNIGITnhwO7Mhhfr0zVUFcRGeQyliDc/T/cfYK7Twb+Afiru38AWABcEYrNAe4Jj+8N24Tn/xoyXU46dG0IJQgRGaSyMQ/ieuA6M1sDjAJuCftvAUaF/dcBn89CbCnTkhsiMtgVdV+k99x9IbAwPF4LnJGkTCPwnv6Ipy9oLoSIDHaaSZ2m48YMwQzW7WygubU92+GIiPQ5JYg0lRUXMnFkBW3tzvqdB7IdjohIn1OC6IVpuvyoiAxiShC9oH4IERnMlCB6YarOIERkEFOC6IVDCWKrhrqKyOCjBNELsQSxdscB2tpzdk6fiEhalCB6YWhZMeOHldHc2s6GXQ3ZDkdEpE8pQfSSltwQkcFKCaKXjhujjmoRGZyUIHpJZxAiMlgpQfTStLFDAVijRftEZJBRguil+MlyObw6uYhIjylB9FJVZQmjKks40NzGzoNatE9EBg8liD5wXDiL2FjfmuVIRET6jhJEH5imBCEig1CPE4SZjTSzGZkIZqCK9UNsUIIQkUEkpQRhZgvNbJiZVQHLgFvN7IbMhjZwxEYybdqnBCEig0eqZxDD3b0euBy41d1rgIsyF9bAEn8GoZFMIjJYpJogisysGrgSmJfBeAakccNKGVpaxP5mZ+eB5myHIyLSJ1JNEF8F/gKscfdnzOxYYHXmwhpYzIypYUb1C5vrsxyNiEjfSDVBbHb3Ge7+SQB3XwuoDyLOGVOqAHjo+a1ZjkREpG+kmiBuTHFf3rp0xlEA3L9iM61tmjAnIgNfUVdPmtlZwNnAGDO7Lu6pYUBhJgMbaE46ahjVQwrZvL+ZJ9fu4pxpo7MdkohIr3R3BlECDCFKJEPjbvXAFZkNbWAxM94wsQyAuctey3I0IiK91+UZhLs/CjxqZr909/X9FNOAdc4x5fzxhQM8sHIzX3/XyZQUaaK6iAxcqX6DlZrZzWb2kJn9NXbLaGQD0MRhRUwfN5T6xlYWrdme7XBERHqlyzOIOH8AfgL8HGjLXDgD36WnVvPiQ/uYu2wzF54wLtvhiIikLdUziFZ3v8ndn3b3JbFbVy8wszIze9rMlpnZ82b21bB/ipk9ZWarzez3ZlYS9peG7TXh+cm9emdZckkYzfTQ81tobFEuFZGBK9UEMdfMPmlm1WZWFbt185om4EJ3PxU4DXirmc0CvgP8wN2nAbuBq0P5q4Hd7j4V+EEoN+BMHl3JKUcP50BzGwtf3JbtcERE0pZqgpgDfA54AlgSbnVdvcAjsQs1F4ebAxcCfwz7bwPeFR6/M2wTnp9tZpZifDnl0lOrAZi7bHOWIxERSZ9lcnE5MyskSiZTgf8Dvgs8Gc4SMLOJwAPufrKZrQTe6u4bw3MvA2e6+46EOq8BrgGorq6umTt3blqxNTQ0UFFR0adlY+W2N7Tx8fu2U1IIv3jHWMqTjGbqaZ2ZiFN1qk7VOfjrTKa2tnaJu9d2W9Ddu70BVyW7pfLa8PoRwALgjUTrOcX2TwRWhMfPAxPinnsZGNVVvTU1NZ6uurq6Pi8bX+7yHz/uk66f539+bmOf1dlXZVWn6lSd+VNnMkCdp/DdnWoT0+lxtzcCXwHekeJrcfc9wEJgFjDCzGKjpyYAsVllG0PCIDw/HNiV6jFyzaUz1MwkIgNbSgnC3T8Td/sY8HqiWdadMrMxZjYiPC4nun7EC0RnErFZ2HOAe8Lje8M24fm/hkw3IF18SjVm8NhL29l7sCXb4YiI9Fi6U30bgGndlKkGFpjZcuAZ4GF3nwdcD1xnZmuAUcAtofwtwKiw/zrg82nGlhPGDitj1pRRNLe189DzW7IdjohIj6U0Uc7M5hKNQIJokb7XAXd29Rp3X050ppG4fy1wRpL9jcB7UolnoLjk1GoWr93JvOWbeU/txGyHIyLSI6nOpP5e3ONWYL2H0UbSubedXM2X73meRWt2sOtAM1WVXbbKiYjklFT7IB4FVhGt5DoS0HU1U1BVWcI5U0fT1u48sFKd1SIysKSUIMzsSuBpoiagK4GnzEzLfafgkjCaaZ5GM4nIAJNqJ/UXgNPdfY67X0XUh/ClzIU1eLz5pPGUFBbw5Cs72VbfmO1wRERSlmqCKHD3+IWFdvbgtXlteHkx500fgzvct0JnESIycKT6Jf+gmf3FzD5sZh8G7gPuz1xYg8uhZqblShAiMnB0d03qqcA4d/+cmV0OnAMYsBj4TT/ENyhc9LpxlBUXsGT9bjbtOcjRI8qzHZKISLe6O4P4IbAPwN3vcvfr3P1fiM4efpjp4AaLytIiZoeLB923XNerFpGBobsEMTlMeOvA3euAyRmJaJDSEuAiMtB0lyDKunhO7SQ9cP70sQwpLWLFpr2s23Eg2+GIiHSruwTxjJl9LHGnmV1NdJ0HSVFZcSFvOjFqZpqnZiYRGQC6W2rjWuBuM/sAhxNCLdFKrpdlMrDB6NJTq7n7uU3MXbaZs86tzHY4IiJd6jJBuPtW4GwzuwA4Oey+z93/mvHIBqFzpo5heHkxL27dx6t7S6jJdkAiIl1IdS2mBe5+Y7gpOaSppKiAt540HoDHN2hWtYjkNs2G7meXnnoUAA+tPagLCYlITlOC6GdvmDqK0yePpL6pne8/9GK2wxER6ZQSRD8zM77+rpMpMPj1k+tZuWlvtkMSEUlKCSILThg/jIunVtDu8MU/r6S9fcBeeltEBjEliCx570lDGDu0lKUb9nBn3YZshyMicgQliCypKC7gi5ecCMB3HlzF7gO6SJ+I5BYliCy6dEY1Zx07it0NLfz3X1ZlOxwRkQ6UILIo6rA+ieJC43fPbOC5V3dnOyQRkUOUILJs6tihXH3OsbjDl+5ZSZs6rEUkRyhB5IDPXDiV6uFlrNxUzx1Prc92OCIigBJETqgsLeLLocP6u395kR37m7IckYiIEkTOeOvJ4zn3+DHUN7byrfvVYS0i2acEkSPMjK++4yRKCgv407MbeWbdrmyHJCJ5Tgkih0wZXck/nXcsAF/680pa29qzHJGI5LOMJQgzm2hmC8zsBTN73sw+G/ZXmdnDZrY63I8M+83MfmRma8xsuZnNzFRsueyT509lwshyVm3Zx22L1WEtItmTyTOIVuBf3f11wCzgU2Z2IvB5YL67TwPmh22AtwHTwu0a4KYMxpazyksK+cqlJwHwg4dfYtfBtixHJCL5qrtLjqbN3TcDm8PjfWb2AnA08E7g/FDsNmAhcH3Yf7u7O/CkmY0ws+pQT1656MRxzD5hLPNXbeMXS+uZcFx9Sq/bWN/K2F0NlJcUUlZcSFlRAUWFakUUkfRY9H2c4YOYTQYeI7ps6avuPiLuud3uPtLM5gHfdvdFYf984Hp3r0uo6xqiMwyqq6tr5s6dm1ZMDQ0NVFRU9GnZvqxz64FWrn1wB8297IYoNCgptHCLHo8qg387exSVJd0nj2y8d9WpOlVn39WZTG1t7RJ3r+2uXMbOIGLMbAjwJ+Bad683s06LJtl3RPZy95uBmwFqa2u9pia9KzsvWbKEVF+batm+rnN/+QZ+/MjfKSsv77Y+d9h3oAEKi2lsbaexpY3GljbaHA62OgdbD/8oN+2DDQVj+UDNpD6JsyflVKfqVJ39W2dvZDRBmFkxUXL4jbvfFXZvjTUdmVk1sC3s3whMjHv5BOC1TMaX6648fSLHFWxL+wPj7rS0OY2tUbJoamnnvhWb+fYDq5j/wjY+cGb3CUJE8lcmRzEZcAvwgrvfEPfUvcCc8HgOcE/c/qvCaKZZwN587H/oS2ZGSVEBw8qKGTu0jIlVFbx75gQMWLRmBw3NrdkOUURyWCZ7MN8AfAi40MyWhtvFwLeBN5nZauBNYRvgfmAtsAb4GfDJDMaWt8YMLWVqVTHNre0sWr0j2+GISA7L5CimRSTvVwCYnaS8A5/KVDxyWO1Rpaze1cL8F7bx5pPGZzscEclRGgOZh04/qhSA+au26XrYItIpJYg8dMywIo4eUc6O/U0s27gn2+GISI5SgshDZsZFrxsLwCMvbM1yNCKSq5Qg8tRFJ44DYP4L27opKSL5SgkiT505ZRRDSotYtWUfG3Y1ZDscEclBShB5qqSogHOPHw3AfDUziUgSShB57KLXhWamVWpmEpEjKUHksQumj6XA4Mm1O9nX2JLtcEQkxyhB5LGRlSXUTqqipc157CXNqhaRjpQg8tzsMNxV/RAikkgJIs/NDv0QC17cpmtgi0gHShB57rgxlUwZXcnuhhaefVWzqkXkMCWIPGdmzD5BzUwiciQlCDnUzKRlN0QknhKEUDt5JMPLi3l5+wFe2XEg2+GISI5QghCKCws4f/oYQM1MInKYEoQAamYSkSMpQQgA5x0/hqIC45l1u9nboFnVIqIEIcHw8mLOmFJFW7uz8CWtzSQiShAS56JDzUxKECKiBCFxYgli4YvbaNGsapG8pwQhhxwzqoJpY4ewr7GVZ17Zle1wRCTLlCCkg9ilSB/WaCaRvKcEIR1cdGh11224e5ajEZFsUoKQDk6bOJKqyhJe3dXAmm37sx2OiGSREoR0UFhgXBgW71Mzk0h+U4KQI8Q3M4lI/lKCkCO8cdoYSgoLePbV3ext0nBXkXyVsQRhZr8ws21mtjJuX5WZPWxmq8P9yLDfzOxHZrbGzJab2cxMxSXdqywt4qzjRuEOSzY3ZjscEcmSogzW/Uvgf4Hb4/Z9Hpjv7t82s8+H7euBtwHTwu1M4KZwL1ly0evG8uhL2/lpXT0PvLKQyaMrmTSqgsmjDt8fPbKc4kKdhIoMVhlLEO7+mJlNTtj9TuD88Pg2YCFRgngncLtH4yqfNLMRZlbt7pszFZ907e0zjuL2xetZvW0/a3ccYG2S60QUFhgTRpYzaVQlVdbAUVMPUj28PAvRikgmWCbHuocEMc/dTw7be9x9RNzzu919pJnNA77t7ovC/vnA9e5el6TOa4BrAKqrq2vmzp2bVmwNDQ1UVFT0adnBWOeu+gPs8xK27G9j8/42tuxvjR7va2XHwY79E8UF8LapFVx2whCGlXZ+ZjFQ3rvqVJ2Doc5kamtrl7h7bbcF3T1jN2AysDJue0/C87vD/X3AOXH75wM13dVfU1Pj6aqrq+vzsvlW58HmVn9pS73/ZeVmf9+Nj/ik6+f5pOvn+UlfftB/+PBLvq+xJSfiVJ2qM5/rTAao8xS+w/u7AXmrmVUDhPvYOMqNwMS4chOA1/o5NumhsuJCpo0byptPGs+/njWCuZ8+h3OPH8P+plZ+8MhLnPffC7hl0Ss0trRlO1QRSUN/J4h7gTnh8Rzgnrj9V4XRTLOAva7+hwHnlAnDuf2jZ/C7a2Yx85gR7DzQzNfn/Z3Z33+UO+s20KoVYkUGlEwOc/0tsBiYbmYbzexq4NvAm8xsNfCmsA1wP7AWWAP8DPhkpuKSzJt17Cj+9Imz+dlVtUwfN5RNew7y739czlt++BgPrNisNZ5EBohMjmJ6XydPzU5S1oFPZSoW6X9mxptOHMeFJ4zl3mWbuOHhl3h5+wE+8ZtnGVpi1Kx4mlMnjOC0iSOYMWE4o4aUZjtkEUmQyXkQIhQWGJe9fgJvP+Uofv/Mq9y08GVe29vIwhe3s/DF7YfKTawqP5QwTp04gpOPGp7FqEUElCCkn5QUFfChsybzwVmTePBvz9A+YiLLNu5h6YY9rNi4lw27DrJh10HmLY+6ngoLjGOGFXLepueZdWwVZ0wZRVVlSZbfhUh+UYKQfmVmjK0spGZGNW+fUQ1Aa1s7a7bvZ9mGPSzdsJdlG/bw4tZ9vLKnlVeeWMcvn1gHwPRxQznz2CrOnDKKM6ZUMWaomqVEMkkJQrKuqLCAE8YP44Txw3jv6dG+g81t3Dn/KXYXj+aptbt49tXdvLh1Hy9u3cfti9cDcNyYSs48dhRnTqli/45mSjftpay4gNKiQsqKCykrLqCsuFDLgYikSQlCclJ5SSEnjy2lpuZ4AJpa21i+cS9Prd3JU6/som7dbl7efoCXtx/gjqdejV60YFHSugoLjLKiKFkMK27n2sJNXDrjKAoKrL/ejsiApAQhA0JpUSGnT67i9MlVfBpoaWuPEsYrO1mybjcbt++msKScxtY2mlraaWxpi26t7bS1Owea2zjQ3MZO4LO/W8pNC1/mc2+ZzoUnjMVMiUIkGSUIGZCKCwuomTSSmkkjAViyZAk1NTVJy7a0xRJGO7f+5WnuXt3Mqi37uPq2OmonjeTf33oCZ0yp6s/wRQYENc7KoFdcWMDQsmLGDC1l9pQKFvzb+Xzx7a+jqrKEuvW7ufKni/nwrU/z/Gt7sx2qSE5RgpC8U1ZcyD++8Vge/dz5XHvRNCpLCln44nbe/qNFfPqOZ3klydLmIvlITUySt4aWFXPtRcfzoVmT+PHCl/nVk+uZt3wzD6zcwpW1ExhvB9lU1P2akRs3NVIwZjfjh5cxZkgpRRo1JYOEEoTkvVFDSvnSJSdy9TlT+J9HVvOHJRv47dMboiefei61Sp54AgAzGD2klPHDyhg3rIzxw+Mfl7GzvpVpjS0MLS1S57jkPCUIkeCoEeV854oZfOzcY7ntiXWs3bSVkSNHdvu6Ldt30VRQxpb6Rnbsb2L7vui2YlMnfRp/eYiKkkLGDy9j/LDoNi48jiWSLftb2bCrIaW4dx1sY29DC6XFBZQWFSjxSJ9RghBJMHXsEL7+rpNZsqSJmpqZ3ZaPH0HV0tbO9n1NbKlvZOveRrbUN3Z4vH7bXvY0QUNzG2u3H2Dt9i76Ox5YkHrQ8x4CojOY0jDno6zo8GTB0uJCaG7glA0rwnXFK5k8qoKJVRWUFRemfhzJK0oQIn2ouLCAo0aUc9SI5NfmXrJkCTNnzqS+sZWt9Y1s2dsxgWyNJZTd+ykpSW0pkQONTbRRQFNLO81t7TS2RDdoOaLssq2vdtg2g+phZVHCGF3BpFGVTKqqYOu2Jli/+1CCiRJOAeUlUeLRJMP8oAQh0s/MjOHlxQwvL+b4cUOTlulqXkdXZdvanabWtpAk2g7N/2hsbeOZZX+neORRrN95gHU7G1i/8wAbdh/ktb2NvLa3kcVrd3as+NEnOj1mSWEBpcUFmLdT8sDDKcVZ5G2MW7yI4RUljCgvZkRFMSPKiztuVxSzfm8LQ7bs67Y+M9jd2EZ7uythZYgShMggUlhgVJQUUZFk4VvfXkZNzZQO+1ra2nltz8FDCWPdjgZe3XWAzTt2U1RWSVNCkok9bm6LbgA0Nacc35YDKc41eeixlOssuu8Bxg4tPdSPMz7uflzo42loaaelrV3rcvWQEoRIHisuLIialUZVAmMO7e/qDMbdaWptp6mlnbrnnmPGjFO7PY6789Szyzj62OPZ29DCnoPN7GloYU9DC3sPtrCnoZk9B6PtnXv3U15e1m2d7Q7b9hygvtkPnQV16c8PdFiXK+qbKejQV1NWXMjB/XupXrOU0rgFHzuWie43bGxkW0lqV0Ze/1ojB4ft6Nhkl9B8l4vDo5UgRKRHzOzQF9uIssKUl10/amgRM4/pflRYT5vXTppxGtvqm44YEBD/eEd9I83t3mFdri5t3JTS8Vn8bGrlAB5/qsuniwqM4gKofODhsCJxQjIpiktqxYWMaGsgxR9T2pQgRGRAKysu5JhRFRwzqqLTMrHBAS1tfqip7PCijh2bz154aTXVEybR2NpOU0sbTa0J/TlhEchtO3YyckT3Cc9xtu/cQ2nFkHCc9sNNd3F1t7Y7re1wcH9qTXZvmNj9WVZvKUGISF4wM0qKjJKiAoaVFXdarurgRmpqJnZbX7oDCZJxd1ranCfrlnDCSad0mrziVylu2pniWU4vKEGIiGRZLHlVFhcwdmhqZwZLluzIcFRarE9ERDqhBCEiIkkpQYiISFJKECIikpQShIiIJKUEISIiSSlBiIhIUubu2Y4hbWa2HVif5stHA6kOJE61rOpUnapTdeZanclMcvcx3ZZy97y8AXV9XVZ1qk7VqTpzrc7e3NTEJCIiSSlBiIhIUvmcIG7OQFnVqTpVp+rMtTrTNqA7qUVEJHPy+QxCRES6oAQhIiJJKUGIiEhSumBQisysGtjl7k39fNxKoNHd2xL2m3fTgZRKmT6IbyQwDTh0lRN3fyyTx5TBy8yGuPv+8Hiqu6/JdkwxZlaa+PefbN9gkvdnEGY2PsWivwJWmdn3elj/r8L9Z1MsX2Bm7zez+8xsG7AK2Gxmz5vZd81sWii6wMw+Y2bHJLy+xMwuNLPbgDk9iTWujnFmdkm4je2i3D8CjwF/Ab4a7r/SRfnY8XItAAAXCElEQVSzw3u7KnZLUqbCzL5kZj8L29PM7JJ03kdcnW8IiRYz+6CZ3WBmk9KNMa7s+K62MynVOM3sT2b2djPr9G/dzArN7Nc9OHZVklvn1/DsmcfN7M9mdiXR5ylt4bN0asK+Y8zs6C5eU2tmJZ08vTjFfZjZ8WY238xWhu0ZZvbFFON+LpVy/SHvRzGZ2X3u/vYUyxpwors/H7dvHPBN4Ch3f5uZnQic5e63hOf/DrwNuBc4H7D4Ot19V8IxHgUeAe4BVrp7e9hfBVwAvB+4G/gj8FHgA8AUYA/Rf/GFwEPA/7n7UjPbB3T6S3b3YQnHvxL4LrAwxPpG4HPu/sckP48VwOnAk+5+mpmdAHzV3d+bpOyvgOOApUDsbMjd/Z8Tyv0eWAJc5e4nm1k5sNjdT0tS54+SvKW9RDNM74krtxw4FZhBlOhvAS539/PSiTGufIfPTg8/Sx9x91vD4+u6KuvuN6Qbp5ldBHwEmAX8Afilu69KUu4vwKXu3pxC7OuAicBuos/ICGAzsA34mLsvCeVGAFcBk4lrrYiP08wqgGZ3b43b9wngf4F/cPc/JBz7v4FvAAeBB4l+r9e6+xEJLiStVcAMdz8Q9j0E/Ke71yUpX020dM9H4+sLif9o4NdEf3+xv+FhwE/c/YQkdT0KfA74qbu/Puxb6e4nJ5ZNVU/ee5/J9FTtwX4DHgCuBJaF7SJgRdzz/wy8ADQBa+NurwBrk9RXnMIxixO3gWpgRBev+RrwSWAo0Qf7E8C/Jym3DBgbtz0m9t6SlH0m3C8FSmOPOyn7AuEfkm7eW124fy4+pk7K3kx0BvOZcFsI/B9RMv5hXLlnw/2Xgavj96UTYx99bl6Ne/xf4XYHsBr4fri9BPy8L+IEhgMfBzYATxAljeK4538KPAN8Cbguduukrp8Ab4nbfjNwA1ESeipu/xNh/0eIzmbnAHMS6noSGB+3fRmwHLgIuC/JsZfGlbsNqOrs8xHKfY/oCx/gmPjPVZKynwd+BixM2D8HWADsC/ex271E/2h09bfxXGLs4fE+oL6T2/bwc5ndm/feFzf1QfTeaHe/08z+A8DdW83sUH+Bu/8I+JGZ3UT0h3VueOoxd1+WWJm7t3R3wMQyYXtzNy97i7ufGbd9k5k9Bfx3QrkCd98Wt72TzpsiN4b/Ev8MPGxmu4HXOim7EhifQpzN4azBAczsOKLkmsxU4EIP/32Gn/FDwJuAFXHl9oXfzweBc82skCipphtjSsKZS9KngHGxDXf/aij/EDDT3feF7a8Q/dffqzjNbBTwIaL3/xzwG+Acoi++80Ox18KtgOifiK7UuvvH4+J/yMy+6e7XmVlpXLkyd+/y7Agod/ctIc5rgI8RfTFuN7NvJykf+71dDPzW3XdFJ/ad+jnRl/4viM5mbu2i7IeA84B7zew4d385vL/bgNvM7N3u/qdu3k/MjvDZjX2OryDu9+Xunf6Mw+fzZKLfU/wZR0/fe68pQfTegfAHGPsgzCJq5ki0iugU9S6iL4hfmdnP3P3Gfoqzzcw+APwuxPo+DjdPxHsgNDf8Nmy/F7g/WYXufll4+BUzW0D0X+qDnRx/NPB3M3uauC98d39HQrmvhDommtlvgDcQ/QeazNFAJYd/3pVETX1tZhafVN5L1DRwtbtvsajf5ru9iDFV44C3EDXFxDOi/64THQPEN/E0EzXPRC8ym0v0uxuaapxmdhdwAlHT2iWxL2Pg92ZWF/faWJIaGm1GHcWd2GVm1xN9liD6+e4OX2ztceV+ZWYfA+YlxBnfrLrTzP6LqMnqcmB6SA7VQLK+gLlmtoqomeWTZjYGaOwsUHdfZWaY2fFEn/lzkpUzswuAVe6+w8xuBa4G/jOhrj+Z2duBk+g4KONrSar8FNEZ7glmtomoxeCDncWZcJw2YJmZJX439Oi994W874PoLTObCdxIlOlXEjXJXOHuyxPKLSfqm4i1hVYSta3P6Kc4JwP/Q/SF68DjRO2X6xLKfQd4iugPyYiacGa5+/W9PP55yfa7+6NJyo4iaq4wov6NpEsam9nVwBc53F9yLlF/0G+Br7j75zIVY4r13QLc6u6Lkjx3h7u/P2HfF4iaK+8m+h1dBtzp7t/sKr6u4jSzi4ETiX7v7cAi4CZ3b0wodzJREqkKu3YQ9QM9TwIzG03UJBb7jCwiGqSwFzjGw8gjM/sU8P+I+sdiXzTu7sfG1TWKqLmzGXiZ6Et5GVF/2xfc/Y4kxx8J1Id/BCqAYXGJ7whm9mGi/rpN7v6+Tsr8GrjD3e83s2FE/WDTPfQBhjI/ASpCbD8HrgCedveruzh2JdFZ+b7OyvREwnuvBIZ29d57fTwliN4zsyJgOtEfy4vJmoliHbqxP0wzKyNqpzylX4Pthpk96+4zE/Yt78dENt/dZ3e3L+65auAMop/90+7+Wtxzi9z9nCQd9Ub0RTWMBBYNOjg9bD6d0NyWcWZWw+H/ch9z9yNGtJjZdxITdrJ9Yf+dRO3avwm73geMdPf3JJR7gugLeUHYPh/4pruf3Yv38jJwZmcJvpPXHEWUzJa7+4udlDmbIzu+b++izgqi5p13u/sjSZ4fAdQB0zx8IYaBAL9393lx5Za7+4y4+yHAXe7+5iR1lgLvThJnsrONlIT3cR1REr7GohGN0+Nj7GtqYuobZ3D4gzDTzJJ9YG8FnjKzu8P2u4hG0/SLcDr6MY78wH40PP8Jok7sYxPazocSnW2ke9yUvqRDwqwARof/kuJHihzVxSEKiDr1ioCpFo2dfyy8t3PCfXdt6rFYE0dw3WhmSUdwZdBSoi+zohDTMe7+akKZNwGJyeBtSfZB9AUSP9RzgZkd0fcFVMaSA4C7Lwz/oR5iZj9092vjmro6SNLE9TzQkORYnQoJPlm/SyyGpCO4gE4ThLs3EDV/dvb8HqL+rPh9H0pS9GC4bwiJbCfRCMJk7iE6o1pC531oPXVrqC+WtDcS/ayUIHJVqh9Yd7/BzBZy+LT8I8n+O8yge4C/EQ2hTdb3cAfRiKxvEY3miNmX0GbcIz34kv4n4FqiZLCEwwminmhk0hFCc9h7ib6IYk0BTtQslo4vEJ3lbQv1jyH6efVLgjCzzxA13Wwl+h0Z0fuZEZ5PJ4k/Z2az3P3JUMeZnZRda2ZfImpmgqi9/JWEMrHnUp0L1AYsDf1T8X0QSYcNp6iWaKh5Npo+5oWzje8CzxL9bn7eSdkJ7v7WPj7+ce7+XjN7H4C7H7QM91KriamXzOwFsveBTZmZLfUkcwlyjZl9JtWOezN7kWiMe5/8h2ZmK+Kb/CyaXLasv5oBzWwNUZPMzk6eHw6MJIUkHpo0nWjky3Tg1bA9Cfi7h/H4ZvYrd/+QRXMxJnP4H5hHiea07I6rM9nZTFfvJ+lETY9GBaXFzP4A/LO798lIs17EUUo0SivZgBTM7GbgRndfkez5NI/5BDAbeNzdZ1o0Suq37n5GXx0jkc4geq9Ph0Zm0Dwzu9jdk45IyhXufmPoMD2RjiNFkjUhrCX6AuyrU/gHLcURXBmygeQj4GLc3deFzt8OzKwqIUmkOvu8xqJZ5XOIOl9jZy0Aif+d/hmYGY73J3d/d1cVu/ttFs1KPj7sSto/10N9PdKsRxL7PxKbky2aOd0env+Ima0NccaaVHvTl/cVUh/h1yd0BpEm6zjk8DQgKx/YVIU+gEqiGFvooqM2mywa8ng+UYK4n6htfZG7X5Gk7J+IZpPOp4+aMMzs3UR/eEbUSXx3Ny/pM2HU03TgPjq+nxvC8/Pc/RIze4Xosxf/Bd5hdFAPjvnPRKOIjgU2xT+VWKeZPeeHZwUfetxF3ecTTehaF+qbSDRRLu21uvp6pFkPj93tDHaL5gJ1eqbu7ut7GUNKI/z6ihJEmsIH1YDvAP8e/xTwHe84KS0nWLRcR+LCehn/w+qJ0DRyKtEM1FPDqKKfu/ulScr2eRNGNoXkeAQPcxTiyv2KqJ/lb55k2Yw0j32Tu3+imzKHRrglG+2WpPwS4P2x0UgWzUX4rbvX9DLWrIw0S6U5OZWfSy+O36MRfn1BTUxpin2xmllx4pesRTOBc4pFC+t9FphA9B/QLKLJWhn7cKXpoLu3m1mrRePRtxH9d3uEvkoE6QyHzYTERNCFW4n6Cm40s2OJZkf/zd3/pxfH7jI5BKeaWT3Rz6U8PIbOf07FHjdU1d1fsl4u6pflkWapNCePtS7W1vKEdbVS0YsRfr2mBJGmTA0LzaDPcnhhvQssLKyX5ZiSqQsjRX5GNJppP1Hz3SFmdqe7XxnXEdtBT9t5ezocNlPCaJ9k7+fChO2/WrQY3OlE/QYfJ5qomXaCSIW7F/bwJXWh2Sw2+ukDRL/T3uj3kWbWsxnshcAQjuy/6Y0ej/DrK2piSlNPRpTkAjN7xt1PN7OlRCNlmnJ9ZJNFs7+H+ZGz0qvdfbN1smR3uu28sRE93e3LFIsmycWUEU20anX3f08oN5+oP2kx0dDlRf3VzNITYaTPp+g4K//HvRl1lo2RZj1pTs5wE1PKI/z6is4g0hSGt+0lmpk6EPRkYb2siW9T9bAMSGI7a2yIY287/JI4KSGWIqBX7eU94WGZ7DiPhzOFRMuJ4jqZ6DO4x8wWu/vBJGWzwqJ1mW5x9w8SrejaV/p9pFkPm5MzNi+hhyP8+oQSRJ7wni2s1+960s6apK+gg572GVi00ut/cmS7ejPRgmv9IgwiiCkgSgJHXITI3f8llB9CNMzx1lCuNLFstni0VtAYMyvxFK4x0YN6P5cw0uzmTI8062Fzcsb69Dob4UcXs8h7fUw1MUkusOiKe7F21k0cHo+/j+hL4Ii2VjP7GrCFqI3biNq4h7p74hLmqcbwLXf/j/TeQe8lDF9tJZrJ/DVPWOzPzD5NdCGnGqIL3MRGNP21fyPumpn9lGjexL3Agdj+dDpqsylXmpN7MsKvz46pBCG5xMy+THSxn3qLln6YCXzd3Z9NUvapxOHEyfb18Pg5f41tM/scUVJY4nFXYssVdnh29h7gB4nP92C0VnydOTHSLJvM7Gl3PyMMH76A6J+nle5+UjcvTZuamCTXXOHuXzOzc4gWpfs+cBOQ7Es/1WtcpKSTocCLgQu7el1fCUNAP8Hhi0otJLpkZeIFopJdyyKXxGZnv0q0FH6v5cpIsyzrdoRfX9MZhOSU2AxdM/sW0aVb7+hs1q6leI2LHhw75WtsZ4KZ/Zxo6ZDY/I4PAW3u/o/9cfy+YodnZ0+h40CII2Znp1F3Vkea5YrORvj1+XGUICSXmNk8oj6Ii4ja2A8SzZY9tcsX9s2xszoU2MyWJb7PZPsGilRmZ6dRZ4dhpGGk2XJ3P7Evj5OLNJNaJLqi2luB77n7HosuCJT0ynDWzTUu0pDtocBtFnct5DBLOu0ms2zry+SQKyPNsiGbM6l1BiEDlkXLH/+NqD320Bepp35h+a7qPo8wFLgvh2l2c8zZRENW14Zdk4muG7Kg0xflmWyPNMuGdEb49dmxlSBkoOqr5h8zGxZGTVUle76/hjKG/xT/lcNj6R8GfuAJ14/OdwNhpFkm9GSEX58dUwlCBioz+wbwhPfyGheWfBntQ/e96VTtYRwpXT86n3U20ixxvarByA5fC/sc4JtEI/z+szfDurs9phKEDFQ2QK5xkarB1kmdCdkeaZZNPRnh11cKMlWxSKa5+1B3L3D3cncfFrbTTg5mdlmYNRvbHmFm7+qbaFPynJnNijt+Z9ePzmeNsSY3Myv16HoY07McU3/ZFGanXwncHxZDzOh3uM4gZMAxsxPcfZWZJV01M9022WR9Gpn+Dy0cI6XrRwuY2d1E609dSzSBcTfRdScuzmpg/cDMKohG+K1w99VhhN8p7v5Qxo6pBCEDjZnd7O7X2JHXT4g1MaXVHh1r403Y12F56UywTpYtj/G+X7V2UMjGSLN8owQhA1ZYavmTRNcbcKIhrzelO+rHzH4B7CG6CIsDnyHqJP5wnwQsaetshFlMfy6al0+UIGTA6mTUzwh3vzLN+iqBLxHN4jbgIeAb7n6gyxdKxiWMMEvUbyPN8o0ShAxYmRr1Y9G1sNvdfX+vAhQZ4LTUhgxkz5nZLHd/Eno/6sfMTiG6+EpV2N4BzHH3lX0RrPSemZ2bbH8+TJTLBp1ByICTqVE/YemOL8SWtjCz84FvuvvZfRG39J6ZzY3bLAPOILouxqCfKJcNOoOQgeiSDNVbGb/ukbsvDP0SkiM84eppZjYRSOsKgtI9JQgZcDI47HNtWOPmV2H7g0SX/ZTctRHQPJEMUYIQOeyjwFeBu4hGyzxKNClLcoSZ3cjhuS8FwGnAsuxFNLipD0IkCTMrJGpyqu+2sPQbM5sTt9kKrHN3LUeSIUoQIoGZ3QF8nOjaEkuIZuneMACuAS2SEVqsT+SwE8MZw7uA+4FjiK4LLTnCzC4xs+fMbJeZ1ZvZvrgrzEkfUx+EyGHFZlZMlCD+191bzEyn2Lnlh8DlRAvW6XeTYTqDEDnsp8A6omtMPBYW0dN/p7llA7BSyaF/qA9CpAtmVuTurdmOQyJmdjrwdaIRZk2x/e5+Q9aCGsR0BiESmNlnzWyYRW4xs2eJrjkgueP/AQ1Es6iHxt0kA9QHIXLYR939f8zsLcAYojkQtxKt6iq5ocrd35ztIPKFziBEDostJX0xcKu7LyP58tKSPY+YmRJEP1EfhEhgZrcCRwNTgFOBQmChu9dkNTA5xMz2EQ0iaAJaOHwVwbSvRS6dU4IQCcwstnRDMVAKjAaOdvcbsxqYSJYoQYgEZvaPwGeBCcBSYBawWEtJZ5+ZneDuq8xsZrLn3f3Z/o4pHyhBiAThOhOnA0+6+2lmdgLwVXd/b5ZDy3tmdrO7X2NmC+J2H/ryUhLPDHVSixzW6O6NAGZW6u6riC5IJFnm7teEhzcB73T3C4AFwF7g37IW2CCnBCFy2EYzGwH8GXjYzO4BXstyTNLRF9293szOAd4E/JIoaUgGqIlJJAkzO49oNdcH3b052/FIxMyec/fXm9m3iNZjuiO2L9uxDUZKECIyYJjZPGATcBFQAxwEnnb3U7Ma2CClBCEiA4aZVQBvJTp7WG1m1cAp7q7Z7hmgBCEiIkmpk1pERJJSghARkaSUIEQCM/uCmT1vZsvNbKmZnZnBYy00s9pM1S/SF7TctwhgZmcBlwAz3b3JzEYDJVkOSySrdAYhEqkGdrh7E4C773D318zsy2b2jJmtNLObzczg0BnAD8zsMTN7wcxON7O7zGy1mX0jlJlsZqvM7LZwVvLHMAqnAzN7s5ktNrNnzewPZjYk7P+2mf09vPZ7/fizEAGUIERiHgImmtlLZvbjMFEO4H/d/XR3PxkoJzrLiGl293OBnwD3AJ8CTgY+bGajQpnpwM3uPoPo+tafjD9oOFP5InCRu88E6oDrzKwKuAw4Kbz2Gxl4zyJdUoIQAdx9P9HEq2uA7cDvzezDwAVm9lRYyO9C4KS4l90b7lcAz7v75nAGshaYGJ7b4O6Ph8e/Bs5JOPQs4ETgcTNbCswBJhElk0bg52Z2OdFlNkX6lfogRAJ3bwMWAgtDQvgnYAZQ6+4bzOwrRNdCjmkK9+1xj2Pbsb+txIlGidsGPOzu70uMx8zOAGYD/wB8Gl0fW/qZziBEADObbmbT4nadBrwYHu8I/QJXpFH1MaEDHOB9wKKE558E3mBmU0McFWZ2fDjecHe/H7g2xCPSr3QGIRIZAtwYVnNtBdYQNTftIWpCWgc8k0a9LwBzzOynwGoSVh519+2hKeu3ZlYadn8R2AfcY2ZlRGcZ/5LGsUV6RUttiGSImU0G5oUObpEBR01MIiKSlM4gREQkKZ1BiIhIUkoQIiKSlBKEiIgkpQQhIiJJKUGIiEhS/x8O2wbvFWFCBAAAAABJRU5ErkJggg==\n",
      "text/plain": [
       "<Figure size 432x288 with 1 Axes>"
      ]
     },
     "metadata": {
      "needs_background": "light"
     },
     "output_type": "display_data"
    }
   ],
   "source": [
    "# plot frequency of words in text\n",
    "text.plot(30)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "states limit models separation desolvation assembly mutant set\n",
      "structure parts definition concentration near-native density fig\n",
      "formation order comparison discretization association\n"
     ]
    }
   ],
   "source": [
    "# find similar words according to context\n",
    "text.similar('simulation')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'<Text: Complete protein-protein association kinetics in atomic detail revealed...>'"
      ]
     },
     "execution_count": 38,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "text.unicode_repr()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "FreqDist({'.': 569, ',': 503, 'the': 328, 'of': 245, ')': 198, '(': 198, 'and': 197, 'to': 103, 'a': 99, 'in': 84, ...})"
      ]
     },
     "execution_count": 39,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# analyze text vocabulary\n",
    "vocabulary = text.vocab()\n",
    "vocabulary"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0.0025909532548850263"
      ]
     },
     "execution_count": 40,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# get frequency of given word\n",
    "vocabulary.freq('simulation')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[('.', 569), (',', 503), ('the', 328), ('of', 245), (')', 198)]"
      ]
     },
     "execution_count": 41,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# get most common words in vocabulary\n",
    "vocabulary.most_common(n=5)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "dict_keys(['rate-limiting4', 'atoms', 'loop', 'barnase–barstar', 'Commission', 'Markov', 'pose', 'Tang', '37', 'Multiscale', 'FEFF00550073006100720065002000710075006500730074006500200069006d0070006f007300740061007a0069006f006e00690020007000650072002000630072006500610072006500200064006f00630075006d0065006e007400690020005000440046002000610064006100740074006900200070006500720020006c00610020007300740061006d00700061002000650020006c0061002000760069007300750061006c0069007a007a0061007a0069006f006e006500200064006900200064006f00630075006d0065006e0074006900200061007a00690065006e00640061006c0069002e0020004900200064006f00630075006d0065006e00740069002000500044004600200070006f00730073006f006e006f0020006500730073006500720065002000610070006500720074006900200063006f006e0020004100630072006f00620061007400200065002000520065006100640065007200200035002e003000200065002000760065007200730069006f006e006900200073007500630063006500730073006900760065002e', 'accurate', 'Fabra', '/DownsampleMonoImages', 'lacking', '42', 'mutagenesis', 'His102', 'modelling14–19', 'determine', 'reactions', 'www.nature.com/naturechemistry', '797', 'clockwise', '/CCITTFaxEncode', 'requests', 'dissociation', '5a', 'micromolar', '/CalGrayProfile', 'Frank', 'Despite', 'transitions', 'rates', 'Wild-type', '21', 'happens21,36', '/ESP', 'dimeric', 'respect', '143', '1.000000', 'pcCells', '383–386', 'mech-', '2Computational', 'diffusion', 'grouping', 'self-consistent', 'pocket', 'overlay', 'substrate', 'vice', 'on', '335', 'thus', 'relaxation', 'Correspondence', 'Trp', 'availability', 'describing', 'fast-folding', 'Daumke', '184114', 'Both', 'indicates', 'Code', 'exchange', 'N.', 'confirms', 'diffusion-', 'kon', 'consequences', 'idea', 'Notes', 'Doerr2', 'additional', 'area', 'factor', 'model12', 'Efficient', 'hidden', 'longest', 'method47,48', 'om', 'However', 'actions', '33', '155106', '/MonoImageResolution', '15', '–1', 'dihedral', 'leads', '/AutoPositionEPSFiles', 'speed-ups', 'precisely', '/ENU', 'automatically', 'crispr–cas9-', 'Buch', 'Fig', 'wild', '/EmbedJobOptions', 'NVT', '1632–1639', 'ligands', 'activation', 'interacts', 'plots', 'microsecond', '5525–5542', 'editing2', 'Macmillan', 'Eaton', '3257–3273', 'tightly', 'N.P', 'Multiple', 'transient', '16', 'concentrations', 'Dynamics', 'thank', 'U.S', '/GrayImageMinDownsampleDepth', 'over', 'competing', 'complex', '/SVE', 'high-throughput', 'interface', 'intermediates', 'Courant', '4.00', 'however', 'An', 'correlation', 'field40', 'Selected', 'was', '/AutoRotatePages', '1997', 'balancing', 'associates', 'how', 'nanomaterials', 'quite', 'several', 'speed-up', 'mutant/kon', 'B', 'bacterial', '18', 'Helms', 'rearrange', '12', 'high-', '1.7', 'Total', '35.29000', 'set', 'kinetically', 'Fabritiis', '100', '/Error', 'Several', '248', 'developed/implemented', '/TileWidth', 'automated', '41', '/VSamples', 'Table', 'feedback', 'groove', 'challenged', 'moved', 'previous', 'Scherer', 'ERC', 'sciences', 'association', 'full', 'consists', '10–1', 'hours', '/ParseICCProfilesInComments', '1983', 'identified', 'liquid', 'signal', 'geminate', 'milliseconds', '336', 'μ', 'Madura', 'trajectories25–27', 'Based', '08010', 'ready-to-dock', 'C/', 'first', 'FEFF004200720075006700200064006900730073006500200069006e0064007300740069006c006c0069006e006700650072002000740069006c0020006100740020006f0070007200650074007400650020005000440046002d0064006f006b0075006d0065006e007400650072002c0020006400650072002000650072002000650067006e006500640065002000740069006c0020007000e5006c006900640065006c006900670020007600690073006e0069006e00670020006f00670020007500640073006b007200690076006e0069006e006700200061006600200066006f0072007200650074006e0069006e006700730064006f006b0075006d0065006e007400650072002e0020005000440046002d0064006f006b0075006d0065006e007400650072006e00650020006b0061006e002000e50062006e006500730020006d006500640020004100630072006f0062006100740020006f0067002000520065006100640065007200200035002e00300020006f00670020006e0079006500720065002e', 'pH', '20', 'lifetime', 'avoid', 'TICA', 'Reprints', 'GPUGRID', 'magenta', 'condition', 'random', 'ACEMD', 'Bayesian', '/GrayACSImageDict', 'Berlin', 'non-adaptive', 'most', 'ΔΔGon', 'up', '74', ',', 'minimalist', '/PDFXOutputCondition', 'Allison', 'iv', 'Results', 'pre-', 'associ-', 'periods', 'sequences', '/EmbedAllFonts', 'separation', '.', 'complementarity', '2001', 'Skidmore', 'accuracy', 'so', 'structure', 'follows', '40', 'any', 'Stefan', 'assignment', 'barnase/barstar', 'produce', 'will', 'explained', 'inspiring', 'Three', 'concentration-independent', 'principles', 'protein–', 'Rebinding', 'selecting', 'shown', 'modulation', 'bimolecular', 'programs', 'adaptively', 'Lys', 'wild-type', '/DetectBlends', 'increase', 'anti-', 'Dror', 'event', 'ADVANCE', '/AntiAliasGrayImages', 'Biophysics', '24', '3789–3798', 'theory', 'U1A–SL2', 'funnel', '1Department', 'crystal', 'ability', 'Competing', '44', 'immune', '7653', 'toward', '30', 'interactions', 'systems', 'mapped', 'Misbound', '1917–1929', '174101', 'contacts', '⇌', 'None', 'long', 'How', 'e1002054', 'ff99SB', '/TransferFunctionInfo', '8.50000', 'one', 'generated', 'Additional', 'functionalities', '/AllowTransparency', 'based', 'energies', 'mixture', 'description', '/CreateJDFFile', '/DAN', '130', 'Northrup', 'compared', 'near', 'Mathematics', 'promising', 'Adaptive', '0.2', 'native-state', 'favour', '0.4', 'giving', 'activity', 'run', 'long-timescale', 'described', 'larger', 'conditions', 'rates30', 'quickly', '334', 'We', ')', '/QFactor', 'cellular', '/EncodeColorImages', '[', 'predicted', 'experiences', 'Deutsche', '/ColorImageMinResolution', 'Arrow', 'Optimal', '17', 'I', 'error', 'them', '896–908', 's–1', 'F.', '/PreserveEPSInfo', 'loops', '/False', 'Giupponi', '2c', '/PDFXBleedBoxToTrimBoxOffset', '72', '0.8', 'Drug', 'C', 'fitted', 'Science', 't+50', 'correctly', 'model', 'evaluated', 'Barcelona', 'broken', 'Bonomi', 'studied', 'observed', '/ColorImageMinDownsampleDepth', 'ensembles', 'faster-associating', '5', '65', 'Lluis', 'Einstein', 'more', 'Separation', 'constructed', 'unspecific', 'comparison', 'prototypical', 'learning', '//acellera.com', '408', 'play', 'Bound', 'construction', 'scopic', '/sRGBProfile', 'even', 'X', '/PDFX3Check', 'computing', 'variationally50', 'rate-limiting', 'state', 'x', 'msmtools', '478–486', 'HMMs', 'Iwahara', 'χ2', '/DCTEncode', '0.10', '48', 'type', 'parameters', 'probability', 'required', 'mediated', 'MD', 'extended', 'Simulation', 'host', 'processes', 'Biology', 'previously', 'measurements', 'microseconds', 'network', 'Options', '0.598', 'performance', 'cycles', 'models', '19181–19191', 'Recerca', 'encounter', 'state23', 'concentration-dependent', 'Σjkji', 'docking', 'acids', 'simple', 'Universität', 'Directed', 'Y.', 'sampled', 'necessarily', 'immediately', 'Wade', 'explored', 'superimposed', '2007', '0.06', 'E73A', '129', '13', 'role', 'slowest', 'contributions', 'short-', 'outline', '/GrayImageDict', 'statistical', 'ϕ', 'reduces', 'C40,82A', '6', 'system', 'Chipot', 'deeper', 'Supplementary', 'y', 'surrounding', '/PDFX1aCheck', 'Rapid', '926–935', 'back', '/LeaveColorUnchanged', 'information', 'door', '2g', 'major', '0.00', 'setpagedevice', '958/A4', 'energy', 'clouds', 'no', 'Research', 'kij', 'developed', 'claims', 'tested', 'NpT', 'manipulating', 'comprising', '5145–5150', 'anchor', 'trajectory', 'duplicate', 'protein–DNA', '756', 'top', 'remains', '371', 'observe', 'Conformational', ':', 'W.', 'A', '/ColorImageFilter', 'coupled', 'fu-berlin.de', 'used', '‘', 'predicts', '87', 'on-rate', 'boxes', 'steering', 'make', '/PDFXTrapped', 'fold', '1.00000', 'only', 'substate', 'nc', 'rights', 'v2', '150', 'snippets', 'range', 'Lett', 'E', 'paper', 'By', 'barstar', 'steps', 's–1M–1', 'Levy', '/SubsetFonts', 'Rational', 'transduction', 'Abell', 'Ala', 'recognition', 'successfully', 'significant', 'dissociated', 'population', '1517–1524', 'function', 'removing', 'frames', 'Association', '0.04', '444', 'USA', 'not', 'screening', 'few', 'study31', 'd', 'approaches', 'Mathematical', 'ribonuclease', 'Laboratory', 'mol–1', 'Chem', 'thickness', 'rounds', 'Sarich', '47', 'rate', 'could', '104', 'reproducibly', '/GrayImageMinResolutionPolicy', 'Prakash', 'predict', '2014102337', '015102', 'k-means', '1.6–1.8', 'Spain', 'excluding', 'Biotechnol', 'starting', 'September', 'χ1', 'salt', '–5', 'www.nature.com/naturechemistry2', 'uncertainties', 'Inf', 'stability', 'to', 'minor', 'path', 'interacting', 'www.nature.com/reprints', 'et', 'Introduction', 'Protein–protein', 'mechanisms', 'disappears', 'recovery', 'epigenetic', '–kTlnμ', 'In', '/ITA', 'nucleic', 'seen', 'drugs1', 'diffusional', 'slow', '/Binding', 'Germany', 'identify', 'green', 'between', 'BIO2014-53095-P', 'whereas', 'estimate', 'denotes', '108', 'initial', '/AutoFilterGrayImages', 'McCammon', 'kji', '36', 'smaller', 'loosely', 'estimating', 'Fast', 'thousand', 'depends', 'parts', 'definition', 'than', 'part', 'plot', '308', 'transitions37', 'AssociatedDissociated', 'modelling', 'g', 'permissions', 'materials3', 'mol', 'electrostatic', 'many', '427–431', '/SUO', 'proceeded', 'Only', 'reproduced', 't+150', 'next', '32', 'values', 'intervals', 'point', 'validated', 'approach', 'consensus', 'macromolecular', 'examine', 'majority', '/PageSize', 'get', 'J.', 'likely', 'jurisdictional', 'Native', 'desolvation', 'As', 'nearest-neighbour', 'it', 'Shown', 'microscopy', 'alanine', 'anchors', '/ColorACSImageDict', 'vertical', 'contain', 'resolution', 'around', 'Molgedey', 'length', 'Scheme', 'faster', 'pathways', 'fast', 'regions', 'contact', '787–794', 'pro-', 'Dissociated', 'known', 'native', 'Interactions', 'er', '0.25', 'inhibition', 'peptide', 'passage', 'therefore', '©', 'M', 'lifetimes', 'Inference', 'S.', 'nm', '101', '/OPM', 'as', '1,892', 'tech-', 'cluster', '//www.nature.com/reprints', '16.8', '/Description', 'Louis', 'Anunciado', 'constants', 'line', '’', 'discovery', '/AntiAliasColorImages', 'experimen-', '/Bicubic', 'RNA', 'unbinding', 'Asp', '/OK', 'plasticity', 'at', 'Of', 'currently', 'Gabdoulline', 'atom', 'adaptive', 'accessibility', 'that', 'online', '3.4', '1913–1924', 'understand', 'opens', 'ψ', '80', 're', 'right', 'latter', '12935–12942', 'At', 'when', 'fact', 'reached', 'Lindorff-Larsen', 'PUBLISHED', '/ConvertImagesToIndexed', 'round', 'fraction', 'aggregate', 'heavy', 'experiments', '3', 'Compute', 'further', 'Arg', 'barnase', 'possible', '/EmitDSCWarnings', 'Adds', 'this', 'On', 'assembly', 'complexes', 'Universitat', 'al', 's', 'deep', 'diffusion-map-directed', 'Sosa-Peinado', 'c', '1.3', 'Although', 'T.', '1171–1174', 'built', 'preference', '/EncodeMonoImages', 'Lao', 'techniques', 'gets', '3634–3637', 'sample', 'transition', 'spatial', 'difference', '×', 'process', 'Compatibility', 'Wehmeyer', 'respectively', 'attached', '1994', '/TileHeight', '1668–1688', 'Acad', 'picoseconds', 'Application', '/PTB', 'phosphate', 'pairs', 'Passeig', 'non-native', 'type23', 'II', 'TheE73Amodel', 'bind', 'FEFF00560065007200770065006e00640065006e0020005300690065002000640069006500730065002000450069006e007300740065006c006c0075006e00670065006e0020007a0075006d002000450072007300740065006c006c0065006e00200076006f006e0020005000440046002d0044006f006b0075006d0065006e00740065006e002c00200075006d002000650069006e00650020007a0075007600650072006c00e40073007300690067006500200041006e007a006500690067006500200075006e00640020004100750073006700610062006500200076006f006e00200047006500730063006800e40066007400730064006f006b0075006d0065006e00740065006e0020007a0075002000650072007a00690065006c0065006e002e00200044006900650020005000440046002d0044006f006b0075006d0065006e007400650020006b00f6006e006e0065006e0020006d006900740020004100630072006f0062006100740020006f0064006500720020006d00690074002000640065006d002000520065006100640065007200200035002e003000200075006e00640020006800f600680065007200200067006500f600660066006e00650074002000770065007200640065006e002e', 'gy', 'Binding', 'wide', 'Exacycle', 'barrier', 'European', '/Apply', 'randomly', 'coordinates', '08003', 'Comp', '2.63', 'itt', 'Knocking', 'targets', 'cycle', '43', '/DefaultRenderingIntent', 'DOI', '/GrayImageFilter', 'Time', '110', 'Following', 'Case', 'gene', 'bringing', 'Pande', 'quantitatively', 'conformational', 'assemblies', 'excellent', 'Google', '/CompressObjects', 'Brownian', '/CropGrayImages', 'estimator', '1BRS', 'resulting', 'lit', 'nology', 'msm', 'Identification', 'initially', 'runs', 'U-tautomer', 'Piana', 'probed', 'Huang', '9', 'Smoluchowski', '4th', 'discretization', '2a', 'interest', '2010', 'Cohen', 'copies', '?', '109', 'en', 'metric14', '/MonoImageDepth', 'until', '113', 'the', 'FEFF005500740069006c0069007a006500200065007300740061007300200063006f006e00660069006700750072006100e700f5006500730020007000610072006100200063007200690061007200200064006f00630075006d0065006e0074006f0073002000500044004600200063006f006d00200075006d0061002000760069007300750061006c0069007a006100e700e3006f0020006500200069006d0070007200650073007300e3006f00200061006400650071007500610064006100730020007000610072006100200064006f00630075006d0065006e0074006f007300200063006f006d0065007200630069006100690073002e0020004f007300200064006f00630075006d0065006e0074006f0073002000500044004600200070006f00640065006d0020007300650072002000610062006500720074006f007300200063006f006d0020006f0020004100630072006f006200610074002c002000520065006100640065007200200035002e00300020006500200070006f00730074006500720069006f0072002e', 'inactivation', 'intermediate', 'Barducci', 'Trim', 'Wolynes', 'consisting', 'theory33', 'www.nature.com/naturechemistry6', '/Off', 'King', 'select', 'Zhou', 'future', 'Enhanced', 'biological', 'Plattner', 'NATURE', 'Coated', 'S.D', '5.0', 'a', 'within', 'American', 'Acknowledgements', 'functions', '665.858', 'self-assembling', 'per', 'Intermediate', 'Olsson', 'f', '35–41', 'interact', '/JPN', 'S6', 'Schwarz', 'yet', '0', 'cover', 'power', 'strongly', 'Struct', 'un-protonated', '10–5', 'trajectories', 'either', 'Proteins', 'very', 'low', 'signals', '4,094', 'helix', '69–77', '1984', 'Publisher', 'conducted', 'regular', 'Sampling', 'Forschungsgemeinschaft', 'continues', 'minimized', 'other', 'K.', 'statistics', '5d', 'viously23,32', '8', '/ColorImageAutoFilterStrategy', 'times', 'langevin', 'cloud', 'References', 'Schematic', 'fields', 'k', 'Medicine', 'fixed', '/ColorImageDepth', '8878–8889', 'variety', '59', 'various', 'Adhesive', 'ΔΔGTS', 'published', 'size', '/Left', 'April', 'simulations6–9', 'expect', 'Single-molecule', '35', 'early', 'raw', 'arise', 'structures', 'started', '2.0-Å', 'distribution', 'b→', 'small', 'complete', 'Doctor', '90', 'plan', 'has', 'Network', '2d', 'demonstrated', 'financial', 'Here', 'saving', 'FEFF004200720075006b00200064006900730073006500200069006e006e007300740069006c006c0069006e00670065006e0065002000740069006c002000e50020006f00700070007200650074007400650020005000440046002d0064006f006b0075006d0065006e00740065007200200073006f006d002000700061007300730065007200200066006f00720020007000e5006c006900740065006c006900670020007600690073006e0069006e00670020006f00670020007500740073006b007200690066007400200061007600200066006f0072007200650074006e0069006e006700730064006f006b0075006d0065006e007400650072002e0020005000440046002d0064006f006b0075006d0065006e00740065006e00650020006b0061006e002000e50070006e006500730020006d006500640020004100630072006f0062006100740020006f0067002000520065006100640065007200200035002e00300020006f0067002000730065006e006500720065002e', 'pivotal', '/MonoImageFilter', 'Model', '/PDFXNoTrimBoxError', 'This', 'well-characterized', '/GrayImageResolution', '1.0', 'Accelerated', 'map', 'collectively', 'flexible', 'once', 'are', 'swing', 'every', 'unbiased', 'subdivided', 'Kohlhoff', 'rapidly', 'Ahmad', '2014', 'During', 'residue', 'correlations', '83', 'observation', 'calculating', '/ColorImageDownsampleThreshold', 'seeded', 'dynamics', 'G.', 'misbound', '/Default', '/HSamples', 'critically', 'profound', 'desirable', 'simulation', 'barriers', 'predictions22', 'independent', 'development', 'proxy', '/MonoImageDict', 'together', '1993', 'Impey', 'inserted', 'quantities', 'particular', 'radius', 't+100', 'ensemble', 'bepredicted', 'Barstar', 'Proc', 'atomic', '68', 'Perez-Hernandez', 'Gumbart', 'occur', 'passes', 'each', 'reweighting', 'Pre-bound', '0.21', 'Two', 'continuously', 'H.-X', 'When', '0.02', 'Computer', 'Atomistic', '/ColorImageResolution', 'chosen', 'Hornak', 'Nuria', 'Pompeu', '88', 'rendering', '27', 'contour', 'i', 'E.', 'entro-', 'c→', 'sense', '1a', '839–860', 'lengths', '/PDFXOutputConditionIdentifier', 'equilibrium', 'flexibly', 'difficult', 'ol', 'measured', 'energetics', '0.3', 'fundamental', 'co-folding', '34', 'f∞', 'High-throughput', 'F', 'hundred', 'Gu', 'limited', 'Dhar', 'protein', 'count', 'whose', 'Vol', 'Removing', 'funnels', '86', 'minimal', 'interval', '10184–10189', 'backbone', 'PyEMMA', 'simulated', 'Equilibrium', '/EndPage', 'ref', '/CompressPages', '/PreserveHalftoneInfo', 'electrostatically', 'kcal', ';', '397–403', '/PDFXCompliantPDFOnly', '/UCRandBGInfo', 'hour', 'Web', 'highly', 'Geyer', 'Rev', 'Projected', '139', 'N-terminal', 'number', 'much', 'G', 'binding', 'I.', 'note', 'IEC61966-2.1', 'reduce', 'Baranger', '/DSCReportingLevel', 'improved', 'V.', 'xp', '3a→', 'determined', 'distributed', 'diffusive', 'concentration', '/CropMonoImages', 'multiple', '1.5–4', 'apparent', '–2', 'details', 'topology', 'translation', 'or', 'Doerr', 'reserved', 'basal', 'Free', 'features', 'Ensign', '@', '2,000', 'free', 'andpost-translational', 'slight', '73', 'far-reaching', 'sgrnas', '0.6', '/ASCII85EncodePages', 'barstar22,23', 'field', 'pre-bound', 'Society', 'almost', 'Glu', 'Phys', 'separated', 'combining', 'been', 'bound', '2008', 'after', 'Feature', 'enhanced', 'blue/red', 'active', 'english', 'ta', 'left', 'NPG', 'succeeded', 'kBT=', '-1', 'transitioning', 'natively', 'd→', 'sequence', 'Silva', 'Limited', '1–10', 'ΔΔ', 'Unbinding', 'requires', '102', 'total', 'residue–residue', 'locations', '/GrayImageMinResolution', 'Timescale', 'both', 'made', 'mutation24', 'aspects', 'state22', 'Doench', 'manuscript', 'biomolecular', 'surface', 'studies', 'averaged', 'Molecular', 'some', 'Wild', 'validation', 'de', 'dissociate', 'design', 'additive', '/PreserveCopyPage', '0.1000', 'counting', 'sets', '/CheckCompliance', 'NMR5', 'Dammer', 'red', 'commonly', '/ImageMemory', 'con-', 'rare', 'effect', 'origin', '0.1', 'sampling', '2013', '/DownsampleGrayImages', 'new', 'Companys', '/UsePrologue', 'understanding', '/AllowPSXObjects', 'Author', 'Paul', '2005', '77–85', 'interaction', 'individual', 'Multiensemble', 'were', 'previously10', 'ro', '343', 'observable', 'drug', 'before6,7', 'chains16,19,20', 'far', '1262–1267', 'Complete', 'neutral', 'higher', 'dynamic', 'accepted', '2006', 'analysed', 't', 'detect-', '/JPEG2000ColorImageDict', 'stabilized', 'different', 'Aiguader', 'Schluttig', 'uncertainty', 'stay', '/MonoImageMinResolution', 'proteins', 'false', 'μs–1', 'ku', '1996', 'considered', 'special', 'Giorgino', 'unknown', '18,000', '2,867', 'data', 'Multistep', 'mis-bound', '19', 'refer', '*e-mail', 'linear', 'GPUgrid', 'E3221–E3230', 'StG', 'RMSD', 'approximately', 'Preto', 'Data', 'flexibility', 'box', 'nuria.plattner', 'Publishers', 'then', 'populations', '2064–2069', 'show', 'Lifetime', '5–12', 'present', 'b', 'Energetics', 'imply', 'Frisch', 'key', 'probe', 'which', 'mandatory', 'Kinetic', 'mediates', '/LockDistillerParams', 'domain', 'Best', 'standard', 'Catalana', 'Theory', 'Harvey', 'Schuster', 'GRIB-IMIM', 'four', 'Foundation', 'limits', 'away', 'control', '//pyemma.org', 'Mol', 'strategies', 'Ser', 'mixing', 'HMM', '854.929', 'basin', '–kBTln', '90°', 'effort', 'proportional', '/Remove', 'achieved', 'selection', 'via', 'about', 'formed', 'kc', 'challenge', 'Structurally', 'pharmaceutics', 'parallel', 'eds', 'PUBLICATION', 'pseudo', '200', 'Experiments', 'expected', 'explains', 'also', 'exist', 'Schütte', '450', 'affiliations', 'well', 'setdistillerparams', 'FEFF0055007300650020006500730074006100730020006f007000630069006f006e006500730020007000610072006100200063007200650061007200200064006f00630075006d0065006e0074006f0073002000500044004600200071007500650020007000650072006d006900740061006e002000760069007300750061006c0069007a006100720020006500200069006d007000720069006d0069007200200063006f007200720065006300740061006d0065006e0074006500200064006f00630075006d0065006e0074006f007300200065006d00700072006500730061007200690061006c00650073002e0020004c006f007300200064006f00630075006d0065006e0074006f00730020005000440046002000730065002000700075006500640065006e00200061006200720069007200200063006f006e0020004100630072006f00620061007400200079002000520065006100640065007200200035002e003000200079002000760065007200730069006f006e0065007300200070006f00730074006500720069006f007200650073002e', 'timescale', '2400', 'down', 'O.', 'The', 'defined', 'reflects', 'G.D.F', 'micro-', 'yield', '45', '90,000', 'experimental', 'sRGB', 'ment', 'period', 'states', '117', 'dataset', '/PDFXTrimBoxToMediaBoxOffset', '103', '10.1038/NCHEM.2785', 'SFB', 'Clementi', 'CHEMISTRY', '28', 'PRBB', 'trapped', '–', 'enabled', 'bi', 'these', 'becomes', '”', 'density', 'study', 'conformation', 'move', '/PDFXSetBleedBoxToMediaBox', 'Life', 'characterized', 'D', 'software', 'ID', 'Alamanova', 'thermal', 'due', 'cost', 'robust', 'Clark', 'protein–protein', '46', 'lagtime', 'black-box', 'candidate', 'Sci', 'double-mutant', '0.0', 'lower', '/PDFXRegistryName', 'solvent', '/AntiAliasMonoImages', 'reproduce', 'http', 'displays', '/PreserveFlatness', 'FEFF3053306e8a2d5b9a306f300130d330b830cd30b9658766f8306e8868793a304a3088307353705237306b90693057305f00200050004400460020658766f830924f5c62103059308b3068304d306b4f7f75283057307e305930023053306e8a2d5b9a30674f5c62103057305f00200050004400460020658766f8306f0020004100630072006f0062006100740020304a30883073002000520065006100640065007200200035002e003000204ee5964d30678868793a3067304d307e30593002', 'ARTICLES', '2011', 'Sukenik', 'time', 'available', '23', 'order', '4c', 'events', 'during', 'FEFF0041006e007600e4006e00640020006400650020006800e4007200200069006e0073007400e4006c006c006e0069006e006700610072006e00610020006e00e40072002000640075002000760069006c006c00200073006b0061007000610020005000440046002d0064006f006b0075006d0065006e007400200073006f006d00200070006100730073006100720020006600f600720020007000e5006c00690074006c006900670020007600690073006e0069006e00670020006f006300680020007500740073006b0072006900660074002000610076002000610066006600e4007200730064006f006b0075006d0065006e0074002e0020005000440046002d0064006f006b0075006d0065006e00740065006e0020006b0061006e002000f600700070006e006100730020006d006500640020004100630072006f0062006100740020006f00630068002000520065006100640065007200200035002e003000200065006c006c00650072002000730065006e006100720065002e', '740/D7', 'researchers', '10s', 'Advances', 'summing', '1-σ', 'known10', 'Hartley', 'water', 'occurs', 'feature', 'less', '//htmd.org', 'delayed', 'ns', 'Simul', 'ONLINE', 'enzyme-', 'state⇌', 'have', 'reduced7,22', 'life', '2004', 'prevented', 'attachment', '712–725', 'To', 'pose21', '/MonoImageMinResolutionPolicy', 'involved', 'Crystal', 'orange', '118', '/JPEG2000GrayImageDict', 'picture', 'W', 'rebuild', 'efficiently25–27', 'true', 'Initial', 'Roux', 'folding', 'indeed', 'P.', 'class', 'in', 'DNA', 'Natl', '0.50', 'would', 'Parrinello', 'E4708–E4713', 'minimum', 'materials', 'combination', 'kinetics', 'All-atom', 'D.', 'dependent', 'group', 'landscape', 'Prinz', 'TIP3P', 'R', 'networks', 'roughly', '/CreateJobTicket', 'assigned', '10–4', 'methods', '1845–1852', 'an', 'P', 'samples', 're-', 'Buckle', '38', 'Bleed', 'able', 'Comparison', 'n', 'lag', 'stable', 'site', '100s', 'single', 'none', 'extensive', 'Fundamental', '/PreserveOverprintSettings', 'estimated', '/PDFXOutputIntentProfile', 'such', 'molecules', 'discussion', '4πDb', 'but', 'with34,35', '|', 'before', 'had', '3.5', 'average', 'scaffolds', 'efficient', 'macromolecules', 'Interaction', 'relevant', '1.5', '/CalRGBProfile', 'beyond', 'and', 'mechanism', '3–7', 'disc', 'atomic-detail', '11', 'S.D.', 'Methods', 'level', '36.28000', 'E73', '174105', 'discretized', 'HTMD45', 'combinations', 'CHARMM36', 'probabilities', '49', 'involve', '/None', 'chain', 'escape', 'Free-energy', 'onset', 'N.P.', 'downhill', '256', ']', 'is', 'corresponding', 'double', 'sphere', 'still', 'step', 'frame', '/PreserveDICMYKValues', 'Park', 'detail', 'states4', 'pathway', 'determination', 'opposite', 'indicated', 'including', 'Chung', 'molecular', 'white', 'D.-A.', 'B.', '2015', 'patterns', 'μs', 'Hidden', 'C.', 'μM', 'atomistic', 'obtained', 'shifts', 'polar', 'amino', 'carried', 'reproducing', 'effective', 'Thus', 'stabilize', '2.5', 'e–g', 'from', 'accessible', 'computed', '/ColorSettingsFile', 'orientation', 'Metastability', '2017', 'analysis', 'experiment', '2-σ', 'Microscopic', 'loose–tight', 'computational', 'temperature', '2009', 'From', 'rotation', 'intrinsic', 't+200', '0.5', '/ColorImageDownsampleType', 'experimentally', 'modeling', 'combine', 'See', 'they', 'For', 'preferentially', '2.0', 'limit', 'Discov', 'Cloud-based', 'mutations', 'fit', 'contains', 'towards', 'engineering', 'Experimental', 'dissociates', 'These', 'GPCR', 'predictions', '–4', 'advances', 'Theo', 'formation', 'ii', 'takes', 'residues', 'associate', 'Wu', 'results', '/CompatibilityLevel', 'out', '28.5', 'bottleneck', 'recrossing', 'flux', 'Anderson', '39', 'faces', 'CHARMM', 'stateswhere', '14195', 'U.', 'Biomedical', 'jump', '/EncodeGrayImages', '7', 'Protonation', '//dx.doi.org/10.1038/nchem.2785', 'simulating', '/AutoFilterColorImages', 'institutional', 'metadynamics', 'Nüske', 'individually', 'Matysiak', 'rise', 'Figure', 'Long', 'high', 'compute', 'we', '10', 'Due', 'suffered', 'thermodynamics', '/Optimize', '/CalCMYKProfile', '&', 'FEFF004f007000740069006f006e00730020007000650072006d0065007400740061006e007400200064006500200063007200e900650072002000640065007300200064006f00630075006d0065006e007400730020005000440046002000700072006f00660065007300730069006f006e006e0065006c007300200066006900610062006c0065007300200070006f007500720020006c0061002000760069007300750061006c00690073006100740069006f006e0020006500740020006c00270069006d007000720065007300730069006f006e002e00200049006c002000650073007400200070006f0073007300690062006c0065002000640027006f00750076007200690072002000630065007300200064006f00630075006d0065006e007400730020005000440046002000640061006e00730020004100630072006f0062006100740020006500740020005200650061006400650072002c002000760065007200730069006f006e002000200035002e00300020006f007500200075006c007400e9007200690065007500720065002e', '/DetectCurves', 'include', 'interfaces', 'late', '76', 'method', 'timescales', 'network43', 'large', 'polypeptide', 'J', 'setup', '≥0.2ms–1', 'Jorgensen', 'collective', 'using', '300', 'hydrophobic', 'Diffusional', '/DoThumbnails', 'ms', 'selected', 'Prebound', 'R.', 'donating', 'ligand', 'Grey', 'find', 'Statistical', 'Computational', 'A.', 'reconstructed', 'staying', 'Rather', 'mechanistic', 'bottom', 'change', 'inspired', 'Plattner1*', 'fully', 'Springer', 'shorter', '\\\\050SWOP\\\\051', '134', 'O', '(', 'inhibitor', 'Gly', '28,29', 'inter-', '4a', 'potential', 'discussions', 'Freie', 'FEFF004e00e4006900640065006e002000610073006500740075007300740065006e0020006100760075006c006c006100200076006f006900740020006c0075006f006400610020006a0061002000740075006c006f00730074006100610020005000440046002d0061007300690061006b00690072006a006f006a0061002c0020006a006f006900640065006e0020006500730069006b0061007400730065006c00750020006e00e400790074007400e400e40020006c0075006f00740065007400740061007600610073007400690020006c006f00700070007500740075006c006f006b00730065006e002e0020005000440046002d0061007300690061006b00690072006a0061007400200076006f0069006400610061006e0020006100760061007400610020004100630072006f006200610074002d0020006a0061002000520065006100640065007200200035002e00300020002d006f0068006a0065006c006d0061006c006c0061002000740061006900200075007500640065006d006d0061006c006c0061002000760065007200730069006f006c006c0061002e', 'simulations', 'previously30', 'PRACE', 'proceeds', 'recent', 'Spaar', '/StartPage', 'narrow', 'combined', 'determines', 'provides', 'go', 'applications', 'Previous', 'rapid', 'iii', 'reach', 'given', 'our', 'Commun', 'remaining', 'bridge', '/NLD', 'representative', 'protein-protein', 'perturbation', 'exceed', 'Estimation', '22', 'does', 'Visualization', 'predictor', 'bulk', 'Models', 'dominant', 'be', 'kT', 'tryptophan', 'side-chain', '/DownsampleColorImages', 'pair', 'Haran', 'www.nature.com/naturechemistry4', 'on-pathway', 'Optimization', '%', 'Tiwary', '1b', 'although', '/ParseDSCCommentsForDocInfo', 'Amber', 'research', '/PassThroughJPEGImages', 'mailto', 'biotechnology', 'broad', '261', 'iating', 'distance', 'response', 'shows', 'typically', 'located', '10–6', 'Discrete', 'statistically', 'exploration', 'OFCOM_PO_P1_F60', 'vi', 'etc', 'all-atom', 'October', 'fluorescence', 'can', 'kinetic', 'influenced', '/AlwaysEmbed', 'designed', 'under', '5c', 'longer', 'appropriate', 'roles', 'reconstruction', 'Figs', 'building', 'similar', '26', 'itself', 'binding–unbinding', '2016', 'HTMD', 'crystallographic', 'Upon', 'committor', 'depend', 'concentration-', 'mM', 'J.-H.', 'big', '533–550', 'charge', 'figurations', 'being', 'confidence', 'exploited', 'declare', 'Overall', 'interests', '111', 'mutants', 'dynamical', '79', '1,500', 'pages', 'equilibrated', 'program42', 'Changes', '/K', 'by', 'near-native', 'Colour', 'manner', 'reported', 'applicable', '29', 'maps', 'offer', '31', 'Gruebele', 'Bowman', 'Conclusion', 'volunteers', '2012', '=', 'ΔΔG', '/PreserveOPIComments', 'lies', 'its', '/DEU', 'substates', '1200', 'nature', 'Scott', 'assisted', 'functional', 'possibly', 'metastable', 'anism', 'coarse-grained', 'transition34', 'nanoseconds', 'viral', 'employed', '“', 'optimally', 'facilitate', 'Lecture', 'properties', 'v', '1c', 'PRINT', '/MonoImageDownsampleThreshold', '1048576', 'specific', 'wt', 'sufficiently', 'points', 'type24', 'reaction', 'Structures', 'large-barrier', 'angles', 'should', 'because', 'Nature', 'associ-ation', 'undergoes', 'Surprisingly', 'Minimal', '4.7', 'membrane', 'recently', 'M.', 'force', '5978–5984', 'Chandrasekhar', 'weight', '/NeverEmbed', 'comparable', 'centered', 'of', 'prediction', 'may', '/ColorConversionStrategy', '235–248', '<', '0.08', 'Early', 'subsequently', 'with', 'Bank', 'Avançats', '/NOR', 'Harel', 'Small', 'where', 'Their', 'Bayly', 'Fersht', 'acknowledged', 'counted', 'SoOPic', '3b→', '25', '0.15', '10–100', 'distances', '4', 'After', 'populates', 'oligomerization', '5b', '14', 'consistent', 'Such', 'fourth', '50', 'Loosely', 'insight', '1', 'appear', '/Quality', '/HWResolution', 'frank.noe', 'magnitude', 'tens', '1.00', '635–655', 'absolute', 'through', 'model41', 'Onuchic', 'nm2', '2', 'PDF', 'Acellera', 'involves', 'associated', 'showed', 'Estudis', 'Klein', 'direct', 'rebinding', 'version', 'regard', 'previously49', '230602', 'AmberTools39', 'variational', 'Biochemistry', 'While', 'evolution', 'FEDER', 'image', 'Biophys', 'for', 'root', 'reduced', 'Fabritiis2,3', 'June', 'addressed', 'committed', 'Gianni', 'lie', 'S', 'presented', 'provided', '/CannotEmbedFontPolicy', 'grey', '/JPEG2000GrayACSImageDict', 'same', 'Received', 'participate', 'trapping', '4.8', 'ating', 'ICREA', 'Biol', 'reversible', 'while', '2.00', 'bio-molecular', 'al.31', '/GrayImageDepth', 'package', 'increased', 'simulations13', 'Uncertainties', 'launched', 'two', 'variables11', 'counterclockwise/clockwise', 'conformations', 'all', 'inmed-', '>', 'PLoS', 'State', 'On-the-fly', 'strategy', '45°', 'landscapes', 'overall', 'Arnimallee', 'uncertain', '2b', 'immersed', 'along', 'stochastic', 'clusterClientba', '/ParseDSCComments', 'project', '/ColorImageMinResolutionPolicy', '4b', 'ceeds', 'Tasks', 'perturbations', '517–520', 'diffusion-limited', 'mutated', 'M–1', 'Nat', 'obtain', 'Ltd.', 'T.Weikl', 'edge', 'Schreiber', 'modifications', 'tally', 'estimation', '/GrayImageAutoFilterStrategy', 'finally', 'heavy-atom', 'seven', 'Late', 'scheme', 'their', 'hence', '∼10', 'removed', '101,465', 'y-positions', '/MaxSubsetPct', 'example', '15–21', 'mutant', 'perform', '/SyntheticBoldness', '/GrayImageDownsampleThreshold', 'Clore', '95', 'andmaterial', 'patches', 'ba', 'All', 'affinities', '39Lys', 'stuck', 'wrote', 'traps', 'importance', 'analyses', 'height', '/EmbedOpenType', 'Trajectories', 'additionally', 'Different', 'overlapping', 'NO', 'anassociation', 'last', '981–984', 'clustering', '1995', 'into', '2c–f', 'revealed', 'confirm', 'characterization', 'reveal', 'forming', 'well-defined', 'reveals', 'form', '/MonoImageDownsampleType', 'Giogino', 'Trendelkamp-Schroer', 'H.', 'calculation', 'authors', '180–196', 'connected', 'Comput', 'induced', 'im', 'i.e', '/FRA', 'association–dissociation', 'Job', 'Funding', 'in23,32', 'stabilizing', 'Mutational', '2.65', '/JPEG2000ColorACSImageDict', 'structural', 'm', '//www.nature.com/naturechemistry', 'Shaw', 'De', 'upside', 'JUNE', '511–516', 'found', 'governed', 'F.N', '/GrayImageDownsampleType', 'account', 'Noé1*', 'versa', 'Protein', 'Tightly', '/ColorImageDict', 'microscopic', 'indicating', 'see', 'compare', 'hitting', 'short', 'Noé', 'pre-defined', 'µs', 'grateful', 'L.', 'space', 'following', 'FEFF004700650062007200750069006b002000640065007a006500200069006e007300740065006c006c0069006e00670065006e0020006f006d0020005000440046002d0064006f00630075006d0065006e00740065006e0020007400650020006d0061006b0065006e00200064006900650020006700650073006300680069006b00740020007a0069006a006e0020006f006d0020007a0061006b0065006c0069006a006b006500200064006f00630075006d0065006e00740065006e00200062006500740072006f0075007700620061006100720020007700650065007200200074006500200067006500760065006e00200065006e0020006100660020007400650020006400720075006b006b0065006e002e0020004400650020005000440046002d0064006f00630075006d0065006e00740065006e0020006b0075006e006e0065006e00200077006f007200640065006e002000670065006f00700065006e00640020006d006500740020004100630072006f00620061007400200065006e002000520065006100640065007200200035002e003000200065006e00200068006f006700650072002e', 'directly', 'final', 'recruit-', '/JPEG', 'demonstrate', 'equation', '3.0', 'MINECO', 'generation', 'exceeded', 'pic', 'changes', '825/3-1', 'position', 'Having', 'constant', 'breakthroughs', 'e', '3Institució', 'square', '/CropColorImages', 'targeting'])"
      ]
     },
     "execution_count": 42,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# get all vocabulary terms\n",
    "vocabulary.keys()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "24"
      ]
     },
     "execution_count": 43,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# get occurrence of given word in vocabulary\n",
    "vocabulary['simulation']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.5.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
