{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Import PDF and analyze content\n",
    "\n",
    "Example notebook for how to read content and metadata of  PDF files"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Python environment installation instructions\n",
    "\n",
    "General Packages:\n",
    "\n",
    "    conda install numpy scipy matplotlib jupyter\n",
    "\n",
    "PDF-specific packages:\n",
    "\n",
    "    pip install pypdf2\n",
    "    conda install tika"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Example PDF file\n",
    "\n",
    "The PDF file used in the example below can be downloaded from:\n",
    "\n",
    "    https://doi.org/10.1073/pnas.1117201109"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "# set PDF filename/filepath parameter; this PDF file will be used in all examples; \n",
    "pdf_name = '12980.full.pdf'"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Get Text and Metadata from PDF using PyPDF2\n",
    "\n",
    "https://pythonhosted.org/PyPDF2/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "# import PyPDF2\n",
    "import PyPDF2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "# read the example pdf defined above\n",
    "file = open(pdf_name, 'rb')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "# creating a pdf reader object\n",
    "fileReader = PyPDF2.PdfFileReader(file)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "6\n"
     ]
    }
   ],
   "source": [
    "# print the number of pages in pdf file\n",
    "print(fileReader.numPages)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'/CreationDate': \"D:20120727203916-04'00'\",\n",
       " '/Creator': 'Arbortext Advanced Print Publisher 9.1.405/W Unicode',\n",
       " '/ModDate': \"D:20200423000344-07'00'\",\n",
       " '/Producer': 'Acrobat Distiller 6.0.1 (Windows)',\n",
       " '/Title': '201117201 12980..12985'}"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# get document metadata\n",
    "fileReader.documentInfo"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "page = fileReader.getPage(0)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "page_content = page.extractText()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "b'Aquantitativequasispeciestheory-basedmodelof\\nvirusescapemutationunderimmuneselection\\nHyung-JuneWooandJaquesReifman\\n1BiotechnologyHighPerformanceComputingSoftwareApplicationsInstitute,TelemedicineandAdvancedTechnologyResearchCenter,USArmyMedica\\nlResearchandMaterielCommand,FortDetrick,MD21702\\nEditedbyPeterSchuster,UniversityofVienna,Vienna,andapprovedJune28,2012(receivedforreviewOctober18,2011)\\nViralinfectionsinvolveacomplexinterplayoftheimmune\\nresponseandescapemutationofthevirusquasispeciesinsidea\\nsinglehost.Althoughfundamentalaspectsofsuchabalanceof\\n\\nmutationandselectionpressurehavebeenestablishedbythequa-\\nsispeciestheorydecadesago,itsimplicationshavelargelyre-\\nmainedqualitative.Here,wepresentaquantitativeapproachto\\n\\nmodelthevirusevolutionundercytotoxicT-lymphocyteimmune\\nresponse.Thevirusquasispeciesdynamicsareexplicitlyrepre-\\nsentedbymutationsinthecombinedsequencespaceofasetof\\n\\nepitopeswithintheviralgenome.Westochasticallysimulatedthe\\n\\ngrowthofaviralpopulationoriginatingfromasinglewild-type\\nfoundervirusanditsrecognitionandclearancebytheimmunere-\\nsponse,aswellastheexpansionofitsgeneticdiversity.Appliedto\\n\\ntheimmuneescapeofasimianimmunodeficiencyvirusepitope,\\nmodelpredictionswerequantitativelycomparabletotheexperi-\\nmentaldata.Withinthemodelparameterspace,wefoundtwo\\n\\nqualitativelydifferentregimesofinfectiousdiseasepathogenesis,\\n\\neachrepresentingalternativefatesoftheimmuneresponse:Itcan\\ncleartheinfectioninfinitetimeoreventuallybeoverwhelmedby\\nviralgrowthandescapemutation.Thelatterregimeexhibitsthe\\n\\ncharacteristicdiseaseprogressionpatternofhumanimmunodefi-\\nciencyvirus,whiletheformerisboundedbymaximummutation\\nratesthatcanbesuppressedbytheimmuneresponse.Ourresults\\n\\ndemonstratethat,byexplicitlyrepresentingepitopemutations\\nandthusprovidingagenotype\\n\\xc5\\x92phenotypemap,thequasispecies\\ntheorycanformthebasisofadetailedsequence-specificmodelof\\nreal-worldviralpathogensevolvingunderimmuneselection.\\npopulationdynamics\\nnext-generationsequencing\\nstochasticsimulation\\nHIVViruseswithRNAgenomes,suchashumanimmunodeficiency\\nvirustype1(HIV-1),havehighratesofmutationandevolve\\nrapidlyinresponsetohostimmuneselectionpressure.Oneofthe\\nconsequencesofsuchrapidmutationsistheerrorcatastrophe\\n(1),whereaviruspopulationisdriventoextinctionwhenitsmu-\\ntationrateexceedsathreshold.Theexistenceofsuchathreshold\\n\\nisacentralpredictionofthequasispeciestheorypioneeredby\\n\\nEigen(1)andSwetinaandSchuster(2).Therecentexperimental\\ndemonstrationoflethalmutagenesis(3\\n\\xc5\\x925),inwhichanerrorcat-\\nastrophetransitioniscausedbymutagens,demonstratesthatkey\\nfeaturesofthebalanceofmutationandselectioninvirusesare\\nelegantlycapturedbythequasispeciestheory.\\nAlthoughinsightsprovidedbythequasispeciestheoryhave\\ngreatlyexpandedourunderstandingofvirusbehavior,applica-\\ntionssofarhavebeenlimitedtoaconceptuallevel,partiallydue\\ntothelackofexperimentalinformationontheevolutionarydy-\\nnamicsinsequencespace.Next-generationsequencingtechni-\\n\\nques(6)havethepotentialtochangethissituation.Tobuild\\nquantitativemodelsimplementingthequasispeciesdynamics\\nanddescribesuchexperimentaldata,itisimportanttorealisti-\\n\\ncallyspecifythenatureofselectionpressure.Virusesinanimal\\nhostsevolveunderimmunepressure,andtheircapacityforrapid\\nescapemutationunderliesmanyofthedifficultiesincombating\\n\\npathogens,includingHIV-1.Inatypicaldiseasepathogenesisof\\n\\nHIV-1,theacuteviremiaafteraninitialinfectioniscurbedby\\nCD8\\xc3\\xbecytotoxicT-lymphocyte(CTL)responsesaswellassubse-\\nquentantibodyactions,leadingtoanasymptomaticchronicinfec-\\ntionstagethatcanlastupto10y(7).However,thisapparent\\n\\ncontrolofviremiaisnevercomplete,andwithoutantiviralther-\\n\\napy,thechronicinfectioneventuallyleadstotheonsetofdisease.\\n\\nThischronicinfectionstageinvolvescontinuousescapemutation-\\nCTLresponsecycles,whosedetailedcharacteristicsarebeingun-\\ncoveredbyultradeepsequencing(8\\n\\xc5\\x9212).CTLsrecognizespecific\\nviralepitopes(approximately10aminoacidslong)presentedon\\n\\nthesurfaceofinfectedcellsbyclassIhumanleukocyteantigen\\n\\n(HLA).TheepitoperecognitiondependssensitivelyonHLAal-\\n\\nleles,leadingtodifferentialpatternsofimmuneresponseamong\\npatients(13\\n\\xc5\\x9215),whilecharacteristicsofimmuneresponseduring\\nearlyinfectionoftenshapeandinfluencetheoveralldiseasepro-\\ngression(16,17).Quantitativesequence-basedmodelsofvirus-\\n\\nCTLdynamicswillgreatlyfacilitatetheinterpretationofexperi-\\n\\nmentaldata.\\nInaseriesofpioneeringworks(18\\n\\xc5\\x9220),Nowakandcoworkers\\nintroducedpopulationdynamicsconceptsthatcaptureadiverse\\n\\nrangeofimmuneresponseandescapemutation.Similarap-\\n\\nproachesfocusedmoreonescapedynamicshavealsobeenpro-\\nposed(21,22).Thesemodels,however,donotdescribemutations\\ninsequencespaceexplicitly.Othermathematicalmodelsof\\n\\nsequenceevolutionfocusonsequencedivergenceduringthevery\\n\\nearlystagesofinfection,whileignoringoronlyimplicitlyincluding\\n\\ntheeffectofselectionpressure(23,24).Inthispaper,wedescribe\\naquantitativequasispecies-basedmodelofvirusdynamicsunder\\nTcell-basedimmunepressure,explicitlymappinggenotype\\n\\xc5\\x92phe-notyperelationships.Ourapproachcombinesthedescriptionof\\n\\nvirusevolutionarydynamicsprovidedbythequasispeciestheory\\n\\nandthepopulationdynamicsmodelsofNowakandcoworkers.\\n\\nWeshowthatthemodelnotonlycapturesthesalientfeatures\\nobservedinsequencingdataofsimianimmunodeficiencyvirus\\n(SIV)(9)butalsorevealsgeneralqualitativefeaturesofviral\\n\\ninfectiondiseasepathogenesis:Theimmunesystemcanclear\\n\\ntheinfectionwithintimescalesrangingfromdaystothosemuch\\n\\nlongerthanpatientlifetimes,orbeoverwhelmedbyimmunees-\\n\\ncape.Theviralloadprogressioninthelatterregimeclosely\\nmatchestheobservedHIV-1diseasepathogenesispattern.\\nResultsandDiscussion\\nStochasticQuasispeciesDynamics.\\nDuringacuteinfection,thefoun-\\ndervirus(oftenasinglevirion)(7,11,17,25)undergoesreplica-\\ntionstoproduceanexponentiallygrowingpopulation,whichmay\\nbedescribedbythequasispeciesdynamicswithoutdegradation,\\nAuthorcontributions:H.-J.W.designedresearch;H.-J.W.performedresearch;H.-J.W.and\\nJ.R.analyzeddata;andH.-J.W.andJ.R.wrotethepaper.\\nTheauthorsdeclarenoconflictofinterest.\\nThisarticleisaPNASDirectSubmission.\\nFreelyavailableonlinethroughthePNASopenaccessoption.\\n1Towhomcorrespondenceshouldbeaddressed.E-mail:jaques.reifman@us.army.mil.\\nThisarticlecontainssupportinginformationonlineat\\nwww.pnas.org/lookup/suppl/\\ndoi:10.1073/pnas.1117201109/-/DCSupplemental\\n.12980\\xc5\\x9212985PNASAugust7,2012\\nvol.109\\nno.32www.pnas.org/cgi/doi/10.1073/pnas.1117201109\\nDownloaded by guest on April 23, 2020 '\n"
     ]
    }
   ],
   "source": [
    "print (page_content.encode('utf-8'))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Get text and metadata from PDF using TIKA\n",
    "\n",
    "General information about TIKA\n",
    "https://cwiki.apache.org/confluence/display/TIKA/TikaServer\n",
    "\n",
    "TIKA python API\n",
    "https://github.com/chrismattmann/tika-python"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "# import parser from TIKA\n",
    "from tika import parser"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "# read example PDF definded above\n",
    "parsedPDF = parser.from_file(pdf_name)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "{'status': 200, 'content': '\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n201117201 12980..12985\\n\\n\\nA quantitative quasispecies theory-based model of\\nvirus escape mutation under immune selection\\nHyung-June Woo and Jaques Reifman1\\n\\nBiotechnology High Performance Computing Software Applications Institute, Telemedicine and Advanced Technology Research Center, US Army Medical\\nResearch and Materiel Command, Fort Detrick, MD 21702\\n\\nEdited by Peter Schuster, University of Vienna, Vienna, and approved June 28, 2012 (received for review October 18, 2011)\\n\\nViral infections involve a complex interplay of the immune\\nresponse and escape mutation of the virus quasispecies inside a\\nsingle host. Although fundamental aspects of such a balance of\\nmutation and selection pressure have been established by the qua-\\nsispecies theory decades ago, its implications have largely re-\\nmained qualitative. Here, we present a quantitative approach to\\nmodel the virus evolution under cytotoxic T-lymphocyte immune\\nresponse. The virus quasispecies dynamics are explicitly repre-\\nsented by mutations in the combined sequence space of a set of\\nepitopes within the viral genome. We stochastically simulated the\\ngrowth of a viral population originating from a single wild-type\\nfounder virus and its recognition and clearance by the immune re-\\nsponse, as well as the expansion of its genetic diversity. Applied to\\nthe immune escape of a simian immunodeficiency virus epitope,\\nmodel predictions were quantitatively comparable to the experi-\\nmental data. Within the model parameter space, we found two\\nqualitatively different regimes of infectious disease pathogenesis,\\neach representing alternative fates of the immune response: It can\\nclear the infection in finite time or eventually be overwhelmed by\\nviral growth and escape mutation. The latter regime exhibits the\\ncharacteristic disease progression pattern of human immunodefi-\\nciency virus, while the former is bounded by maximum mutation\\nrates that can be suppressed by the immune response. Our results\\ndemonstrate that, by explicitly representing epitope mutations\\nand thus providing a genotype–phenotype map, the quasispecies\\ntheory can form the basis of a detailed sequence-specific model of\\nreal-world viral pathogens evolving under immune selection.\\n\\npopulation dynamics ∣ next-generation sequencing ∣ stochastic simulation ∣\\nHIV\\n\\nViruses with RNA genomes, such as human immunodeficiencyvirus type 1 (HIV-1), have high rates of mutation and evolve\\nrapidly in response to host immune selection pressure. One of the\\nconsequences of such rapid mutations is the error catastrophe\\n(1), where a virus population is driven to extinction when its mu-\\ntation rate exceeds a threshold. The existence of such a threshold\\nis a central prediction of the quasispecies theory pioneered by\\nEigen (1) and Swetina and Schuster (2). The recent experimental\\ndemonstration of lethal mutagenesis (3–5), in which an error cat-\\nastrophe transition is caused by mutagens, demonstrates that key\\nfeatures of the balance of mutation and selection in viruses are\\nelegantly captured by the quasispecies theory.\\n\\nAlthough insights provided by the quasispecies theory have\\ngreatly expanded our understanding of virus behavior, applica-\\ntions so far have been limited to a conceptual level, partially due\\nto the lack of experimental information on the evolutionary dy-\\nnamics in sequence space. Next-generation sequencing techni-\\nques (6) have the potential to change this situation. To build\\nquantitative models implementing the quasispecies dynamics\\nand describe such experimental data, it is important to realisti-\\ncally specify the nature of selection pressure. Viruses in animal\\nhosts evolve under immune pressure, and their capacity for rapid\\nescape mutation underlies many of the difficulties in combating\\npathogens, including HIV-1. In a typical disease pathogenesis of\\nHIV-1, the acute viremia after an initial infection is curbed by\\n\\nCD8þ cytotoxic T-lymphocyte (CTL) responses as well as subse-\\nquent antibody actions, leading to an asymptomatic chronic infec-\\ntion stage that can last up to 10 y (7). However, this apparent\\ncontrol of viremia is never complete, and without antiviral ther-\\napy, the chronic infection eventually leads to the onset of disease.\\nThis chronic infection stage involves continuous escape mutation-\\nCTL response cycles, whose detailed characteristics are being un-\\ncovered by ultradeep sequencing (8–12). CTLs recognize specific\\nviral epitopes (approximately 10 amino acids long) presented on\\nthe surface of infected cells by class I human leukocyte antigen\\n(HLA). The epitope recognition depends sensitively on HLA al-\\nleles, leading to differential patterns of immune response among\\npatients (13–15), while characteristics of immune response during\\nearly infection often shape and influence the overall disease pro-\\ngression (16, 17). Quantitative sequence-based models of virus-\\nCTL dynamics will greatly facilitate the interpretation of experi-\\nmental data.\\n\\nIn a series of pioneering works (18–20), Nowak and coworkers\\nintroduced population dynamics concepts that capture a diverse\\nrange of immune response and escape mutation. Similar ap-\\nproaches focused more on escape dynamics have also been pro-\\nposed (21, 22). These models, however, do not describe mutations\\nin sequence space explicitly. Other mathematical models of\\nsequence evolution focus on sequence divergence during the very\\nearly stages of infection, while ignoring or only implicitly including\\nthe effect of selection pressure (23, 24). In this paper, we describe\\na quantitative quasispecies-based model of virus dynamics under\\nT cell-based immune pressure, explicitly mapping genotype–phe-\\nnotype relationships. Our approach combines the description of\\nvirus evolutionary dynamics provided by the quasispecies theory\\nand the population dynamics models of Nowak and coworkers.\\nWe show that the model not only captures the salient features\\nobserved in sequencing data of simian immunodeficiency virus\\n(SIV) (9) but also reveals general qualitative features of viral\\ninfection disease pathogenesis: The immune system can clear\\nthe infection within time scales ranging from days to those much\\nlonger than patient lifetimes, or be overwhelmed by immune es-\\ncape. The viral load progression in the latter regime closely\\nmatches the observed HIV-1 disease pathogenesis pattern.\\n\\nResults and Discussion\\nStochastic Quasispecies Dynamics.During acute infection, the foun-\\nder virus (often a single virion) (7, 11, 17, 25) undergoes replica-\\ntions to produce an exponentially growing population, which may\\nbe described by the quasispecies dynamics without degradation,\\n\\nAuthor contributions: H.-J.W. designed research; H.-J.W. performed research; H.-J.W. and\\nJ.R. analyzed data; and H.-J.W. and J.R. wrote the paper.\\n\\nThe authors declare no conflict of interest.\\n\\nThis article is a PNAS Direct Submission.\\n\\nFreely available online through the PNAS open access option.\\n1To whom correspondence should be addressed. E-mail: jaques.reifman@us.army.mil.\\n\\nThis article contains supporting information online at www.pnas.org/lookup/suppl/\\ndoi:10.1073/pnas.1117201109/-/DCSupplemental.\\n\\n12980–12985 ∣ PNAS ∣ August 7, 2012 ∣ vol. 109 ∣ no. 32 www.pnas.org/cgi/doi/10.1073/pnas.1117201109\\n\\nD\\now\\n\\nnl\\noa\\n\\nde\\nd \\n\\nby\\n g\\n\\nue\\nst\\n\\n o\\nn \\n\\nA\\npr\\n\\nil \\n23\\n\\n, 2\\n02\\n\\n0 \\n\\nhttp://www.pnas.org/lookup/suppl/doi:10.1073/pnas.1117201109/-/DCSupplemental\\nhttp://www.pnas.org/lookup/suppl/doi:10.1073/pnas.1117201109/-/DCSupplemental\\nhttp://www.pnas.org/lookup/suppl/doi:10.1073/pnas.1117201109/-/DCSupplemental\\nhttp://www.pnas.org/lookup/suppl/doi:10.1073/pnas.1117201109/-/DCSupplemental\\nhttp://www.pnas.org/lookup/suppl/doi:10.1073/pnas.1117201109/-/DCSupplemental\\nhttp://www.pnas.org/lookup/suppl/doi:10.1073/pnas.1117201109/-/DCSupplemental\\n\\n\\n_nj ¼ ∑\\ni\\n\\nQjirini; [1]\\n\\nwhere ni is the number of virions with genotype i, ri is the repli-\\ncation rate of i, andQji ¼ ð1 − μÞL−dðj;iÞðμ∕3Þdðj;iÞ is the mutation\\nprobability from i to genotype j, with mean mutation rate μ,\\ngenome length L (in bp), and Hamming distance (the number\\nof nucleotides that are different) dðj; iÞ between j and i. A more\\ncomplete description for early stages of infection, where ni is of\\nthe order of 1 and restricted to integer values, is given by the che-\\nmical reaction representation\\n\\nRi →\\nQjiri\\n\\nRi þRj; [2]\\n\\nwhere Ri is a virion with genotype i. Eq. 2 can be simulated\\nstochastically with the Gillespie algorithm (26). A major advan-\\ntage of such a stochastic formulation is that it allows us to avoid\\nexhaustive enumerations of all possible genotypes. This is signif-\\nicant even for a single epitope of viral proteins: An amino acid\\nsequence with La ¼ 10, whose genome length is L ¼ 3La ¼\\n30 bp, still has 4L ∼ 1018 genotypes. Nowak and Schuster (27)\\nperformed Gillespie algorithm simulations of the standard qua-\\nsispecies model under the single-peak fitness landscape (ri ¼ 1∕τ\\nfor the wild-type (WT) sequence and ri ¼ 1∕Aτ for all other mu-\\ntants, whereA > 1 is the relative fitness of theWTand τ is a char-\\nacteristic time scale). In this case, distributions of individuals\\namong only two groups (WT and mutants) need to be tracked.\\n\\nIn our numerical simulations, the initial genotype (referred to\\nhere as the WT) replicates with mutation rate μ, and the gener-\\nated mutants are compared with the existing list of sequences,\\nwhich is updated dynamically when new sequences arise. We ver-\\nified that this numerical scheme sampled the relevant sequence\\nspace sufficiently (SI Text, Figs. S1–S3). The initial condition we\\nused (single WT) and the discreteness of ni imply that ni should\\nbe interpreted as the total number of virions within a finite sys-\\ntem. The HIV-1 viral load during the acute infection phase can\\nreach up to 104 ∼ 106 RNA copies/mL (7). Accounting for the\\nvolume of blood in the body of an average adult (approximately\\n5 L), the total population size Nv ¼ Σi ni would be up to about\\n109. We found simulations to slow down significantly as Nv be-\\ncame larger than about 106. For computational efficiency, we\\ntherefore regarded the system as a small representative volume\\n(1 mL) of blood, and the viral loads and CTL levels reported refer\\nto RNA copy numbers and cell counts within this volume.\\n\\nCTL Response. Within an infected host, the action of the cellular\\nimmune response provides the major force countering the acute\\nviremia. In this work, we combine the population dynamics ap-\\nproach of virus-immune system dynamics by Nowak and cowor-\\nkers (18, 19) with the classic quasispecies theory. Accordingly,\\nEq. 2 is modified as follows:\\n\\nRi →\\nQjiri\\n\\nRi þRj; ðreplication with mutationÞ [3A]\\n\\nRi→\\nb\\nRi þ∑\\n\\nl\\n\\nk¼1\\nCðαikÞ; ðstimulationÞ [3B]\\n\\nRi þ CðαikÞ→\\np\\nCðαikÞ; ðclearanceÞ [3C]\\n\\nCðαÞ→\\ng\\n0; ðdecayÞ [3D]\\n\\nwhere the genotype i is now a member of the combined sequence\\nspace of l epitopes. The amino acid sequence (“phenotype”) of\\nepitope k in genotype i is denoted as αik, and CðαÞ represents a\\n\\nCTL specific to phenotype α. Eq. 3B represents a reaction in\\nwhich, with a rate b, virions of sequence i stimulate the produc-\\ntion of a set of CTLs CðαikÞ corresponding to the phenotypes of\\nits epitopes. Eq. 3C denotes the reaction where cells infected with\\nviruses with genotype i are cleared by CTLs with phenotypes that\\nmatch one of its epitopes with a rate p, and Eq. 3D represents the\\nnatural decay of T cells with a rate g. Under the alternative de-\\nterministic continuum approximation, one may write:\\n\\n_nj ¼ ∑\\ni\\n\\nQjirini − p∑\\nl\\n\\nk¼1\\ncðαjkÞnj; [4A]\\n\\n_cðαÞ ¼ bnα − gcðαÞ; [4B]\\n\\nwhere cðαÞ is the number of Tcells CðαÞ and nα is the total num-\\nber of virions containing epitopes with phenotype α.\\n\\nThe qualitative features of the infection-clearance dynamics\\ncan be illustrated by setting the mutation rate μ to zero. Numer-\\nical integrations of Eq. 4 yield the continuum approximation re-\\nsult (Fig. S4 for μ ¼ 0 and l ¼ 1), which showed reasonable\\nagreement with the stochastic simulation results. For nonvanish-\\ning mutation rates, numerical integration rapidly becomes prohi-\\nbitive with increasing L because of the high dimensionality of the\\nsequence space. The inverse of b is a measure of the time delay of\\nthe growth of T cells compared to that of viruses. Here, we\\nadopted b values of approximately 0.01 d−1. The efficiency of\\nCTLs in recognizing cells presenting the epitope and killing them\\nis represented by parameter p, the clearance rate. We found\\np values of 10−4 ∼ 10−3 d−1 to give good fits to experiments\\n(see below).\\n\\nImmune Escape Dynamics. To make comparisons with experiments,\\nit is necessary to consider that fitness is determined by amino acid\\nsequences, and many mutations are synonymous. Therefore, the\\nfull immune escape dynamics represented by Eq. 3 distinguishes\\nthe amino acid sequence αik corresponding to the nucleotide\\nsequence of epitope k in genotype i. We refer to the set of nu-\\ncleotide and amino acid sequences as “genotypes” and “pheno-\\ntypes,” respectively. The fitness ri is a function of phenotypes\\nonly. During simulations, each genotype is translated into a phe-\\nnotype, and a new mutant is checked against the existing pheno-\\ntype/genotype list, which is updated and expanded. The nature of\\nthis genotype–phenotype map plays important roles in evolution-\\nary dynamics, leading to key signatures of selective pressure, such\\nas the ratio of synonymous to nonsynonymous mutations (28).\\nQuantitative insights to the effects of this mapping on molecular\\nevolution have been provided by Manrubia and coworkers (29),\\nwho studied short RNA sequences for which fitness can be esti-\\nmated by secondary structure predictions. In our case, a simpli-\\nfied overview of the genotype–phenotype map can be gleaned\\nfrom a network representation of the genetic code (Fig. S5): If\\nall mutations were neutral and amino acids within an epitope in-\\ndependent, this map would be sufficient to determine the statis-\\ntics of evolutionary drifts. The accessibility of certain mutations\\nfrom a phenotype, in particular, has been shown to affect the\\nevolvability of viruses (30).\\n\\nAnother major ingredient for a realistic model of immune es-\\ncape is the fitness landscape beyond the level of the single-peak\\nmodel. Much progress has been made recently in understanding\\nthe nature of fitness landscapes (31–34). Explicit fitness measure-\\nments of viral clones (35, 36) and biochemical assays of proteins\\n(37) both indicate that single-nucleotide substitutions lead to a\\nbroad distribution of fitness changes, most of which are deleter-\\nious. Therefore, one may assume that the fitness of a mutant is a\\nrandom variable centered around a mean fitness value f . We ex-\\npect this mean fitness to be a decreasing function of distance d to\\nthe WT (the number of amino acids that are different), which we\\n\\nWoo and Reifman PNAS ∣ August 7, 2012 ∣ vol. 109 ∣ no. 32 ∣ 12981\\n\\nBI\\nO\\nPH\\n\\nYS\\nIC\\nS\\nA\\nN\\nD\\n\\nCO\\nM\\nPU\\n\\nTA\\nTI\\nO\\nN\\nA\\nL\\nBI\\nO\\nLO\\n\\nG\\nY\\n\\nD\\now\\n\\nnl\\noa\\n\\nde\\nd \\n\\nby\\n g\\n\\nue\\nst\\n\\n o\\nn \\n\\nA\\npr\\n\\nil \\n23\\n\\n, 2\\n02\\n\\n0 \\n\\nhttp://www.pnas.org/lookup/suppl/doi:10.1073/pnas.1117201109/-/DCSupplemental/pnas.1117201109_SI.pdf?targetid=STXT\\nhttp://www.pnas.org/lookup/suppl/doi:10.1073/pnas.1117201109/-/DCSupplemental/pnas.1117201109_SI.pdf?targetid=SF1\\nhttp://www.pnas.org/lookup/suppl/doi:10.1073/pnas.1117201109/-/DCSupplemental/pnas.1117201109_SI.pdf?targetid=SF3\\nhttp://www.pnas.org/lookup/suppl/doi:10.1073/pnas.1117201109/-/DCSupplemental/pnas.1117201109_SI.pdf?targetid=SF4\\nhttp://www.pnas.org/lookup/suppl/doi:10.1073/pnas.1117201109/-/DCSupplemental/pnas.1117201109_SI.pdf?targetid=SF4\\nhttp://www.pnas.org/lookup/suppl/doi:10.1073/pnas.1117201109/-/DCSupplemental/pnas.1117201109_SI.pdf?targetid=SF5\\nhttp://www.pnas.org/lookup/suppl/doi:10.1073/pnas.1117201109/-/DCSupplemental/pnas.1117201109_SI.pdf?targetid=SF5\\n\\n\\nassume to be exponential: f ðdÞ ¼ expð−d∕ξÞ, where ξ denotes a\\ncharacteristic distance (Methods).\\n\\nAn empirical evidence for this choice can be found from a re-\\ncent experimental study by Fernández et al. (37), who measured\\nthe fitness landscapes of HIV-1 protease quasispecies for three\\npatients. We plotted the distribution of their reported fitness\\nvalues as a function of distance to the dominant phenotype\\nand found near-exponential dependence for two quasispecies\\n(Fig. S6). Our fitness function therefore models both the de-\\ncrease of the mean fitness away from WT and the distribution\\nof neutral, deleterious, and beneficial mutants for a given dis-\\ntance. In simulations, these fitness values were assigned dynami-\\ncally to newly encountered phenotypes. It is important to note,\\nhowever, that this landscape chosen is still an approximation that\\nignores many potentially important effects, such as the heteroge-\\nneity of the mutational neighborhoods within the phenotype\\nspace.\\n\\nWe tested this quasispecies model with the ultradeep sequen-\\ncing data (9, 22) of the SIVepitope Tat28–35SL8 (38). Fig. 1 shows\\nthe time dependence of the viral load and WT frequency from\\nsimulations of the single-epitope version of Eq. 3 compared with\\nthe experimental data (22). The initial rapid growth of the viral\\nload, its subsequent decrease as CTLs are activated, and the more\\ngradual increase as mutants appear are all captured quantita-\\ntively. The WT frequency decrease agrees with experimental\\ntrends (Fig. 1B) with signatures of two distinct time scales (22).\\nThe mutation rate we used (μ ¼ 2 × 10−7 bp−1) was chosen to\\nobtain the best overall agreement for WT frequency with the sin-\\ngle-epitope version of the model. For multiepitope applications,\\nlarger mutation rate values close to the experimental estimate\\nfor HIV-1 (μ ¼ 3.4 × 10−5 bp−1) (39) gave realistic dynamics\\n(Fig. 2).\\n\\nFig. 1B also shows the frequency of the most dominant escape\\nmutant (EM) at each time point. As mutations occur, a new strain\\nproliferates temporarily before being curbed down, and the total\\nnumber of phenotypes continues to increase. The increase in via-\\nbility of populations with multiple but lower fitness peaks has\\nbeen described first by Schuster and Swetina (40) and has later\\nbeen known as the “survival of the flattest” effect (41). Further\\ninsights into the role this expansion in sequence space plays in the\\ngrowth of viral loads can be gained from the deterministic con-\\ntinuum approximation, Eq. 4, without mutation: The immune re-\\nsponse to WT leads to damped oscillations of viral load and CTL\\n\\nlevel (Fig. S4), reaching a steady state n� and c�. This process is\\nrepeated for each new mutant, leading to continued increases\\nin n� and c� roughly proportional to the number of phenotypes\\n(18). However, the timing, distribution, and the probability of\\nthe appearance of mutants are highly nontrivial functions of\\nthe characteristics of the system, including genome length, gen-\\notype–phenotype map, and fitness landscape. Our quantitative\\nquasispecies-based model, Eq. 3, provides a realistic description\\nof this complex diversification process via the single parameter μ.\\nWe note that, for a single epitope, this interpretation ignores the\\neffects of competition among viral strains because each CTL\\nis specific to only one phenotype. In reality, in addition to com-\\npeting directly for host cells, different viral strains share their vul-\\nnerability to CTLs specific to common epitopes within their\\ngenomes. By considering multiple epitopes within a strain, our\\nmodel takes into account this interdependence of viral strains\\nand their indirect competition arising from shared epitope phe-\\nnotypes (see below).\\n\\nAs a quantitative measure of the relative degree of WT\\npersistence during immune escape, we also examined the time\\n(half-life) required for WT frequency to reach one-half (Fig. 1B)\\n(22) with varying stimulation rate b and clearance rate p (Fig. S7).\\nThe SIV epitopes studied by Bimber et al. (9) were estimated to\\nhave a WT half-life of about 20 d (22). We found that, within our\\nmodel, the WT half-life increases with increasing p and b in this\\nrange: An increase in the magnitude of selection pressure, rather\\nthan accelerating immune escape, suppresses viral population\\ngrowth more effectively and reduces the chances of escape muta-\\ntion (Fig. 2). The immune escape thus becomes more pronounced\\nwhen viral loads in the chronic phase are larger with more fre-\\nquent mutations.\\n\\nDisease Progression Under Immune Response.We examined general\\ntrends of viral infection-immune response dynamics within our\\nquasispecies model on multiple-epitope levels (l ¼ 3) with varia-\\ntions of key model parameters. The evolutionary dynamics, in\\nparticular, is critically affected by p, the clearance rate represent-\\ning the effectiveness of immune response, and the mutation\\nrate μ. We identified two qualitatively different behaviors based\\non the eventual fate of viral load/CTL level: in one regime\\n\\n0 50 100 150\\n10\\n\\n0\\n\\n10\\n2\\n\\n10\\n4\\n\\n10\\n6\\n\\n10\\n8\\n\\nV\\nira\\n\\nl l\\noa\\n\\nd\\n\\n0 50 100 150\\nTime (d)\\n\\n10\\n-2\\n\\n10\\n-1\\n\\n10\\n0\\n\\nF\\nre\\n\\nqu\\nen\\n\\ncy\\n\\nWT\\nDominant EM\\n\\nA\\n\\nB\\n\\nFig. 1. Simulation of full immune escape dynamics of a single SIV epitope\\nTat28–35SL8. (A) Viral load. (B) Frequencies of WT and the dominant escape\\nmutants (EM). Symbols represent experimental data from refs. 9 and 22, with\\neach symbol corresponding to one of four different animals. Solid lines are\\nfrom our stochastic quasispecies model with τ ¼ 0.8 d, μ ¼ 2.0 × 10−7 bp−1,\\nb ¼ 0.01 d−1, g ¼ 0.2 d−1, and p ¼ 2.0 × 10−4 d−1. The fitness function para-\\nmeters were σ ¼ 0.1 and ξ ¼ 1 (Methods). The EM frequency is for a single\\ntrajectory while the rest are averages.\\n\\n10\\n1\\n\\n10\\n2\\n\\n10\\n3\\n\\n10\\n4\\n\\nV\\nira\\n\\nl l\\noa\\n\\nd\\n\\n10\\n0\\n\\n10\\n1\\n\\n10\\n2\\n\\n10\\n3\\n\\nTime (d)\\n\\n10\\n0\\n\\n10\\n1\\n\\nN\\no.\\n\\n o\\nf p\\n\\nhe\\nno\\n\\nty\\npe\\n\\ns\\n\\n10\\n2\\n\\n10\\n3\\n\\n10\\n4\\n\\nC\\nT\\n\\nL\\n\\np = 1.0x10\\n-3\\n\\n, µ = 1.0x10-5\\n\\np = 5.0x10\\n-4\\n\\n, µ = 1.0x10-5\\n\\np = 1.0x10\\n-3\\n\\n, µ = 1.0x10-6\\n\\n0\\n\\n4000\\n\\n8000\\n\\nAcute\\n\\nChronic\\n\\n100\\n\\nA\\n\\nB\\n\\nC\\n\\nDisease\\n\\n101 102 103\\n\\nFig. 2. Typical disease progression patterns in the runaway regime from the\\nmultiepitope quasispecies model. (A) Viral load. (B) Total CTL levels. (C) Num-\\nber of distinct phenotypes per epitope present in the population. The inset in\\nA shows the viral load in linear scale for p ¼ 1.0 × 10−3 d−1 and μ ¼ 1.0 ×\\n10−5 bp−1. Other parameter values were b ¼ 0.02 d−1, g ¼ 0.1 d−1, τ ¼ 1 d,\\nσ ¼ 0.2, and ξ ¼ 1. The units of p and μ in the legends are d−1 and bp−1,\\nrespectively. All data represent averages over trajectories.\\n\\n12982 ∣ www.pnas.org/cgi/doi/10.1073/pnas.1117201109 Woo and Reifman\\n\\nD\\now\\n\\nnl\\noa\\n\\nde\\nd \\n\\nby\\n g\\n\\nue\\nst\\n\\n o\\nn \\n\\nA\\npr\\n\\nil \\n23\\n\\n, 2\\n02\\n\\n0 \\n\\nhttp://www.pnas.org/lookup/suppl/doi:10.1073/pnas.1117201109/-/DCSupplemental/pnas.1117201109_SI.pdf?targetid=SF6\\nhttp://www.pnas.org/lookup/suppl/doi:10.1073/pnas.1117201109/-/DCSupplemental/pnas.1117201109_SI.pdf?targetid=SF6\\nhttp://www.pnas.org/lookup/suppl/doi:10.1073/pnas.1117201109/-/DCSupplemental/pnas.1117201109_SI.pdf?targetid=SF4\\nhttp://www.pnas.org/lookup/suppl/doi:10.1073/pnas.1117201109/-/DCSupplemental/pnas.1117201109_SI.pdf?targetid=SF4\\nhttp://www.pnas.org/lookup/suppl/doi:10.1073/pnas.1117201109/-/DCSupplemental/pnas.1117201109_SI.pdf?targetid=SF7\\nhttp://www.pnas.org/lookup/suppl/doi:10.1073/pnas.1117201109/-/DCSupplemental/pnas.1117201109_SI.pdf?targetid=SF7\\n\\n\\n(“runaway,” Fig. 2), the chronic phase after the resolution of the\\nacute infection is eventually followed by a runaway growth in viral\\nloads (Fig. 2A). The level of CTLs also rises consistently during\\nthe chronic phase (Fig. 2B). This disease progression is accom-\\npanied by an increase in the diversity of quasispecies, or the\\n“spreading of clouds” (Fig. 2C). The time scale for the duration\\nof the chronic phase, in particular, varied sensitively with p and μ\\n(from a few months to years) (Fig. 2A). We found that the total\\nnumber of phenotypes per epitope (Fig. 2C) generally goes down\\nas the number of epitopes increases: A new escape mutant has\\nmuch lower probability for survival with multiple epitopes be-\\ncause it shares WT phenotypes on other parts of its genome with\\nexisting strains, and therefore is susceptible to CTLs that are al-\\nready present.\\n\\nThe disease progression pattern in the runaway regime corre-\\nsponds to situations where the immune system is overwhelmed by\\nviral growth, either shortly after infection (there is no discernible\\nchronic phase for sufficiently small p) or after a long chronic\\nphase and accumulation of escape mutants. The latter case clo-\\nsely matches the characteristic HIV-1 disease pathogenesis (7),\\nrevealing the major role played by the uncontrolled diversifica-\\ntion of the quasispecies that overwhelms the CTL response. This\\nfeature was first suggested by Nowak and May (19), who coined\\nthe term “diversity threshold” based on a model assuming ran-\\ndom appearances of mutants. Our results show that the basic qua-\\nsispecies dynamics under selection pressure provide a realistic\\ndescription of this phenomenon.\\n\\nIf p is sufficiently large and μ sufficiently small, one enters a\\ndifferent regime (“extinction,” Fig. 3), where the acute and\\nchronic phases lead to viral loads that decrease either rapidly\\nor asymptotically with time. As populations shrink in size, their\\ndynamics become increasingly stochastic, and extinction (a com-\\nplete clearance of infection) occurs when the viral load reaches\\nzero. This behavior is inherently stochastic and cannot be cap-\\ntured by the deterministic approximation in Eq. 4. The time re-\\nquired for clearing the infection (“extinction time”) typically has\\na broad distribution, increasing with decreasing p while becoming\\ninfinite above a threshold μ (Fig. 4A). The rapid clearance of in-\\nfection within approximately 10 d corresponds to the normal state\\nof affairs in a healthy immune system against viruses such as in-\\nfluenza. As shown in Fig. 3A (p ¼ 0.01 d−1), however, the extinc-\\ntion time can also reach time scales approaching a patient’s\\n\\nlifetime. This feature may be relevant in understanding the basis\\nfor some chronic viral infections with long and stable setpoints\\n(e.g., hepatitis C), although the high variability of extinction times\\nwe observed (Fig. 4A) likely may not correspond to actual pos-\\nsible clearances of such chronic infections.\\n\\nThe threshold separating the runaway (Fig. 2) and extinction\\n(Fig. 3) regimes can be identified by the maximum mutation\\nrate for which the extinction time remains finite (Fig. 4A). This\\nerror threshold for the immune control of infection is analo-\\ngous to Eigen’s threshold for genomic stability: The latter is\\nthe error rate above which replicators cannot maintain a stable\\nmaster sequence. The former is a maximum mutation rate the\\ngiven immune system can suppress. Fig. 4B shows this threshold\\nas a function of p and μ. A given virus would have roughly the\\nsame mutation rate, while p would vary with individual patients.\\nThe sensitivity of disease progression with p is consistent with\\nthe observation that even when the fatality rate of a certain\\ninfection is known, the course of infection in a patient is often\\nunpredictable.\\n\\nThe case for HIV-1 is unique in the sense that without treat-\\nment, most patients eventually progress to the disease stage. One\\nof the special characteristics of HIV-1 is that it targets CD4þ\\n(helper) Tcells, which play critical roles in eliciting and mediating\\nCTL responses. Within the context of our model, the destruction\\nof CD4þ Tcells would lead to a gradual decrease in both the sti-\\nmulation rate b and the clearance rate p. One therefore expects a\\ncontinual deterioration of any initial effective control of infec-\\ntion, represented by p values that decrease over the course of dis-\\nease progression. In Fig. 4B, therefore, even though patients with\\nstronger immune systems may have p values initially in the extinc-\\ntion regime, the continued depletion of CD4þ cells would lower p\\nand cause them to cross the threshold.\\n\\nConclusions\\nIn this paper, we demonstrated that the classic quasispecies the-\\nory can form the basis of a quantitative model of virus evolution\\nunder immune selection. One of the apparent challenges in de-\\nveloping such a model is that viral genomes, while relatively small\\n(approximately 104 bp), still constitute a huge sequence space.\\nHere, we showed that a set of epitopes (La ∼ 10 amino acids)\\ncan be considered as minimal units of genomic segments on which\\nselection acts. In standard quasispecies theory, competition be-\\ntween different strains arises indirectly from the constraint of\\nfixed total population size. In our model, the degradation and\\nremoval of virions occur via immune response, which can be re-\\ngarded as a concrete specification of the selection forces. In this\\ncase, indirect competition occurs because most viral strains share\\n\\n10\\n1\\n\\n10\\n2\\n\\n10\\n3\\n\\n10\\n4\\n\\nV\\nira\\n\\nl l\\noa\\n\\nd\\n\\n10\\n0\\n\\n10\\n1\\n\\n10\\n2\\n\\n10\\n3\\n\\n10\\n4\\n\\nTime (d)\\n\\n1\\n\\n2\\n\\n3\\n\\n4\\n\\nN\\no.\\n\\n o\\nf p\\n\\nhe\\nno\\n\\nty\\npe\\n\\ns\\n\\np = 0.01 d\\n-1\\n\\n10\\n1\\n\\n10\\n2\\n\\n10\\n3\\n\\nC\\nT\\n\\nL\\n\\nAcute\\nChronic\\n\\nExtinction\\n\\nA\\n\\nB\\n\\nC\\n\\n} p = 0.02 d\\n-1\\n\\nFig. 3. Typical disease progression patterns in the extinction regime from\\nthe multi-epitope quasispecies model. (A) Viral load. (B) Total CTL levels.\\n(C) Number of distinct phenotypes per epitope present in the population.\\nThe parameter values other than p and μ ¼ 1.0 × 10−5 bp−1 were the same\\nas in Fig. 2. The data shown are individual trajectories.\\n\\n10\\n-6\\n\\n10\\n-5\\n\\n10\\n-4\\n\\n10\\n-3\\n\\n10\\n-2\\n\\n10\\n-1\\n\\nMutation rate µ (bp-1)\\n\\n10\\n-2\\n\\n10\\n-1\\n\\nC\\nle\\n\\nar\\nan\\n\\nce\\n r\\n\\nat\\ne \\n\\np \\n(d\\n\\n-1\\n)\\n\\n10\\n-6\\n\\n10\\n-5\\n\\n10\\n-4\\n\\n10\\n-3\\n\\n10\\n-2\\n\\n10\\n-1\\n\\nMutation rate µ (bp-1)\\n\\n10\\n0\\n\\n10\\n1\\n\\n10\\n2\\n\\n10\\n3\\n\\nE\\nxt\\n\\nin\\nct\\n\\nio\\nn \\n\\ntim\\ne \\n\\n(d\\n)\\n\\np = 0.10 d\\n-1\\n\\np = 0.02 d\\n-1\\n\\nA B\\nExtinction\\n\\nRunaway\\nHIV-1\\n\\nFig. 4. Crossover between the runaway and extinction regimes. (A) Varia-\\ntion of extinction time with mutation rate μ for different clearance rates\\np in the extinction regime. Error bars represent one standard deviation. Near\\nthe threshold, the extinction events occur increasingly in the edge of viral\\nload fluctuations (Fig. 3A). (B) The estimated threshold between the runaway\\nand extinction regimes in the p − μ parameter space (defined as the maxi-\\nmum μ for which the extinction time converges for a given p). The arrow\\nillustrates the dynamic deterioration of immune response during HIV-1 infec-\\ntion. Other parameter values were the same as in Fig. 2.\\n\\nWoo and Reifman PNAS ∣ August 7, 2012 ∣ vol. 109 ∣ no. 32 ∣ 12983\\n\\nBI\\nO\\nPH\\n\\nYS\\nIC\\nS\\nA\\nN\\nD\\n\\nCO\\nM\\nPU\\n\\nTA\\nTI\\nO\\nN\\nA\\nL\\nBI\\nO\\nLO\\n\\nG\\nY\\n\\nD\\now\\n\\nnl\\noa\\n\\nde\\nd \\n\\nby\\n g\\n\\nue\\nst\\n\\n o\\nn \\n\\nA\\npr\\n\\nil \\n23\\n\\n, 2\\n02\\n\\n0 \\n\\n\\n\\nidentical phenotypes in a large fraction of their epitopes and are\\nconstrained by common groups of CTLs. However, the model\\ndoes not account for the effects of direct competitions for re-\\nsources (limited number of host cells available for infection).\\nWe have additionally explored an extended model that includes\\nhost-cell dynamics (SI Text), whose analysis suggests that the main\\nqualitative results remain valid (Fig. S8).\\n\\nOur finding that the relative importance of immune escape is\\nmainly determined by the viral load in chronic infection may have\\nimplications to vaccine design (42): Strategies aimed at prevent-\\ning immune escape, such as targeting epitopes with higher fitness\\ncosts to mutations, may not confer more benefits than those at-\\ntempting to boost the overall immune response. We also note,\\nhowever, that more realistic representations of T-cell stimulation\\nand clearance will have to take their dependence on phenotypes\\ninto account (p and b should depend on amino acid sequences),\\nbecause mutations affect HLA-epitope-CTL binding affinities.\\nOur model also ignores complex epistatic effects coupling muta-\\ntions on the same and different epitopes (43). The approach pre-\\nsented here likely can provide a foundation for a more compre-\\nhensive modeling framework to tackle such global genome-wide\\neffects. In addition, although we interpreted Eqs. 3B–3D strictly\\nas the CTL-mediated immune response, we expect similar ap-\\nproaches to be applicable for antibody-based responses.\\n\\nMethods\\nStochastic Evolutionary Dynamics. Simulations are carried out by applying the\\nGillespie algorithm (26) to Eq. 3. At a given time, a dynamic list of genotypes\\nand phenotypes (N and M in total, respectively), starting with a single WT at\\nthe initial condition, is kept and updated, instead of enumerating all possible\\nsequences. A genotype consists of the nucleotide sequence of a set of epi-\\ntopes (l in total), each with the same length L, such that the total length be-\\ncomes l × L. We define the following quantities\\n\\naðrÞi ¼ ∑\\ni\\n\\nj¼1\\nrjnj; a\\n\\nðsÞ\\ni ¼ ∑\\n\\ni\\n\\nj¼1\\nðrj þ bÞnj;\\n\\naðcÞi ¼ ∑\\ni\\n\\nj¼1\\nðrj þ bþ pcjÞnj; aðdÞα ¼ aðcÞN þ g∑\\n\\nα\\n\\nβ¼1\\ncðβÞ;\\n\\neach corresponding to the replication, stimulation, clearance, and decay re-\\nactions (Eq. 3), respectively, where cj ¼ ΣkcðαjkÞ is the total number of CTLs\\ncorresponding to any of the epitope phenotypes belonging to the genotype\\nj. A random number 0 < r < 1 is drawn to determine the time interval Δt ¼\\nð1∕aÞ lnð1∕rÞ before the next reaction event, where a ¼ a ðdÞM . A second ran-\\ndom number r is drawn, and if aðxÞi−1 < ar < a\\n\\nðxÞ\\ni for x ¼ r, s, c, the replication,\\n\\nstimulation, or clearance reaction is chosen for the genotype i, respectively. If\\na ðdÞα−1 < ar < a\\n\\nðdÞ\\nα instead, the decay reaction is carried out for the CTL with phe-\\n\\nnotype α. In this algorithm, therefore, a given CTL contributes equally to the\\nclearance probabilities of all viral strains containing its phenotype. This pre-\\nscription generates unbiased trajectories of two classes of objects, the viruses\\nwith genotype i and phenotype αik and CTLs with phenotype α. For replica-\\ntion reactions, the chosen genotype i is mutated with probability μ per bp.\\nFurther details on implementing the genotype-phenotype lists can be found\\nin SI Text.\\n\\nSimulation Conditions. The standard genetic code was used to translate\\nmutant genotypes into phenotypes. The reduced fitness sα ¼ rατ of a pheno-\\ntype αwas taken as follows: sα ¼ 1 for theWT, and for all other phenotypes sα\\nis a random variable with the following Gaussian distribution: PðsαÞ ¼\\nexpf−½sα − fðdαÞ�2�∕2σ2g∕ð2πÞ1∕2σ, where the position of the maximum\\nfðdαÞ ¼ expð−dα∕ξÞ is an exponentially decreasing function of the distance\\ndα ≥ 1 (summed over all epitopes; in units of amino acids) of the phenotype\\nα to the WT. The parameters ξ and σ are the measures of the fitness costs of\\nmutations and of randomness in fitness distributions, respectively. During si-\\nmulations, a newly encountered phenotype is dynamically assigned a fitness\\nvalue based on this distributionwith negative rα values replaced by zero. Phe-\\nnotypes containing stop codons were assigned zero fitness. In Fig. 1, the SIV\\nepitope Tat28–35SL8 with the WT sequence of 5′-TCC ACT CCA GAA TCG GCC\\nAAC CTG-3′ (STPESANL) (38) was used (La ¼ 8 and l ¼ 1). In Figs. 2–4, random\\nsequences were used for the WT and simulations were performed with La ¼\\n10 and l ¼ 3. Typically, quantities were averaged over 102 ∼ 103 trajectories\\nunless otherwise specified.\\n\\nACKNOWLEDGMENTS. This work was sponsored by both the US Army’s\\nNetwork Science Initiative, managed by the Military Operational Medicine\\nResearch Program of the US Army Medical Research and Materiel Command,\\nFort Detrick, Maryland, and by a competitive In-House Laboratory Indepen-\\ndent Research award by the US Army Assistant Secretary of the Army for\\nAcquisition, Logistics, and Technology. The opinions and assertions contained\\nherein are the private views of the authors and are not to be construed as\\nofficial or as reflecting the views of the US Army or of the US Department\\nof Defense. This paper has been approved for public release with unlimited\\ndistribution.\\n\\n1. Eigen M (1971) Selforganization of matter and the evolution of biological macromo-\\nlecules. Naturwissenschaften 58:465–523.\\n\\n2. Swetina J, Schuster P (1982) Self-replication with errors: A model for polynucleotide\\nreplication. Biophys Chem 16:329–345.\\n\\n3. Loeb LA, et al. (1999) Lethal mutagenesis of HIV with mutagenic nucleoside analogs.\\nProc Natl Acad Sci USA 96:1492–1497.\\n\\n4. Iranzo J, Perales C, Domingo E, Manrubia SC (2011) Tempo and mode of inhibitor-\\nmutagen antiviral therapies: A multidisciplinary approach. Proc Natl Acad Sci USA\\n108:16008–16013.\\n\\n5. Domingo E, Holland JJ (1997) RNA virus mutations and fitness for survival. Annu Rev\\nMicrobiol 51:151–178.\\n\\n6. Mardis ER (2008) Next-generation DNA sequencing methods. Annu Rev Genomics\\nHum Genet 9:387–402.\\n\\n7. McMichael AJ, Borrow P, Tomaras GD, Goodnetilleke N, Haynes BF (2010) The immune\\nresponse during acute HIV-1 infection: Clues for vaccine development. Nat Rev Immu-\\nnol 10:11–23.\\n\\n8. Mild M, Hedskog C, Jernberg J, Albert J (2011) Performance of ultradeep pyrosequen-\\ncing in analysis of HIV-1 pol gene variation. PLoS One 6:e22741.\\n\\n9. Bimber BN, et al. (2009) Ultradeep pyrosequencing detects complex patterns of CD8þ\\n\\nT-lymphocyte escape in simian immunodeficiency virus-infected macaques. J Virol\\n83:8247–8253.\\n\\n10. Bimber BN, et al. (2010) Whole-genome characterization of human and simian\\nimmunodeficiency virus intrahost diversity by ultradeep pyrosequencing. J Virol\\n84:12087–12092.\\n\\n11. Fischer W, et al. (2010) Transmission of single HIV-1 genomes and dynamics of early\\nimmune escape revealed by ultradeep sequencing. PLoS One 5:e12303.\\n\\n12. Cale EM, et al. (2011) Epitope-specific CD8þ T lymphocytes cross-recognize mutant\\nsimian immunodefieciency virus (SIV) sequences but fail to contain very early evolu-\\ntion and eventual fixation of epitope escape mutations during SIV infection. J Virol\\n85:3746–3757.\\n\\n13. Moore CB, et al. (2002) Evidence of HIV-1 adaptation to HLA-restricted immune\\nresponses at a population level. Science 296:1439–1443.\\n\\n14. Bhattacharya T, et al. (2007) Founder effects in the assessment of HIV polymorphisms\\nand HLA allele associations. Science 315:1583–1586.\\n\\n15. Kawashima Y, et al. (2009) Adaptation of HIV-1 to human leukocyte antigen class I.\\nNature 458:641–645.\\n\\n16. Addo MM, et al. (2003) Comprehensive epitope analysis of human immunodeficiency\\nvirus type 1 (HIV-1)-specific T-cell responses directed against the entire expressed HIV-1\\ngenome demonstrate broadly directed responses, but no correlation to viral load.\\nJ Virol 77:2081–2092.\\n\\n17. Goodnetilleke N, et al. (2009) The first T-cell response to transmitted/founder\\nvirus contributes to the control of acute viremia in HIV-1 infection. J Exp Med\\n206:1253–1272.\\n\\n18. Nowak MA, Bangham CR (1996) Population dynamics of immune responses to persis-\\ntent viruses. Science 272:74–79.\\n\\n19. Nowak MA, May RM (2001) Virus Dynamics: Mathematical Principles of Immunology\\nand Virology (Oxford Univ Press, New York).\\n\\n20. Nowak MA, et al. (1995) Antigenic oscillations and shifting immunodominance in\\nHIV-1 infections. Nature 375:606–611.\\n\\n21. Althaus CL, De Boer RJ (2008) Dynamics of immune escape during HIV/SIV infection.\\nPLoS Comput Biol 4:e1000103.\\n\\n22. Love TM, Thurston SW, Keefer MC, Dewhurst S, Lee HY (2010) Mathematical modeling\\nof ultradeep sequencing data reveals that acute CD8þ T-lymphocyte responses exert\\nstrong selective pressure in simian immunodeficiency virus-infected macaques but still\\nfail to clear founder epitope sequences. J Virol 84:5802–5814.\\n\\n23. Lee HY, et al. (2009) Modeling sequence evolution in acute HIV-1 infection. J Theor Biol\\n261:341–360.\\n\\n24. Lee HY, Perelson AS, Park S-C, Leitner T (2008) Dynamic correlation between intrahost\\nHIV-1 quasispecies evolution and disease progression. PLoS Comput Biol 4:e1000240.\\n\\n25. Keele BF, et al. (2008) Identification and characterization of transmitted and\\nearly founder virus envelopes in primary HIV-1 infection. Proc Natl Acad Sci USA\\n105:7552–7557.\\n\\n26. Gillespie DT (1977) Exact stochastic simulation of coupled chemical reactions. J Phys\\nChem 81:2340–2361.\\n\\n27. Nowak M, Schuster P (1989) Error thresholds of replication in finite populations muta-\\ntion frequencies and the onset of Muller’s ratchet. J Theor Biol 137:375–395.\\n\\n28. Yang Z, Bielawski JP (2000) Statistical methods for detecting molecular adaptation.\\nTrends Ecol Evol 15:496–503.\\n\\n12984 ∣ www.pnas.org/cgi/doi/10.1073/pnas.1117201109 Woo and Reifman\\n\\nD\\now\\n\\nnl\\noa\\n\\nde\\nd \\n\\nby\\n g\\n\\nue\\nst\\n\\n o\\nn \\n\\nA\\npr\\n\\nil \\n23\\n\\n, 2\\n02\\n\\n0 \\n\\nhttp://www.pnas.org/lookup/suppl/doi:10.1073/pnas.1117201109/-/DCSupplemental/pnas.1117201109_SI.pdf?targetid=STXT\\nhttp://www.pnas.org/lookup/suppl/doi:10.1073/pnas.1117201109/-/DCSupplemental/pnas.1117201109_SI.pdf?targetid=SF8\\nhttp://www.pnas.org/lookup/suppl/doi:10.1073/pnas.1117201109/-/DCSupplemental/pnas.1117201109_SI.pdf?targetid=STXT\\n\\n\\n29. Aguirre J, Buldu JM, Stich M, Manrubia SC (2011) Topological structure of the space of\\nphenotypes: The case of RNA neutral networks. PLoS One 6:e26324.\\n\\n30. Burch CL, Chao L (2000) Evolvability of an RNA virus is determined by its mutational\\nneighborhood. Nature 406:625–628.\\n\\n31. Schuster P, Fontana W (1999) Chance and necessity in evolution: Lessons from RNA.\\nPhysica D 133:427–452.\\n\\n32. Hietpas RT, Jensen JD, Bolon DN (2011) Experimental illumination of a fitness land-\\nscape. Proc Natl Acad Sci USA 108:7896–7901.\\n\\n33. Stich M, Briones C, Manrubia SC (2008) On the structural repertoire of pools of short,\\nrandom RNA sequences. J Theor Biol 252:750–763.\\n\\n34. Pitt JN, Ferre-D’Amare AR (2010) Rapid construction of empirical RNA fitness land-\\nscapes. Science 330:376–379.\\n\\n35. Duarte EA, et al. (1994) Subclonal components of consensus fitness in an RNA virus\\nclone. J Virol 68:4295–4301.\\n\\n36. Sanjuan R, Moya A, Elena SF (2004) The distribution of fitness effects caused by single-\\nnucleotide substitutions in an RNA virus. Proc Natl Acad Sci USA 101:8396–8401.\\n\\n37. Fernandez G, Clotet B, Martinez MA (2007) Fitness landscape of human immunode-\\nficiency virus type 1 protease quasispecies. J Virol 81:2485–2496.\\n\\n38. Allen TM, et al. (2000) Tat-specific cytotoxic T lymphocytes select for SIV escape\\nvariants during resolution of primary viraemia. Nature 407:386–390.\\n\\n39. Mansky LM, Temin HM (1995) Lower in vivo mutation rate of human immunodefi-\\nciency virus type 1 than that predicted from the fidelity of purified reverse transcrip-\\ntase. J Virol 69:5087–5094.\\n\\n40. Schuster P, Swetina J (1988) Stationary mutant distributions and evolutionary optimi-\\nzation. Bull Math Biol 50:635–660.\\n\\n41. Wilke CO,Wang JL, Ofria C, Lenski RE, Adami C (2001) Evolution of digital organisms at\\nhigh mutation rates leads to survival of the flattest. Nature 412:331–333.\\n\\n42. Gaschen B, et al. (2002) Diversity considerations in HIV-1 vaccine selection. Science\\n296:2354–2360.\\n\\n43. Dahirel V, et al. (2011) Coordinate linkage of HIV evolution reveals regions of immu-\\nnological vulnerability. Proc Natl Acad Sci USA 108:11530–11535.\\n\\nWoo and Reifman PNAS ∣ August 7, 2012 ∣ vol. 109 ∣ no. 32 ∣ 12985\\n\\nBI\\nO\\nPH\\n\\nYS\\nIC\\nS\\nA\\nN\\nD\\n\\nCO\\nM\\nPU\\n\\nTA\\nTI\\nO\\nN\\nA\\nL\\nBI\\nO\\nLO\\n\\nG\\nY\\n\\nD\\now\\n\\nnl\\noa\\n\\nde\\nd \\n\\nby\\n g\\n\\nue\\nst\\n\\n o\\nn \\n\\nA\\npr\\n\\nil \\n23\\n\\n, 2\\n02\\n\\n0 \\n\\n\\n', 'metadata': {'xmpTPg:NPages': '6', 'Last-Save-Date': '2020-04-23T07:03:44Z', 'Creation-Date': '2012-07-28T00:39:16Z', 'Content-Type': 'application/pdf', 'X-Parsed-By': ['org.apache.tika.parser.DefaultParser', 'org.apache.tika.parser.pdf.PDFParser'], 'date': '2020-04-23T07:03:44Z', 'pdf:docinfo:modified': '2020-04-23T07:03:44Z', 'dc:title': '201117201 12980..12985', 'created': '2012-07-28T00:39:16Z', 'access_permission:assemble_document': 'true', 'access_permission:modify_annotations': 'true', 'access_permission:extract_for_accessibility': 'true', 'access_permission:fill_in_form': 'true', 'xmpMM:DocumentID': 'uuid:cb69f4d0-b7da-45a8-a0bc-3d44c8259269', 'resourceName': '12980.full.pdf', 'dc:format': 'application/pdf; version=1.4', 'dcterms:created': '2012-07-28T00:39:16Z', 'title': '201117201 12980..12985', 'access_permission:can_modify': 'true', 'Last-Modified': '2020-04-23T07:03:44Z', 'pdf:encrypted': 'false', 'producer': 'Acrobat Distiller 6.0.1 (Windows)', 'pdf:docinfo:creator_tool': 'Arbortext Advanced Print Publisher 9.1.405/W Unicode', 'access_permission:can_print_degraded': 'true', 'access_permission:can_print': 'true', 'pdf:docinfo:producer': 'Acrobat Distiller 6.0.1 (Windows)', 'meta:save-date': '2020-04-23T07:03:44Z', 'xmp:CreatorTool': 'Arbortext Advanced Print Publisher 9.1.405/W Unicode', 'X-TIKA:parse_time_millis': '70', 'access_permission:extract_content': 'true', 'meta:creation-date': '2012-07-28T00:39:16Z', 'dcterms:modified': '2020-04-23T07:03:44Z', 'pdf:PDFVersion': '1.4', 'pdf:docinfo:title': '201117201 12980..12985', 'modified': '2020-04-23T07:03:44Z', 'pdf:docinfo:created': '2012-07-28T00:39:16Z'}}\n"
     ]
    }
   ],
   "source": [
    "print(parsedPDF)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "dict_keys(['status', 'content', 'metadata'])"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# parsed pdf is stored as a dictionary; here we get the keys\n",
    "parsedPDF.keys()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "201117201 12980..12985\n",
      "\n",
      "\n",
      "A quantitative quasispecies theory-based model of\n",
      "virus escape mutation under immune selection\n",
      "Hyung-June Woo and Jaques Reifman1\n",
      "\n",
      "Biotechnology High Performance Computing Software Applications Institute, Telemedicine and Advanced Technology Research Center, US Army Medical\n",
      "Research and Materiel Command, Fort Detrick, MD 21702\n",
      "\n",
      "Edited by Peter Schuster, University of Vienna, Vienna, and approved June 28, 2012 (received for review October 18, 2011)\n",
      "\n",
      "Viral infections involve a complex interplay of the immune\n",
      "response and escape mutation of the virus quasispecies inside a\n",
      "single host. Although fundamental aspects of such a balance of\n",
      "mutation and selection pressure have been established by the qua-\n",
      "sispecies theory decades ago, its implications have largely re-\n",
      "mained qualitative. Here, we present a quantitative approach to\n",
      "model the virus evolution under cytotoxic T-lymphocyte immune\n",
      "response. The virus quasispecies dynamics are explicitly repre-\n",
      "sented by mutations in the combined sequence space of a set of\n",
      "epitopes within the viral genome. We stochastically simulated the\n",
      "growth of a viral population originating from a single wild-type\n",
      "founder virus and its recognition and clearance by the immune re-\n",
      "sponse, as well as the expansion of its genetic diversity. Applied to\n",
      "the immune escape of a simian immunodeficiency virus epitope,\n",
      "model predictions were quantitatively comparable to the experi-\n",
      "mental data. Within the model parameter space, we found two\n",
      "qualitatively different regimes of infectious disease pathogenesis,\n",
      "each representing alternative fates of the immune response: It can\n",
      "clear the infection in finite time or eventually be overwhelmed by\n",
      "viral growth and escape mutation. The latter regime exhibits the\n",
      "characteristic disease progression pattern of human immunodefi-\n",
      "ciency virus, while the former is bounded by maximum mutation\n",
      "rates that can be suppressed by the immune response. Our results\n",
      "demonstrate that, by explicitly representing epitope mutations\n",
      "and thus providing a genotype–phenotype map, the quasispecies\n",
      "theory can form the basis of a detailed sequence-specific model of\n",
      "real-world viral pathogens evolving under immune selection.\n",
      "\n",
      "population dynamics ∣ next-generation sequencing ∣ stochastic simulation ∣\n",
      "HIV\n",
      "\n",
      "Viruses with RNA genomes, such as human immunodeficiencyvirus type 1 (HIV-1), have high rates of mutation and evolve\n",
      "rapidly in response to host immune selection pressure. One of the\n",
      "consequences of such rapid mutations is the error catastrophe\n",
      "(1), where a virus population is driven to extinction when its mu-\n",
      "tation rate exceeds a threshold. The existence of such a threshold\n",
      "is a central prediction of the quasispecies theory pioneered by\n",
      "Eigen (1) and Swetina and Schuster (2). The recent experimental\n",
      "demonstration of lethal mutagenesis (3–5), in which an error cat-\n",
      "astrophe transition is caused by mutagens, demonstrates that key\n",
      "features of the balance of mutation and selection in viruses are\n",
      "elegantly captured by the quasispecies theory.\n",
      "\n",
      "Although insights provided by the quasispecies theory have\n",
      "greatly expanded our understanding of virus behavior, applica-\n",
      "tions so far have been limited to a conceptual level, partially due\n",
      "to the lack of experimental information on the evolutionary dy-\n",
      "namics in sequence space. Next-generation sequencing techni-\n",
      "ques (6) have the potential to change this situation. To build\n",
      "quantitative models implementing the quasispecies dynamics\n",
      "and describe such experimental data, it is important to realisti-\n",
      "cally specify the nature of selection pressure. Viruses in animal\n",
      "hosts evolve under immune pressure, and their capacity for rapid\n",
      "escape mutation underlies many of the difficulties in combating\n",
      "pathogens, including HIV-1. In a typical disease pathogenesis of\n",
      "HIV-1, the acute viremia after an initial infection is curbed by\n",
      "\n",
      "CD8þ cytotoxic T-lymphocyte (CTL) responses as well as subse-\n",
      "quent antibody actions, leading to an asymptomatic chronic infec-\n",
      "tion stage that can last up to 10 y (7). However, this apparent\n",
      "control of viremia is never complete, and without antiviral ther-\n",
      "apy, the chronic infection eventually leads to the onset of disease.\n",
      "This chronic infection stage involves continuous escape mutation-\n",
      "CTL response cycles, whose detailed characteristics are being un-\n",
      "covered by ultradeep sequencing (8–12). CTLs recognize specific\n",
      "viral epitopes (approximately 10 amino acids long) presented on\n",
      "the surface of infected cells by class I human leukocyte antigen\n",
      "(HLA). The epitope recognition depends sensitively on HLA al-\n",
      "leles, leading to differential patterns of immune response among\n",
      "patients (13–15), while characteristics of immune response during\n",
      "early infection often shape and influence the overall disease pro-\n",
      "gression (16, 17). Quantitative sequence-based models of virus-\n",
      "CTL dynamics will greatly facilitate the interpretation of experi-\n",
      "mental data.\n",
      "\n",
      "In a series of pioneering works (18–20), Nowak and coworkers\n",
      "introduced population dynamics concepts that capture a diverse\n",
      "range of immune response and escape mutation. Similar ap-\n",
      "proaches focused more on escape dynamics have also been pro-\n",
      "posed (21, 22). These models, however, do not describe mutations\n",
      "in sequence space explicitly. Other mathematical models of\n",
      "sequence evolution focus on sequence divergence during the very\n",
      "early stages of infection, while ignoring or only implicitly including\n",
      "the effect of selection pressure (23, 24). In this paper, we describe\n",
      "a quantitative quasispecies-based model of virus dynamics under\n",
      "T cell-based immune pressure, explicitly mapping genotype–phe-\n",
      "notype relationships. Our approach combines the description of\n",
      "virus evolutionary dynamics provided by the quasispecies theory\n",
      "and the population dynamics models of Nowak and coworkers.\n",
      "We show that the model not only captures the salient features\n",
      "observed in sequencing data of simian immunodeficiency virus\n",
      "(SIV) (9) but also reveals general qualitative features of viral\n",
      "infection disease pathogenesis: The immune system can clear\n",
      "the infection within time scales ranging from days to those much\n",
      "longer than patient lifetimes, or be overwhelmed by immune es-\n",
      "cape. The viral load progression in the latter regime closely\n",
      "matches the observed HIV-1 disease pathogenesis pattern.\n",
      "\n",
      "Results and Discussion\n",
      "Stochastic Quasispecies Dynamics.During acute infection, the foun-\n",
      "der virus (often a single virion) (7, 11, 17, 25) undergoes replica-\n",
      "tions to produce an exponentially growing population, which may\n",
      "be described by the quasispecies dynamics without degradation,\n",
      "\n",
      "Author contributions: H.-J.W. designed research; H.-J.W. performed research; H.-J.W. and\n",
      "J.R. analyzed data; and H.-J.W. and J.R. wrote the paper.\n",
      "\n",
      "The authors declare no conflict of interest.\n",
      "\n",
      "This article is a PNAS Direct Submission.\n",
      "\n",
      "Freely available online through the PNAS open access option.\n",
      "1To whom correspondence should be addressed. E-mail: jaques.reifman@us.army.mil.\n",
      "\n",
      "This article contains supporting information online at www.pnas.org/lookup/suppl/\n",
      "doi:10.1073/pnas.1117201109/-/DCSupplemental.\n",
      "\n",
      "12980–12985 ∣ PNAS ∣ August 7, 2012 ∣ vol. 109 ∣ no. 32 www.pnas.org/cgi/doi/10.1073/pnas.1117201109\n",
      "\n",
      "D\n",
      "ow\n",
      "\n",
      "nl\n",
      "oa\n",
      "\n",
      "de\n",
      "d \n",
      "\n",
      "by\n",
      " g\n",
      "\n",
      "ue\n",
      "st\n",
      "\n",
      " o\n",
      "n \n",
      "\n",
      "A\n",
      "pr\n",
      "\n",
      "il \n",
      "23\n",
      "\n",
      ", 2\n",
      "02\n",
      "\n",
      "0 \n",
      "\n",
      "http://www.pnas.org/lookup/suppl/doi:10.1073/pnas.1117201109/-/DCSupplemental\n",
      "http://www.pnas.org/lookup/suppl/doi:10.1073/pnas.1117201109/-/DCSupplemental\n",
      "http://www.pnas.org/lookup/suppl/doi:10.1073/pnas.1117201109/-/DCSupplemental\n",
      "http://www.pnas.org/lookup/suppl/doi:10.1073/pnas.1117201109/-/DCSupplemental\n",
      "http://www.pnas.org/lookup/suppl/doi:10.1073/pnas.1117201109/-/DCSupplemental\n",
      "http://www.pnas.org/lookup/suppl/doi:10.1073/pnas.1117201109/-/DCSupplemental\n",
      "\n",
      "\n",
      "_nj ¼ ∑\n",
      "i\n",
      "\n",
      "Qjirini; [1]\n",
      "\n",
      "where ni is the number of virions with genotype i, ri is the repli-\n",
      "cation rate of i, andQji ¼ ð1 − μÞL−dðj;iÞðμ∕3Þdðj;iÞ is the mutation\n",
      "probability from i to genotype j, with mean mutation rate μ,\n",
      "genome length L (in bp), and Hamming distance (the number\n",
      "of nucleotides that are different) dðj; iÞ between j and i. A more\n",
      "complete description for early stages of infection, where ni is of\n",
      "the order of 1 and restricted to integer values, is given by the che-\n",
      "mical reaction representation\n",
      "\n",
      "Ri →\n",
      "Qjiri\n",
      "\n",
      "Ri þRj; [2]\n",
      "\n",
      "where Ri is a virion with genotype i. Eq. 2 can be simulated\n",
      "stochastically with the Gillespie algorithm (26). A major advan-\n",
      "tage of such a stochastic formulation is that it allows us to avoid\n",
      "exhaustive enumerations of all possible genotypes. This is signif-\n",
      "icant even for a single epitope of viral proteins: An amino acid\n",
      "sequence with La ¼ 10, whose genome length is L ¼ 3La ¼\n",
      "30 bp, still has 4L ∼ 1018 genotypes. Nowak and Schuster (27)\n",
      "performed Gillespie algorithm simulations of the standard qua-\n",
      "sispecies model under the single-peak fitness landscape (ri ¼ 1∕τ\n",
      "for the wild-type (WT) sequence and ri ¼ 1∕Aτ for all other mu-\n",
      "tants, whereA > 1 is the relative fitness of theWTand τ is a char-\n",
      "acteristic time scale). In this case, distributions of individuals\n",
      "among only two groups (WT and mutants) need to be tracked.\n",
      "\n",
      "In our numerical simulations, the initial genotype (referred to\n",
      "here as the WT) replicates with mutation rate μ, and the gener-\n",
      "ated mutants are compared with the existing list of sequences,\n",
      "which is updated dynamically when new sequences arise. We ver-\n",
      "ified that this numerical scheme sampled the relevant sequence\n",
      "space sufficiently (SI Text, Figs. S1–S3). The initial condition we\n",
      "used (single WT) and the discreteness of ni imply that ni should\n",
      "be interpreted as the total number of virions within a finite sys-\n",
      "tem. The HIV-1 viral load during the acute infection phase can\n",
      "reach up to 104 ∼ 106 RNA copies/mL (7). Accounting for the\n",
      "volume of blood in the body of an average adult (approximately\n",
      "5 L), the total population size Nv ¼ Σi ni would be up to about\n",
      "109. We found simulations to slow down significantly as Nv be-\n",
      "came larger than about 106. For computational efficiency, we\n",
      "therefore regarded the system as a small representative volume\n",
      "(1 mL) of blood, and the viral loads and CTL levels reported refer\n",
      "to RNA copy numbers and cell counts within this volume.\n",
      "\n",
      "CTL Response. Within an infected host, the action of the cellular\n",
      "immune response provides the major force countering the acute\n",
      "viremia. In this work, we combine the population dynamics ap-\n",
      "proach of virus-immune system dynamics by Nowak and cowor-\n",
      "kers (18, 19) with the classic quasispecies theory. Accordingly,\n",
      "Eq. 2 is modified as follows:\n",
      "\n",
      "Ri →\n",
      "Qjiri\n",
      "\n",
      "Ri þRj; ðreplication with mutationÞ [3A]\n",
      "\n",
      "Ri→\n",
      "b\n",
      "Ri þ∑\n",
      "\n",
      "l\n",
      "\n",
      "k¼1\n",
      "CðαikÞ; ðstimulationÞ [3B]\n",
      "\n",
      "Ri þ CðαikÞ→\n",
      "p\n",
      "CðαikÞ; ðclearanceÞ [3C]\n",
      "\n",
      "CðαÞ→\n",
      "g\n",
      "0; ðdecayÞ [3D]\n",
      "\n",
      "where the genotype i is now a member of the combined sequence\n",
      "space of l epitopes. The amino acid sequence (“phenotype”) of\n",
      "epitope k in genotype i is denoted as αik, and CðαÞ represents a\n",
      "\n",
      "CTL specific to phenotype α. Eq. 3B represents a reaction in\n",
      "which, with a rate b, virions of sequence i stimulate the produc-\n",
      "tion of a set of CTLs CðαikÞ corresponding to the phenotypes of\n",
      "its epitopes. Eq. 3C denotes the reaction where cells infected with\n",
      "viruses with genotype i are cleared by CTLs with phenotypes that\n",
      "match one of its epitopes with a rate p, and Eq. 3D represents the\n",
      "natural decay of T cells with a rate g. Under the alternative de-\n",
      "terministic continuum approximation, one may write:\n",
      "\n",
      "_nj ¼ ∑\n",
      "i\n",
      "\n",
      "Qjirini − p∑\n",
      "l\n",
      "\n",
      "k¼1\n",
      "cðαjkÞnj; [4A]\n",
      "\n",
      "_cðαÞ ¼ bnα − gcðαÞ; [4B]\n",
      "\n",
      "where cðαÞ is the number of Tcells CðαÞ and nα is the total num-\n",
      "ber of virions containing epitopes with phenotype α.\n",
      "\n",
      "The qualitative features of the infection-clearance dynamics\n",
      "can be illustrated by setting the mutation rate μ to zero. Numer-\n",
      "ical integrations of Eq. 4 yield the continuum approximation re-\n",
      "sult (Fig. S4 for μ ¼ 0 and l ¼ 1), which showed reasonable\n",
      "agreement with the stochastic simulation results. For nonvanish-\n",
      "ing mutation rates, numerical integration rapidly becomes prohi-\n",
      "bitive with increasing L because of the high dimensionality of the\n",
      "sequence space. The inverse of b is a measure of the time delay of\n",
      "the growth of T cells compared to that of viruses. Here, we\n",
      "adopted b values of approximately 0.01 d−1. The efficiency of\n",
      "CTLs in recognizing cells presenting the epitope and killing them\n",
      "is represented by parameter p, the clearance rate. We found\n",
      "p values of 10−4 ∼ 10−3 d−1 to give good fits to experiments\n",
      "(see below).\n",
      "\n",
      "Immune Escape Dynamics. To make comparisons with experiments,\n",
      "it is necessary to consider that fitness is determined by amino acid\n",
      "sequences, and many mutations are synonymous. Therefore, the\n",
      "full immune escape dynamics represented by Eq. 3 distinguishes\n",
      "the amino acid sequence αik corresponding to the nucleotide\n",
      "sequence of epitope k in genotype i. We refer to the set of nu-\n",
      "cleotide and amino acid sequences as “genotypes” and “pheno-\n",
      "types,” respectively. The fitness ri is a function of phenotypes\n",
      "only. During simulations, each genotype is translated into a phe-\n",
      "notype, and a new mutant is checked against the existing pheno-\n",
      "type/genotype list, which is updated and expanded. The nature of\n",
      "this genotype–phenotype map plays important roles in evolution-\n",
      "ary dynamics, leading to key signatures of selective pressure, such\n",
      "as the ratio of synonymous to nonsynonymous mutations (28).\n",
      "Quantitative insights to the effects of this mapping on molecular\n",
      "evolution have been provided by Manrubia and coworkers (29),\n",
      "who studied short RNA sequences for which fitness can be esti-\n",
      "mated by secondary structure predictions. In our case, a simpli-\n",
      "fied overview of the genotype–phenotype map can be gleaned\n",
      "from a network representation of the genetic code (Fig. S5): If\n",
      "all mutations were neutral and amino acids within an epitope in-\n",
      "dependent, this map would be sufficient to determine the statis-\n",
      "tics of evolutionary drifts. The accessibility of certain mutations\n",
      "from a phenotype, in particular, has been shown to affect the\n",
      "evolvability of viruses (30).\n",
      "\n",
      "Another major ingredient for a realistic model of immune es-\n",
      "cape is the fitness landscape beyond the level of the single-peak\n",
      "model. Much progress has been made recently in understanding\n",
      "the nature of fitness landscapes (31–34). Explicit fitness measure-\n",
      "ments of viral clones (35, 36) and biochemical assays of proteins\n",
      "(37) both indicate that single-nucleotide substitutions lead to a\n",
      "broad distribution of fitness changes, most of which are deleter-\n",
      "ious. Therefore, one may assume that the fitness of a mutant is a\n",
      "random variable centered around a mean fitness value f . We ex-\n",
      "pect this mean fitness to be a decreasing function of distance d to\n",
      "the WT (the number of amino acids that are different), which we\n",
      "\n",
      "Woo and Reifman PNAS ∣ August 7, 2012 ∣ vol. 109 ∣ no. 32 ∣ 12981\n",
      "\n",
      "BI\n",
      "O\n",
      "PH\n",
      "\n",
      "YS\n",
      "IC\n",
      "S\n",
      "A\n",
      "N\n",
      "D\n",
      "\n",
      "CO\n",
      "M\n",
      "PU\n",
      "\n",
      "TA\n",
      "TI\n",
      "O\n",
      "N\n",
      "A\n",
      "L\n",
      "BI\n",
      "O\n",
      "LO\n",
      "\n",
      "G\n",
      "Y\n",
      "\n",
      "D\n",
      "ow\n",
      "\n",
      "nl\n",
      "oa\n",
      "\n",
      "de\n",
      "d \n",
      "\n",
      "by\n",
      " g\n",
      "\n",
      "ue\n",
      "st\n",
      "\n",
      " o\n",
      "n \n",
      "\n",
      "A\n",
      "pr\n",
      "\n",
      "il \n",
      "23\n",
      "\n",
      ", 2\n",
      "02\n",
      "\n",
      "0 \n",
      "\n",
      "http://www.pnas.org/lookup/suppl/doi:10.1073/pnas.1117201109/-/DCSupplemental/pnas.1117201109_SI.pdf?targetid=STXT\n",
      "http://www.pnas.org/lookup/suppl/doi:10.1073/pnas.1117201109/-/DCSupplemental/pnas.1117201109_SI.pdf?targetid=SF1\n",
      "http://www.pnas.org/lookup/suppl/doi:10.1073/pnas.1117201109/-/DCSupplemental/pnas.1117201109_SI.pdf?targetid=SF3\n",
      "http://www.pnas.org/lookup/suppl/doi:10.1073/pnas.1117201109/-/DCSupplemental/pnas.1117201109_SI.pdf?targetid=SF4\n",
      "http://www.pnas.org/lookup/suppl/doi:10.1073/pnas.1117201109/-/DCSupplemental/pnas.1117201109_SI.pdf?targetid=SF4\n",
      "http://www.pnas.org/lookup/suppl/doi:10.1073/pnas.1117201109/-/DCSupplemental/pnas.1117201109_SI.pdf?targetid=SF5\n",
      "http://www.pnas.org/lookup/suppl/doi:10.1073/pnas.1117201109/-/DCSupplemental/pnas.1117201109_SI.pdf?targetid=SF5\n",
      "\n",
      "\n",
      "assume to be exponential: f ðdÞ ¼ expð−d∕ξÞ, where ξ denotes a\n",
      "characteristic distance (Methods).\n",
      "\n",
      "An empirical evidence for this choice can be found from a re-\n",
      "cent experimental study by Fernández et al. (37), who measured\n",
      "the fitness landscapes of HIV-1 protease quasispecies for three\n",
      "patients. We plotted the distribution of their reported fitness\n",
      "values as a function of distance to the dominant phenotype\n",
      "and found near-exponential dependence for two quasispecies\n",
      "(Fig. S6). Our fitness function therefore models both the de-\n",
      "crease of the mean fitness away from WT and the distribution\n",
      "of neutral, deleterious, and beneficial mutants for a given dis-\n",
      "tance. In simulations, these fitness values were assigned dynami-\n",
      "cally to newly encountered phenotypes. It is important to note,\n",
      "however, that this landscape chosen is still an approximation that\n",
      "ignores many potentially important effects, such as the heteroge-\n",
      "neity of the mutational neighborhoods within the phenotype\n",
      "space.\n",
      "\n",
      "We tested this quasispecies model with the ultradeep sequen-\n",
      "cing data (9, 22) of the SIVepitope Tat28–35SL8 (38). Fig. 1 shows\n",
      "the time dependence of the viral load and WT frequency from\n",
      "simulations of the single-epitope version of Eq. 3 compared with\n",
      "the experimental data (22). The initial rapid growth of the viral\n",
      "load, its subsequent decrease as CTLs are activated, and the more\n",
      "gradual increase as mutants appear are all captured quantita-\n",
      "tively. The WT frequency decrease agrees with experimental\n",
      "trends (Fig. 1B) with signatures of two distinct time scales (22).\n",
      "The mutation rate we used (μ ¼ 2 × 10−7 bp−1) was chosen to\n",
      "obtain the best overall agreement for WT frequency with the sin-\n",
      "gle-epitope version of the model. For multiepitope applications,\n",
      "larger mutation rate values close to the experimental estimate\n",
      "for HIV-1 (μ ¼ 3.4 × 10−5 bp−1) (39) gave realistic dynamics\n",
      "(Fig. 2).\n",
      "\n",
      "Fig. 1B also shows the frequency of the most dominant escape\n",
      "mutant (EM) at each time point. As mutations occur, a new strain\n",
      "proliferates temporarily before being curbed down, and the total\n",
      "number of phenotypes continues to increase. The increase in via-\n",
      "bility of populations with multiple but lower fitness peaks has\n",
      "been described first by Schuster and Swetina (40) and has later\n",
      "been known as the “survival of the flattest” effect (41). Further\n",
      "insights into the role this expansion in sequence space plays in the\n",
      "growth of viral loads can be gained from the deterministic con-\n",
      "tinuum approximation, Eq. 4, without mutation: The immune re-\n",
      "sponse to WT leads to damped oscillations of viral load and CTL\n",
      "\n",
      "level (Fig. S4), reaching a steady state n� and c�. This process is\n",
      "repeated for each new mutant, leading to continued increases\n",
      "in n� and c� roughly proportional to the number of phenotypes\n",
      "(18). However, the timing, distribution, and the probability of\n",
      "the appearance of mutants are highly nontrivial functions of\n",
      "the characteristics of the system, including genome length, gen-\n",
      "otype–phenotype map, and fitness landscape. Our quantitative\n",
      "quasispecies-based model, Eq. 3, provides a realistic description\n",
      "of this complex diversification process via the single parameter μ.\n",
      "We note that, for a single epitope, this interpretation ignores the\n",
      "effects of competition among viral strains because each CTL\n",
      "is specific to only one phenotype. In reality, in addition to com-\n",
      "peting directly for host cells, different viral strains share their vul-\n",
      "nerability to CTLs specific to common epitopes within their\n",
      "genomes. By considering multiple epitopes within a strain, our\n",
      "model takes into account this interdependence of viral strains\n",
      "and their indirect competition arising from shared epitope phe-\n",
      "notypes (see below).\n",
      "\n",
      "As a quantitative measure of the relative degree of WT\n",
      "persistence during immune escape, we also examined the time\n",
      "(half-life) required for WT frequency to reach one-half (Fig. 1B)\n",
      "(22) with varying stimulation rate b and clearance rate p (Fig. S7).\n",
      "The SIV epitopes studied by Bimber et al. (9) were estimated to\n",
      "have a WT half-life of about 20 d (22). We found that, within our\n",
      "model, the WT half-life increases with increasing p and b in this\n",
      "range: An increase in the magnitude of selection pressure, rather\n",
      "than accelerating immune escape, suppresses viral population\n",
      "growth more effectively and reduces the chances of escape muta-\n",
      "tion (Fig. 2). The immune escape thus becomes more pronounced\n",
      "when viral loads in the chronic phase are larger with more fre-\n",
      "quent mutations.\n",
      "\n",
      "Disease Progression Under Immune Response.We examined general\n",
      "trends of viral infection-immune response dynamics within our\n",
      "quasispecies model on multiple-epitope levels (l ¼ 3) with varia-\n",
      "tions of key model parameters. The evolutionary dynamics, in\n",
      "particular, is critically affected by p, the clearance rate represent-\n",
      "ing the effectiveness of immune response, and the mutation\n",
      "rate μ. We identified two qualitatively different behaviors based\n",
      "on the eventual fate of viral load/CTL level: in one regime\n",
      "\n",
      "0 50 100 150\n",
      "10\n",
      "\n",
      "0\n",
      "\n",
      "10\n",
      "2\n",
      "\n",
      "10\n",
      "4\n",
      "\n",
      "10\n",
      "6\n",
      "\n",
      "10\n",
      "8\n",
      "\n",
      "V\n",
      "ira\n",
      "\n",
      "l l\n",
      "oa\n",
      "\n",
      "d\n",
      "\n",
      "0 50 100 150\n",
      "Time (d)\n",
      "\n",
      "10\n",
      "-2\n",
      "\n",
      "10\n",
      "-1\n",
      "\n",
      "10\n",
      "0\n",
      "\n",
      "F\n",
      "re\n",
      "\n",
      "qu\n",
      "en\n",
      "\n",
      "cy\n",
      "\n",
      "WT\n",
      "Dominant EM\n",
      "\n",
      "A\n",
      "\n",
      "B\n",
      "\n",
      "Fig. 1. Simulation of full immune escape dynamics of a single SIV epitope\n",
      "Tat28–35SL8. (A) Viral load. (B) Frequencies of WT and the dominant escape\n",
      "mutants (EM). Symbols represent experimental data from refs. 9 and 22, with\n",
      "each symbol corresponding to one of four different animals. Solid lines are\n",
      "from our stochastic quasispecies model with τ ¼ 0.8 d, μ ¼ 2.0 × 10−7 bp−1,\n",
      "b ¼ 0.01 d−1, g ¼ 0.2 d−1, and p ¼ 2.0 × 10−4 d−1. The fitness function para-\n",
      "meters were σ ¼ 0.1 and ξ ¼ 1 (Methods). The EM frequency is for a single\n",
      "trajectory while the rest are averages.\n",
      "\n",
      "10\n",
      "1\n",
      "\n",
      "10\n",
      "2\n",
      "\n",
      "10\n",
      "3\n",
      "\n",
      "10\n",
      "4\n",
      "\n",
      "V\n",
      "ira\n",
      "\n",
      "l l\n",
      "oa\n",
      "\n",
      "d\n",
      "\n",
      "10\n",
      "0\n",
      "\n",
      "10\n",
      "1\n",
      "\n",
      "10\n",
      "2\n",
      "\n",
      "10\n",
      "3\n",
      "\n",
      "Time (d)\n",
      "\n",
      "10\n",
      "0\n",
      "\n",
      "10\n",
      "1\n",
      "\n",
      "N\n",
      "o.\n",
      "\n",
      " o\n",
      "f p\n",
      "\n",
      "he\n",
      "no\n",
      "\n",
      "ty\n",
      "pe\n",
      "\n",
      "s\n",
      "\n",
      "10\n",
      "2\n",
      "\n",
      "10\n",
      "3\n",
      "\n",
      "10\n",
      "4\n",
      "\n",
      "C\n",
      "T\n",
      "\n",
      "L\n",
      "\n",
      "p = 1.0x10\n",
      "-3\n",
      "\n",
      ", µ = 1.0x10-5\n",
      "\n",
      "p = 5.0x10\n",
      "-4\n",
      "\n",
      ", µ = 1.0x10-5\n",
      "\n",
      "p = 1.0x10\n",
      "-3\n",
      "\n",
      ", µ = 1.0x10-6\n",
      "\n",
      "0\n",
      "\n",
      "4000\n",
      "\n",
      "8000\n",
      "\n",
      "Acute\n",
      "\n",
      "Chronic\n",
      "\n",
      "100\n",
      "\n",
      "A\n",
      "\n",
      "B\n",
      "\n",
      "C\n",
      "\n",
      "Disease\n",
      "\n",
      "101 102 103\n",
      "\n",
      "Fig. 2. Typical disease progression patterns in the runaway regime from the\n",
      "multiepitope quasispecies model. (A) Viral load. (B) Total CTL levels. (C) Num-\n",
      "ber of distinct phenotypes per epitope present in the population. The inset in\n",
      "A shows the viral load in linear scale for p ¼ 1.0 × 10−3 d−1 and μ ¼ 1.0 ×\n",
      "10−5 bp−1. Other parameter values were b ¼ 0.02 d−1, g ¼ 0.1 d−1, τ ¼ 1 d,\n",
      "σ ¼ 0.2, and ξ ¼ 1. The units of p and μ in the legends are d−1 and bp−1,\n",
      "respectively. All data represent averages over trajectories.\n",
      "\n",
      "12982 ∣ www.pnas.org/cgi/doi/10.1073/pnas.1117201109 Woo and Reifman\n",
      "\n",
      "D\n",
      "ow\n",
      "\n",
      "nl\n",
      "oa\n",
      "\n",
      "de\n",
      "d \n",
      "\n",
      "by\n",
      " g\n",
      "\n",
      "ue\n",
      "st\n",
      "\n",
      " o\n",
      "n \n",
      "\n",
      "A\n",
      "pr\n",
      "\n",
      "il \n",
      "23\n",
      "\n",
      ", 2\n",
      "02\n",
      "\n",
      "0 \n",
      "\n",
      "http://www.pnas.org/lookup/suppl/doi:10.1073/pnas.1117201109/-/DCSupplemental/pnas.1117201109_SI.pdf?targetid=SF6\n",
      "http://www.pnas.org/lookup/suppl/doi:10.1073/pnas.1117201109/-/DCSupplemental/pnas.1117201109_SI.pdf?targetid=SF6\n",
      "http://www.pnas.org/lookup/suppl/doi:10.1073/pnas.1117201109/-/DCSupplemental/pnas.1117201109_SI.pdf?targetid=SF4\n",
      "http://www.pnas.org/lookup/suppl/doi:10.1073/pnas.1117201109/-/DCSupplemental/pnas.1117201109_SI.pdf?targetid=SF4\n",
      "http://www.pnas.org/lookup/suppl/doi:10.1073/pnas.1117201109/-/DCSupplemental/pnas.1117201109_SI.pdf?targetid=SF7\n",
      "http://www.pnas.org/lookup/suppl/doi:10.1073/pnas.1117201109/-/DCSupplemental/pnas.1117201109_SI.pdf?targetid=SF7\n",
      "\n",
      "\n",
      "(“runaway,” Fig. 2), the chronic phase after the resolution of the\n",
      "acute infection is eventually followed by a runaway growth in viral\n",
      "loads (Fig. 2A). The level of CTLs also rises consistently during\n",
      "the chronic phase (Fig. 2B). This disease progression is accom-\n",
      "panied by an increase in the diversity of quasispecies, or the\n",
      "“spreading of clouds” (Fig. 2C). The time scale for the duration\n",
      "of the chronic phase, in particular, varied sensitively with p and μ\n",
      "(from a few months to years) (Fig. 2A). We found that the total\n",
      "number of phenotypes per epitope (Fig. 2C) generally goes down\n",
      "as the number of epitopes increases: A new escape mutant has\n",
      "much lower probability for survival with multiple epitopes be-\n",
      "cause it shares WT phenotypes on other parts of its genome with\n",
      "existing strains, and therefore is susceptible to CTLs that are al-\n",
      "ready present.\n",
      "\n",
      "The disease progression pattern in the runaway regime corre-\n",
      "sponds to situations where the immune system is overwhelmed by\n",
      "viral growth, either shortly after infection (there is no discernible\n",
      "chronic phase for sufficiently small p) or after a long chronic\n",
      "phase and accumulation of escape mutants. The latter case clo-\n",
      "sely matches the characteristic HIV-1 disease pathogenesis (7),\n",
      "revealing the major role played by the uncontrolled diversifica-\n",
      "tion of the quasispecies that overwhelms the CTL response. This\n",
      "feature was first suggested by Nowak and May (19), who coined\n",
      "the term “diversity threshold” based on a model assuming ran-\n",
      "dom appearances of mutants. Our results show that the basic qua-\n",
      "sispecies dynamics under selection pressure provide a realistic\n",
      "description of this phenomenon.\n",
      "\n",
      "If p is sufficiently large and μ sufficiently small, one enters a\n",
      "different regime (“extinction,” Fig. 3), where the acute and\n",
      "chronic phases lead to viral loads that decrease either rapidly\n",
      "or asymptotically with time. As populations shrink in size, their\n",
      "dynamics become increasingly stochastic, and extinction (a com-\n",
      "plete clearance of infection) occurs when the viral load reaches\n",
      "zero. This behavior is inherently stochastic and cannot be cap-\n",
      "tured by the deterministic approximation in Eq. 4. The time re-\n",
      "quired for clearing the infection (“extinction time”) typically has\n",
      "a broad distribution, increasing with decreasing p while becoming\n",
      "infinite above a threshold μ (Fig. 4A). The rapid clearance of in-\n",
      "fection within approximately 10 d corresponds to the normal state\n",
      "of affairs in a healthy immune system against viruses such as in-\n",
      "fluenza. As shown in Fig. 3A (p ¼ 0.01 d−1), however, the extinc-\n",
      "tion time can also reach time scales approaching a patient’s\n",
      "\n",
      "lifetime. This feature may be relevant in understanding the basis\n",
      "for some chronic viral infections with long and stable setpoints\n",
      "(e.g., hepatitis C), although the high variability of extinction times\n",
      "we observed (Fig. 4A) likely may not correspond to actual pos-\n",
      "sible clearances of such chronic infections.\n",
      "\n",
      "The threshold separating the runaway (Fig. 2) and extinction\n",
      "(Fig. 3) regimes can be identified by the maximum mutation\n",
      "rate for which the extinction time remains finite (Fig. 4A). This\n",
      "error threshold for the immune control of infection is analo-\n",
      "gous to Eigen’s threshold for genomic stability: The latter is\n",
      "the error rate above which replicators cannot maintain a stable\n",
      "master sequence. The former is a maximum mutation rate the\n",
      "given immune system can suppress. Fig. 4B shows this threshold\n",
      "as a function of p and μ. A given virus would have roughly the\n",
      "same mutation rate, while p would vary with individual patients.\n",
      "The sensitivity of disease progression with p is consistent with\n",
      "the observation that even when the fatality rate of a certain\n",
      "infection is known, the course of infection in a patient is often\n",
      "unpredictable.\n",
      "\n",
      "The case for HIV-1 is unique in the sense that without treat-\n",
      "ment, most patients eventually progress to the disease stage. One\n",
      "of the special characteristics of HIV-1 is that it targets CD4þ\n",
      "(helper) Tcells, which play critical roles in eliciting and mediating\n",
      "CTL responses. Within the context of our model, the destruction\n",
      "of CD4þ Tcells would lead to a gradual decrease in both the sti-\n",
      "mulation rate b and the clearance rate p. One therefore expects a\n",
      "continual deterioration of any initial effective control of infec-\n",
      "tion, represented by p values that decrease over the course of dis-\n",
      "ease progression. In Fig. 4B, therefore, even though patients with\n",
      "stronger immune systems may have p values initially in the extinc-\n",
      "tion regime, the continued depletion of CD4þ cells would lower p\n",
      "and cause them to cross the threshold.\n",
      "\n",
      "Conclusions\n",
      "In this paper, we demonstrated that the classic quasispecies the-\n",
      "ory can form the basis of a quantitative model of virus evolution\n",
      "under immune selection. One of the apparent challenges in de-\n",
      "veloping such a model is that viral genomes, while relatively small\n",
      "(approximately 104 bp), still constitute a huge sequence space.\n",
      "Here, we showed that a set of epitopes (La ∼ 10 amino acids)\n",
      "can be considered as minimal units of genomic segments on which\n",
      "selection acts. In standard quasispecies theory, competition be-\n",
      "tween different strains arises indirectly from the constraint of\n",
      "fixed total population size. In our model, the degradation and\n",
      "removal of virions occur via immune response, which can be re-\n",
      "garded as a concrete specification of the selection forces. In this\n",
      "case, indirect competition occurs because most viral strains share\n",
      "\n",
      "10\n",
      "1\n",
      "\n",
      "10\n",
      "2\n",
      "\n",
      "10\n",
      "3\n",
      "\n",
      "10\n",
      "4\n",
      "\n",
      "V\n",
      "ira\n",
      "\n",
      "l l\n",
      "oa\n",
      "\n",
      "d\n",
      "\n",
      "10\n",
      "0\n",
      "\n",
      "10\n",
      "1\n",
      "\n",
      "10\n",
      "2\n",
      "\n",
      "10\n",
      "3\n",
      "\n",
      "10\n",
      "4\n",
      "\n",
      "Time (d)\n",
      "\n",
      "1\n",
      "\n",
      "2\n",
      "\n",
      "3\n",
      "\n",
      "4\n",
      "\n",
      "N\n",
      "o.\n",
      "\n",
      " o\n",
      "f p\n",
      "\n",
      "he\n",
      "no\n",
      "\n",
      "ty\n",
      "pe\n",
      "\n",
      "s\n",
      "\n",
      "p = 0.01 d\n",
      "-1\n",
      "\n",
      "10\n",
      "1\n",
      "\n",
      "10\n",
      "2\n",
      "\n",
      "10\n",
      "3\n",
      "\n",
      "C\n",
      "T\n",
      "\n",
      "L\n",
      "\n",
      "Acute\n",
      "Chronic\n",
      "\n",
      "Extinction\n",
      "\n",
      "A\n",
      "\n",
      "B\n",
      "\n",
      "C\n",
      "\n",
      "} p = 0.02 d\n",
      "-1\n",
      "\n",
      "Fig. 3. Typical disease progression patterns in the extinction regime from\n",
      "the multi-epitope quasispecies model. (A) Viral load. (B) Total CTL levels.\n",
      "(C) Number of distinct phenotypes per epitope present in the population.\n",
      "The parameter values other than p and μ ¼ 1.0 × 10−5 bp−1 were the same\n",
      "as in Fig. 2. The data shown are individual trajectories.\n",
      "\n",
      "10\n",
      "-6\n",
      "\n",
      "10\n",
      "-5\n",
      "\n",
      "10\n",
      "-4\n",
      "\n",
      "10\n",
      "-3\n",
      "\n",
      "10\n",
      "-2\n",
      "\n",
      "10\n",
      "-1\n",
      "\n",
      "Mutation rate µ (bp-1)\n",
      "\n",
      "10\n",
      "-2\n",
      "\n",
      "10\n",
      "-1\n",
      "\n",
      "C\n",
      "le\n",
      "\n",
      "ar\n",
      "an\n",
      "\n",
      "ce\n",
      " r\n",
      "\n",
      "at\n",
      "e \n",
      "\n",
      "p \n",
      "(d\n",
      "\n",
      "-1\n",
      ")\n",
      "\n",
      "10\n",
      "-6\n",
      "\n",
      "10\n",
      "-5\n",
      "\n",
      "10\n",
      "-4\n",
      "\n",
      "10\n",
      "-3\n",
      "\n",
      "10\n",
      "-2\n",
      "\n",
      "10\n",
      "-1\n",
      "\n",
      "Mutation rate µ (bp-1)\n",
      "\n",
      "10\n",
      "0\n",
      "\n",
      "10\n",
      "1\n",
      "\n",
      "10\n",
      "2\n",
      "\n",
      "10\n",
      "3\n",
      "\n",
      "E\n",
      "xt\n",
      "\n",
      "in\n",
      "ct\n",
      "\n",
      "io\n",
      "n \n",
      "\n",
      "tim\n",
      "e \n",
      "\n",
      "(d\n",
      ")\n",
      "\n",
      "p = 0.10 d\n",
      "-1\n",
      "\n",
      "p = 0.02 d\n",
      "-1\n",
      "\n",
      "A B\n",
      "Extinction\n",
      "\n",
      "Runaway\n",
      "HIV-1\n",
      "\n",
      "Fig. 4. Crossover between the runaway and extinction regimes. (A) Varia-\n",
      "tion of extinction time with mutation rate μ for different clearance rates\n",
      "p in the extinction regime. Error bars represent one standard deviation. Near\n",
      "the threshold, the extinction events occur increasingly in the edge of viral\n",
      "load fluctuations (Fig. 3A). (B) The estimated threshold between the runaway\n",
      "and extinction regimes in the p − μ parameter space (defined as the maxi-\n",
      "mum μ for which the extinction time converges for a given p). The arrow\n",
      "illustrates the dynamic deterioration of immune response during HIV-1 infec-\n",
      "tion. Other parameter values were the same as in Fig. 2.\n",
      "\n",
      "Woo and Reifman PNAS ∣ August 7, 2012 ∣ vol. 109 ∣ no. 32 ∣ 12983\n",
      "\n",
      "BI\n",
      "O\n",
      "PH\n",
      "\n",
      "YS\n",
      "IC\n",
      "S\n",
      "A\n",
      "N\n",
      "D\n",
      "\n",
      "CO\n",
      "M\n",
      "PU\n",
      "\n",
      "TA\n",
      "TI\n",
      "O\n",
      "N\n",
      "A\n",
      "L\n",
      "BI\n",
      "O\n",
      "LO\n",
      "\n",
      "G\n",
      "Y\n",
      "\n",
      "D\n",
      "ow\n",
      "\n",
      "nl\n",
      "oa\n",
      "\n",
      "de\n",
      "d \n",
      "\n",
      "by\n",
      " g\n",
      "\n",
      "ue\n",
      "st\n",
      "\n",
      " o\n",
      "n \n",
      "\n",
      "A\n",
      "pr\n",
      "\n",
      "il \n",
      "23\n",
      "\n",
      ", 2\n",
      "02\n",
      "\n",
      "0 \n",
      "\n",
      "\n",
      "\n",
      "identical phenotypes in a large fraction of their epitopes and are\n",
      "constrained by common groups of CTLs. However, the model\n",
      "does not account for the effects of direct competitions for re-\n",
      "sources (limited number of host cells available for infection).\n",
      "We have additionally explored an extended model that includes\n",
      "host-cell dynamics (SI Text), whose analysis suggests that the main\n",
      "qualitative results remain valid (Fig. S8).\n",
      "\n",
      "Our finding that the relative importance of immune escape is\n",
      "mainly determined by the viral load in chronic infection may have\n",
      "implications to vaccine design (42): Strategies aimed at prevent-\n",
      "ing immune escape, such as targeting epitopes with higher fitness\n",
      "costs to mutations, may not confer more benefits than those at-\n",
      "tempting to boost the overall immune response. We also note,\n",
      "however, that more realistic representations of T-cell stimulation\n",
      "and clearance will have to take their dependence on phenotypes\n",
      "into account (p and b should depend on amino acid sequences),\n",
      "because mutations affect HLA-epitope-CTL binding affinities.\n",
      "Our model also ignores complex epistatic effects coupling muta-\n",
      "tions on the same and different epitopes (43). The approach pre-\n",
      "sented here likely can provide a foundation for a more compre-\n",
      "hensive modeling framework to tackle such global genome-wide\n",
      "effects. In addition, although we interpreted Eqs. 3B–3D strictly\n",
      "as the CTL-mediated immune response, we expect similar ap-\n",
      "proaches to be applicable for antibody-based responses.\n",
      "\n",
      "Methods\n",
      "Stochastic Evolutionary Dynamics. Simulations are carried out by applying the\n",
      "Gillespie algorithm (26) to Eq. 3. At a given time, a dynamic list of genotypes\n",
      "and phenotypes (N and M in total, respectively), starting with a single WT at\n",
      "the initial condition, is kept and updated, instead of enumerating all possible\n",
      "sequences. A genotype consists of the nucleotide sequence of a set of epi-\n",
      "topes (l in total), each with the same length L, such that the total length be-\n",
      "comes l × L. We define the following quantities\n",
      "\n",
      "aðrÞi ¼ ∑\n",
      "i\n",
      "\n",
      "j¼1\n",
      "rjnj; a\n",
      "\n",
      "ðsÞ\n",
      "i ¼ ∑\n",
      "\n",
      "i\n",
      "\n",
      "j¼1\n",
      "ðrj þ bÞnj;\n",
      "\n",
      "aðcÞi ¼ ∑\n",
      "i\n",
      "\n",
      "j¼1\n",
      "ðrj þ bþ pcjÞnj; aðdÞα ¼ aðcÞN þ g∑\n",
      "\n",
      "α\n",
      "\n",
      "β¼1\n",
      "cðβÞ;\n",
      "\n",
      "each corresponding to the replication, stimulation, clearance, and decay re-\n",
      "actions (Eq. 3), respectively, where cj ¼ ΣkcðαjkÞ is the total number of CTLs\n",
      "corresponding to any of the epitope phenotypes belonging to the genotype\n",
      "j. A random number 0 < r < 1 is drawn to determine the time interval Δt ¼\n",
      "ð1∕aÞ lnð1∕rÞ before the next reaction event, where a ¼ a ðdÞM . A second ran-\n",
      "dom number r is drawn, and if aðxÞi−1 < ar < a\n",
      "\n",
      "ðxÞ\n",
      "i for x ¼ r, s, c, the replication,\n",
      "\n",
      "stimulation, or clearance reaction is chosen for the genotype i, respectively. If\n",
      "a ðdÞα−1 < ar < a\n",
      "\n",
      "ðdÞ\n",
      "α instead, the decay reaction is carried out for the CTL with phe-\n",
      "\n",
      "notype α. In this algorithm, therefore, a given CTL contributes equally to the\n",
      "clearance probabilities of all viral strains containing its phenotype. This pre-\n",
      "scription generates unbiased trajectories of two classes of objects, the viruses\n",
      "with genotype i and phenotype αik and CTLs with phenotype α. For replica-\n",
      "tion reactions, the chosen genotype i is mutated with probability μ per bp.\n",
      "Further details on implementing the genotype-phenotype lists can be found\n",
      "in SI Text.\n",
      "\n",
      "Simulation Conditions. The standard genetic code was used to translate\n",
      "mutant genotypes into phenotypes. The reduced fitness sα ¼ rατ of a pheno-\n",
      "type αwas taken as follows: sα ¼ 1 for theWT, and for all other phenotypes sα\n",
      "is a random variable with the following Gaussian distribution: PðsαÞ ¼\n",
      "expf−½sα − fðdαÞ�2�∕2σ2g∕ð2πÞ1∕2σ, where the position of the maximum\n",
      "fðdαÞ ¼ expð−dα∕ξÞ is an exponentially decreasing function of the distance\n",
      "dα ≥ 1 (summed over all epitopes; in units of amino acids) of the phenotype\n",
      "α to the WT. The parameters ξ and σ are the measures of the fitness costs of\n",
      "mutations and of randomness in fitness distributions, respectively. During si-\n",
      "mulations, a newly encountered phenotype is dynamically assigned a fitness\n",
      "value based on this distributionwith negative rα values replaced by zero. Phe-\n",
      "notypes containing stop codons were assigned zero fitness. In Fig. 1, the SIV\n",
      "epitope Tat28–35SL8 with the WT sequence of 5′-TCC ACT CCA GAA TCG GCC\n",
      "AAC CTG-3′ (STPESANL) (38) was used (La ¼ 8 and l ¼ 1). In Figs. 2–4, random\n",
      "sequences were used for the WT and simulations were performed with La ¼\n",
      "10 and l ¼ 3. Typically, quantities were averaged over 102 ∼ 103 trajectories\n",
      "unless otherwise specified.\n",
      "\n",
      "ACKNOWLEDGMENTS. This work was sponsored by both the US Army’s\n",
      "Network Science Initiative, managed by the Military Operational Medicine\n",
      "Research Program of the US Army Medical Research and Materiel Command,\n",
      "Fort Detrick, Maryland, and by a competitive In-House Laboratory Indepen-\n",
      "dent Research award by the US Army Assistant Secretary of the Army for\n",
      "Acquisition, Logistics, and Technology. The opinions and assertions contained\n",
      "herein are the private views of the authors and are not to be construed as\n",
      "official or as reflecting the views of the US Army or of the US Department\n",
      "of Defense. This paper has been approved for public release with unlimited\n",
      "distribution.\n",
      "\n",
      "1. Eigen M (1971) Selforganization of matter and the evolution of biological macromo-\n",
      "lecules. Naturwissenschaften 58:465–523.\n",
      "\n",
      "2. Swetina J, Schuster P (1982) Self-replication with errors: A model for polynucleotide\n",
      "replication. Biophys Chem 16:329–345.\n",
      "\n",
      "3. Loeb LA, et al. (1999) Lethal mutagenesis of HIV with mutagenic nucleoside analogs.\n",
      "Proc Natl Acad Sci USA 96:1492–1497.\n",
      "\n",
      "4. Iranzo J, Perales C, Domingo E, Manrubia SC (2011) Tempo and mode of inhibitor-\n",
      "mutagen antiviral therapies: A multidisciplinary approach. Proc Natl Acad Sci USA\n",
      "108:16008–16013.\n",
      "\n",
      "5. Domingo E, Holland JJ (1997) RNA virus mutations and fitness for survival. Annu Rev\n",
      "Microbiol 51:151–178.\n",
      "\n",
      "6. Mardis ER (2008) Next-generation DNA sequencing methods. Annu Rev Genomics\n",
      "Hum Genet 9:387–402.\n",
      "\n",
      "7. McMichael AJ, Borrow P, Tomaras GD, Goodnetilleke N, Haynes BF (2010) The immune\n",
      "response during acute HIV-1 infection: Clues for vaccine development. Nat Rev Immu-\n",
      "nol 10:11–23.\n",
      "\n",
      "8. Mild M, Hedskog C, Jernberg J, Albert J (2011) Performance of ultradeep pyrosequen-\n",
      "cing in analysis of HIV-1 pol gene variation. PLoS One 6:e22741.\n",
      "\n",
      "9. Bimber BN, et al. (2009) Ultradeep pyrosequencing detects complex patterns of CD8þ\n",
      "\n",
      "T-lymphocyte escape in simian immunodeficiency virus-infected macaques. J Virol\n",
      "83:8247–8253.\n",
      "\n",
      "10. Bimber BN, et al. (2010) Whole-genome characterization of human and simian\n",
      "immunodeficiency virus intrahost diversity by ultradeep pyrosequencing. J Virol\n",
      "84:12087–12092.\n",
      "\n",
      "11. Fischer W, et al. (2010) Transmission of single HIV-1 genomes and dynamics of early\n",
      "immune escape revealed by ultradeep sequencing. PLoS One 5:e12303.\n",
      "\n",
      "12. Cale EM, et al. (2011) Epitope-specific CD8þ T lymphocytes cross-recognize mutant\n",
      "simian immunodefieciency virus (SIV) sequences but fail to contain very early evolu-\n",
      "tion and eventual fixation of epitope escape mutations during SIV infection. J Virol\n",
      "85:3746–3757.\n",
      "\n",
      "13. Moore CB, et al. (2002) Evidence of HIV-1 adaptation to HLA-restricted immune\n",
      "responses at a population level. Science 296:1439–1443.\n",
      "\n",
      "14. Bhattacharya T, et al. (2007) Founder effects in the assessment of HIV polymorphisms\n",
      "and HLA allele associations. Science 315:1583–1586.\n",
      "\n",
      "15. Kawashima Y, et al. (2009) Adaptation of HIV-1 to human leukocyte antigen class I.\n",
      "Nature 458:641–645.\n",
      "\n",
      "16. Addo MM, et al. (2003) Comprehensive epitope analysis of human immunodeficiency\n",
      "virus type 1 (HIV-1)-specific T-cell responses directed against the entire expressed HIV-1\n",
      "genome demonstrate broadly directed responses, but no correlation to viral load.\n",
      "J Virol 77:2081–2092.\n",
      "\n",
      "17. Goodnetilleke N, et al. (2009) The first T-cell response to transmitted/founder\n",
      "virus contributes to the control of acute viremia in HIV-1 infection. J Exp Med\n",
      "206:1253–1272.\n",
      "\n",
      "18. Nowak MA, Bangham CR (1996) Population dynamics of immune responses to persis-\n",
      "tent viruses. Science 272:74–79.\n",
      "\n",
      "19. Nowak MA, May RM (2001) Virus Dynamics: Mathematical Principles of Immunology\n",
      "and Virology (Oxford Univ Press, New York).\n",
      "\n",
      "20. Nowak MA, et al. (1995) Antigenic oscillations and shifting immunodominance in\n",
      "HIV-1 infections. Nature 375:606–611.\n",
      "\n",
      "21. Althaus CL, De Boer RJ (2008) Dynamics of immune escape during HIV/SIV infection.\n",
      "PLoS Comput Biol 4:e1000103.\n",
      "\n",
      "22. Love TM, Thurston SW, Keefer MC, Dewhurst S, Lee HY (2010) Mathematical modeling\n",
      "of ultradeep sequencing data reveals that acute CD8þ T-lymphocyte responses exert\n",
      "strong selective pressure in simian immunodeficiency virus-infected macaques but still\n",
      "fail to clear founder epitope sequences. J Virol 84:5802–5814.\n",
      "\n",
      "23. Lee HY, et al. (2009) Modeling sequence evolution in acute HIV-1 infection. J Theor Biol\n",
      "261:341–360.\n",
      "\n",
      "24. Lee HY, Perelson AS, Park S-C, Leitner T (2008) Dynamic correlation between intrahost\n",
      "HIV-1 quasispecies evolution and disease progression. PLoS Comput Biol 4:e1000240.\n",
      "\n",
      "25. Keele BF, et al. (2008) Identification and characterization of transmitted and\n",
      "early founder virus envelopes in primary HIV-1 infection. Proc Natl Acad Sci USA\n",
      "105:7552–7557.\n",
      "\n",
      "26. Gillespie DT (1977) Exact stochastic simulation of coupled chemical reactions. J Phys\n",
      "Chem 81:2340–2361.\n",
      "\n",
      "27. Nowak M, Schuster P (1989) Error thresholds of replication in finite populations muta-\n",
      "tion frequencies and the onset of Muller’s ratchet. J Theor Biol 137:375–395.\n",
      "\n",
      "28. Yang Z, Bielawski JP (2000) Statistical methods for detecting molecular adaptation.\n",
      "Trends Ecol Evol 15:496–503.\n",
      "\n",
      "12984 ∣ www.pnas.org/cgi/doi/10.1073/pnas.1117201109 Woo and Reifman\n",
      "\n",
      "D\n",
      "ow\n",
      "\n",
      "nl\n",
      "oa\n",
      "\n",
      "de\n",
      "d \n",
      "\n",
      "by\n",
      " g\n",
      "\n",
      "ue\n",
      "st\n",
      "\n",
      " o\n",
      "n \n",
      "\n",
      "A\n",
      "pr\n",
      "\n",
      "il \n",
      "23\n",
      "\n",
      ", 2\n",
      "02\n",
      "\n",
      "0 \n",
      "\n",
      "http://www.pnas.org/lookup/suppl/doi:10.1073/pnas.1117201109/-/DCSupplemental/pnas.1117201109_SI.pdf?targetid=STXT\n",
      "http://www.pnas.org/lookup/suppl/doi:10.1073/pnas.1117201109/-/DCSupplemental/pnas.1117201109_SI.pdf?targetid=SF8\n",
      "http://www.pnas.org/lookup/suppl/doi:10.1073/pnas.1117201109/-/DCSupplemental/pnas.1117201109_SI.pdf?targetid=STXT\n",
      "\n",
      "\n",
      "29. Aguirre J, Buldu JM, Stich M, Manrubia SC (2011) Topological structure of the space of\n",
      "phenotypes: The case of RNA neutral networks. PLoS One 6:e26324.\n",
      "\n",
      "30. Burch CL, Chao L (2000) Evolvability of an RNA virus is determined by its mutational\n",
      "neighborhood. Nature 406:625–628.\n",
      "\n",
      "31. Schuster P, Fontana W (1999) Chance and necessity in evolution: Lessons from RNA.\n",
      "Physica D 133:427–452.\n",
      "\n",
      "32. Hietpas RT, Jensen JD, Bolon DN (2011) Experimental illumination of a fitness land-\n",
      "scape. Proc Natl Acad Sci USA 108:7896–7901.\n",
      "\n",
      "33. Stich M, Briones C, Manrubia SC (2008) On the structural repertoire of pools of short,\n",
      "random RNA sequences. J Theor Biol 252:750–763.\n",
      "\n",
      "34. Pitt JN, Ferre-D’Amare AR (2010) Rapid construction of empirical RNA fitness land-\n",
      "scapes. Science 330:376–379.\n",
      "\n",
      "35. Duarte EA, et al. (1994) Subclonal components of consensus fitness in an RNA virus\n",
      "clone. J Virol 68:4295–4301.\n",
      "\n",
      "36. Sanjuan R, Moya A, Elena SF (2004) The distribution of fitness effects caused by single-\n",
      "nucleotide substitutions in an RNA virus. Proc Natl Acad Sci USA 101:8396–8401.\n",
      "\n",
      "37. Fernandez G, Clotet B, Martinez MA (2007) Fitness landscape of human immunode-\n",
      "ficiency virus type 1 protease quasispecies. J Virol 81:2485–2496.\n",
      "\n",
      "38. Allen TM, et al. (2000) Tat-specific cytotoxic T lymphocytes select for SIV escape\n",
      "variants during resolution of primary viraemia. Nature 407:386–390.\n",
      "\n",
      "39. Mansky LM, Temin HM (1995) Lower in vivo mutation rate of human immunodefi-\n",
      "ciency virus type 1 than that predicted from the fidelity of purified reverse transcrip-\n",
      "tase. J Virol 69:5087–5094.\n",
      "\n",
      "40. Schuster P, Swetina J (1988) Stationary mutant distributions and evolutionary optimi-\n",
      "zation. Bull Math Biol 50:635–660.\n",
      "\n",
      "41. Wilke CO,Wang JL, Ofria C, Lenski RE, Adami C (2001) Evolution of digital organisms at\n",
      "high mutation rates leads to survival of the flattest. Nature 412:331–333.\n",
      "\n",
      "42. Gaschen B, et al. (2002) Diversity considerations in HIV-1 vaccine selection. Science\n",
      "296:2354–2360.\n",
      "\n",
      "43. Dahirel V, et al. (2011) Coordinate linkage of HIV evolution reveals regions of immu-\n",
      "nological vulnerability. Proc Natl Acad Sci USA 108:11530–11535.\n",
      "\n",
      "Woo and Reifman PNAS ∣ August 7, 2012 ∣ vol. 109 ∣ no. 32 ∣ 12985\n",
      "\n",
      "BI\n",
      "O\n",
      "PH\n",
      "\n",
      "YS\n",
      "IC\n",
      "S\n",
      "A\n",
      "N\n",
      "D\n",
      "\n",
      "CO\n",
      "M\n",
      "PU\n",
      "\n",
      "TA\n",
      "TI\n",
      "O\n",
      "N\n",
      "A\n",
      "L\n",
      "BI\n",
      "O\n",
      "LO\n",
      "\n",
      "G\n",
      "Y\n",
      "\n",
      "D\n",
      "ow\n",
      "\n",
      "nl\n",
      "oa\n",
      "\n",
      "de\n",
      "d \n",
      "\n",
      "by\n",
      " g\n",
      "\n",
      "ue\n",
      "st\n",
      "\n",
      " o\n",
      "n \n",
      "\n",
      "A\n",
      "pr\n",
      "\n",
      "il \n",
      "23\n",
      "\n",
      ", 2\n",
      "02\n",
      "\n",
      "0 \n",
      "\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# one of the keys provides the content\n",
    "print(parsedPDF['content'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'Content-Type': 'application/pdf',\n",
       " 'Creation-Date': '2012-07-28T00:39:16Z',\n",
       " 'Last-Modified': '2020-04-23T07:03:44Z',\n",
       " 'Last-Save-Date': '2020-04-23T07:03:44Z',\n",
       " 'X-Parsed-By': ['org.apache.tika.parser.DefaultParser',\n",
       "  'org.apache.tika.parser.pdf.PDFParser'],\n",
       " 'X-TIKA:parse_time_millis': '70',\n",
       " 'access_permission:assemble_document': 'true',\n",
       " 'access_permission:can_modify': 'true',\n",
       " 'access_permission:can_print': 'true',\n",
       " 'access_permission:can_print_degraded': 'true',\n",
       " 'access_permission:extract_content': 'true',\n",
       " 'access_permission:extract_for_accessibility': 'true',\n",
       " 'access_permission:fill_in_form': 'true',\n",
       " 'access_permission:modify_annotations': 'true',\n",
       " 'created': '2012-07-28T00:39:16Z',\n",
       " 'date': '2020-04-23T07:03:44Z',\n",
       " 'dc:format': 'application/pdf; version=1.4',\n",
       " 'dc:title': '201117201 12980..12985',\n",
       " 'dcterms:created': '2012-07-28T00:39:16Z',\n",
       " 'dcterms:modified': '2020-04-23T07:03:44Z',\n",
       " 'meta:creation-date': '2012-07-28T00:39:16Z',\n",
       " 'meta:save-date': '2020-04-23T07:03:44Z',\n",
       " 'modified': '2020-04-23T07:03:44Z',\n",
       " 'pdf:PDFVersion': '1.4',\n",
       " 'pdf:docinfo:created': '2012-07-28T00:39:16Z',\n",
       " 'pdf:docinfo:creator_tool': 'Arbortext Advanced Print Publisher 9.1.405/W Unicode',\n",
       " 'pdf:docinfo:modified': '2020-04-23T07:03:44Z',\n",
       " 'pdf:docinfo:producer': 'Acrobat Distiller 6.0.1 (Windows)',\n",
       " 'pdf:docinfo:title': '201117201 12980..12985',\n",
       " 'pdf:encrypted': 'false',\n",
       " 'producer': 'Acrobat Distiller 6.0.1 (Windows)',\n",
       " 'resourceName': '12980.full.pdf',\n",
       " 'title': '201117201 12980..12985',\n",
       " 'xmp:CreatorTool': 'Arbortext Advanced Print Publisher 9.1.405/W Unicode',\n",
       " 'xmpMM:DocumentID': 'uuid:cb69f4d0-b7da-45a8-a0bc-3d44c8259269',\n",
       " 'xmpTPg:NPages': '6'}"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# one of the keys provides the metadata\n",
    "parsedPDF['metadata']"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Analyze PDF text with generic tools"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [],
   "source": [
    "# store content in variable\n",
    "cont = parsedPDF['content']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n201117201 12980..12985\\n\\n\\nA quantitative quasispecies theory-based model of\\nvirus escape mutation under immune selection\\nHyung-June Woo and Jaques R',\n",
       " 'fman1\\n\\nBiotechnology High Performance Computing Software Applications Institute, Telemedicine and Advanced Technology Research Center, US Army Medical\\nResearch and Materiel Command, Fort Detrick, MD 21702\\n\\nEdited by Peter Schuster, University of Vienna, Vienna, and approved June 28, 2012 (rec',\n",
       " 'ved for review October 18, 2011)\\n\\nViral infections involve a complex interplay of the immune\\nresponse and escape mutation of the virus quasispecies inside a\\nsingle host. Although fundamental aspects of such a balance of\\nmutation and selection pressure have been established by the qua-\\nsispecies theory decades ago, its implications have largely re-\\nmained qualitative. Here, we present a quantitative approach to\\nmodel the virus evolution under cytotoxic T-lymphocyte immune\\nresponse. The virus quasispecies dynamics are explicitly repre-\\nsented by mutations in the combined sequence space of a set of\\nepitopes within the viral genome. We stochastically simulated the\\ngrowth of a viral population originating from a single wild-type\\nfounder virus and its recognition and clearance by the immune re-\\nsponse, as well as the expansion of its genetic diversity. Applied to\\nthe immune escape of a simian immunodeficiency virus epitope,\\nmodel predictions were quantitatively comparable to the experi-\\nmental data. Within the model parameter space, we found two\\nqualitatively different regimes of infectious disease pathogenesis,\\neach representing alternative fates of the immune response: It can\\nclear the infection in finite time or eventually be overwhelmed by\\nviral growth and escape mutation. The latter regime exhibits the\\ncharacteristic disease progression pattern of human immunodefi-\\nciency virus, while the former is bounded by maximum mutation\\nrates that can be suppressed by the immune response. Our results\\ndemonstrate that, by explicitly representing epitope mutations\\nand thus providing a genotype–phenotype map, the quasispecies\\ntheory can form the basis of a detailed sequence-specific model of\\nreal-world viral pathogens evolving under immune selection.\\n\\npopulation dynamics ∣ next-generation sequencing ∣ stochastic simulation ∣\\nHIV\\n\\nViruses with RNA genomes, such as human immunodeficiencyvirus type 1 (HIV-1), have high rates of mutation and evolve\\nrapidly in response to host immune selection pressure. One of the\\nconsequences of such rapid mutations is the error catastrophe\\n(1), where a virus population is driven to extinction when its mu-\\ntation rate exceeds a threshold. The existence of such a threshold\\nis a central prediction of the quasispecies theory pioneered by\\nEigen (1) and Swetina and Schuster (2). The recent experimental\\ndemonstration of lethal mutagenesis (3–5), in which an error cat-\\nastrophe transition is caused by mutagens, demonstrates that key\\nfeatures of the balance of mutation and selection in viruses are\\nelegantly captured by the quasispecies theory.\\n\\nAlthough insights provided by the quasispecies theory have\\ngreatly expanded our understanding of virus behavior, applica-\\ntions so far have been limited to a conceptual level, partially due\\nto the lack of experimental information on the evolutionary dy-\\nnamics in sequence space. Next-generation sequencing techni-\\nques (6) have the potential to change this situation. To build\\nquantitative models implementing the quasispecies dynamics\\nand describe such experimental data, it is important to realisti-\\ncally specify the nature of selection pressure. Viruses in animal\\nhosts evolve under immune pressure, and th',\n",
       " 'r capacity for rapid\\nescape mutation underlies many of the difficulties in combating\\npathogens, including HIV-1. In a typical disease pathogenesis of\\nHIV-1, the acute viremia after an initial infection is curbed by\\n\\nCD8þ cytotoxic T-lymphocyte (CTL) responses as well as subse-\\nquent antibody actions, leading to an asymptomatic chronic infec-\\ntion stage that can last up to 10 y (7). However, this apparent\\ncontrol of viremia is never complete, and without antiviral ther-\\napy, the chronic infection eventually leads to the onset of disease.\\nThis chronic infection stage involves continuous escape mutation-\\nCTL response cycles, whose detailed characteristics are b',\n",
       " 'ng un-\\ncovered by ultradeep sequencing (8–12). CTLs recognize specific\\nviral epitopes (approximately 10 amino acids long) presented on\\nthe surface of infected cells by class I human leukocyte antigen\\n(HLA). The epitope recognition depends sensitively on HLA al-\\nleles, leading to differential patterns of immune response among\\npatients (13–15), while characteristics of immune response during\\nearly infection often shape and influence the overall disease pro-\\ngression (16, 17). Quantitative sequence-based models of virus-\\nCTL dynamics will greatly facilitate the interpretation of experi-\\nmental data.\\n\\nIn a series of pioneering works (18–20), Nowak and coworkers\\nintroduced population dynamics concepts that capture a diverse\\nrange of immune response and escape mutation. Similar ap-\\nproaches focused more on escape dynamics have also been pro-\\nposed (21, 22). These models, however, do not describe mutations\\nin sequence space explicitly. Other mathematical models of\\nsequence evolution focus on sequence divergence during the very\\nearly stages of infection, while ignoring or only implicitly including\\nthe effect of selection pressure (23, 24). In this paper, we describe\\na quantitative quasispecies-based model of virus dynamics under\\nT cell-based immune pressure, explicitly mapping genotype–phe-\\nnotype relationships. Our approach combines the description of\\nvirus evolutionary dynamics provided by the quasispecies theory\\nand the population dynamics models of Nowak and coworkers.\\nWe show that the model not only captures the salient features\\nobserved in sequencing data of simian immunodeficiency virus\\n(SIV) (9) but also reveals general qualitative features of viral\\ninfection disease pathogenesis: The immune system can clear\\nthe infection within time scales ranging from days to those much\\nlonger than patient lifetimes, or be overwhelmed by immune es-\\ncape. The viral load progression in the latter regime closely\\nmatches the observed HIV-1 disease pathogenesis pattern.\\n\\nResults and Discussion\\nStochastic Quasispecies Dynamics.During acute infection, the foun-\\nder virus (often a single virion) (7, 11, 17, 25) undergoes replica-\\ntions to produce an exponentially growing population, which may\\nbe described by the quasispecies dynamics without degradation,\\n\\nAuthor contributions: H.-J.W. designed research; H.-J.W. performed research; H.-J.W. and\\nJ.R. analyzed data; and H.-J.W. and J.R. wrote the paper.\\n\\nThe authors declare no conflict of interest.\\n\\nThis article is a PNAS Direct Submission.\\n\\nFreely available online through the PNAS open access option.\\n1To whom correspondence should be addressed. E-mail: jaques.r',\n",
       " 'fman@us.army.mil.\\n\\nThis article contains supporting information online at www.pnas.org/lookup/suppl/\\ndoi:10.1073/pnas.1117201109/-/DCSupplemental.\\n\\n12980–12985 ∣ PNAS ∣ August 7, 2012 ∣ vol. 109 ∣ no. 32 www.pnas.org/cgi/doi/10.1073/pnas.1117201109\\n\\nD\\now\\n\\nnl\\noa\\n\\nde\\nd \\n\\nby\\n g\\n\\nue\\nst\\n\\n o\\nn \\n\\nA\\npr\\n\\nil \\n23\\n\\n, 2\\n02\\n\\n0 \\n\\nhttp://www.pnas.org/lookup/suppl/doi:10.1073/pnas.1117201109/-/DCSupplemental\\nhttp://www.pnas.org/lookup/suppl/doi:10.1073/pnas.1117201109/-/DCSupplemental\\nhttp://www.pnas.org/lookup/suppl/doi:10.1073/pnas.1117201109/-/DCSupplemental\\nhttp://www.pnas.org/lookup/suppl/doi:10.1073/pnas.1117201109/-/DCSupplemental\\nhttp://www.pnas.org/lookup/suppl/doi:10.1073/pnas.1117201109/-/DCSupplemental\\nhttp://www.pnas.org/lookup/suppl/doi:10.1073/pnas.1117201109/-/DCSupplemental\\n\\n\\n_nj ¼ ∑\\ni\\n\\nQjirini; [1]\\n\\nwhere ni is the number of virions with genotype i, ri is the repli-\\ncation rate of i, andQji ¼ ð1 − μÞL−dðj;iÞðμ∕3Þdðj;iÞ is the mutation\\nprobability from i to genotype j, with mean mutation rate μ,\\ngenome length L (in bp), and Hamming distance (the number\\nof nucleotides that are different) dðj; iÞ between j and i. A more\\ncomplete description for early stages of infection, where ni is of\\nthe order of 1 and restricted to integer values, is given by the che-\\nmical reaction representation\\n\\nRi →\\nQjiri\\n\\nRi þRj; [2]\\n\\nwhere Ri is a virion with genotype i. Eq. 2 can be simulated\\nstochastically with the Gillespie algorithm (26). A major advan-\\ntage of such a stochastic formulation is that it allows us to avoid\\nexhaustive enumerations of all possible genotypes. This is signif-\\nicant even for a single epitope of viral prot',\n",
       " 'ns: An amino acid\\nsequence with La ¼ 10, whose genome length is L ¼ 3La ¼\\n30 bp, still has 4L ∼ 1018 genotypes. Nowak and Schuster (27)\\nperformed Gillespie algorithm simulations of the standard qua-\\nsispecies model under the single-peak fitness landscape (ri ¼ 1∕τ\\nfor the wild-type (WT) sequence and ri ¼ 1∕Aτ for all other mu-\\ntants, whereA > 1 is the relative fitness of theWTand τ is a char-\\nacteristic time scale). In this case, distributions of individuals\\namong only two groups (WT and mutants) need to be tracked.\\n\\nIn our numerical simulations, the initial genotype (referred to\\nhere as the WT) replicates with mutation rate μ, and the gener-\\nated mutants are compared with the existing list of sequences,\\nwhich is updated dynamically when new sequences arise. We ver-\\nified that this numerical scheme sampled the relevant sequence\\nspace sufficiently (SI Text, Figs. S1–S3). The initial condition we\\nused (single WT) and the discreteness of ni imply that ni should\\nbe interpreted as the total number of virions within a finite sys-\\ntem. The HIV-1 viral load during the acute infection phase can\\nreach up to 104 ∼ 106 RNA copies/mL (7). Accounting for the\\nvolume of blood in the body of an average adult (approximately\\n5 L), the total population size Nv ¼ Σi ni would be up to about\\n109. We found simulations to slow down significantly as Nv be-\\ncame larger than about 106. For computational efficiency, we\\ntherefore regarded the system as a small representative volume\\n(1 mL) of blood, and the viral loads and CTL levels reported refer\\nto RNA copy numbers and cell counts within this volume.\\n\\nCTL Response. Within an infected host, the action of the cellular\\nimmune response provides the major force countering the acute\\nviremia. In this work, we combine the population dynamics ap-\\nproach of virus-immune system dynamics by Nowak and cowor-\\nkers (18, 19) with the classic quasispecies theory. Accordingly,\\nEq. 2 is modified as follows:\\n\\nRi →\\nQjiri\\n\\nRi þRj; ðreplication with mutationÞ [3A]\\n\\nRi→\\nb\\nRi þ∑\\n\\nl\\n\\nk¼1\\nCðαikÞ; ðstimulationÞ [3B]\\n\\nRi þ CðαikÞ→\\np\\nCðαikÞ; ðclearanceÞ [3C]\\n\\nCðαÞ→\\ng\\n0; ðdecayÞ [3D]\\n\\nwhere the genotype i is now a member of the combined sequence\\nspace of l epitopes. The amino acid sequence (“phenotype”) of\\nepitope k in genotype i is denoted as αik, and CðαÞ represents a\\n\\nCTL specific to phenotype α. Eq. 3B represents a reaction in\\nwhich, with a rate b, virions of sequence i stimulate the produc-\\ntion of a set of CTLs CðαikÞ corresponding to the phenotypes of\\nits epitopes. Eq. 3C denotes the reaction where cells infected with\\nviruses with genotype i are cleared by CTLs with phenotypes that\\nmatch one of its epitopes with a rate p, and Eq. 3D represents the\\nnatural decay of T cells with a rate g. Under the alternative de-\\nterministic continuum approximation, one may write:\\n\\n_nj ¼ ∑\\ni\\n\\nQjirini − p∑\\nl\\n\\nk¼1\\ncðαjkÞnj; [4A]\\n\\n_cðαÞ ¼ bnα − gcðαÞ; [4B]\\n\\nwhere cðαÞ is the number of Tcells CðαÞ and nα is the total num-\\nber of virions containing epitopes with phenotype α.\\n\\nThe qualitative features of the infection-clearance dynamics\\ncan be illustrated by setting the mutation rate μ to zero. Numer-\\nical integrations of Eq. 4 yield the continuum approximation re-\\nsult (Fig. S4 for μ ¼ 0 and l ¼ 1), which showed reasonable\\nagreement with the stochastic simulation results. For nonvanish-\\ning mutation rates, numerical integration rapidly becomes prohi-\\nbitive with increasing L because of the high dimensionality of the\\nsequence space. The inverse of b is a measure of the time delay of\\nthe growth of T cells compared to that of viruses. Here, we\\nadopted b values of approximately 0.01 d−1. The efficiency of\\nCTLs in recognizing cells presenting the epitope and killing them\\nis represented by parameter p, the clearance rate. We found\\np values of 10−4 ∼ 10−3 d−1 to give good fits to experiments\\n(see below).\\n\\nImmune Escape Dynamics. To make comparisons with experiments,\\nit is necessary to consider that fitness is determined by amino acid\\nsequences, and many mutations are synonymous. Therefore, the\\nfull immune escape dynamics represented by Eq. 3 distinguishes\\nthe amino acid sequence αik corresponding to the nucleotide\\nsequence of epitope k in genotype i. We refer to the set of nu-\\ncleotide and amino acid sequences as “genotypes” and “pheno-\\ntypes,” respectively. The fitness ri is a function of phenotypes\\nonly. During simulations, each genotype is translated into a phe-\\nnotype, and a new mutant is checked against the existing pheno-\\ntype/genotype list, which is updated and expanded. The nature of\\nthis genotype–phenotype map plays important roles in evolution-\\nary dynamics, leading to key signatures of selective pressure, such\\nas the ratio of synonymous to nonsynonymous mutations (28).\\nQuantitative insights to the effects of this mapping on molecular\\nevolution have been provided by Manrubia and coworkers (29),\\nwho studied short RNA sequences for which fitness can be esti-\\nmated by secondary structure predictions. In our case, a simpli-\\nfied overview of the genotype–phenotype map can be gleaned\\nfrom a network representation of the genetic code (Fig. S5): If\\nall mutations were neutral and amino acids within an epitope in-\\ndependent, this map would be sufficient to determine the statis-\\ntics of evolutionary drifts. The accessibility of certain mutations\\nfrom a phenotype, in particular, has been shown to affect the\\nevolvability of viruses (30).\\n\\nAnother major ingredient for a realistic model of immune es-\\ncape is the fitness landscape beyond the level of the single-peak\\nmodel. Much progress has been made recently in understanding\\nthe nature of fitness landscapes (31–34). Explicit fitness measure-\\nments of viral clones (35, 36) and biochemical assays of prot',\n",
       " 'ns\\n(37) both indicate that single-nucleotide substitutions lead to a\\nbroad distribution of fitness changes, most of which are deleter-\\nious. Therefore, one may assume that the fitness of a mutant is a\\nrandom variable centered around a mean fitness value f . We ex-\\npect this mean fitness to be a decreasing function of distance d to\\nthe WT (the number of amino acids that are different), which we\\n\\nWoo and R',\n",
       " 'fman PNAS ∣ August 7, 2012 ∣ vol. 109 ∣ no. 32 ∣ 12981\\n\\nBI\\nO\\nPH\\n\\nYS\\nIC\\nS\\nA\\nN\\nD\\n\\nCO\\nM\\nPU\\n\\nTA\\nTI\\nO\\nN\\nA\\nL\\nBI\\nO\\nLO\\n\\nG\\nY\\n\\nD\\now\\n\\nnl\\noa\\n\\nde\\nd \\n\\nby\\n g\\n\\nue\\nst\\n\\n o\\nn \\n\\nA\\npr\\n\\nil \\n23\\n\\n, 2\\n02\\n\\n0 \\n\\nhttp://www.pnas.org/lookup/suppl/doi:10.1073/pnas.1117201109/-/DCSupplemental/pnas.1117201109_SI.pdf?targetid=STXT\\nhttp://www.pnas.org/lookup/suppl/doi:10.1073/pnas.1117201109/-/DCSupplemental/pnas.1117201109_SI.pdf?targetid=SF1\\nhttp://www.pnas.org/lookup/suppl/doi:10.1073/pnas.1117201109/-/DCSupplemental/pnas.1117201109_SI.pdf?targetid=SF3\\nhttp://www.pnas.org/lookup/suppl/doi:10.1073/pnas.1117201109/-/DCSupplemental/pnas.1117201109_SI.pdf?targetid=SF4\\nhttp://www.pnas.org/lookup/suppl/doi:10.1073/pnas.1117201109/-/DCSupplemental/pnas.1117201109_SI.pdf?targetid=SF4\\nhttp://www.pnas.org/lookup/suppl/doi:10.1073/pnas.1117201109/-/DCSupplemental/pnas.1117201109_SI.pdf?targetid=SF5\\nhttp://www.pnas.org/lookup/suppl/doi:10.1073/pnas.1117201109/-/DCSupplemental/pnas.1117201109_SI.pdf?targetid=SF5\\n\\n\\nassume to be exponential: f ðdÞ ¼ expð−d∕ξÞ, where ξ denotes a\\ncharacteristic distance (Methods).\\n\\nAn empirical evidence for this choice can be found from a re-\\ncent experimental study by Fernández et al. (37), who measured\\nthe fitness landscapes of HIV-1 protease quasispecies for three\\npatients. We plotted the distribution of th',\n",
       " 'r reported fitness\\nvalues as a function of distance to the dominant phenotype\\nand found near-exponential dependence for two quasispecies\\n(Fig. S6). Our fitness function therefore models both the de-\\ncrease of the mean fitness away from WT and the distribution\\nof neutral, deleterious, and beneficial mutants for a given dis-\\ntance. In simulations, these fitness values were assigned dynami-\\ncally to newly encountered phenotypes. It is important to note,\\nhowever, that this landscape chosen is still an approximation that\\nignores many potentially important effects, such as the heteroge-\\nn',\n",
       " 'ty of the mutational n',\n",
       " 'ghborhoods within the phenotype\\nspace.\\n\\nWe tested this quasispecies model with the ultradeep sequen-\\ncing data (9, 22) of the SIVepitope Tat28–35SL8 (38). Fig. 1 shows\\nthe time dependence of the viral load and WT frequency from\\nsimulations of the single-epitope version of Eq. 3 compared with\\nthe experimental data (22). The initial rapid growth of the viral\\nload, its subsequent decrease as CTLs are activated, and the more\\ngradual increase as mutants appear are all captured quantita-\\ntively. The WT frequency decrease agrees with experimental\\ntrends (Fig. 1B) with signatures of two distinct time scales (22).\\nThe mutation rate we used (μ ¼ 2 × 10−7 bp−1) was chosen to\\nobtain the best overall agreement for WT frequency with the sin-\\ngle-epitope version of the model. For multiepitope applications,\\nlarger mutation rate values close to the experimental estimate\\nfor HIV-1 (μ ¼ 3.4 × 10−5 bp−1) (39) gave realistic dynamics\\n(Fig. 2).\\n\\nFig. 1B also shows the frequency of the most dominant escape\\nmutant (EM) at each time point. As mutations occur, a new strain\\nproliferates temporarily before b',\n",
       " 'ng curbed down, and the total\\nnumber of phenotypes continues to increase. The increase in via-\\nbility of populations with multiple but lower fitness peaks has\\nbeen described first by Schuster and Swetina (40) and has later\\nbeen known as the “survival of the flattest” effect (41). Further\\ninsights into the role this expansion in sequence space plays in the\\ngrowth of viral loads can be gained from the deterministic con-\\ntinuum approximation, Eq. 4, without mutation: The immune re-\\nsponse to WT leads to damped oscillations of viral load and CTL\\n\\nlevel (Fig. S4), reaching a steady state n� and c�. This process is\\nrepeated for each new mutant, leading to continued increases\\nin n� and c� roughly proportional to the number of phenotypes\\n(18). However, the timing, distribution, and the probability of\\nthe appearance of mutants are highly nontrivial functions of\\nthe characteristics of the system, including genome length, gen-\\notype–phenotype map, and fitness landscape. Our quantitative\\nquasispecies-based model, Eq. 3, provides a realistic description\\nof this complex diversification process via the single parameter μ.\\nWe note that, for a single epitope, this interpretation ignores the\\neffects of competition among viral strains because each CTL\\nis specific to only one phenotype. In reality, in addition to com-\\npeting directly for host cells, different viral strains share th',\n",
       " 'r vul-\\nnerability to CTLs specific to common epitopes within th',\n",
       " 'r\\ngenomes. By considering multiple epitopes within a strain, our\\nmodel takes into account this interdependence of viral strains\\nand th',\n",
       " 'r indirect competition arising from shared epitope phe-\\nnotypes (see below).\\n\\nAs a quantitative measure of the relative degree of WT\\npersistence during immune escape, we also examined the time\\n(half-life) required for WT frequency to reach one-half (Fig. 1B)\\n(22) with varying stimulation rate b and clearance rate p (Fig. S7).\\nThe SIV epitopes studied by Bimber et al. (9) were estimated to\\nhave a WT half-life of about 20 d (22). We found that, within our\\nmodel, the WT half-life increases with increasing p and b in this\\nrange: An increase in the magnitude of selection pressure, rather\\nthan accelerating immune escape, suppresses viral population\\ngrowth more effectively and reduces the chances of escape muta-\\ntion (Fig. 2). The immune escape thus becomes more pronounced\\nwhen viral loads in the chronic phase are larger with more fre-\\nquent mutations.\\n\\nDisease Progression Under Immune Response.We examined general\\ntrends of viral infection-immune response dynamics within our\\nquasispecies model on multiple-epitope levels (l ¼ 3) with varia-\\ntions of key model parameters. The evolutionary dynamics, in\\nparticular, is critically affected by p, the clearance rate represent-\\ning the effectiveness of immune response, and the mutation\\nrate μ. We identified two qualitatively different behaviors based\\non the eventual fate of viral load/CTL level: in one regime\\n\\n0 50 100 150\\n10\\n\\n0\\n\\n10\\n2\\n\\n10\\n4\\n\\n10\\n6\\n\\n10\\n8\\n\\nV\\nira\\n\\nl l\\noa\\n\\nd\\n\\n0 50 100 150\\nTime (d)\\n\\n10\\n-2\\n\\n10\\n-1\\n\\n10\\n0\\n\\nF\\nre\\n\\nqu\\nen\\n\\ncy\\n\\nWT\\nDominant EM\\n\\nA\\n\\nB\\n\\nFig. 1. Simulation of full immune escape dynamics of a single SIV epitope\\nTat28–35SL8. (A) Viral load. (B) Frequencies of WT and the dominant escape\\nmutants (EM). Symbols represent experimental data from refs. 9 and 22, with\\neach symbol corresponding to one of four different animals. Solid lines are\\nfrom our stochastic quasispecies model with τ ¼ 0.8 d, μ ¼ 2.0 × 10−7 bp−1,\\nb ¼ 0.01 d−1, g ¼ 0.2 d−1, and p ¼ 2.0 × 10−4 d−1. The fitness function para-\\nmeters were σ ¼ 0.1 and ξ ¼ 1 (Methods). The EM frequency is for a single\\ntrajectory while the rest are averages.\\n\\n10\\n1\\n\\n10\\n2\\n\\n10\\n3\\n\\n10\\n4\\n\\nV\\nira\\n\\nl l\\noa\\n\\nd\\n\\n10\\n0\\n\\n10\\n1\\n\\n10\\n2\\n\\n10\\n3\\n\\nTime (d)\\n\\n10\\n0\\n\\n10\\n1\\n\\nN\\no.\\n\\n o\\nf p\\n\\nhe\\nno\\n\\nty\\npe\\n\\ns\\n\\n10\\n2\\n\\n10\\n3\\n\\n10\\n4\\n\\nC\\nT\\n\\nL\\n\\np = 1.0x10\\n-3\\n\\n, µ = 1.0x10-5\\n\\np = 5.0x10\\n-4\\n\\n, µ = 1.0x10-5\\n\\np = 1.0x10\\n-3\\n\\n, µ = 1.0x10-6\\n\\n0\\n\\n4000\\n\\n8000\\n\\nAcute\\n\\nChronic\\n\\n100\\n\\nA\\n\\nB\\n\\nC\\n\\nDisease\\n\\n101 102 103\\n\\nFig. 2. Typical disease progression patterns in the runaway regime from the\\nmultiepitope quasispecies model. (A) Viral load. (B) Total CTL levels. (C) Num-\\nber of distinct phenotypes per epitope present in the population. The inset in\\nA shows the viral load in linear scale for p ¼ 1.0 × 10−3 d−1 and μ ¼ 1.0 ×\\n10−5 bp−1. Other parameter values were b ¼ 0.02 d−1, g ¼ 0.1 d−1, τ ¼ 1 d,\\nσ ¼ 0.2, and ξ ¼ 1. The units of p and μ in the legends are d−1 and bp−1,\\nrespectively. All data represent averages over trajectories.\\n\\n12982 ∣ www.pnas.org/cgi/doi/10.1073/pnas.1117201109 Woo and R',\n",
       " 'fman\\n\\nD\\now\\n\\nnl\\noa\\n\\nde\\nd \\n\\nby\\n g\\n\\nue\\nst\\n\\n o\\nn \\n\\nA\\npr\\n\\nil \\n23\\n\\n, 2\\n02\\n\\n0 \\n\\nhttp://www.pnas.org/lookup/suppl/doi:10.1073/pnas.1117201109/-/DCSupplemental/pnas.1117201109_SI.pdf?targetid=SF6\\nhttp://www.pnas.org/lookup/suppl/doi:10.1073/pnas.1117201109/-/DCSupplemental/pnas.1117201109_SI.pdf?targetid=SF6\\nhttp://www.pnas.org/lookup/suppl/doi:10.1073/pnas.1117201109/-/DCSupplemental/pnas.1117201109_SI.pdf?targetid=SF4\\nhttp://www.pnas.org/lookup/suppl/doi:10.1073/pnas.1117201109/-/DCSupplemental/pnas.1117201109_SI.pdf?targetid=SF4\\nhttp://www.pnas.org/lookup/suppl/doi:10.1073/pnas.1117201109/-/DCSupplemental/pnas.1117201109_SI.pdf?targetid=SF7\\nhttp://www.pnas.org/lookup/suppl/doi:10.1073/pnas.1117201109/-/DCSupplemental/pnas.1117201109_SI.pdf?targetid=SF7\\n\\n\\n(“runaway,” Fig. 2), the chronic phase after the resolution of the\\nacute infection is eventually followed by a runaway growth in viral\\nloads (Fig. 2A). The level of CTLs also rises consistently during\\nthe chronic phase (Fig. 2B). This disease progression is accom-\\npanied by an increase in the diversity of quasispecies, or the\\n“spreading of clouds” (Fig. 2C). The time scale for the duration\\nof the chronic phase, in particular, varied sensitively with p and μ\\n(from a few months to years) (Fig. 2A). We found that the total\\nnumber of phenotypes per epitope (Fig. 2C) generally goes down\\nas the number of epitopes increases: A new escape mutant has\\nmuch lower probability for survival with multiple epitopes be-\\ncause it shares WT phenotypes on other parts of its genome with\\nexisting strains, and therefore is susceptible to CTLs that are al-\\nready present.\\n\\nThe disease progression pattern in the runaway regime corre-\\nsponds to situations where the immune system is overwhelmed by\\nviral growth, ',\n",
       " 'ther shortly after infection (there is no discernible\\nchronic phase for sufficiently small p) or after a long chronic\\nphase and accumulation of escape mutants. The latter case clo-\\nsely matches the characteristic HIV-1 disease pathogenesis (7),\\nrevealing the major role played by the uncontrolled diversifica-\\ntion of the quasispecies that overwhelms the CTL response. This\\nfeature was first suggested by Nowak and May (19), who coined\\nthe term “diversity threshold” based on a model assuming ran-\\ndom appearances of mutants. Our results show that the basic qua-\\nsispecies dynamics under selection pressure provide a realistic\\ndescription of this phenomenon.\\n\\nIf p is sufficiently large and μ sufficiently small, one enters a\\ndifferent regime (“extinction,” Fig. 3), where the acute and\\nchronic phases lead to viral loads that decrease ',\n",
       " 'ther rapidly\\nor asymptotically with time. As populations shrink in size, th',\n",
       " 'r\\ndynamics become increasingly stochastic, and extinction (a com-\\nplete clearance of infection) occurs when the viral load reaches\\nzero. This behavior is inherently stochastic and cannot be cap-\\ntured by the deterministic approximation in Eq. 4. The time re-\\nquired for clearing the infection (“extinction time”) typically has\\na broad distribution, increasing with decreasing p while becoming\\ninfinite above a threshold μ (Fig. 4A). The rapid clearance of in-\\nfection within approximately 10 d corresponds to the normal state\\nof affairs in a healthy immune system against viruses such as in-\\nfluenza. As shown in Fig. 3A (p ¼ 0.01 d−1), however, the extinc-\\ntion time can also reach time scales approaching a patient’s\\n\\nlifetime. This feature may be relevant in understanding the basis\\nfor some chronic viral infections with long and stable setpoints\\n(e.g., hepatitis C), although the high variability of extinction times\\nwe observed (Fig. 4A) likely may not correspond to actual pos-\\nsible clearances of such chronic infections.\\n\\nThe threshold separating the runaway (Fig. 2) and extinction\\n(Fig. 3) regimes can be identified by the maximum mutation\\nrate for which the extinction time remains finite (Fig. 4A). This\\nerror threshold for the immune control of infection is analo-\\ngous to Eigen’s threshold for genomic stability: The latter is\\nthe error rate above which replicators cannot maintain a stable\\nmaster sequence. The former is a maximum mutation rate the\\ngiven immune system can suppress. Fig. 4B shows this threshold\\nas a function of p and μ. A given virus would have roughly the\\nsame mutation rate, while p would vary with individual patients.\\nThe sensitivity of disease progression with p is consistent with\\nthe observation that even when the fatality rate of a certain\\ninfection is known, the course of infection in a patient is often\\nunpredictable.\\n\\nThe case for HIV-1 is unique in the sense that without treat-\\nment, most patients eventually progress to the disease stage. One\\nof the special characteristics of HIV-1 is that it targets CD4þ\\n(helper) Tcells, which play critical roles in eliciting and mediating\\nCTL responses. Within the context of our model, the destruction\\nof CD4þ Tcells would lead to a gradual decrease in both the sti-\\nmulation rate b and the clearance rate p. One therefore expects a\\ncontinual deterioration of any initial effective control of infec-\\ntion, represented by p values that decrease over the course of dis-\\nease progression. In Fig. 4B, therefore, even though patients with\\nstronger immune systems may have p values initially in the extinc-\\ntion regime, the continued depletion of CD4þ cells would lower p\\nand cause them to cross the threshold.\\n\\nConclusions\\nIn this paper, we demonstrated that the classic quasispecies the-\\nory can form the basis of a quantitative model of virus evolution\\nunder immune selection. One of the apparent challenges in de-\\nveloping such a model is that viral genomes, while relatively small\\n(approximately 104 bp), still constitute a huge sequence space.\\nHere, we showed that a set of epitopes (La ∼ 10 amino acids)\\ncan be considered as minimal units of genomic segments on which\\nselection acts. In standard quasispecies theory, competition be-\\ntween different strains arises indirectly from the constraint of\\nfixed total population size. In our model, the degradation and\\nremoval of virions occur via immune response, which can be re-\\ngarded as a concrete specification of the selection forces. In this\\ncase, indirect competition occurs because most viral strains share\\n\\n10\\n1\\n\\n10\\n2\\n\\n10\\n3\\n\\n10\\n4\\n\\nV\\nira\\n\\nl l\\noa\\n\\nd\\n\\n10\\n0\\n\\n10\\n1\\n\\n10\\n2\\n\\n10\\n3\\n\\n10\\n4\\n\\nTime (d)\\n\\n1\\n\\n2\\n\\n3\\n\\n4\\n\\nN\\no.\\n\\n o\\nf p\\n\\nhe\\nno\\n\\nty\\npe\\n\\ns\\n\\np = 0.01 d\\n-1\\n\\n10\\n1\\n\\n10\\n2\\n\\n10\\n3\\n\\nC\\nT\\n\\nL\\n\\nAcute\\nChronic\\n\\nExtinction\\n\\nA\\n\\nB\\n\\nC\\n\\n} p = 0.02 d\\n-1\\n\\nFig. 3. Typical disease progression patterns in the extinction regime from\\nthe multi-epitope quasispecies model. (A) Viral load. (B) Total CTL levels.\\n(C) Number of distinct phenotypes per epitope present in the population.\\nThe parameter values other than p and μ ¼ 1.0 × 10−5 bp−1 were the same\\nas in Fig. 2. The data shown are individual trajectories.\\n\\n10\\n-6\\n\\n10\\n-5\\n\\n10\\n-4\\n\\n10\\n-3\\n\\n10\\n-2\\n\\n10\\n-1\\n\\nMutation rate µ (bp-1)\\n\\n10\\n-2\\n\\n10\\n-1\\n\\nC\\nle\\n\\nar\\nan\\n\\nce\\n r\\n\\nat\\ne \\n\\np \\n(d\\n\\n-1\\n)\\n\\n10\\n-6\\n\\n10\\n-5\\n\\n10\\n-4\\n\\n10\\n-3\\n\\n10\\n-2\\n\\n10\\n-1\\n\\nMutation rate µ (bp-1)\\n\\n10\\n0\\n\\n10\\n1\\n\\n10\\n2\\n\\n10\\n3\\n\\nE\\nxt\\n\\nin\\nct\\n\\nio\\nn \\n\\ntim\\ne \\n\\n(d\\n)\\n\\np = 0.10 d\\n-1\\n\\np = 0.02 d\\n-1\\n\\nA B\\nExtinction\\n\\nRunaway\\nHIV-1\\n\\nFig. 4. Crossover between the runaway and extinction regimes. (A) Varia-\\ntion of extinction time with mutation rate μ for different clearance rates\\np in the extinction regime. Error bars represent one standard deviation. Near\\nthe threshold, the extinction events occur increasingly in the edge of viral\\nload fluctuations (Fig. 3A). (B) The estimated threshold between the runaway\\nand extinction regimes in the p − μ parameter space (defined as the maxi-\\nmum μ for which the extinction time converges for a given p). The arrow\\nillustrates the dynamic deterioration of immune response during HIV-1 infec-\\ntion. Other parameter values were the same as in Fig. 2.\\n\\nWoo and R',\n",
       " 'fman PNAS ∣ August 7, 2012 ∣ vol. 109 ∣ no. 32 ∣ 12983\\n\\nBI\\nO\\nPH\\n\\nYS\\nIC\\nS\\nA\\nN\\nD\\n\\nCO\\nM\\nPU\\n\\nTA\\nTI\\nO\\nN\\nA\\nL\\nBI\\nO\\nLO\\n\\nG\\nY\\n\\nD\\now\\n\\nnl\\noa\\n\\nde\\nd \\n\\nby\\n g\\n\\nue\\nst\\n\\n o\\nn \\n\\nA\\npr\\n\\nil \\n23\\n\\n, 2\\n02\\n\\n0 \\n\\n\\n\\nidentical phenotypes in a large fraction of th',\n",
       " 'r epitopes and are\\nconstrained by common groups of CTLs. However, the model\\ndoes not account for the effects of direct competitions for re-\\nsources (limited number of host cells available for infection).\\nWe have additionally explored an extended model that includes\\nhost-cell dynamics (SI Text), whose analysis suggests that the main\\nqualitative results remain valid (Fig. S8).\\n\\nOur finding that the relative importance of immune escape is\\nmainly determined by the viral load in chronic infection may have\\nimplications to vaccine design (42): Strategies aimed at prevent-\\ning immune escape, such as targeting epitopes with higher fitness\\ncosts to mutations, may not confer more benefits than those at-\\ntempting to boost the overall immune response. We also note,\\nhowever, that more realistic representations of T-cell stimulation\\nand clearance will have to take th',\n",
       " 'r dependence on phenotypes\\ninto account (p and b should depend on amino acid sequences),\\nbecause mutations affect HLA-epitope-CTL binding affinities.\\nOur model also ignores complex epistatic effects coupling muta-\\ntions on the same and different epitopes (43). The approach pre-\\nsented here likely can provide a foundation for a more compre-\\nhensive modeling framework to tackle such global genome-wide\\neffects. In addition, although we interpreted Eqs. 3B–3D strictly\\nas the CTL-mediated immune response, we expect similar ap-\\nproaches to be applicable for antibody-based responses.\\n\\nMethods\\nStochastic Evolutionary Dynamics. Simulations are carried out by applying the\\nGillespie algorithm (26) to Eq. 3. At a given time, a dynamic list of genotypes\\nand phenotypes (N and M in total, respectively), starting with a single WT at\\nthe initial condition, is kept and updated, instead of enumerating all possible\\nsequences. A genotype consists of the nucleotide sequence of a set of epi-\\ntopes (l in total), each with the same length L, such that the total length be-\\ncomes l × L. We define the following quantities\\n\\naðrÞi ¼ ∑\\ni\\n\\nj¼1\\nrjnj; a\\n\\nðsÞ\\ni ¼ ∑\\n\\ni\\n\\nj¼1\\nðrj þ bÞnj;\\n\\naðcÞi ¼ ∑\\ni\\n\\nj¼1\\nðrj þ bþ pcjÞnj; aðdÞα ¼ aðcÞN þ g∑\\n\\nα\\n\\nβ¼1\\ncðβÞ;\\n\\neach corresponding to the replication, stimulation, clearance, and decay re-\\nactions (Eq. 3), respectively, where cj ¼ ΣkcðαjkÞ is the total number of CTLs\\ncorresponding to any of the epitope phenotypes belonging to the genotype\\nj. A random number 0 < r < 1 is drawn to determine the time interval Δt ¼\\nð1∕aÞ lnð1∕rÞ before the next reaction event, where a ¼ a ðdÞM . A second ran-\\ndom number r is drawn, and if aðxÞi−1 < ar < a\\n\\nðxÞ\\ni for x ¼ r, s, c, the replication,\\n\\nstimulation, or clearance reaction is chosen for the genotype i, respectively. If\\na ðdÞα−1 < ar < a\\n\\nðdÞ\\nα instead, the decay reaction is carried out for the CTL with phe-\\n\\nnotype α. In this algorithm, therefore, a given CTL contributes equally to the\\nclearance probabilities of all viral strains containing its phenotype. This pre-\\nscription generates unbiased trajectories of two classes of objects, the viruses\\nwith genotype i and phenotype αik and CTLs with phenotype α. For replica-\\ntion reactions, the chosen genotype i is mutated with probability μ per bp.\\nFurther details on implementing the genotype-phenotype lists can be found\\nin SI Text.\\n\\nSimulation Conditions. The standard genetic code was used to translate\\nmutant genotypes into phenotypes. The reduced fitness sα ¼ rατ of a pheno-\\ntype αwas taken as follows: sα ¼ 1 for theWT, and for all other phenotypes sα\\nis a random variable with the following Gaussian distribution: PðsαÞ ¼\\nexpf−½sα − fðdαÞ�2�∕2σ2g∕ð2πÞ1∕2σ, where the position of the maximum\\nfðdαÞ ¼ expð−dα∕ξÞ is an exponentially decreasing function of the distance\\ndα ≥ 1 (summed over all epitopes; in units of amino acids) of the phenotype\\nα to the WT. The parameters ξ and σ are the measures of the fitness costs of\\nmutations and of randomness in fitness distributions, respectively. During si-\\nmulations, a newly encountered phenotype is dynamically assigned a fitness\\nvalue based on this distributionwith negative rα values replaced by zero. Phe-\\nnotypes containing stop codons were assigned zero fitness. In Fig. 1, the SIV\\nepitope Tat28–35SL8 with the WT sequence of 5′-TCC ACT CCA GAA TCG GCC\\nAAC CTG-3′ (STPESANL) (38) was used (La ¼ 8 and l ¼ 1). In Figs. 2–4, random\\nsequences were used for the WT and simulations were performed with La ¼\\n10 and l ¼ 3. Typically, quantities were averaged over 102 ∼ 103 trajectories\\nunless otherwise specified.\\n\\nACKNOWLEDGMENTS. This work was sponsored by both the US Army’s\\nNetwork Science Initiative, managed by the Military Operational Medicine\\nResearch Program of the US Army Medical Research and Materiel Command,\\nFort Detrick, Maryland, and by a competitive In-House Laboratory Indepen-\\ndent Research award by the US Army Assistant Secretary of the Army for\\nAcquisition, Logistics, and Technology. The opinions and assertions contained\\nher',\n",
       " 'n are the private views of the authors and are not to be construed as\\nofficial or as reflecting the views of the US Army or of the US Department\\nof Defense. This paper has been approved for public release with unlimited\\ndistribution.\\n\\n1. Eigen M (1971) Selforganization of matter and the evolution of biological macromo-\\nlecules. Naturwissenschaften 58:465–523.\\n\\n2. Swetina J, Schuster P (1982) Self-replication with errors: A model for polynucleotide\\nreplication. Biophys Chem 16:329–345.\\n\\n3. Loeb LA, et al. (1999) Lethal mutagenesis of HIV with mutagenic nucleoside analogs.\\nProc Natl Acad Sci USA 96:1492–1497.\\n\\n4. Iranzo J, Perales C, Domingo E, Manrubia SC (2011) Tempo and mode of inhibitor-\\nmutagen antiviral therapies: A multidisciplinary approach. Proc Natl Acad Sci USA\\n108:16008–16013.\\n\\n5. Domingo E, Holland JJ (1997) RNA virus mutations and fitness for survival. Annu Rev\\nMicrobiol 51:151–178.\\n\\n6. Mardis ER (2008) Next-generation DNA sequencing methods. Annu Rev Genomics\\nHum Genet 9:387–402.\\n\\n7. McMichael AJ, Borrow P, Tomaras GD, Goodnetilleke N, Haynes BF (2010) The immune\\nresponse during acute HIV-1 infection: Clues for vaccine development. Nat Rev Immu-\\nnol 10:11–23.\\n\\n8. Mild M, Hedskog C, Jernberg J, Albert J (2011) Performance of ultradeep pyrosequen-\\ncing in analysis of HIV-1 pol gene variation. PLoS One 6:e22741.\\n\\n9. Bimber BN, et al. (2009) Ultradeep pyrosequencing detects complex patterns of CD8þ\\n\\nT-lymphocyte escape in simian immunodeficiency virus-infected macaques. J Virol\\n83:8247–8253.\\n\\n10. Bimber BN, et al. (2010) Whole-genome characterization of human and simian\\nimmunodeficiency virus intrahost diversity by ultradeep pyrosequencing. J Virol\\n84:12087–12092.\\n\\n11. Fischer W, et al. (2010) Transmission of single HIV-1 genomes and dynamics of early\\nimmune escape revealed by ultradeep sequencing. PLoS One 5:e12303.\\n\\n12. Cale EM, et al. (2011) Epitope-specific CD8þ T lymphocytes cross-recognize mutant\\nsimian immunodefieciency virus (SIV) sequences but fail to contain very early evolu-\\ntion and eventual fixation of epitope escape mutations during SIV infection. J Virol\\n85:3746–3757.\\n\\n13. Moore CB, et al. (2002) Evidence of HIV-1 adaptation to HLA-restricted immune\\nresponses at a population level. Science 296:1439–1443.\\n\\n14. Bhattacharya T, et al. (2007) Founder effects in the assessment of HIV polymorphisms\\nand HLA allele associations. Science 315:1583–1586.\\n\\n15. Kawashima Y, et al. (2009) Adaptation of HIV-1 to human leukocyte antigen class I.\\nNature 458:641–645.\\n\\n16. Addo MM, et al. (2003) Comprehensive epitope analysis of human immunodeficiency\\nvirus type 1 (HIV-1)-specific T-cell responses directed against the entire expressed HIV-1\\ngenome demonstrate broadly directed responses, but no correlation to viral load.\\nJ Virol 77:2081–2092.\\n\\n17. Goodnetilleke N, et al. (2009) The first T-cell response to transmitted/founder\\nvirus contributes to the control of acute viremia in HIV-1 infection. J Exp Med\\n206:1253–1272.\\n\\n18. Nowak MA, Bangham CR (1996) Population dynamics of immune responses to persis-\\ntent viruses. Science 272:74–79.\\n\\n19. Nowak MA, May RM (2001) Virus Dynamics: Mathematical Principles of Immunology\\nand Virology (Oxford Univ Press, New York).\\n\\n20. Nowak MA, et al. (1995) Antigenic oscillations and shifting immunodominance in\\nHIV-1 infections. Nature 375:606–611.\\n\\n21. Althaus CL, De Boer RJ (2008) Dynamics of immune escape during HIV/SIV infection.\\nPLoS Comput Biol 4:e1000103.\\n\\n22. Love TM, Thurston SW, Keefer MC, Dewhurst S, Lee HY (2010) Mathematical modeling\\nof ultradeep sequencing data reveals that acute CD8þ T-lymphocyte responses exert\\nstrong selective pressure in simian immunodeficiency virus-infected macaques but still\\nfail to clear founder epitope sequences. J Virol 84:5802–5814.\\n\\n23. Lee HY, et al. (2009) Modeling sequence evolution in acute HIV-1 infection. J Theor Biol\\n261:341–360.\\n\\n24. Lee HY, Perelson AS, Park S-C, L',\n",
       " 'tner T (2008) Dynamic correlation between intrahost\\nHIV-1 quasispecies evolution and disease progression. PLoS Comput Biol 4:e1000240.\\n\\n25. Keele BF, et al. (2008) Identification and characterization of transmitted and\\nearly founder virus envelopes in primary HIV-1 infection. Proc Natl Acad Sci USA\\n105:7552–7557.\\n\\n26. Gillespie DT (1977) Exact stochastic simulation of coupled chemical reactions. J Phys\\nChem 81:2340–2361.\\n\\n27. Nowak M, Schuster P (1989) Error thresholds of replication in finite populations muta-\\ntion frequencies and the onset of Muller’s ratchet. J Theor Biol 137:375–395.\\n\\n28. Yang Z, Bielawski JP (2000) Statistical methods for detecting molecular adaptation.\\nTrends Ecol Evol 15:496–503.\\n\\n12984 ∣ www.pnas.org/cgi/doi/10.1073/pnas.1117201109 Woo and R',\n",
       " 'fman\\n\\nD\\now\\n\\nnl\\noa\\n\\nde\\nd \\n\\nby\\n g\\n\\nue\\nst\\n\\n o\\nn \\n\\nA\\npr\\n\\nil \\n23\\n\\n, 2\\n02\\n\\n0 \\n\\nhttp://www.pnas.org/lookup/suppl/doi:10.1073/pnas.1117201109/-/DCSupplemental/pnas.1117201109_SI.pdf?targetid=STXT\\nhttp://www.pnas.org/lookup/suppl/doi:10.1073/pnas.1117201109/-/DCSupplemental/pnas.1117201109_SI.pdf?targetid=SF8\\nhttp://www.pnas.org/lookup/suppl/doi:10.1073/pnas.1117201109/-/DCSupplemental/pnas.1117201109_SI.pdf?targetid=STXT\\n\\n\\n29. Aguirre J, Buldu JM, Stich M, Manrubia SC (2011) Topological structure of the space of\\nphenotypes: The case of RNA neutral networks. PLoS One 6:e26324.\\n\\n30. Burch CL, Chao L (2000) Evolvability of an RNA virus is determined by its mutational\\nn',\n",
       " 'ghborhood. Nature 406:625–628.\\n\\n31. Schuster P, Fontana W (1999) Chance and necessity in evolution: Lessons from RNA.\\nPhysica D 133:427–452.\\n\\n32. Hietpas RT, Jensen JD, Bolon DN (2011) Experimental illumination of a fitness land-\\nscape. Proc Natl Acad Sci USA 108:7896–7901.\\n\\n33. Stich M, Briones C, Manrubia SC (2008) On the structural repertoire of pools of short,\\nrandom RNA sequences. J Theor Biol 252:750–763.\\n\\n34. Pitt JN, Ferre-D’Amare AR (2010) Rapid construction of empirical RNA fitness land-\\nscapes. Science 330:376–379.\\n\\n35. Duarte EA, et al. (1994) Subclonal components of consensus fitness in an RNA virus\\nclone. J Virol 68:4295–4301.\\n\\n36. Sanjuan R, Moya A, Elena SF (2004) The distribution of fitness effects caused by single-\\nnucleotide substitutions in an RNA virus. Proc Natl Acad Sci USA 101:8396–8401.\\n\\n37. Fernandez G, Clotet B, Martinez MA (2007) Fitness landscape of human immunode-\\nficiency virus type 1 protease quasispecies. J Virol 81:2485–2496.\\n\\n38. Allen TM, et al. (2000) Tat-specific cytotoxic T lymphocytes select for SIV escape\\nvariants during resolution of primary viraemia. Nature 407:386–390.\\n\\n39. Mansky LM, Temin HM (1995) Lower in vivo mutation rate of human immunodefi-\\nciency virus type 1 than that predicted from the fidelity of purified reverse transcrip-\\ntase. J Virol 69:5087–5094.\\n\\n40. Schuster P, Swetina J (1988) Stationary mutant distributions and evolutionary optimi-\\nzation. Bull Math Biol 50:635–660.\\n\\n41. Wilke CO,Wang JL, Ofria C, Lenski RE, Adami C (2001) Evolution of digital organisms at\\nhigh mutation rates leads to survival of the flattest. Nature 412:331–333.\\n\\n42. Gaschen B, et al. (2002) Diversity considerations in HIV-1 vaccine selection. Science\\n296:2354–2360.\\n\\n43. Dahirel V, et al. (2011) Coordinate linkage of HIV evolution reveals regions of immu-\\nnological vulnerability. Proc Natl Acad Sci USA 108:11530–11535.\\n\\nWoo and R',\n",
       " 'fman PNAS ∣ August 7, 2012 ∣ vol. 109 ∣ no. 32 ∣ 12985\\n\\nBI\\nO\\nPH\\n\\nYS\\nIC\\nS\\nA\\nN\\nD\\n\\nCO\\nM\\nPU\\n\\nTA\\nTI\\nO\\nN\\nA\\nL\\nBI\\nO\\nLO\\n\\nG\\nY\\n\\nD\\now\\n\\nnl\\noa\\n\\nde\\nd \\n\\nby\\n g\\n\\nue\\nst\\n\\n o\\nn \\n\\nA\\npr\\n\\nil \\n23\\n\\n, 2\\n02\\n\\n0 \\n\\n\\n']"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# separate content by a specified sequence of characters\n",
    "cont.split('ei')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['',\n",
       " '',\n",
       " '',\n",
       " '',\n",
       " '',\n",
       " '',\n",
       " '',\n",
       " '',\n",
       " '',\n",
       " '',\n",
       " '',\n",
       " '',\n",
       " '',\n",
       " '',\n",
       " '',\n",
       " '',\n",
       " '',\n",
       " '',\n",
       " '',\n",
       " '',\n",
       " '',\n",
       " '',\n",
       " '',\n",
       " '',\n",
       " '',\n",
       " '',\n",
       " '',\n",
       " '',\n",
       " '',\n",
       " '',\n",
       " '',\n",
       " '',\n",
       " '',\n",
       " '',\n",
       " '',\n",
       " '',\n",
       " '',\n",
       " '201117201 12980..12985',\n",
       " '',\n",
       " '',\n",
       " 'A quantitative quasispecies theory-based model of',\n",
       " 'virus escape mutation under immune selection',\n",
       " 'Hyung-June Woo and Jaques Reifman1',\n",
       " '',\n",
       " 'Biotechnology High Performance Computing Software Applications Institute, Telemedicine and Advanced Technology Research Center, US Army Medical',\n",
       " 'Research and Materiel Command, Fort Detrick, MD 21702',\n",
       " '',\n",
       " 'Edited by Peter Schuster, University of Vienna, Vienna, and approved June 28, 2012 (received for review October 18, 2011)',\n",
       " '',\n",
       " 'Viral infections involve a complex interplay of the immune',\n",
       " 'response and escape mutation of the virus quasispecies inside a',\n",
       " 'single host. Although fundamental aspects of such a balance of',\n",
       " 'mutation and selection pressure have been established by the qua-',\n",
       " 'sispecies theory decades ago, its implications have largely re-',\n",
       " 'mained qualitative. Here, we present a quantitative approach to',\n",
       " 'model the virus evolution under cytotoxic T-lymphocyte immune',\n",
       " 'response. The virus quasispecies dynamics are explicitly repre-',\n",
       " 'sented by mutations in the combined sequence space of a set of',\n",
       " 'epitopes within the viral genome. We stochastically simulated the',\n",
       " 'growth of a viral population originating from a single wild-type',\n",
       " 'founder virus and its recognition and clearance by the immune re-',\n",
       " 'sponse, as well as the expansion of its genetic diversity. Applied to',\n",
       " 'the immune escape of a simian immunodeficiency virus epitope,',\n",
       " 'model predictions were quantitatively comparable to the experi-',\n",
       " 'mental data. Within the model parameter space, we found two',\n",
       " 'qualitatively different regimes of infectious disease pathogenesis,',\n",
       " 'each representing alternative fates of the immune response: It can',\n",
       " 'clear the infection in finite time or eventually be overwhelmed by',\n",
       " 'viral growth and escape mutation. The latter regime exhibits the',\n",
       " 'characteristic disease progression pattern of human immunodefi-',\n",
       " 'ciency virus, while the former is bounded by maximum mutation',\n",
       " 'rates that can be suppressed by the immune response. Our results',\n",
       " 'demonstrate that, by explicitly representing epitope mutations',\n",
       " 'and thus providing a genotype–phenotype map, the quasispecies',\n",
       " 'theory can form the basis of a detailed sequence-specific model of',\n",
       " 'real-world viral pathogens evolving under immune selection.',\n",
       " '',\n",
       " 'population dynamics ∣ next-generation sequencing ∣ stochastic simulation ∣',\n",
       " 'HIV',\n",
       " '',\n",
       " 'Viruses with RNA genomes, such as human immunodeficiencyvirus type 1 (HIV-1), have high rates of mutation and evolve',\n",
       " 'rapidly in response to host immune selection pressure. One of the',\n",
       " 'consequences of such rapid mutations is the error catastrophe',\n",
       " '(1), where a virus population is driven to extinction when its mu-',\n",
       " 'tation rate exceeds a threshold. The existence of such a threshold',\n",
       " 'is a central prediction of the quasispecies theory pioneered by',\n",
       " 'Eigen (1) and Swetina and Schuster (2). The recent experimental',\n",
       " 'demonstration of lethal mutagenesis (3–5), in which an error cat-',\n",
       " 'astrophe transition is caused by mutagens, demonstrates that key',\n",
       " 'features of the balance of mutation and selection in viruses are',\n",
       " 'elegantly captured by the quasispecies theory.',\n",
       " '',\n",
       " 'Although insights provided by the quasispecies theory have',\n",
       " 'greatly expanded our understanding of virus behavior, applica-',\n",
       " 'tions so far have been limited to a conceptual level, partially due',\n",
       " 'to the lack of experimental information on the evolutionary dy-',\n",
       " 'namics in sequence space. Next-generation sequencing techni-',\n",
       " 'ques (6) have the potential to change this situation. To build',\n",
       " 'quantitative models implementing the quasispecies dynamics',\n",
       " 'and describe such experimental data, it is important to realisti-',\n",
       " 'cally specify the nature of selection pressure. Viruses in animal',\n",
       " 'hosts evolve under immune pressure, and their capacity for rapid',\n",
       " 'escape mutation underlies many of the difficulties in combating',\n",
       " 'pathogens, including HIV-1. In a typical disease pathogenesis of',\n",
       " 'HIV-1, the acute viremia after an initial infection is curbed by',\n",
       " '',\n",
       " 'CD8þ cytotoxic T-lymphocyte (CTL) responses as well as subse-',\n",
       " 'quent antibody actions, leading to an asymptomatic chronic infec-',\n",
       " 'tion stage that can last up to 10 y (7). However, this apparent',\n",
       " 'control of viremia is never complete, and without antiviral ther-',\n",
       " 'apy, the chronic infection eventually leads to the onset of disease.',\n",
       " 'This chronic infection stage involves continuous escape mutation-',\n",
       " 'CTL response cycles, whose detailed characteristics are being un-',\n",
       " 'covered by ultradeep sequencing (8–12). CTLs recognize specific',\n",
       " 'viral epitopes (approximately 10 amino acids long) presented on',\n",
       " 'the surface of infected cells by class I human leukocyte antigen',\n",
       " '(HLA). The epitope recognition depends sensitively on HLA al-',\n",
       " 'leles, leading to differential patterns of immune response among',\n",
       " 'patients (13–15), while characteristics of immune response during',\n",
       " 'early infection often shape and influence the overall disease pro-',\n",
       " 'gression (16, 17). Quantitative sequence-based models of virus-',\n",
       " 'CTL dynamics will greatly facilitate the interpretation of experi-',\n",
       " 'mental data.',\n",
       " '',\n",
       " 'In a series of pioneering works (18–20), Nowak and coworkers',\n",
       " 'introduced population dynamics concepts that capture a diverse',\n",
       " 'range of immune response and escape mutation. Similar ap-',\n",
       " 'proaches focused more on escape dynamics have also been pro-',\n",
       " 'posed (21, 22). These models, however, do not describe mutations',\n",
       " 'in sequence space explicitly. Other mathematical models of',\n",
       " 'sequence evolution focus on sequence divergence during the very',\n",
       " 'early stages of infection, while ignoring or only implicitly including',\n",
       " 'the effect of selection pressure (23, 24). In this paper, we describe',\n",
       " 'a quantitative quasispecies-based model of virus dynamics under',\n",
       " 'T cell-based immune pressure, explicitly mapping genotype–phe-',\n",
       " 'notype relationships. Our approach combines the description of',\n",
       " 'virus evolutionary dynamics provided by the quasispecies theory',\n",
       " 'and the population dynamics models of Nowak and coworkers.',\n",
       " 'We show that the model not only captures the salient features',\n",
       " 'observed in sequencing data of simian immunodeficiency virus',\n",
       " '(SIV) (9) but also reveals general qualitative features of viral',\n",
       " 'infection disease pathogenesis: The immune system can clear',\n",
       " 'the infection within time scales ranging from days to those much',\n",
       " 'longer than patient lifetimes, or be overwhelmed by immune es-',\n",
       " 'cape. The viral load progression in the latter regime closely',\n",
       " 'matches the observed HIV-1 disease pathogenesis pattern.',\n",
       " '',\n",
       " 'Results and Discussion',\n",
       " 'Stochastic Quasispecies Dynamics.During acute infection, the foun-',\n",
       " 'der virus (often a single virion) (7, 11, 17, 25) undergoes replica-',\n",
       " 'tions to produce an exponentially growing population, which may',\n",
       " 'be described by the quasispecies dynamics without degradation,',\n",
       " '',\n",
       " 'Author contributions: H.-J.W. designed research; H.-J.W. performed research; H.-J.W. and',\n",
       " 'J.R. analyzed data; and H.-J.W. and J.R. wrote the paper.',\n",
       " '',\n",
       " 'The authors declare no conflict of interest.',\n",
       " '',\n",
       " 'This article is a PNAS Direct Submission.',\n",
       " '',\n",
       " 'Freely available online through the PNAS open access option.',\n",
       " '1To whom correspondence should be addressed. E-mail: jaques.reifman@us.army.mil.',\n",
       " '',\n",
       " 'This article contains supporting information online at www.pnas.org/lookup/suppl/',\n",
       " 'doi:10.1073/pnas.1117201109/-/DCSupplemental.',\n",
       " '',\n",
       " '12980–12985 ∣ PNAS ∣ August 7, 2012 ∣ vol. 109 ∣ no. 32 www.pnas.org/cgi/doi/10.1073/pnas.1117201109',\n",
       " '',\n",
       " 'D',\n",
       " 'ow',\n",
       " '',\n",
       " 'nl',\n",
       " 'oa',\n",
       " '',\n",
       " 'de',\n",
       " 'd ',\n",
       " '',\n",
       " 'by',\n",
       " ' g',\n",
       " '',\n",
       " 'ue',\n",
       " 'st',\n",
       " '',\n",
       " ' o',\n",
       " 'n ',\n",
       " '',\n",
       " 'A',\n",
       " 'pr',\n",
       " '',\n",
       " 'il ',\n",
       " '23',\n",
       " '',\n",
       " ', 2',\n",
       " '02',\n",
       " '',\n",
       " '0 ',\n",
       " '',\n",
       " 'http://www.pnas.org/lookup/suppl/doi:10.1073/pnas.1117201109/-/DCSupplemental',\n",
       " 'http://www.pnas.org/lookup/suppl/doi:10.1073/pnas.1117201109/-/DCSupplemental',\n",
       " 'http://www.pnas.org/lookup/suppl/doi:10.1073/pnas.1117201109/-/DCSupplemental',\n",
       " 'http://www.pnas.org/lookup/suppl/doi:10.1073/pnas.1117201109/-/DCSupplemental',\n",
       " 'http://www.pnas.org/lookup/suppl/doi:10.1073/pnas.1117201109/-/DCSupplemental',\n",
       " 'http://www.pnas.org/lookup/suppl/doi:10.1073/pnas.1117201109/-/DCSupplemental',\n",
       " '',\n",
       " '',\n",
       " '_nj ¼ ∑',\n",
       " 'i',\n",
       " '',\n",
       " 'Qjirini; [1]',\n",
       " '',\n",
       " 'where ni is the number of virions with genotype i, ri is the repli-',\n",
       " 'cation rate of i, andQji ¼ ð1 − μÞL−dðj;iÞðμ∕3Þdðj;iÞ is the mutation',\n",
       " 'probability from i to genotype j, with mean mutation rate μ,',\n",
       " 'genome length L (in bp), and Hamming distance (the number',\n",
       " 'of nucleotides that are different) dðj; iÞ between j and i. A more',\n",
       " 'complete description for early stages of infection, where ni is of',\n",
       " 'the order of 1 and restricted to integer values, is given by the che-',\n",
       " 'mical reaction representation',\n",
       " '',\n",
       " 'Ri →',\n",
       " 'Qjiri',\n",
       " '',\n",
       " 'Ri þRj; [2]',\n",
       " '',\n",
       " 'where Ri is a virion with genotype i. Eq. 2 can be simulated',\n",
       " 'stochastically with the Gillespie algorithm (26). A major advan-',\n",
       " 'tage of such a stochastic formulation is that it allows us to avoid',\n",
       " 'exhaustive enumerations of all possible genotypes. This is signif-',\n",
       " 'icant even for a single epitope of viral proteins: An amino acid',\n",
       " 'sequence with La ¼ 10, whose genome length is L ¼ 3La ¼',\n",
       " '30 bp, still has 4L ∼ 1018 genotypes. Nowak and Schuster (27)',\n",
       " 'performed Gillespie algorithm simulations of the standard qua-',\n",
       " 'sispecies model under the single-peak fitness landscape (ri ¼ 1∕τ',\n",
       " 'for the wild-type (WT) sequence and ri ¼ 1∕Aτ for all other mu-',\n",
       " 'tants, whereA > 1 is the relative fitness of theWTand τ is a char-',\n",
       " 'acteristic time scale). In this case, distributions of individuals',\n",
       " 'among only two groups (WT and mutants) need to be tracked.',\n",
       " '',\n",
       " 'In our numerical simulations, the initial genotype (referred to',\n",
       " 'here as the WT) replicates with mutation rate μ, and the gener-',\n",
       " 'ated mutants are compared with the existing list of sequences,',\n",
       " 'which is updated dynamically when new sequences arise. We ver-',\n",
       " 'ified that this numerical scheme sampled the relevant sequence',\n",
       " 'space sufficiently (SI Text, Figs. S1–S3). The initial condition we',\n",
       " 'used (single WT) and the discreteness of ni imply that ni should',\n",
       " 'be interpreted as the total number of virions within a finite sys-',\n",
       " 'tem. The HIV-1 viral load during the acute infection phase can',\n",
       " 'reach up to 104 ∼ 106 RNA copies/mL (7). Accounting for the',\n",
       " 'volume of blood in the body of an average adult (approximately',\n",
       " '5 L), the total population size Nv ¼ Σi ni would be up to about',\n",
       " '109. We found simulations to slow down significantly as Nv be-',\n",
       " 'came larger than about 106. For computational efficiency, we',\n",
       " 'therefore regarded the system as a small representative volume',\n",
       " '(1 mL) of blood, and the viral loads and CTL levels reported refer',\n",
       " 'to RNA copy numbers and cell counts within this volume.',\n",
       " '',\n",
       " 'CTL Response. Within an infected host, the action of the cellular',\n",
       " 'immune response provides the major force countering the acute',\n",
       " 'viremia. In this work, we combine the population dynamics ap-',\n",
       " 'proach of virus-immune system dynamics by Nowak and cowor-',\n",
       " 'kers (18, 19) with the classic quasispecies theory. Accordingly,',\n",
       " 'Eq. 2 is modified as follows:',\n",
       " '',\n",
       " 'Ri →',\n",
       " 'Qjiri',\n",
       " '',\n",
       " 'Ri þRj; ðreplication with mutationÞ [3A]',\n",
       " '',\n",
       " 'Ri→',\n",
       " 'b',\n",
       " 'Ri þ∑',\n",
       " '',\n",
       " 'l',\n",
       " '',\n",
       " 'k¼1',\n",
       " 'CðαikÞ; ðstimulationÞ [3B]',\n",
       " '',\n",
       " 'Ri þ CðαikÞ→',\n",
       " 'p',\n",
       " 'CðαikÞ; ðclearanceÞ [3C]',\n",
       " '',\n",
       " 'CðαÞ→',\n",
       " 'g',\n",
       " '0; ðdecayÞ [3D]',\n",
       " '',\n",
       " 'where the genotype i is now a member of the combined sequence',\n",
       " 'space of l epitopes. The amino acid sequence (“phenotype”) of',\n",
       " 'epitope k in genotype i is denoted as αik, and CðαÞ represents a',\n",
       " '',\n",
       " 'CTL specific to phenotype α. Eq. 3B represents a reaction in',\n",
       " 'which, with a rate b, virions of sequence i stimulate the produc-',\n",
       " 'tion of a set of CTLs CðαikÞ corresponding to the phenotypes of',\n",
       " 'its epitopes. Eq. 3C denotes the reaction where cells infected with',\n",
       " 'viruses with genotype i are cleared by CTLs with phenotypes that',\n",
       " 'match one of its epitopes with a rate p, and Eq. 3D represents the',\n",
       " 'natural decay of T cells with a rate g. Under the alternative de-',\n",
       " 'terministic continuum approximation, one may write:',\n",
       " '',\n",
       " '_nj ¼ ∑',\n",
       " 'i',\n",
       " '',\n",
       " 'Qjirini − p∑',\n",
       " 'l',\n",
       " '',\n",
       " 'k¼1',\n",
       " 'cðαjkÞnj; [4A]',\n",
       " '',\n",
       " '_cðαÞ ¼ bnα − gcðαÞ; [4B]',\n",
       " '',\n",
       " 'where cðαÞ is the number of Tcells CðαÞ and nα is the total num-',\n",
       " 'ber of virions containing epitopes with phenotype α.',\n",
       " '',\n",
       " 'The qualitative features of the infection-clearance dynamics',\n",
       " 'can be illustrated by setting the mutation rate μ to zero. Numer-',\n",
       " 'ical integrations of Eq. 4 yield the continuum approximation re-',\n",
       " 'sult (Fig. S4 for μ ¼ 0 and l ¼ 1), which showed reasonable',\n",
       " 'agreement with the stochastic simulation results. For nonvanish-',\n",
       " 'ing mutation rates, numerical integration rapidly becomes prohi-',\n",
       " 'bitive with increasing L because of the high dimensionality of the',\n",
       " 'sequence space. The inverse of b is a measure of the time delay of',\n",
       " 'the growth of T cells compared to that of viruses. Here, we',\n",
       " 'adopted b values of approximately 0.01 d−1. The efficiency of',\n",
       " 'CTLs in recognizing cells presenting the epitope and killing them',\n",
       " 'is represented by parameter p, the clearance rate. We found',\n",
       " 'p values of 10−4 ∼ 10−3 d−1 to give good fits to experiments',\n",
       " '(see below).',\n",
       " '',\n",
       " 'Immune Escape Dynamics. To make comparisons with experiments,',\n",
       " 'it is necessary to consider that fitness is determined by amino acid',\n",
       " 'sequences, and many mutations are synonymous. Therefore, the',\n",
       " 'full immune escape dynamics represented by Eq. 3 distinguishes',\n",
       " 'the amino acid sequence αik corresponding to the nucleotide',\n",
       " 'sequence of epitope k in genotype i. We refer to the set of nu-',\n",
       " 'cleotide and amino acid sequences as “genotypes” and “pheno-',\n",
       " 'types,” respectively. The fitness ri is a function of phenotypes',\n",
       " 'only. During simulations, each genotype is translated into a phe-',\n",
       " 'notype, and a new mutant is checked against the existing pheno-',\n",
       " 'type/genotype list, which is updated and expanded. The nature of',\n",
       " 'this genotype–phenotype map plays important roles in evolution-',\n",
       " 'ary dynamics, leading to key signatures of selective pressure, such',\n",
       " 'as the ratio of synonymous to nonsynonymous mutations (28).',\n",
       " 'Quantitative insights to the effects of this mapping on molecular',\n",
       " 'evolution have been provided by Manrubia and coworkers (29),',\n",
       " 'who studied short RNA sequences for which fitness can be esti-',\n",
       " 'mated by secondary structure predictions. In our case, a simpli-',\n",
       " 'fied overview of the genotype–phenotype map can be gleaned',\n",
       " 'from a network representation of the genetic code (Fig. S5): If',\n",
       " 'all mutations were neutral and amino acids within an epitope in-',\n",
       " 'dependent, this map would be sufficient to determine the statis-',\n",
       " 'tics of evolutionary drifts. The accessibility of certain mutations',\n",
       " 'from a phenotype, in particular, has been shown to affect the',\n",
       " 'evolvability of viruses (30).',\n",
       " '',\n",
       " 'Another major ingredient for a realistic model of immune es-',\n",
       " 'cape is the fitness landscape beyond the level of the single-peak',\n",
       " 'model. Much progress has been made recently in understanding',\n",
       " 'the nature of fitness landscapes (31–34). Explicit fitness measure-',\n",
       " 'ments of viral clones (35, 36) and biochemical assays of proteins',\n",
       " '(37) both indicate that single-nucleotide substitutions lead to a',\n",
       " 'broad distribution of fitness changes, most of which are deleter-',\n",
       " 'ious. Therefore, one may assume that the fitness of a mutant is a',\n",
       " 'random variable centered around a mean fitness value f . We ex-',\n",
       " 'pect this mean fitness to be a decreasing function of distance d to',\n",
       " 'the WT (the number of amino acids that are different), which we',\n",
       " '',\n",
       " 'Woo and Reifman PNAS ∣ August 7, 2012 ∣ vol. 109 ∣ no. 32 ∣ 12981',\n",
       " '',\n",
       " 'BI',\n",
       " 'O',\n",
       " 'PH',\n",
       " '',\n",
       " 'YS',\n",
       " 'IC',\n",
       " 'S',\n",
       " 'A',\n",
       " 'N',\n",
       " 'D',\n",
       " '',\n",
       " 'CO',\n",
       " 'M',\n",
       " 'PU',\n",
       " '',\n",
       " 'TA',\n",
       " 'TI',\n",
       " 'O',\n",
       " 'N',\n",
       " 'A',\n",
       " 'L',\n",
       " 'BI',\n",
       " 'O',\n",
       " 'LO',\n",
       " '',\n",
       " 'G',\n",
       " 'Y',\n",
       " '',\n",
       " 'D',\n",
       " 'ow',\n",
       " '',\n",
       " 'nl',\n",
       " 'oa',\n",
       " '',\n",
       " 'de',\n",
       " 'd ',\n",
       " '',\n",
       " 'by',\n",
       " ' g',\n",
       " '',\n",
       " 'ue',\n",
       " 'st',\n",
       " '',\n",
       " ' o',\n",
       " 'n ',\n",
       " '',\n",
       " 'A',\n",
       " 'pr',\n",
       " '',\n",
       " 'il ',\n",
       " '23',\n",
       " '',\n",
       " ', 2',\n",
       " '02',\n",
       " '',\n",
       " '0 ',\n",
       " '',\n",
       " 'http://www.pnas.org/lookup/suppl/doi:10.1073/pnas.1117201109/-/DCSupplemental/pnas.1117201109_SI.pdf?targetid=STXT',\n",
       " 'http://www.pnas.org/lookup/suppl/doi:10.1073/pnas.1117201109/-/DCSupplemental/pnas.1117201109_SI.pdf?targetid=SF1',\n",
       " 'http://www.pnas.org/lookup/suppl/doi:10.1073/pnas.1117201109/-/DCSupplemental/pnas.1117201109_SI.pdf?targetid=SF3',\n",
       " 'http://www.pnas.org/lookup/suppl/doi:10.1073/pnas.1117201109/-/DCSupplemental/pnas.1117201109_SI.pdf?targetid=SF4',\n",
       " 'http://www.pnas.org/lookup/suppl/doi:10.1073/pnas.1117201109/-/DCSupplemental/pnas.1117201109_SI.pdf?targetid=SF4',\n",
       " 'http://www.pnas.org/lookup/suppl/doi:10.1073/pnas.1117201109/-/DCSupplemental/pnas.1117201109_SI.pdf?targetid=SF5',\n",
       " 'http://www.pnas.org/lookup/suppl/doi:10.1073/pnas.1117201109/-/DCSupplemental/pnas.1117201109_SI.pdf?targetid=SF5',\n",
       " '',\n",
       " '',\n",
       " 'assume to be exponential: f ðdÞ ¼ expð−d∕ξÞ, where ξ denotes a',\n",
       " 'characteristic distance (Methods).',\n",
       " '',\n",
       " 'An empirical evidence for this choice can be found from a re-',\n",
       " 'cent experimental study by Fernández et al. (37), who measured',\n",
       " 'the fitness landscapes of HIV-1 protease quasispecies for three',\n",
       " 'patients. We plotted the distribution of their reported fitness',\n",
       " 'values as a function of distance to the dominant phenotype',\n",
       " 'and found near-exponential dependence for two quasispecies',\n",
       " '(Fig. S6). Our fitness function therefore models both the de-',\n",
       " 'crease of the mean fitness away from WT and the distribution',\n",
       " 'of neutral, deleterious, and beneficial mutants for a given dis-',\n",
       " 'tance. In simulations, these fitness values were assigned dynami-',\n",
       " 'cally to newly encountered phenotypes. It is important to note,',\n",
       " 'however, that this landscape chosen is still an approximation that',\n",
       " 'ignores many potentially important effects, such as the heteroge-',\n",
       " 'neity of the mutational neighborhoods within the phenotype',\n",
       " 'space.',\n",
       " '',\n",
       " 'We tested this quasispecies model with the ultradeep sequen-',\n",
       " 'cing data (9, 22) of the SIVepitope Tat28–35SL8 (38). Fig. 1 shows',\n",
       " 'the time dependence of the viral load and WT frequency from',\n",
       " 'simulations of the single-epitope version of Eq. 3 compared with',\n",
       " 'the experimental data (22). The initial rapid growth of the viral',\n",
       " 'load, its subsequent decrease as CTLs are activated, and the more',\n",
       " 'gradual increase as mutants appear are all captured quantita-',\n",
       " 'tively. The WT frequency decrease agrees with experimental',\n",
       " 'trends (Fig. 1B) with signatures of two distinct time scales (22).',\n",
       " 'The mutation rate we used (μ ¼ 2 × 10−7 bp−1) was chosen to',\n",
       " 'obtain the best overall agreement for WT frequency with the sin-',\n",
       " 'gle-epitope version of the model. For multiepitope applications,',\n",
       " 'larger mutation rate values close to the experimental estimate',\n",
       " 'for HIV-1 (μ ¼ 3.4 × 10−5 bp−1) (39) gave realistic dynamics',\n",
       " '(Fig. 2).',\n",
       " '',\n",
       " 'Fig. 1B also shows the frequency of the most dominant escape',\n",
       " 'mutant (EM) at each time point. As mutations occur, a new strain',\n",
       " 'proliferates temporarily before being curbed down, and the total',\n",
       " 'number of phenotypes continues to increase. The increase in via-',\n",
       " 'bility of populations with multiple but lower fitness peaks has',\n",
       " 'been described first by Schuster and Swetina (40) and has later',\n",
       " 'been known as the “survival of the flattest” effect (41). Further',\n",
       " 'insights into the role this expansion in sequence space plays in the',\n",
       " 'growth of viral loads can be gained from the deterministic con-',\n",
       " 'tinuum approximation, Eq. 4, without mutation: The immune re-',\n",
       " 'sponse to WT leads to damped oscillations of viral load and CTL',\n",
       " '',\n",
       " 'level (Fig. S4), reaching a steady state n� and c�. This process is',\n",
       " 'repeated for each new mutant, leading to continued increases',\n",
       " 'in n� and c� roughly proportional to the number of phenotypes',\n",
       " '(18). However, the timing, distribution, and the probability of',\n",
       " 'the appearance of mutants are highly nontrivial functions of',\n",
       " 'the characteristics of the system, including genome length, gen-',\n",
       " 'otype–phenotype map, and fitness landscape. Our quantitative',\n",
       " 'quasispecies-based model, Eq. 3, provides a realistic description',\n",
       " 'of this complex diversification process via the single parameter μ.',\n",
       " 'We note that, for a single epitope, this interpretation ignores the',\n",
       " 'effects of competition among viral strains because each CTL',\n",
       " 'is specific to only one phenotype. In reality, in addition to com-',\n",
       " 'peting directly for host cells, different viral strains share their vul-',\n",
       " 'nerability to CTLs specific to common epitopes within their',\n",
       " 'genomes. By considering multiple epitopes within a strain, our',\n",
       " 'model takes into account this interdependence of viral strains',\n",
       " 'and their indirect competition arising from shared epitope phe-',\n",
       " 'notypes (see below).',\n",
       " '',\n",
       " 'As a quantitative measure of the relative degree of WT',\n",
       " 'persistence during immune escape, we also examined the time',\n",
       " '(half-life) required for WT frequency to reach one-half (Fig. 1B)',\n",
       " '(22) with varying stimulation rate b and clearance rate p (Fig. S7).',\n",
       " 'The SIV epitopes studied by Bimber et al. (9) were estimated to',\n",
       " 'have a WT half-life of about 20 d (22). We found that, within our',\n",
       " 'model, the WT half-life increases with increasing p and b in this',\n",
       " 'range: An increase in the magnitude of selection pressure, rather',\n",
       " 'than accelerating immune escape, suppresses viral population',\n",
       " 'growth more effectively and reduces the chances of escape muta-',\n",
       " 'tion (Fig. 2). The immune escape thus becomes more pronounced',\n",
       " 'when viral loads in the chronic phase are larger with more fre-',\n",
       " 'quent mutations.',\n",
       " '',\n",
       " 'Disease Progression Under Immune Response.We examined general',\n",
       " 'trends of viral infection-immune response dynamics within our',\n",
       " 'quasispecies model on multiple-epitope levels (l ¼ 3) with varia-',\n",
       " 'tions of key model parameters. The evolutionary dynamics, in',\n",
       " 'particular, is critically affected by p, the clearance rate represent-',\n",
       " 'ing the effectiveness of immune response, and the mutation',\n",
       " 'rate μ. We identified two qualitatively different behaviors based',\n",
       " 'on the eventual fate of viral load/CTL level: in one regime',\n",
       " '',\n",
       " '0 50 100 150',\n",
       " '10',\n",
       " '',\n",
       " '0',\n",
       " '',\n",
       " '10',\n",
       " '2',\n",
       " '',\n",
       " '10',\n",
       " '4',\n",
       " '',\n",
       " '10',\n",
       " '6',\n",
       " '',\n",
       " '10',\n",
       " '8',\n",
       " '',\n",
       " 'V',\n",
       " 'ira',\n",
       " '',\n",
       " 'l l',\n",
       " 'oa',\n",
       " '',\n",
       " 'd',\n",
       " '',\n",
       " '0 50 100 150',\n",
       " 'Time (d)',\n",
       " '',\n",
       " '10',\n",
       " '-2',\n",
       " '',\n",
       " '10',\n",
       " '-1',\n",
       " '',\n",
       " '10',\n",
       " '0',\n",
       " '',\n",
       " 'F',\n",
       " 're',\n",
       " '',\n",
       " 'qu',\n",
       " 'en',\n",
       " '',\n",
       " 'cy',\n",
       " '',\n",
       " 'WT',\n",
       " 'Dominant EM',\n",
       " '',\n",
       " 'A',\n",
       " '',\n",
       " 'B',\n",
       " '',\n",
       " 'Fig. 1. Simulation of full immune escape dynamics of a single SIV epitope',\n",
       " 'Tat28–35SL8. (A) Viral load. (B) Frequencies of WT and the dominant escape',\n",
       " 'mutants (EM). Symbols represent experimental data from refs. 9 and 22, with',\n",
       " 'each symbol corresponding to one of four different animals. Solid lines are',\n",
       " 'from our stochastic quasispecies model with τ ¼ 0.8 d, μ ¼ 2.0 × 10−7 bp−1,',\n",
       " 'b ¼ 0.01 d−1, g ¼ 0.2 d−1, and p ¼ 2.0 × 10−4 d−1. The fitness function para-',\n",
       " 'meters were σ ¼ 0.1 and ξ ¼ 1 (Methods). The EM frequency is for a single',\n",
       " 'trajectory while the rest are averages.',\n",
       " '',\n",
       " '10',\n",
       " '1',\n",
       " '',\n",
       " '10',\n",
       " '2',\n",
       " '',\n",
       " '10',\n",
       " '3',\n",
       " '',\n",
       " '10',\n",
       " '4',\n",
       " '',\n",
       " 'V',\n",
       " 'ira',\n",
       " '',\n",
       " 'l l',\n",
       " 'oa',\n",
       " '',\n",
       " 'd',\n",
       " '',\n",
       " '10',\n",
       " '0',\n",
       " '',\n",
       " '10',\n",
       " '1',\n",
       " '',\n",
       " '10',\n",
       " '2',\n",
       " '',\n",
       " '10',\n",
       " '3',\n",
       " '',\n",
       " 'Time (d)',\n",
       " '',\n",
       " '10',\n",
       " '0',\n",
       " '',\n",
       " '10',\n",
       " '1',\n",
       " '',\n",
       " 'N',\n",
       " 'o.',\n",
       " '',\n",
       " ' o',\n",
       " 'f p',\n",
       " '',\n",
       " 'he',\n",
       " 'no',\n",
       " '',\n",
       " 'ty',\n",
       " 'pe',\n",
       " '',\n",
       " 's',\n",
       " '',\n",
       " '10',\n",
       " '2',\n",
       " '',\n",
       " '10',\n",
       " '3',\n",
       " '',\n",
       " '10',\n",
       " '4',\n",
       " '',\n",
       " 'C',\n",
       " 'T',\n",
       " '',\n",
       " 'L',\n",
       " '',\n",
       " 'p = 1.0x10',\n",
       " '-3',\n",
       " '',\n",
       " ', µ = 1.0x10-5',\n",
       " '',\n",
       " 'p = 5.0x10',\n",
       " '-4',\n",
       " '',\n",
       " ', µ = 1.0x10-5',\n",
       " '',\n",
       " 'p = 1.0x10',\n",
       " '-3',\n",
       " '',\n",
       " ', µ = 1.0x10-6',\n",
       " '',\n",
       " '0',\n",
       " '',\n",
       " '4000',\n",
       " '',\n",
       " '8000',\n",
       " '',\n",
       " 'Acute',\n",
       " '',\n",
       " 'Chronic',\n",
       " '',\n",
       " '100',\n",
       " '',\n",
       " 'A',\n",
       " '',\n",
       " 'B',\n",
       " '',\n",
       " 'C',\n",
       " '',\n",
       " 'Disease',\n",
       " '',\n",
       " '101 102 103',\n",
       " '',\n",
       " 'Fig. 2. Typical disease progression patterns in the runaway regime from the',\n",
       " 'multiepitope quasispecies model. (A) Viral load. (B) Total CTL levels. (C) Num-',\n",
       " 'ber of distinct phenotypes per epitope present in the population. The inset in',\n",
       " 'A shows the viral load in linear scale for p ¼ 1.0 × 10−3 d−1 and μ ¼ 1.0 ×',\n",
       " '10−5 bp−1. Other parameter values were b ¼ 0.02 d−1, g ¼ 0.1 d−1, τ ¼ 1 d,',\n",
       " 'σ ¼ 0.2, and ξ ¼ 1. The units of p and μ in the legends are d−1 and bp−1,',\n",
       " 'respectively. All data represent averages over trajectories.',\n",
       " '',\n",
       " '12982 ∣ www.pnas.org/cgi/doi/10.1073/pnas.1117201109 Woo and Reifman',\n",
       " '',\n",
       " 'D',\n",
       " 'ow',\n",
       " '',\n",
       " 'nl',\n",
       " 'oa',\n",
       " '',\n",
       " 'de',\n",
       " 'd ',\n",
       " '',\n",
       " 'by',\n",
       " ' g',\n",
       " '',\n",
       " 'ue',\n",
       " 'st',\n",
       " '',\n",
       " ' o',\n",
       " 'n ',\n",
       " '',\n",
       " 'A',\n",
       " 'pr',\n",
       " '',\n",
       " 'il ',\n",
       " '23',\n",
       " '',\n",
       " ', 2',\n",
       " '02',\n",
       " '',\n",
       " '0 ',\n",
       " '',\n",
       " 'http://www.pnas.org/lookup/suppl/doi:10.1073/pnas.1117201109/-/DCSupplemental/pnas.1117201109_SI.pdf?targetid=SF6',\n",
       " 'http://www.pnas.org/lookup/suppl/doi:10.1073/pnas.1117201109/-/DCSupplemental/pnas.1117201109_SI.pdf?targetid=SF6',\n",
       " 'http://www.pnas.org/lookup/suppl/doi:10.1073/pnas.1117201109/-/DCSupplemental/pnas.1117201109_SI.pdf?targetid=SF4',\n",
       " 'http://www.pnas.org/lookup/suppl/doi:10.1073/pnas.1117201109/-/DCSupplemental/pnas.1117201109_SI.pdf?targetid=SF4',\n",
       " 'http://www.pnas.org/lookup/suppl/doi:10.1073/pnas.1117201109/-/DCSupplemental/pnas.1117201109_SI.pdf?targetid=SF7',\n",
       " 'http://www.pnas.org/lookup/suppl/doi:10.1073/pnas.1117201109/-/DCSupplemental/pnas.1117201109_SI.pdf?targetid=SF7',\n",
       " '',\n",
       " '',\n",
       " '(“runaway,” Fig. 2), the chronic phase after the resolution of the',\n",
       " 'acute infection is eventually followed by a runaway growth in viral',\n",
       " 'loads (Fig. 2A). The level of CTLs also rises consistently during',\n",
       " 'the chronic phase (Fig. 2B). This disease progression is accom-',\n",
       " 'panied by an increase in the diversity of quasispecies, or the',\n",
       " '“spreading of clouds” (Fig. 2C). The time scale for the duration',\n",
       " 'of the chronic phase, in particular, varied sensitively with p and μ',\n",
       " '(from a few months to years) (Fig. 2A). We found that the total',\n",
       " 'number of phenotypes per epitope (Fig. 2C) generally goes down',\n",
       " 'as the number of epitopes increases: A new escape mutant has',\n",
       " 'much lower probability for survival with multiple epitopes be-',\n",
       " 'cause it shares WT phenotypes on other parts of its genome with',\n",
       " 'existing strains, and therefore is susceptible to CTLs that are al-',\n",
       " 'ready present.',\n",
       " '',\n",
       " 'The disease progression pattern in the runaway regime corre-',\n",
       " 'sponds to situations where the immune system is overwhelmed by',\n",
       " 'viral growth, either shortly after infection (there is no discernible',\n",
       " 'chronic phase for sufficiently small p) or after a long chronic',\n",
       " 'phase and accumulation of escape mutants. The latter case clo-',\n",
       " 'sely matches the characteristic HIV-1 disease pathogenesis (7),',\n",
       " 'revealing the major role played by the uncontrolled diversifica-',\n",
       " 'tion of the quasispecies that overwhelms the CTL response. This',\n",
       " 'feature was first suggested by Nowak and May (19), who coined',\n",
       " 'the term “diversity threshold” based on a model assuming ran-',\n",
       " 'dom appearances of mutants. Our results show that the basic qua-',\n",
       " 'sispecies dynamics under selection pressure provide a realistic',\n",
       " 'description of this phenomenon.',\n",
       " '',\n",
       " 'If p is sufficiently large and μ sufficiently small, one enters a',\n",
       " 'different regime (“extinction,” Fig. 3), where the acute and',\n",
       " 'chronic phases lead to viral loads that decrease either rapidly',\n",
       " 'or asymptotically with time. As populations shrink in size, their',\n",
       " 'dynamics become increasingly stochastic, and extinction (a com-',\n",
       " 'plete clearance of infection) occurs when the viral load reaches',\n",
       " 'zero. This behavior is inherently stochastic and cannot be cap-',\n",
       " 'tured by the deterministic approximation in Eq. 4. The time re-',\n",
       " 'quired for clearing the infection (“extinction time”) typically has',\n",
       " 'a broad distribution, increasing with decreasing p while becoming',\n",
       " 'infinite above a threshold μ (Fig. 4A). The rapid clearance of in-',\n",
       " 'fection within approximately 10 d corresponds to the normal state',\n",
       " 'of affairs in a healthy immune system against viruses such as in-',\n",
       " 'fluenza. As shown in Fig. 3A (p ¼ 0.01 d−1), however, the extinc-',\n",
       " 'tion time can also reach time scales approaching a patient’s',\n",
       " '',\n",
       " 'lifetime. This feature may be relevant in understanding the basis',\n",
       " 'for some chronic viral infections with long and stable setpoints',\n",
       " '(e.g., hepatitis C), although the high variability of extinction times',\n",
       " 'we observed (Fig. 4A) likely may not correspond to actual pos-',\n",
       " 'sible clearances of such chronic infections.',\n",
       " '',\n",
       " 'The threshold separating the runaway (Fig. 2) and extinction',\n",
       " '(Fig. 3) regimes can be identified by the maximum mutation',\n",
       " 'rate for which the extinction time remains finite (Fig. 4A). This',\n",
       " 'error threshold for the immune control of infection is analo-',\n",
       " 'gous to Eigen’s threshold for genomic stability: The latter is',\n",
       " 'the error rate above which replicators cannot maintain a stable',\n",
       " 'master sequence. The former is a maximum mutation rate the',\n",
       " 'given immune system can suppress. Fig. 4B shows this threshold',\n",
       " 'as a function of p and μ. A given virus would have roughly the',\n",
       " 'same mutation rate, while p would vary with individual patients.',\n",
       " 'The sensitivity of disease progression with p is consistent with',\n",
       " 'the observation that even when the fatality rate of a certain',\n",
       " 'infection is known, the course of infection in a patient is often',\n",
       " 'unpredictable.',\n",
       " '',\n",
       " 'The case for HIV-1 is unique in the sense that without treat-',\n",
       " 'ment, most patients eventually progress to the disease stage. One',\n",
       " 'of the special characteristics of HIV-1 is that it targets CD4þ',\n",
       " '(helper) Tcells, which play critical roles in eliciting and mediating',\n",
       " 'CTL responses. Within the context of our model, the destruction',\n",
       " 'of CD4þ Tcells would lead to a gradual decrease in both the sti-',\n",
       " 'mulation rate b and the clearance rate p. One therefore expects a',\n",
       " 'continual deterioration of any initial effective control of infec-',\n",
       " 'tion, represented by p values that decrease over the course of dis-',\n",
       " 'ease progression. In Fig. 4B, therefore, even though patients with',\n",
       " 'stronger immune systems may have p values initially in the extinc-',\n",
       " 'tion regime, the continued depletion of CD4þ cells would lower p',\n",
       " 'and cause them to cross the threshold.',\n",
       " '',\n",
       " 'Conclusions',\n",
       " 'In this paper, we demonstrated that the classic quasispecies the-',\n",
       " 'ory can form the basis of a quantitative model of virus evolution',\n",
       " 'under immune selection. One of the apparent challenges in de-',\n",
       " 'veloping such a model is that viral genomes, while relatively small',\n",
       " '(approximately 104 bp), still constitute a huge sequence space.',\n",
       " 'Here, we showed that a set of epitopes (La ∼ 10 amino acids)',\n",
       " 'can be considered as minimal units of genomic segments on which',\n",
       " 'selection acts. In standard quasispecies theory, competition be-',\n",
       " 'tween different strains arises indirectly from the constraint of',\n",
       " 'fixed total population size. In our model, the degradation and',\n",
       " 'removal of virions occur via immune response, which can be re-',\n",
       " 'garded as a concrete specification of the selection forces. In this',\n",
       " 'case, indirect competition occurs because most viral strains share',\n",
       " '',\n",
       " '10',\n",
       " '1',\n",
       " '',\n",
       " '10',\n",
       " '2',\n",
       " '',\n",
       " '10',\n",
       " '3',\n",
       " '',\n",
       " '10',\n",
       " '4',\n",
       " '',\n",
       " 'V',\n",
       " 'ira',\n",
       " '',\n",
       " 'l l',\n",
       " 'oa',\n",
       " '',\n",
       " 'd',\n",
       " '',\n",
       " '10',\n",
       " '0',\n",
       " '',\n",
       " '10',\n",
       " '1',\n",
       " '',\n",
       " '10',\n",
       " '2',\n",
       " '',\n",
       " '10',\n",
       " '3',\n",
       " '',\n",
       " '10',\n",
       " '4',\n",
       " '',\n",
       " 'Time (d)',\n",
       " '',\n",
       " '1',\n",
       " '',\n",
       " '2',\n",
       " '',\n",
       " '3',\n",
       " '',\n",
       " '4',\n",
       " '',\n",
       " 'N',\n",
       " 'o.',\n",
       " '',\n",
       " ' o',\n",
       " 'f p',\n",
       " '',\n",
       " 'he',\n",
       " 'no',\n",
       " '',\n",
       " 'ty',\n",
       " 'pe',\n",
       " '',\n",
       " 's',\n",
       " '',\n",
       " 'p = 0.01 d',\n",
       " '-1',\n",
       " '',\n",
       " '10',\n",
       " '1',\n",
       " '',\n",
       " '10',\n",
       " '2',\n",
       " '',\n",
       " '10',\n",
       " '3',\n",
       " '',\n",
       " 'C',\n",
       " 'T',\n",
       " '',\n",
       " 'L',\n",
       " '',\n",
       " 'Acute',\n",
       " 'Chronic',\n",
       " '',\n",
       " 'Extinction',\n",
       " '',\n",
       " 'A',\n",
       " '',\n",
       " 'B',\n",
       " '',\n",
       " 'C',\n",
       " '',\n",
       " '} p = 0.02 d',\n",
       " '-1',\n",
       " '',\n",
       " 'Fig. 3. Typical disease progression patterns in the extinction regime from',\n",
       " 'the multi-epitope quasispecies model. (A) Viral load. (B) Total CTL levels.',\n",
       " '(C) Number of distinct phenotypes per epitope present in the population.',\n",
       " 'The parameter values other than p and μ ¼ 1.0 × 10−5 bp−1 were the same',\n",
       " 'as in Fig. 2. The data shown are individual trajectories.',\n",
       " '',\n",
       " '10',\n",
       " '-6',\n",
       " '',\n",
       " '10',\n",
       " '-5',\n",
       " '',\n",
       " '10',\n",
       " '-4',\n",
       " '',\n",
       " '10',\n",
       " '-3',\n",
       " '',\n",
       " '10',\n",
       " '-2',\n",
       " '',\n",
       " '10',\n",
       " '-1',\n",
       " '',\n",
       " 'Mutation rate µ (bp-1)',\n",
       " '',\n",
       " '10',\n",
       " '-2',\n",
       " '',\n",
       " '10',\n",
       " '-1',\n",
       " '',\n",
       " 'C',\n",
       " 'le',\n",
       " '',\n",
       " 'ar',\n",
       " 'an',\n",
       " '',\n",
       " 'ce',\n",
       " ' r',\n",
       " '',\n",
       " 'at',\n",
       " 'e ',\n",
       " '',\n",
       " 'p ',\n",
       " '(d',\n",
       " '',\n",
       " '-1',\n",
       " ')',\n",
       " '',\n",
       " '10',\n",
       " '-6',\n",
       " '',\n",
       " '10',\n",
       " '-5',\n",
       " '',\n",
       " '10',\n",
       " '-4',\n",
       " '',\n",
       " '10',\n",
       " '-3',\n",
       " '',\n",
       " '10',\n",
       " '-2',\n",
       " '',\n",
       " '10',\n",
       " '-1',\n",
       " '',\n",
       " 'Mutation rate µ (bp-1)',\n",
       " '',\n",
       " '10',\n",
       " '0',\n",
       " '',\n",
       " '10',\n",
       " '1',\n",
       " '',\n",
       " '10',\n",
       " '2',\n",
       " '',\n",
       " '10',\n",
       " ...]"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# separate all lines in the PDF file\n",
    "cont.splitlines()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [],
   "source": [
    "# get a conent Partition at a specific word\n",
    "meth = cont.partition('quasispecies')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "201117201 12980..12985\n",
      "\n",
      "\n",
      "A quantitative \n"
     ]
    }
   ],
   "source": [
    "print(meth[0])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.5.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
